
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Genome Paper Figures: DMT analysis Version 2 &#8212; A marmoratus Genome Project 1.0 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to A marmoratus Genome Project’s documentation!" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #303F9F;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #D84315;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 8ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #cfcfcf;
    border-radius: 2px;
    background: #f7f7f7;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.pngmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-bold { font-weight: bold; }
</style>
<div class="section" id="Genome-Paper-Figures:-DMT-analysis-Version-2">
<h1>Genome Paper Figures: DMT analysis Version 2<a class="headerlink" href="#Genome-Paper-Figures:-DMT-analysis-Version-2" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Introduction">
<h2>Introduction<a class="headerlink" href="#Introduction" title="Permalink to this headline">¶</a></h2>
<p>&nbsp;&nbsp;&nbsp;&nbsp; This notebook summarizes the analysis that was done for the
<em>Aspidoscelis marmoratus</em> genome publication. The analysis done for this
project started in 2011 and spans many projects. Many of these side
projects did not make it in entirety into the final publication. Rather
than upload every single sub project, regardless of whether all the
material contained was relevent, we decided instead to summarize the
manuscript in a single, organized notebook. Any additional information
required can be requested by contacting me at <a class="reference external" href="mailto:duncan&#37;&#52;&#48;baumannlab&#46;org">duncan<span>&#64;</span>baumannlab<span>&#46;</span>org</a>.</p>
</div>
<div class="section" id="Imports">
<h2>Imports<a class="headerlink" href="#Imports" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [1]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">print_function</span>
<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">division</span>
<span class="o">%</span><span class="k">matplotlib</span> inline
<span class="o">%</span><span class="k">load_ext</span> ipycache
<span class="kn">import</span> <span class="nn">matplotlib</span>
<span class="kn">import</span> <span class="nn">matplotlib.pylab</span> <span class="kn">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">matplotlib.ticker</span> <span class="kn">as</span> <span class="nn">ticker</span>
<span class="kn">from</span> <span class="nn">matplotlib.ticker</span> <span class="kn">import</span> <span class="n">ScalarFormatter</span>
<span class="kn">from</span> <span class="nn">matplotlib.patches</span> <span class="kn">import</span> <span class="n">Rectangle</span>
<span class="kn">import</span> <span class="nn">matplotlib.gridspec</span> <span class="kn">as</span> <span class="nn">gridspec</span>
<span class="kn">from</span> <span class="nn">matplotlib.collections</span> <span class="kn">import</span> <span class="n">BrokenBarHCollection</span>
<span class="kn">import</span> <span class="nn">matplotlib.patches</span> <span class="kn">as</span> <span class="nn">mpatches</span>
<span class="kn">from</span> <span class="nn">shapely.geometry.polygon</span> <span class="kn">import</span> <span class="n">Polygon</span>
<span class="kn">from</span> <span class="nn">descartes</span> <span class="kn">import</span> <span class="n">PolygonPatch</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">Markdown</span><span class="p">,</span> <span class="n">display</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="kn">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">pandas.tools.plotting</span> <span class="kn">import</span> <span class="n">scatter_matrix</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="kn">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">scipy.stats</span> <span class="kn">as</span> <span class="nn">stats</span>
<span class="kn">import</span> <span class="nn">itertools</span>
<span class="kn">from</span> <span class="nn">itertools</span> <span class="kn">import</span> <span class="n">izip</span>
<span class="kn">import</span> <span class="nn">multiprocessing</span> <span class="kn">as</span> <span class="nn">mp</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="kn">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">glob</span>
<span class="kn">from</span> <span class="nn">pprint</span> <span class="kn">import</span> <span class="n">pprint</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">Image</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span><span class="p">,</span> <span class="n">HTML</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">set_matplotlib_formats</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">Image</span>
<span class="kn">from</span> <span class="nn">matplotlib.ticker</span> <span class="kn">import</span> <span class="n">FuncFormatter</span>
<span class="kn">import</span> <span class="nn">matplotlib</span>
<span class="kn">import</span> <span class="nn">math</span>
<span class="n">plt</span><span class="o">.</span><span class="n">style</span><span class="o">.</span><span class="n">use</span><span class="p">(</span><span class="s1">&#39;seaborn-whitegrid&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="p">[</span><span class="s2">&quot;font.family&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;sans-serif&quot;</span>
<span class="n">plt</span><span class="o">.</span><span class="n">rcParams</span><span class="p">[</span><span class="s2">&quot;font.sans-serif&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;Arial&quot;</span>
<span class="n">matplotlib</span><span class="o">.</span><span class="n">rcParams</span><span class="p">[</span><span class="s1">&#39;pdf.fonttype&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">42</span>
<span class="n">matplotlib</span><span class="o">.</span><span class="n">rcParams</span><span class="p">[</span><span class="s1">&#39;ps.fonttype&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">42</span>
<span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s1">&#39;display.max_colwidth&#39;</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">HTML</span><span class="p">(</span><span class="s2">&quot;&lt;style&gt;.container { width:80% !important; }&lt;/style&gt;&quot;</span><span class="p">))</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;../bin&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">sys</span><span class="o">.</span><span class="n">path</span>
<span class="kn">import</span> <span class="nn">gff3_plotting</span> <span class="kn">as</span> <span class="nn">gff3</span>
<span class="kn">import</span> <span class="nn">fasta_classes</span> <span class="kn">as</span> <span class="nn">fa</span>
<span class="kn">import</span> <span class="nn">blizzard2</span> <span class="kn">as</span> <span class="nn">bliz</span>

</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="stderr output_area docutils container">
<div class="highlight"><pre>
/home/dut/anaconda3/envs/anaconda2/lib/python2.7/site-packages/IPython/config.py:13: ShimWarning: The `IPython.config` package has been deprecated since IPython 4.0. You should import from traitlets.config instead.
  &#34;You should import from traitlets.config instead.&#34;, ShimWarning)
/home/dut/anaconda3/envs/anaconda2/lib/python2.7/site-packages/IPython/utils/traitlets.py:5: UserWarning: IPython.utils.traitlets has moved to a top-level traitlets package.
  warn(&#34;IPython.utils.traitlets has moved to a top-level traitlets package.&#34;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<style>.container { width:80% !important; }</style></div>
</div>
</div>
<div class="section" id="Naming-Colors-and-Constants">
<h2>Naming Colors and Constants<a class="headerlink" href="#Naming-Colors-and-Constants" title="Permalink to this headline">¶</a></h2>
<p>Here I establish some colors and dictionaries to convert sample names
back and forth.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [220]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">colors</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;#543005&#39;</span><span class="p">,</span><span class="s1">&#39;#8c510a&#39;</span><span class="p">,</span><span class="s1">&#39;#bf812d&#39;</span><span class="p">,</span><span class="s1">&#39;#dfc27d&#39;</span><span class="p">,</span><span class="s1">&#39;#f6e8c3&#39;</span><span class="p">,</span><span class="s1">&#39;#c7eae5&#39;</span><span class="p">,</span><span class="s1">&#39;#80cdc1&#39;</span><span class="p">,</span><span class="s1">&#39;#35978f&#39;</span><span class="p">,</span><span class="s1">&#39;#01665e&#39;</span><span class="p">,</span><span class="s1">&#39;#003c30&#39;</span><span class="p">]</span>
<span class="n">animal_names</span>  <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;003&#39;</span><span class="p">,</span>
    <span class="s1">&#39;001&#39;</span><span class="p">,</span>
    <span class="s1">&#39;122&#39;</span><span class="p">,</span>
    <span class="s1">&#39;4278&#39;</span><span class="p">,</span>
    <span class="s1">&#39;9721&#39;</span><span class="p">,</span>
    <span class="s1">&#39;6993&#39;</span><span class="p">,</span>
    <span class="s1">&#39;9177&#39;</span><span class="p">,</span>
    <span class="s1">&#39;8450&#39;</span><span class="p">,</span>
    <span class="s1">&#39;12512&#39;</span><span class="p">,</span>
    <span class="s1">&#39;12513&#39;</span><span class="p">,</span>
<span class="p">]</span>

<span class="n">animal_ids</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;Atig003&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig001&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_122&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_4278&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_9721&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">,</span>
<span class="p">]</span>


<span class="n">og_animal_ids</span> <span class="o">=</span> <span class="p">[</span>
     <span class="s1">&#39;Atig003&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig001&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_122&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">,</span>
<span class="p">]</span>

<span class="n">family_1</span> <span class="o">=</span> <span class="p">[</span>
   <span class="s1">&#39;Atig_122&#39;</span><span class="p">,</span>
   <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">,</span>
<span class="p">]</span>

<span class="n">family_2</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;A.tig_9721&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">,</span>
<span class="p">]</span>

<span class="n">family_3</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;Atig_4278&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">,</span>
<span class="p">]</span>
<span class="n">parth_animals</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">,</span>
<span class="p">]</span>

<span class="c1">#1b9e77</span>
<span class="c1">#d95f02</span>
<span class="c1">#7570b3</span>
<span class="c1">#e7298a</span>
<span class="n">family_colors</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Atig001&#39;</span><span class="p">:</span><span class="s1">&#39;#1b9e77&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig003&#39;</span><span class="p">:</span><span class="s1">&#39;#1b9e77&#39;</span><span class="p">,</span>

    <span class="s1">&#39;Atig_122&#39;</span>    <span class="p">:</span><span class="s1">&#39;#d95f02&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">:</span><span class="s1">&#39;#7570b3&#39;</span><span class="p">,</span>

    <span class="s1">&#39;A.tig_9721&#39;</span> <span class="p">:</span><span class="s1">&#39;#d95f02&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">:</span><span class="s1">&#39;#7570b3&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">:</span><span class="s1">&#39;#e7298a&#39;</span><span class="p">,</span>

    <span class="s1">&#39;Atig_4278&#39;</span><span class="p">:</span><span class="s1">&#39;#d95f02&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">:</span><span class="s1">&#39;#7570b3&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">:</span><span class="s1">&#39;#e7298a&#39;</span><span class="p">,</span>
<span class="p">}</span>


<span class="n">color_names</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">animal_names</span><span class="p">,</span><span class="n">colors</span><span class="p">))</span>
<span class="n">color_ids</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">,</span> <span class="n">colors</span><span class="p">))</span>
<span class="n">id_to_name</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">,</span> <span class="n">animal_names</span><span class="p">))</span>
<span class="n">name_to_id</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">animal_names</span><span class="p">,</span> <span class="n">animal_ids</span><span class="p">))</span>
<span class="n">majorFontSize</span> <span class="o">=</span> <span class="mi">10</span>
<span class="n">minorFontSize</span> <span class="o">=</span> <span class="mi">9</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">style</span><span class="o">=</span><span class="s1">&#39;whitegrid&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [3]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">change_name</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="n">map_dict</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">):</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">map_dict</span><span class="p">[</span><span class="n">name</span><span class="p">]</span>
    <span class="k">return</span> <span class="n">name</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Accesory-Functions">
<h2>Accesory Functions<a class="headerlink" href="#Accesory-Functions" title="Permalink to this headline">¶</a></h2>
<p>Here are a few accessory functions that I use later in the analysis.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [4]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">col_name_for_max</span><span class="p">(</span><span class="n">row</span><span class="p">,</span> <span class="n">cols_to_check</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="n">cols_to_check</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
        <span class="n">max_col</span> <span class="o">=</span> <span class="n">row</span><span class="p">[</span><span class="n">cols_to_check</span><span class="p">]</span><span class="o">.</span><span class="n">argmax</span><span class="p">()</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">max_col</span> <span class="o">=</span> <span class="s1">&#39;Unclassified&#39;</span>

    <span class="k">return</span> <span class="n">max_col</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [5]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">apply_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">func</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">func</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [6]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">apply_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">func</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">func</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">apply_by_multiprocessing</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">func</span><span class="p">,</span> <span class="n">workers</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">):</span>
    <span class="n">pool</span> <span class="o">=</span> <span class="n">mp</span><span class="o">.</span><span class="n">Pool</span><span class="p">(</span><span class="n">processes</span><span class="o">=</span><span class="n">workers</span><span class="p">)</span>
    <span class="n">result</span> <span class="o">=</span> <span class="p">[</span><span class="n">pool</span><span class="o">.</span><span class="n">apply_async</span><span class="p">(</span><span class="n">apply_df</span><span class="p">,</span> <span class="n">args</span> <span class="o">=</span> <span class="p">(</span><span class="n">d</span><span class="p">,</span> <span class="n">func</span><span class="p">)</span> <span class="o">+</span> <span class="n">args</span><span class="p">)</span> <span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">array_split</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">workers</span><span class="p">)]</span>
    <span class="n">output</span> <span class="o">=</span> <span class="p">[</span><span class="n">p</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">result</span><span class="p">]</span>
    <span class="n">pool</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
    <span class="k">return</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">output</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [7]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">round_up_to_even</span><span class="p">(</span><span class="n">f</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">math</span><span class="o">.</span><span class="n">ceil</span><span class="p">(</span><span class="n">f</span> <span class="o">/</span> <span class="mf">2.</span><span class="p">)</span> <span class="o">*</span> <span class="mi">2</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Initial-Microsatellite-Genotyping-of-8450-and-122">
<h2>Initial Microsatellite Genotyping of 8450 and 122<a class="headerlink" href="#Initial-Microsatellite-Genotyping-of-8450-and-122" title="Permalink to this headline">¶</a></h2>
<p>This figure was generated in excel prior to me joining the lab. I have
included it here both to show the data and transfer the image for the
manuscript upload.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [8]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../fig/old<span class="se">\ </span><span class="m">8450</span><span class="se">\ </span>and<span class="se">\ </span><span class="m">122</span><span class="se">\ </span>genotypes.pdf ../fig2/Figure2A.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [9]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/old 8450 and 122 genotypes.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[9]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_18_0.png" src="_images/marmorata_genome_paper_figures_3_18_0.png" />
</div>
</div>
</div>
<div class="section" id="Population-Microsatellite-Analysis-and-Figures">
<h2>Population Microsatellite Analysis and Figures<a class="headerlink" href="#Population-Microsatellite-Analysis-and-Figures" title="Permalink to this headline">¶</a></h2>
<p>The following python class can be used to bin and manipulate
microsatellite genotyping data and calculate internal relatedness. The
data must be formatted in a specific tsv format.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/blizzard2.py</span>
<span class="c1">#!/usr/bin/env python</span>
<span class="c1"># Author: Duncan Tormey</span>
<span class="c1"># Email: dut@stowers.org or duncantormey@gmail.com</span>

<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">print_function</span>
<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">division</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">Counter</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="kn">as</span> <span class="nn">pd</span>


<span class="k">class</span> <span class="nc">MicrosatellitePopulation</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Class for calculating iternal relatedness and homozygosity by loci for a set of individuals</span>
<span class="sd">    micosatellite genotypes. Also provides a a way to filter data prior to analysis.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">data_file_path</span><span class="p">,</span> <span class="n">split_id</span><span class="o">=</span><span class="bp">True</span><span class="p">):</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">data_file_path</span> <span class="o">=</span> <span class="n">data_file_path</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">data_file_path</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">split_id</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;sample_id&#39;</span><span class="p">,</span> <span class="s1">&#39;size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">,</span> <span class="s1">&#39;height_1&#39;</span><span class="p">,</span> <span class="s1">&#39;height_2&#39;</span><span class="p">]</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;sample_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;-&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;sample_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;-&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">])</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span> <span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span> <span class="s1">&#39;size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">,</span> <span class="s1">&#39;height_1&#39;</span><span class="p">,</span> <span class="s1">&#39;height_2&#39;</span><span class="p">]</span>

        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">size_2</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">size_1</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">sat_bins</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">population_intervals</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">binned_population_sizes</span> <span class="o">=</span> <span class="p">{}</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span> <span class="o">=</span> <span class="p">{}</span>


    <span class="k">def</span> <span class="nf">remove_microsattelite</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">micro_sat</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Removes a specific microsatellite from all individuals in population data&quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span> <span class="o">!=</span> <span class="n">micro_sat</span><span class="p">]</span>

    <span class="k">def</span> <span class="nf">remove_sample</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">sample_name</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Removes a individual based on sample name from the population&quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">!=</span> <span class="n">sample_name</span><span class="p">]</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">!=</span> <span class="nb">str</span><span class="p">(</span><span class="n">sample_name</span><span class="p">)]</span>

    <span class="k">def</span> <span class="nf">get_population_sizes</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Populates the population sizes dictionary with all alleles across all animals&quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">sat</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
            <span class="n">size_dist</span> <span class="o">=</span> \
                <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span> <span class="o">==</span> <span class="n">sat</span><span class="p">]</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s1">&#39;size&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">stack</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()[</span>
                    <span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
            <span class="n">size_dist</span> <span class="o">=</span> <span class="p">[</span><span class="nb">round</span><span class="p">(</span><span class="n">size</span><span class="p">)</span> <span class="k">for</span> <span class="n">size</span> <span class="ow">in</span> <span class="n">size_dist</span><span class="p">]</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">]</span> <span class="o">=</span> <span class="n">size_dist</span>

    <span class="k">def</span> <span class="nf">get_sat_bins</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Determines the number of bins for each microsatellite, based on 3 nucleotide windows&quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">sat</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">:</span>
            <span class="n">bins</span> <span class="o">=</span> <span class="nb">round</span><span class="p">(</span><span class="nb">float</span><span class="p">(</span><span class="nb">max</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">])</span> <span class="o">-</span>
                               <span class="nb">min</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">]))</span> <span class="o">/</span> <span class="mf">3.0</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">bins</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">:</span>
                <span class="n">bins</span> <span class="o">=</span> <span class="mi">1</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">sat_bins</span><span class="p">[</span><span class="n">sat</span><span class="p">]</span> <span class="o">=</span> <span class="n">bins</span>

    <span class="k">def</span> <span class="nf">get_population_intervals</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Determines the intervals for the number of bins for each microsatellite based on</span>
<span class="sd">        self.get_sat_bins&quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">sat</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">:</span>
            <span class="n">out</span><span class="p">,</span> <span class="n">bins</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">cut</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">],</span>
                               <span class="n">bins</span><span class="o">=</span><span class="bp">self</span><span class="o">.</span><span class="n">sat_bins</span><span class="p">[</span><span class="n">sat</span><span class="p">],</span>
                               <span class="n">retbins</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

            <span class="n">intervals</span> <span class="o">=</span> <span class="p">[</span>
                <span class="nb">tuple</span><span class="p">(</span>
                    <span class="nb">float</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
                    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">str</span><span class="p">(</span><span class="n">o</span><span class="p">)</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="bp">None</span><span class="p">,</span> <span class="s2">&quot;()[]&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;, &#39;</span><span class="p">)</span>
                <span class="p">)</span>
                <span class="k">for</span> <span class="n">o</span> <span class="ow">in</span> <span class="nb">set</span><span class="p">(</span><span class="n">out</span><span class="p">)</span>
                <span class="p">]</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">population_intervals</span><span class="p">[</span><span class="n">sat</span><span class="p">]</span> <span class="o">=</span> <span class="n">intervals</span>

    <span class="k">def</span> <span class="nf">in_interval</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">size</span><span class="p">,</span> <span class="n">intervals</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;This is method that takes a allele and list of intervals and returns the interval in</span>
<span class="sd">        which the allele resides&quot;&quot;&quot;</span>
        <span class="n">matched</span> <span class="o">=</span> <span class="bp">None</span>
        <span class="k">for</span> <span class="n">interval</span> <span class="ow">in</span> <span class="n">intervals</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">interval</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">&lt;</span> <span class="n">size</span> <span class="o">&lt;=</span> <span class="n">interval</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span>
                <span class="n">matched</span> <span class="o">=</span> <span class="n">interval</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                <span class="k">break</span>
        <span class="k">return</span> <span class="n">matched</span>

    <span class="k">def</span> <span class="nf">get_binned_population_sizes</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Assigns each of the alleles in population_sizes to an interval&quot;&quot;&quot;</span>
        <span class="k">for</span> <span class="n">sat</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">:</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">binned_population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">in_interval</span><span class="p">(</span><span class="n">size</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_intervals</span><span class="p">[</span><span class="n">sat</span><span class="p">])</span>
                                                 <span class="k">for</span> <span class="n">size</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_sizes</span><span class="p">[</span><span class="n">sat</span><span class="p">]]</span>

    <span class="k">def</span> <span class="nf">get_population_size_frequencies</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Determines the frequency of each allele in binned_population_sizes.&quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span> <span class="o">=</span> <span class="p">{</span>
            <span class="n">key</span><span class="p">:</span> <span class="p">{</span>
                <span class="n">k</span><span class="p">:</span> <span class="nb">float</span><span class="p">(</span><span class="n">v</span><span class="p">)</span> <span class="o">/</span> <span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">val</span><span class="p">))</span>
                <span class="k">for</span> <span class="n">k</span><span class="p">,</span> <span class="n">v</span> <span class="ow">in</span> <span class="n">Counter</span><span class="p">(</span><span class="n">val</span><span class="p">)</span><span class="o">.</span><span class="n">items</span><span class="p">()</span>
                <span class="p">}</span>
            <span class="k">for</span> <span class="n">key</span><span class="p">,</span> <span class="n">val</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">binned_population_sizes</span><span class="o">.</span><span class="n">items</span><span class="p">()</span>
            <span class="p">}</span>

    <span class="k">def</span> <span class="nf">bin_data</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Bins thea data for each sample&quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">get_population_sizes</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">get_sat_bins</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">get_population_intervals</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">get_binned_population_sizes</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">get_population_size_frequencies</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;binned_size_1&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span>
                                                                     <span class="bp">self</span><span class="o">.</span><span class="n">in_interval</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="s1">&#39;size_1&#39;</span><span class="p">]),</span>
                                                                                      <span class="bp">self</span><span class="o">.</span><span class="n">population_intervals</span><span class="p">[</span>
                                                                                          <span class="n">x</span><span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">]]),</span>
                                                                     <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="s1">&#39;binned_size_2&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span>
                                                                     <span class="bp">self</span><span class="o">.</span><span class="n">in_interval</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="s1">&#39;size_2&#39;</span><span class="p">]),</span>
                                                                                      <span class="bp">self</span><span class="o">.</span><span class="n">population_intervals</span><span class="p">[</span>
                                                                                          <span class="n">x</span><span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">]]),</span>
                                                                     <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">calc_internal_relatedness</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
        <span class="sd">&quot;&quot;&quot;Calculates internal relatedness and homozygosity by loci for each sample&quot;&quot;&quot;</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">bin_data</span><span class="p">()</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">sample_name</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
            <span class="n">l_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="n">sample_name</span><span class="p">]</span>
            <span class="n">e_homo</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">e_hetero</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">num_hom_loci</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">num_loci</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">l_df</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mf">0.0</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
            <span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">l_df</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>
                <span class="k">if</span> <span class="n">row</span><span class="o">.</span><span class="n">binned_size_1</span> <span class="o">!=</span> <span class="mf">0.0</span> <span class="ow">and</span> <span class="n">row</span><span class="o">.</span><span class="n">binned_size_2</span> <span class="o">!=</span> <span class="mf">0.0</span><span class="p">:</span>
                    <span class="n">num_loci</span> <span class="o">+=</span> <span class="mi">1</span>
                    <span class="k">if</span> <span class="n">row</span><span class="o">.</span><span class="n">binned_size_1</span> <span class="o">==</span> <span class="n">row</span><span class="o">.</span><span class="n">binned_size_2</span><span class="p">:</span>
                        <span class="n">num_hom_loci</span> <span class="o">+=</span> <span class="mi">1</span>
                        <span class="n">e_homo</span> <span class="o">+=</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">micro_sat</span><span class="p">][</span><span class="n">row</span><span class="o">.</span><span class="n">binned_size_1</span><span class="p">]</span>
                        <span class="n">e_homo</span> <span class="o">+=</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">micro_sat</span><span class="p">][</span><span class="n">row</span><span class="o">.</span><span class="n">binned_size_2</span><span class="p">]</span>
                    <span class="k">else</span><span class="p">:</span>
                        <span class="n">e_hetero</span> <span class="o">+=</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">micro_sat</span><span class="p">][</span><span class="n">row</span><span class="o">.</span><span class="n">binned_size_1</span><span class="p">]</span>
                        <span class="n">e_hetero</span> <span class="o">+=</span> <span class="bp">self</span><span class="o">.</span><span class="n">population_size_frequencies</span><span class="p">[</span><span class="n">row</span><span class="o">.</span><span class="n">micro_sat</span><span class="p">][</span><span class="n">row</span><span class="o">.</span><span class="n">binned_size_2</span><span class="p">]</span>

            <span class="n">score</span> <span class="o">=</span> <span class="n">e_homo</span> <span class="o">/</span> <span class="p">(</span><span class="n">e_hetero</span> <span class="o">+</span> <span class="n">e_homo</span><span class="p">)</span>
            <span class="n">ir</span> <span class="o">=</span> <span class="p">(</span><span class="mi">2</span> <span class="o">*</span> <span class="n">num_hom_loci</span> <span class="o">-</span> <span class="p">(</span><span class="n">e_homo</span> <span class="o">+</span> <span class="n">e_hetero</span><span class="p">))</span> <span class="o">/</span> <span class="p">(</span><span class="mi">2</span> <span class="o">*</span> <span class="n">num_loci</span> <span class="o">-</span> <span class="p">(</span><span class="n">e_homo</span> <span class="o">+</span> <span class="n">e_hetero</span><span class="p">))</span>
            <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span><span class="o">.</span><span class="n">append</span><span class="p">({</span><span class="s1">&#39;sample_name&#39;</span><span class="p">:</span> <span class="n">sample_name</span><span class="p">,</span> <span class="s1">&#39;homozygosity_by_loci&#39;</span><span class="p">:</span> <span class="n">score</span><span class="p">,</span>
                               <span class="s1">&#39;internal_relatedness&#39;</span><span class="p">:</span> <span class="n">ir</span><span class="p">,</span> <span class="s1">&#39;num_hom_loci&#39;</span><span class="p">:</span> <span class="n">num_hom_loci</span><span class="p">,</span>
                               <span class="s1">&#39;total_loci&#39;</span><span class="p">:</span> <span class="n">num_loci</span><span class="p">})</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span><span class="p">[[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span> <span class="s1">&#39;homozygosity_by_loci&#39;</span><span class="p">,</span>
                                 <span class="s1">&#39;internal_relatedness&#39;</span><span class="p">,</span> <span class="s1">&#39;num_hom_loci&#39;</span><span class="p">,</span> <span class="s1">&#39;total_loci&#39;</span><span class="p">]]</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;internal_relatedness&#39;</span><span class="p">,</span> <span class="n">ascending</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
        <span class="bp">self</span><span class="o">.</span><span class="n">ir_df</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">drop</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>


<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="k">print</span><span class="p">(</span><span class="s1">&#39;&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<p>First, I loaded in the microsatellite data that was provided to me by
Peter Baumann. This file contains the genotyping data for the more up to
date panel of microsatellites. I restructure the data to work better
with the python class I wrote.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [10]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">populationData</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s1">&#39;../data/marmorata_mirco_test_data.xlsx&#39;</span><span class="p">)</span>
<span class="n">populationData</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[10]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Height 1</th>
      <th>Height 2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>9387-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>9387-Ai5013-LizardMS</td>
      <td>227.66</td>
      <td>253.09</td>
      <td>7229.0</td>
      <td>6420.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9387-Ai5043-LizardMS</td>
      <td>175.87</td>
      <td>NaN</td>
      <td>824.0</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>9387-Cvanu24-LizardMS</td>
      <td>200.09</td>
      <td>NaN</td>
      <td>6737.0</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>4</th>
      <td>9387-Cvanu7-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [11]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span> <span class="o">=</span> <span class="n">bliz</span><span class="o">.</span><span class="n">MicrosatellitePopulation</span><span class="p">(</span><span class="s1">&#39;../data/marmorata_mirco_test_data.xlsx&#39;</span><span class="p">,</span> <span class="n">split_id</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [12]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">len</span><span class="p">(</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[12]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>12
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [13]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">print</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
A105
Ai5013
Ai5043
Cvanu24
Cvanu7
D106
D107
D111
MS1
MS6
MS7
MS8
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [14]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">print</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
9387
9525
95261
9526
9721
9722
115841
12512
125131
137461
137471
16996
16997
17033
17034
17049
17050
17051
17330
176291
17630
176311
19687
116001
11784
14796
16998
17114
184
185
187
195
234
261
267
268
8033
8873
171151
171161
171171
17118
122_tig
8450_tig
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [15]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[15]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>9387-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>1</th>
      <td>9387-Ai5013-LizardMS</td>
      <td>227.66</td>
      <td>253.09</td>
      <td>7229.0</td>
      <td>6420.0</td>
      <td>9387</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9387-Ai5043-LizardMS</td>
      <td>175.87</td>
      <td>175.87</td>
      <td>824.0</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>3</th>
      <td>9387-Cvanu24-LizardMS</td>
      <td>200.09</td>
      <td>200.09</td>
      <td>6737.0</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>4</th>
      <td>9387-Cvanu7-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Cvanu7</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [16]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;sample_name&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">()</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;size_1&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[16]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>micro_sat</th>
    </tr>
    <tr>
      <th>sample_name</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>9526</th>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>9387</th>
      <td>12</td>
      <td>7</td>
      <td>7</td>
      <td>7</td>
      <td>4</td>
      <td>12</td>
    </tr>
    <tr>
      <th>95261</th>
      <td>11</td>
      <td>8</td>
      <td>8</td>
      <td>8</td>
      <td>2</td>
      <td>11</td>
    </tr>
    <tr>
      <th>9525</th>
      <td>12</td>
      <td>10</td>
      <td>10</td>
      <td>10</td>
      <td>5</td>
      <td>12</td>
    </tr>
    <tr>
      <th>122_tig</th>
      <td>12</td>
      <td>11</td>
      <td>11</td>
      <td>11</td>
      <td>5</td>
      <td>12</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [17]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="s2">&quot;9387&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[17]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>9387-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>1</th>
      <td>9387-Ai5013-LizardMS</td>
      <td>227.66</td>
      <td>253.09</td>
      <td>7229.0</td>
      <td>6420.0</td>
      <td>9387</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>2</th>
      <td>9387-Ai5043-LizardMS</td>
      <td>175.87</td>
      <td>175.87</td>
      <td>824.0</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>3</th>
      <td>9387-Cvanu24-LizardMS</td>
      <td>200.09</td>
      <td>200.09</td>
      <td>6737.0</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>4</th>
      <td>9387-Cvanu7-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>5</th>
      <td>9387-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>6</th>
      <td>9387-D107-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>7</th>
      <td>9387-D111-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>9387</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>8</th>
      <td>9387-MS1-LizardMS</td>
      <td>216.81</td>
      <td>216.81</td>
      <td>4940.0</td>
      <td>NaN</td>
      <td>9387</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>9</th>
      <td>9387-MS6-LizardMS</td>
      <td>157.99</td>
      <td>174.19</td>
      <td>2969.0</td>
      <td>3342.0</td>
      <td>9387</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>10</th>
      <td>9387-MS7-LizardMS</td>
      <td>235.80</td>
      <td>259.11</td>
      <td>7798.0</td>
      <td>5524.0</td>
      <td>9387</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>11</th>
      <td>9387-MS8-LizardMS</td>
      <td>109.11</td>
      <td>113.24</td>
      <td>5579.0</td>
      <td>6037.0</td>
      <td>9387</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [18]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="s2">&quot;9526&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[18]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>28</th>
      <td>9526-Cvanu7-LizardMS</td>
      <td>332.55</td>
      <td>344.0</td>
      <td>1058.0</td>
      <td>1000.0</td>
      <td>9526</td>
      <td>Cvanu7</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [19]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="s2">&quot;95261&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[19]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>24</th>
      <td>95261-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>95261</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>25</th>
      <td>95261-Ai5013-LizardMS</td>
      <td>220.42</td>
      <td>253.09</td>
      <td>4216.0</td>
      <td>3469.0</td>
      <td>95261</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>26</th>
      <td>95261-Ai5043-LizardMS</td>
      <td>175.84</td>
      <td>175.84</td>
      <td>10957.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>27</th>
      <td>95261-Cvanu24-LizardMS</td>
      <td>200.00</td>
      <td>200.00</td>
      <td>5789.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>29</th>
      <td>95261-D106-LizardMS</td>
      <td>291.12</td>
      <td>299.20</td>
      <td>2947.0</td>
      <td>3348.0</td>
      <td>95261</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>30</th>
      <td>95261-D107-LizardMS</td>
      <td>127.82</td>
      <td>127.82</td>
      <td>2117.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>31</th>
      <td>95261-D111-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>95261</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>32</th>
      <td>95261-MS1-LizardMS</td>
      <td>216.78</td>
      <td>216.78</td>
      <td>11264.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>33</th>
      <td>95261-MS6-LizardMS</td>
      <td>174.15</td>
      <td>174.15</td>
      <td>4377.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>34</th>
      <td>95261-MS7-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>95261</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>35</th>
      <td>95261-MS8-LizardMS</td>
      <td>109.17</td>
      <td>109.17</td>
      <td>3661.0</td>
      <td>NaN</td>
      <td>95261</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [20]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">remove_sample</span><span class="p">(</span><span class="s1">&#39;9387&#39;</span><span class="p">)</span>
<span class="n">marmPopulation</span><span class="o">.</span><span class="n">remove_sample</span><span class="p">(</span><span class="s1">&#39;9526&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [21]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">print</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
9525 4
95261 5
9721 4
9722 4
115841 6
12512 5
125131 6
137461 6
137471 6
16996 5
16997 5
17033 5
17034 5
17049 5
17050 5
17051 5
17330 5
176291 6
17630 5
176311 6
19687 5
116001 6
11784 5
14796 5
16998 5
17114 5
184 3
185 3
187 3
195 3
234 3
261 3
267 3
268 3
8033 4
8873 4
171151 6
171161 6
171171 6
17118 5
122_tig 7
8450_tig 8
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [22]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">print</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
A105
Ai5013
Ai5043
Cvanu24
Cvanu7
D106
D107
D111
MS1
MS6
MS7
MS8
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [23]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">==</span><span class="s1">&#39;8450_tig&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[23]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>519</th>
      <td>8450_tig-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>520</th>
      <td>8450_tig-Ai5013-LizardMS</td>
      <td>257.14</td>
      <td>257.14</td>
      <td>32277.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>521</th>
      <td>8450_tig-Ai5043-LizardMS</td>
      <td>178.81</td>
      <td>178.81</td>
      <td>7372.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>522</th>
      <td>8450_tig-Cvanu24-LizardMS</td>
      <td>199.92</td>
      <td>199.92</td>
      <td>32614.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>523</th>
      <td>8450_tig-Cvanu7-LizardMS</td>
      <td>328.32</td>
      <td>328.32</td>
      <td>13329.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>524</th>
      <td>8450_tig-D106-LizardMS</td>
      <td>299.26</td>
      <td>299.26</td>
      <td>18096.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>525</th>
      <td>8450_tig-D107-LizardMS</td>
      <td>178.88</td>
      <td>178.88</td>
      <td>114.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>526</th>
      <td>8450_tig-D111-LizardMS</td>
      <td>150.00</td>
      <td>150.00</td>
      <td>10241.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>527</th>
      <td>8450_tig-MS1-LizardMS</td>
      <td>216.99</td>
      <td>216.99</td>
      <td>28176.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>528</th>
      <td>8450_tig-MS6-LizardMS</td>
      <td>174.18</td>
      <td>174.18</td>
      <td>32639.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>529</th>
      <td>8450_tig-MS7-LizardMS</td>
      <td>243.79</td>
      <td>243.79</td>
      <td>14502.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>530</th>
      <td>8450_tig-MS8-LizardMS</td>
      <td>114.31</td>
      <td>114.31</td>
      <td>25818.0</td>
      <td>NaN</td>
      <td>8450_tig</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [24]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">==</span><span class="s1">&#39;122_tig&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[24]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>507</th>
      <td>122_tig-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>508</th>
      <td>122_tig-Ai5013-LizardMS</td>
      <td>213.73</td>
      <td>257.15</td>
      <td>31746.0</td>
      <td>24594.0</td>
      <td>122_tig</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>509</th>
      <td>122_tig-Ai5043-LizardMS</td>
      <td>181.65</td>
      <td>181.65</td>
      <td>11064.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>510</th>
      <td>122_tig-Cvanu24-LizardMS</td>
      <td>200.00</td>
      <td>200.00</td>
      <td>32659.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>511</th>
      <td>122_tig-Cvanu7-LizardMS</td>
      <td>328.41</td>
      <td>351.65</td>
      <td>9057.0</td>
      <td>4693.0</td>
      <td>122_tig</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>512</th>
      <td>122_tig-D106-LizardMS</td>
      <td>295.40</td>
      <td>299.43</td>
      <td>12924.0</td>
      <td>11387.0</td>
      <td>122_tig</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>513</th>
      <td>122_tig-D107-LizardMS</td>
      <td>127.91</td>
      <td>127.91</td>
      <td>665.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>514</th>
      <td>122_tig-D111-LizardMS</td>
      <td>150.00</td>
      <td>150.00</td>
      <td>10981.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>515</th>
      <td>122_tig-MS1-LizardMS</td>
      <td>217.04</td>
      <td>217.04</td>
      <td>30346.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>516</th>
      <td>122_tig-MS6-LizardMS</td>
      <td>174.27</td>
      <td>174.27</td>
      <td>32567.0</td>
      <td>NaN</td>
      <td>122_tig</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>517</th>
      <td>122_tig-MS7-LizardMS</td>
      <td>243.81</td>
      <td>282.56</td>
      <td>16395.0</td>
      <td>14654.0</td>
      <td>122_tig</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>518</th>
      <td>122_tig-MS8-LizardMS</td>
      <td>113.39</td>
      <td>114.43</td>
      <td>30780.0</td>
      <td>25017.0</td>
      <td>122_tig</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [25]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">==</span><span class="s1">&#39;19687&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[25]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>255</th>
      <td>19687-A105-LizardMS</td>
      <td>255.79</td>
      <td>255.79</td>
      <td>37.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>256</th>
      <td>19687-Ai5013-LizardMS</td>
      <td>220.28</td>
      <td>220.28</td>
      <td>6972.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>257</th>
      <td>19687-Ai5043-LizardMS</td>
      <td>175.83</td>
      <td>175.83</td>
      <td>1236.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>258</th>
      <td>19687-Cvanu24-LizardMS</td>
      <td>199.91</td>
      <td>199.91</td>
      <td>7414.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>259</th>
      <td>19687-Cvanu7-LizardMS</td>
      <td>331.56</td>
      <td>331.56</td>
      <td>1350.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>260</th>
      <td>19687-D106-LizardMS</td>
      <td>290.91</td>
      <td>290.91</td>
      <td>2567.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>261</th>
      <td>19687-D107-LizardMS</td>
      <td>170.47</td>
      <td>170.47</td>
      <td>171.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>262</th>
      <td>19687-D111-LizardMS</td>
      <td>150.10</td>
      <td>150.10</td>
      <td>1522.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>263</th>
      <td>19687-MS1-LizardMS</td>
      <td>216.86</td>
      <td>216.86</td>
      <td>8039.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>264</th>
      <td>19687-MS6-LizardMS</td>
      <td>174.16</td>
      <td>174.16</td>
      <td>6526.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>265</th>
      <td>19687-MS7-LizardMS</td>
      <td>290.12</td>
      <td>290.12</td>
      <td>7152.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>266</th>
      <td>19687-MS8-LizardMS</td>
      <td>109.03</td>
      <td>109.03</td>
      <td>6681.0</td>
      <td>NaN</td>
      <td>19687</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [26]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">==</span><span class="s1">&#39;17330&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[26]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>sample_name</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>204</th>
      <td>17330-A105-LizardMS</td>
      <td>255.89</td>
      <td>255.89</td>
      <td>496.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>205</th>
      <td>17330-Ai5013-LizardMS</td>
      <td>220.28</td>
      <td>220.28</td>
      <td>31495.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>206</th>
      <td>17330-Ai5043-LizardMS</td>
      <td>175.95</td>
      <td>175.95</td>
      <td>16340.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>207</th>
      <td>17330-Cvanu24-LizardMS</td>
      <td>199.91</td>
      <td>199.91</td>
      <td>32360.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>208</th>
      <td>17330-Cvanu7-LizardMS</td>
      <td>331.64</td>
      <td>331.64</td>
      <td>9399.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>209</th>
      <td>17330-D106-LizardMS</td>
      <td>294.86</td>
      <td>294.86</td>
      <td>17247.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>210</th>
      <td>17330-D107-LizardMS</td>
      <td>178.59</td>
      <td>268.36</td>
      <td>612.0</td>
      <td>271.0</td>
      <td>17330</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>211</th>
      <td>17330-D111-LizardMS</td>
      <td>150.19</td>
      <td>150.19</td>
      <td>5223.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>212</th>
      <td>17330-MS1-LizardMS</td>
      <td>216.86</td>
      <td>216.86</td>
      <td>28794.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>213</th>
      <td>17330-MS6-LizardMS</td>
      <td>174.12</td>
      <td>174.12</td>
      <td>32581.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>214</th>
      <td>17330-MS7-LizardMS</td>
      <td>259.08</td>
      <td>259.08</td>
      <td>19101.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>215</th>
      <td>17330-MS8-LizardMS</td>
      <td>109.14</td>
      <td>109.14</td>
      <td>32241.0</td>
      <td>NaN</td>
      <td>17330</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [27]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">listsOfAnimalsAndMSUsed</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({</span><span class="s1">&#39;list of microsatellite markers used&#39;</span><span class="p">:</span><span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">(</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">micro_sat</span><span class="o">.</span><span class="n">unique</span><span class="p">()),</span><span class="s1">&#39;list of animals used&#39;</span><span class="p">:</span><span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">(</span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="o">.</span><span class="n">sample_name</span><span class="o">.</span><span class="n">unique</span><span class="p">())})</span>
<span class="n">listsOfAnimalsAndMSUsed</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/listsOfAnimalsAndMSUsed.xlsx&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [28]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">calc_internal_relatedness</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="stderr output_area docutils container">
<div class="highlight"><pre>
/home/dut/anaconda3/envs/anaconda2/lib/python2.7/site-packages/pandas/core/frame.py:2852: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  downcast=downcast, **kwargs)
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [29]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">ir_df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[29]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_name</th>
      <th>homozygosity_by_loci</th>
      <th>internal_relatedness</th>
      <th>num_hom_loci</th>
      <th>total_loci</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>8450_tig</td>
      <td>1.000000</td>
      <td>1.000000</td>
      <td>11</td>
      <td>11</td>
    </tr>
    <tr>
      <th>1</th>
      <td>19687</td>
      <td>1.000000</td>
      <td>1.000000</td>
      <td>12</td>
      <td>12</td>
    </tr>
    <tr>
      <th>2</th>
      <td>17330</td>
      <td>0.989654</td>
      <td>0.800759</td>
      <td>11</td>
      <td>12</td>
    </tr>
    <tr>
      <th>3</th>
      <td>125131</td>
      <td>0.897412</td>
      <td>0.614075</td>
      <td>10</td>
      <td>12</td>
    </tr>
    <tr>
      <th>4</th>
      <td>12512</td>
      <td>0.898693</td>
      <td>0.506762</td>
      <td>9</td>
      <td>12</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [30]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marm_ir</span> <span class="o">=</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">ir_df</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;whitegrid&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">1.2</span><span class="p">))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">distplot</span><span class="p">(</span><span class="n">marm_ir</span><span class="p">[</span><span class="s1">&#39;internal_relatedness&#39;</span><span class="p">],</span> <span class="n">kde</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">rug</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkred&#39;</span><span class="p">,</span> <span class="n">hist_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;alpha&#39;</span><span class="p">:</span><span class="mf">1.0</span><span class="p">,</span><span class="s1">&#39;edgecolor&#39;</span><span class="p">:</span><span class="s2">&quot;k&quot;</span><span class="p">,</span> <span class="s1">&#39;linewidth&#39;</span><span class="p">:</span><span class="mi">1</span><span class="p">})</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mf">8.5</span><span class="p">))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Animals&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Internal Relatedness&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax.set_title(&#39;Internal Relatedness of 44 Aspidoscelis marmorata&#39;)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">arrow</span><span class="p">(</span><span class="mf">0.978</span><span class="p">,</span><span class="mf">2.8</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">head_width</span><span class="o">=</span><span class="mf">0.035</span><span class="p">,</span> <span class="n">head_length</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;#3498db&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">arrow</span><span class="p">(</span><span class="mf">0.335</span> <span class="p">,</span><span class="mf">2.8</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">head_width</span><span class="o">=</span><span class="mf">0.035</span><span class="p">,</span> <span class="n">head_length</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#9b59b6&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mf">1.1</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.15</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure2B.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_44_0.png" src="_images/marmorata_genome_paper_figures_3_44_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [31]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span> <span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span><span class="s1">&#39;size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">,</span> <span class="s1">&#39;height_1&#39;</span><span class="p">,</span> <span class="s1">&#39;height_2&#39;</span><span class="p">,</span> <span class="s1">&#39;binned_size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;binned_size_2&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[31]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_name</th>
      <th>micro_sat</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>height_1</th>
      <th>height_2</th>
      <th>binned_size_1</th>
      <th>binned_size_2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>12</th>
      <td>9525</td>
      <td>A105</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>13</th>
      <td>9525</td>
      <td>Ai5013</td>
      <td>220.34</td>
      <td>252.96</td>
      <td>2195.0</td>
      <td>1931.0</td>
      <td>220.333</td>
      <td>253.944</td>
    </tr>
    <tr>
      <th>14</th>
      <td>9525</td>
      <td>Ai5043</td>
      <td>175.71</td>
      <td>175.71</td>
      <td>7536.0</td>
      <td>NaN</td>
      <td>179.000</td>
      <td>179.000</td>
    </tr>
    <tr>
      <th>15</th>
      <td>9525</td>
      <td>Cvanu24</td>
      <td>200.00</td>
      <td>200.00</td>
      <td>4915.0</td>
      <td>NaN</td>
      <td>200.200</td>
      <td>200.200</td>
    </tr>
    <tr>
      <th>16</th>
      <td>9525</td>
      <td>Cvanu7</td>
      <td>332.59</td>
      <td>352.53</td>
      <td>651.0</td>
      <td>392.0</td>
      <td>334.182</td>
      <td>355.818</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [32]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalTable1</span> <span class="o">=</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span> <span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span><span class="s1">&#39;size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">,</span> <span class="s1">&#39;height_1&#39;</span><span class="p">,</span> <span class="s1">&#39;height_2&#39;</span><span class="p">,</span> <span class="s1">&#39;binned_size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;binned_size_2&#39;</span><span class="p">]]</span>
<span class="n">supplementalTable1</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_1.csv&#39;</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Here I calculate the size ranges for each of the microsatellites in the
population.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [33]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">stack1</span> <span class="o">=</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span> <span class="s1">&#39;size_1&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">stack2</span> <span class="o">=</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">genotypes_df</span><span class="p">[[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">stack1</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span> <span class="s1">&#39;size&#39;</span><span class="p">]</span>
<span class="n">stack2</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">,</span> <span class="s1">&#39;size&#39;</span><span class="p">]</span>
<span class="n">allSizesDf</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">stack1</span><span class="p">,</span> <span class="n">stack2</span><span class="p">])</span>
<span class="n">allSizesDf</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[33]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>micro_sat</th>
      <th>size</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>12</th>
      <td>A105</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Ai5013</td>
      <td>220.34</td>
    </tr>
    <tr>
      <th>14</th>
      <td>Ai5043</td>
      <td>175.71</td>
    </tr>
    <tr>
      <th>15</th>
      <td>Cvanu24</td>
      <td>200.00</td>
    </tr>
    <tr>
      <th>16</th>
      <td>Cvanu7</td>
      <td>332.59</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>I use the group by method and describe to generate the following table.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [34]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">microSatSizeRangesDf</span> <span class="o">=</span> <span class="n">allSizesDf</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">describe</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">microSatSizeRangesDf</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[34]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr>
      <th></th>
      <th>index</th>
      <th>micro_sat</th>
      <th colspan="8" halign="left">size</th>
    </tr>
    <tr>
      <th></th>
      <th></th>
      <th></th>
      <th>count</th>
      <th>mean</th>
      <th>std</th>
      <th>min</th>
      <th>25%</th>
      <th>50%</th>
      <th>75%</th>
      <th>max</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>A105</td>
      <td>74.0</td>
      <td>246.969189</td>
      <td>18.979147</td>
      <td>197.16</td>
      <td>255.7150</td>
      <td>255.835</td>
      <td>255.8975</td>
      <td>256.03</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>Ai5013</td>
      <td>84.0</td>
      <td>231.230714</td>
      <td>16.885507</td>
      <td>201.63</td>
      <td>220.2225</td>
      <td>227.445</td>
      <td>250.9125</td>
      <td>257.15</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>Ai5043</td>
      <td>84.0</td>
      <td>176.634762</td>
      <td>1.591840</td>
      <td>175.50</td>
      <td>175.7700</td>
      <td>175.840</td>
      <td>176.0700</td>
      <td>181.65</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3</td>
      <td>Cvanu24</td>
      <td>84.0</td>
      <td>199.933571</td>
      <td>0.097877</td>
      <td>199.73</td>
      <td>199.9100</td>
      <td>199.910</td>
      <td>200.0000</td>
      <td>200.17</td>
    </tr>
    <tr>
      <th>4</th>
      <td>4</td>
      <td>Cvanu7</td>
      <td>82.0</td>
      <td>337.467683</td>
      <td>9.083376</td>
      <td>328.20</td>
      <td>331.5650</td>
      <td>331.670</td>
      <td>344.2500</td>
      <td>361.60</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5</td>
      <td>D106</td>
      <td>84.0</td>
      <td>300.888214</td>
      <td>21.955367</td>
      <td>274.69</td>
      <td>294.8100</td>
      <td>298.900</td>
      <td>299.3800</td>
      <td>484.02</td>
    </tr>
    <tr>
      <th>6</th>
      <td>6</td>
      <td>D107</td>
      <td>90.0</td>
      <td>158.812222</td>
      <td>36.975351</td>
      <td>89.65</td>
      <td>127.7225</td>
      <td>166.365</td>
      <td>174.5175</td>
      <td>268.36</td>
    </tr>
    <tr>
      <th>7</th>
      <td>7</td>
      <td>D111</td>
      <td>80.0</td>
      <td>152.250875</td>
      <td>6.991427</td>
      <td>149.90</td>
      <td>150.0000</td>
      <td>150.100</td>
      <td>150.2000</td>
      <td>207.90</td>
    </tr>
    <tr>
      <th>8</th>
      <td>8</td>
      <td>MS1</td>
      <td>84.0</td>
      <td>217.545357</td>
      <td>3.686983</td>
      <td>216.62</td>
      <td>216.7600</td>
      <td>216.850</td>
      <td>216.9300</td>
      <td>239.14</td>
    </tr>
    <tr>
      <th>9</th>
      <td>9</td>
      <td>MS6</td>
      <td>84.0</td>
      <td>171.701786</td>
      <td>5.910574</td>
      <td>157.80</td>
      <td>174.0200</td>
      <td>174.130</td>
      <td>174.2400</td>
      <td>176.26</td>
    </tr>
    <tr>
      <th>10</th>
      <td>10</td>
      <td>MS7</td>
      <td>82.0</td>
      <td>268.911098</td>
      <td>16.973769</td>
      <td>235.67</td>
      <td>259.0825</td>
      <td>266.785</td>
      <td>289.0675</td>
      <td>298.10</td>
    </tr>
    <tr>
      <th>11</th>
      <td>11</td>
      <td>MS8</td>
      <td>84.0</td>
      <td>111.705119</td>
      <td>2.947997</td>
      <td>108.80</td>
      <td>109.1100</td>
      <td>109.285</td>
      <td>113.3950</td>
      <td>119.71</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>I then subset to display only min and max values. I then manually copied
this into the excel document that Rob sent me.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [35]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">df</span> <span class="o">=</span> <span class="n">microSatSizeRangesDf</span><span class="p">[</span><span class="s1">&#39;size&#39;</span><span class="p">][[</span> <span class="s1">&#39;min&#39;</span><span class="p">,</span> <span class="s1">&#39;max&#39;</span><span class="p">]]</span>
<span class="n">df</span><span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">microSatSizeRangesDf</span><span class="p">[</span><span class="s1">&#39;micro_sat&#39;</span><span class="p">]</span>
<span class="n">df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[35]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>min</th>
      <th>max</th>
      <th>micro_sat</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>197.16</td>
      <td>256.03</td>
      <td>A105</td>
    </tr>
    <tr>
      <th>1</th>
      <td>201.63</td>
      <td>257.15</td>
      <td>Ai5013</td>
    </tr>
    <tr>
      <th>2</th>
      <td>175.50</td>
      <td>181.65</td>
      <td>Ai5043</td>
    </tr>
    <tr>
      <th>3</th>
      <td>199.73</td>
      <td>200.17</td>
      <td>Cvanu24</td>
    </tr>
    <tr>
      <th>4</th>
      <td>328.20</td>
      <td>361.60</td>
      <td>Cvanu7</td>
    </tr>
    <tr>
      <th>5</th>
      <td>274.69</td>
      <td>484.02</td>
      <td>D106</td>
    </tr>
    <tr>
      <th>6</th>
      <td>89.65</td>
      <td>268.36</td>
      <td>D107</td>
    </tr>
    <tr>
      <th>7</th>
      <td>149.90</td>
      <td>207.90</td>
      <td>D111</td>
    </tr>
    <tr>
      <th>8</th>
      <td>216.62</td>
      <td>239.14</td>
      <td>MS1</td>
    </tr>
    <tr>
      <th>9</th>
      <td>157.80</td>
      <td>176.26</td>
      <td>MS6</td>
    </tr>
    <tr>
      <th>10</th>
      <td>235.67</td>
      <td>298.10</td>
      <td>MS7</td>
    </tr>
    <tr>
      <th>11</th>
      <td>108.80</td>
      <td>119.71</td>
      <td>MS8</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>He I read in the final table, and rewrite it to a file named
’supplemental table 2.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [36]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalTable2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_excel</span><span class="p">(</span><span class="s1">&#39;../data/Microsatellite_primer_pairs_used_for_genotyping.xlsx&#39;</span><span class="p">)</span>
<span class="n">supplementalTable2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[36]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Microsatellite</th>
      <th>Oligo 1 sequence (5’ → 3’)</th>
      <th>Oligo 2 sequence (5’ → 3’)</th>
      <th>Min Size</th>
      <th>Max Size</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>A105</td>
      <td>AATCCTGAACCTACGGTAAGC</td>
      <td>TGCCAGAAAATAGAGGGAAG</td>
      <td>197.16</td>
      <td>256.03</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Ai5013</td>
      <td>AATTAATGTGCAGCACTAT</td>
      <td>GGCAGTTTTTCAGCTAAG</td>
      <td>201.63</td>
      <td>257.15</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Ai5043</td>
      <td>AAAAAGAAAAGGAAGAACTAA</td>
      <td>TGAGACAAGTTGGGTAGA</td>
      <td>175.50</td>
      <td>181.65</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Cvanu24</td>
      <td>TTTAATGCATCCACTGAGTC</td>
      <td>GGAATATAGTGGCATATCAG</td>
      <td>199.73</td>
      <td>200.17</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Cvanu7</td>
      <td>GACCAATAATGTGGAAGCTG</td>
      <td>ACATGGCTGAGTAATTGGTG</td>
      <td>328.20</td>
      <td>361.60</td>
    </tr>
    <tr>
      <th>5</th>
      <td>D106</td>
      <td>TTAAAGCAGAGGTCAGGTTATC</td>
      <td>GATGGAAGAATAGGATGATGAA</td>
      <td>274.69</td>
      <td>484.02</td>
    </tr>
    <tr>
      <th>6</th>
      <td>D107</td>
      <td>TACCCACCTGGAGATGTTTAG</td>
      <td>AGGACGCCTTAAAATAGGAAG</td>
      <td>89.65</td>
      <td>268.36</td>
    </tr>
    <tr>
      <th>7</th>
      <td>D111</td>
      <td>TGGAGGCAGTCTTGGTATC</td>
      <td>GAACATTGACCGCATCAC</td>
      <td>149.90</td>
      <td>207.90</td>
    </tr>
    <tr>
      <th>8</th>
      <td>MS1</td>
      <td>TGCATGATGGAGGAATCTTC</td>
      <td>CTAGTGGTGATAGAAACATGG</td>
      <td>216.62</td>
      <td>239.14</td>
    </tr>
    <tr>
      <th>9</th>
      <td>MS6</td>
      <td>CACACCCATATTATAAGTGG</td>
      <td>CATTCAGATGAAACCTAACC</td>
      <td>157.80</td>
      <td>176.26</td>
    </tr>
    <tr>
      <th>10</th>
      <td>MS7</td>
      <td>AACTAAGTGCTAAGTGTGAC</td>
      <td>ACAGTCTTAGAGATCACAAG</td>
      <td>235.67</td>
      <td>298.10</td>
    </tr>
    <tr>
      <th>11</th>
      <td>MS8</td>
      <td>ACACCCAAAGTCCTCAACAG</td>
      <td>CTAGTACATGTGTAAGGGTG</td>
      <td>108.80</td>
      <td>119.71</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [37]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalTable2</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_2.xls&#39;</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Genomic-Sequencing-Data">
<h2>Genomic Sequencing Data<a class="headerlink" href="#Genomic-Sequencing-Data" title="Permalink to this headline">¶</a></h2>
<p>Here I read in the table that describes all of the sequencing data. This
is referenced when we first describe the genome</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [38]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalTable3</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../../ncbi_genome_delivery/data/supplemental_sequencing_data.csv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">)</span>
<span class="n">supplementalTable3</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[38]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>library id</th>
      <th>sample name</th>
      <th>tissue</th>
      <th>sex</th>
      <th>Instrument Model</th>
      <th>design description</th>
      <th>Purpose</th>
      <th>order type</th>
      <th>selection</th>
      <th>read length</th>
      <th>total reads or read pairs</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>L11871</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>5kb Mate Pair</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>Mate Pair (1ug or 4ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>68124235.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>L11871-1</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>8kb Mate Pair</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>Mate Pair (1ug or 4ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>62138640.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>L11871-2</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>2-15kb Mate Pair</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>Mate Pair (1ug or 4ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>67949265.0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>L13136-2</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>250</td>
      <td>376374707.0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>L15771</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>CHICAGO</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>CHICAGO</td>
      <td>RANDOM</td>
      <td>100</td>
      <td>207139607.0</td>
    </tr>
    <tr>
      <th>5</th>
      <td>L15772</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>CHICAGO</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>CHICAGO</td>
      <td>RANDOM</td>
      <td>100</td>
      <td>52923608.0</td>
    </tr>
    <tr>
      <th>6</th>
      <td>S11870</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina MiSeq</td>
      <td>40kb Mate Pair</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>Mate Pair</td>
      <td>RANDOM</td>
      <td>184</td>
      <td>7325966.0</td>
    </tr>
    <tr>
      <th>7</th>
      <td>S11870</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina MiSeq</td>
      <td>40kb Mate Pair</td>
      <td>DNA Sequencing of Aspidoscelis marmoratus: A_tigris8450</td>
      <td>Mate Pair</td>
      <td>RANDOM</td>
      <td>250</td>
      <td>7578267.0</td>
    </tr>
    <tr>
      <th>8</th>
      <td>L13087</td>
      <td>Atig001</td>
      <td>liver</td>
      <td>not collected</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>96536963.0</td>
    </tr>
    <tr>
      <th>9</th>
      <td>L13088</td>
      <td>Atig003</td>
      <td>liver</td>
      <td>not collect</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>71079632.0</td>
    </tr>
    <tr>
      <th>10</th>
      <td>L13088-1</td>
      <td>Atig003</td>
      <td>liver</td>
      <td>not collect</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>37328268.0</td>
    </tr>
    <tr>
      <th>11</th>
      <td>L13136</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>57367890.0</td>
    </tr>
    <tr>
      <th>12</th>
      <td>L13136-1</td>
      <td>A_tigris8450</td>
      <td>liver</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>46483161.0</td>
    </tr>
    <tr>
      <th>13</th>
      <td>L21676</td>
      <td>Atig_122</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (100ng)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>321903175.0</td>
    </tr>
    <tr>
      <th>14</th>
      <td>L30698</td>
      <td>Atig_9177</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>113372074.0</td>
    </tr>
    <tr>
      <th>15</th>
      <td>L30699</td>
      <td>Atig_6993</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>100116690.0</td>
    </tr>
    <tr>
      <th>16</th>
      <td>L30700</td>
      <td>Atig_4278</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>106787117.0</td>
    </tr>
    <tr>
      <th>17</th>
      <td>L30701</td>
      <td>A.tig_12512</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>102760203.0</td>
    </tr>
    <tr>
      <th>18</th>
      <td>L30702</td>
      <td>A.tig_12513</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>103885909.0</td>
    </tr>
    <tr>
      <th>19</th>
      <td>L30703</td>
      <td>A.tig_9721</td>
      <td>tail</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>DNAseq for Aspidoscelis marmoratus Variant Calling</td>
      <td>DNA-Seq (1ug)</td>
      <td>RANDOM</td>
      <td>150</td>
      <td>101265101.0</td>
    </tr>
    <tr>
      <th>20</th>
      <td>L11223</td>
      <td>9601</td>
      <td>blood</td>
      <td>Female</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>RNAseq for Aspidoscelis marmoratus Genome Annotation</td>
      <td>poly-A Stranded RNA-Seq</td>
      <td>PolyA Selection</td>
      <td>100</td>
      <td>232139414.0</td>
    </tr>
    <tr>
      <th>21</th>
      <td>L11224</td>
      <td>11225</td>
      <td>blood</td>
      <td>Male</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>RNAseq for Aspidoscelis marmoratus Genome Annotation</td>
      <td>poly-A Stranded RNA-Seq</td>
      <td>PolyA Selection</td>
      <td>100</td>
      <td>260779309.0</td>
    </tr>
    <tr>
      <th>22</th>
      <td>L11225</td>
      <td>11677</td>
      <td>blood</td>
      <td>Male</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>RNAseq for Aspidoscelis marmoratus Genome Annotation</td>
      <td>poly-A Stranded RNA-Seq</td>
      <td>PolyA Selection</td>
      <td>100</td>
      <td>189619876.0</td>
    </tr>
    <tr>
      <th>23</th>
      <td>L16900</td>
      <td>S237-2</td>
      <td>embryo</td>
      <td>not collected</td>
      <td>Illumina HiSeq 2500</td>
      <td>Paired End</td>
      <td>RNAseq for Aspidoscelis marmoratus Genome Annotation</td>
      <td>poly-A Stranded RNA-Seq</td>
      <td>PolyA Selection</td>
      <td>100</td>
      <td>253832665.0</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [39]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalTable3</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_3.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Genome-Size-Prediction-with-FACs">
<h2>Genome Size Prediction with FACs<a class="headerlink" href="#Genome-Size-Prediction-with-FACs" title="Permalink to this headline">¶</a></h2>
<p>Next I need to copy over the excel document that contains the analysis
for genome size prediction. I had to manually copy the file over
directly from Google drive to the N drive. Here I rename it as
supplemental table 5.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [40]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../data/09cy569_acb-tableMarch2017.xlsx ../data/supplemental_data/supplemental_table_4.xlsx
</pre></div>
</div>
</div>
</div>
<div class="section" id="All-Scaffolds-Genome-Stats">
<h2>All Scaffolds Genome Stats<a class="headerlink" href="#All-Scaffolds-Genome-Stats" title="Permalink to this headline">¶</a></h2>
<p>The following python module is used to manipulate and gather data from a
fasta formated sequence file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [41]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/fasta_classes.py</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [42]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">dovetailGenomeFile2</span> <span class="o">=</span> <span class="s1">&#39;../../dovetail_genome_delivery/data/fastas/lizard_23Jun2015_piz6a.upper.fasta&#39;</span>
<span class="n">dtGenome2</span> <span class="o">=</span> <span class="n">fa</span><span class="o">.</span><span class="n">Fasta_file</span><span class="p">(</span><span class="n">dovetailGenomeFile2</span><span class="p">)</span>
<span class="n">dtGenome2</span><span class="o">.</span><span class="n">get_bc_stats</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
perc_genome_covered_by_scaffolds_greater_than_1mb 98.4720241117
total bases 1639530780
perc_genome_covered_by_scaffolds_greater_than_100kb 99.4578671466
scaffolds_greater_than_100Mb 0
N90 7979424
scaffolds_greater_than_1Mb 90
median_scaffold 1450
perc_genome_covered_by_scaffolds_greater_than_10kb 99.6264638594
N50 32220929
max_scaffold 85027298
scaffolds_greater_than_10kb 223
mean_scaffold 428523.465761
min_scaffold 927
scaffolds_greater_than_10Mb 45
scaffolds_greater_than_100kb 133
number of scaffolds 3826
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [43]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF2</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">dtGenome2</span><span class="o">.</span><span class="n">stats</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">])</span>
<span class="n">statDF2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[43]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>N50</th>
      <th>N90</th>
      <th>max_scaffold</th>
      <th>mean_scaffold</th>
      <th>median_scaffold</th>
      <th>min_scaffold</th>
      <th>number of scaffolds</th>
      <th>perc_A</th>
      <th>perc_C</th>
      <th>perc_G</th>
      <th>...</th>
      <th>perc_T</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_100kb</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_10kb</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_1mb</th>
      <th>scaffolds_greater_than_100Mb</th>
      <th>scaffolds_greater_than_100kb</th>
      <th>scaffolds_greater_than_10Mb</th>
      <th>scaffolds_greater_than_10kb</th>
      <th>scaffolds_greater_than_1Mb</th>
      <th>total bases</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>statistic</th>
      <td>32220929</td>
      <td>7979424</td>
      <td>85027298</td>
      <td>428523.465761</td>
      <td>1450</td>
      <td>927</td>
      <td>3826</td>
      <td>27.039101</td>
      <td>20.122447</td>
      <td>20.12903</td>
      <td>...</td>
      <td>27.040218</td>
      <td>99.457867</td>
      <td>99.626464</td>
      <td>98.472024</td>
      <td>0</td>
      <td>133</td>
      <td>45</td>
      <td>223</td>
      <td>90</td>
      <td>1639530780</td>
    </tr>
  </tbody>
</table>
<p>1 rows × 21 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [44]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF2</span><span class="p">[</span><span class="s1">&#39;non_n_GC%&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF2</span><span class="p">[[</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">/</span> <span class="n">statDF2</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">*</span> <span class="mf">100.0</span>
<span class="n">statDF2</span><span class="p">[</span><span class="s1">&#39;non_n_AT%&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF2</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">/</span> <span class="n">statDF2</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">*</span> <span class="mf">100.0</span>

<span class="n">statDF2</span> <span class="o">=</span> <span class="n">statDF2</span><span class="o">.</span><span class="n">T</span>
<span class="n">statDF2</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF2</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;</span><span class="si">%.5f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">x</span><span class="p">)</span>
<span class="n">statDF2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[44]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>statistic</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N50</th>
      <td>32220929.00000</td>
    </tr>
    <tr>
      <th>N90</th>
      <td>7979424.00000</td>
    </tr>
    <tr>
      <th>max_scaffold</th>
      <td>85027298.00000</td>
    </tr>
    <tr>
      <th>mean_scaffold</th>
      <td>428523.46576</td>
    </tr>
    <tr>
      <th>median_scaffold</th>
      <td>1450.00000</td>
    </tr>
    <tr>
      <th>min_scaffold</th>
      <td>927.00000</td>
    </tr>
    <tr>
      <th>number of scaffolds</th>
      <td>3826.00000</td>
    </tr>
    <tr>
      <th>perc_A</th>
      <td>27.03910</td>
    </tr>
    <tr>
      <th>perc_C</th>
      <td>20.12245</td>
    </tr>
    <tr>
      <th>perc_G</th>
      <td>20.12903</td>
    </tr>
    <tr>
      <th>perc_N</th>
      <td>5.66920</td>
    </tr>
    <tr>
      <th>perc_T</th>
      <td>27.04022</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_100kb</th>
      <td>99.45787</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_10kb</th>
      <td>99.62646</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_1mb</th>
      <td>98.47202</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_100Mb</th>
      <td>0.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_100kb</th>
      <td>133.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_10Mb</th>
      <td>45.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_10kb</th>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_1Mb</th>
      <td>90.00000</td>
    </tr>
    <tr>
      <th>total bases</th>
      <td>1639530780.00000</td>
    </tr>
    <tr>
      <th>non_n_GC%</th>
      <td>42.67056</td>
    </tr>
    <tr>
      <th>non_n_AT%</th>
      <td>57.32944</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [45]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF2</span> <span class="o">=</span> <span class="n">statDF2</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">statDF2</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">,</span> <span class="s1">&#39;value&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [46]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[46]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>statistic</th>
      <th>value</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>N50</td>
      <td>32220929.00000</td>
    </tr>
    <tr>
      <th>1</th>
      <td>N90</td>
      <td>7979424.00000</td>
    </tr>
    <tr>
      <th>2</th>
      <td>max_scaffold</td>
      <td>85027298.00000</td>
    </tr>
    <tr>
      <th>3</th>
      <td>mean_scaffold</td>
      <td>428523.46576</td>
    </tr>
    <tr>
      <th>4</th>
      <td>median_scaffold</td>
      <td>1450.00000</td>
    </tr>
    <tr>
      <th>5</th>
      <td>min_scaffold</td>
      <td>927.00000</td>
    </tr>
    <tr>
      <th>6</th>
      <td>number of scaffolds</td>
      <td>3826.00000</td>
    </tr>
    <tr>
      <th>7</th>
      <td>perc_A</td>
      <td>27.03910</td>
    </tr>
    <tr>
      <th>8</th>
      <td>perc_C</td>
      <td>20.12245</td>
    </tr>
    <tr>
      <th>9</th>
      <td>perc_G</td>
      <td>20.12903</td>
    </tr>
    <tr>
      <th>10</th>
      <td>perc_N</td>
      <td>5.66920</td>
    </tr>
    <tr>
      <th>11</th>
      <td>perc_T</td>
      <td>27.04022</td>
    </tr>
    <tr>
      <th>12</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_100kb</td>
      <td>99.45787</td>
    </tr>
    <tr>
      <th>13</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_10kb</td>
      <td>99.62646</td>
    </tr>
    <tr>
      <th>14</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_1mb</td>
      <td>98.47202</td>
    </tr>
    <tr>
      <th>15</th>
      <td>scaffolds_greater_than_100Mb</td>
      <td>0.00000</td>
    </tr>
    <tr>
      <th>16</th>
      <td>scaffolds_greater_than_100kb</td>
      <td>133.00000</td>
    </tr>
    <tr>
      <th>17</th>
      <td>scaffolds_greater_than_10Mb</td>
      <td>45.00000</td>
    </tr>
    <tr>
      <th>18</th>
      <td>scaffolds_greater_than_10kb</td>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>19</th>
      <td>scaffolds_greater_than_1Mb</td>
      <td>90.00000</td>
    </tr>
    <tr>
      <th>20</th>
      <td>total bases</td>
      <td>1639530780.00000</td>
    </tr>
    <tr>
      <th>21</th>
      <td>non_n_GC%</td>
      <td>42.67056</td>
    </tr>
    <tr>
      <th>22</th>
      <td>non_n_AT%</td>
      <td>57.32944</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [47]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF2</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_5.csv&#39;</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Large-Scaffold-Genome-Stats">
<h2>Large Scaffold Genome Stats<a class="headerlink" href="#Large-Scaffold-Genome-Stats" title="Permalink to this headline">¶</a></h2>
<p>Here I load in the fasta file containing all scaffolds in the genome
greater than 10kb using the fasta_classes.py module shown above. I use
this module to gather basic statistics about the genome</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [48]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">dovetailGenomeFile</span> <span class="o">=</span> <span class="s1">&#39;../data/gatk6/reference/tigris_scaffolds_filt_10000.fa&#39;</span>
<span class="n">dtGenome</span> <span class="o">=</span> <span class="n">fa</span><span class="o">.</span><span class="n">Fasta_file</span><span class="p">(</span><span class="n">dovetailGenomeFile</span><span class="p">)</span>
<span class="n">dtGenome</span><span class="o">.</span><span class="n">get_bc_stats</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
perc_genome_covered_by_scaffolds_greater_than_1mb 98.8412318344
total bases 1633406540
perc_genome_covered_by_scaffolds_greater_than_100kb 99.8307711563
scaffolds_greater_than_100Mb 0
N90 8340160
scaffolds_greater_than_1Mb 90
median_scaffold 296338
perc_genome_covered_by_scaffolds_greater_than_10kb 100.0
N50 32220929
max_scaffold 85027298
scaffolds_greater_than_10kb 223
mean_scaffold 7324693.00448
min_scaffold 10129
scaffolds_greater_than_10Mb 45
scaffolds_greater_than_100kb 133
number of scaffolds 223
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [49]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">dtGenome</span><span class="o">.</span><span class="n">stats</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">])</span>
<span class="n">statDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[49]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>N50</th>
      <th>N90</th>
      <th>max_scaffold</th>
      <th>mean_scaffold</th>
      <th>median_scaffold</th>
      <th>min_scaffold</th>
      <th>number of scaffolds</th>
      <th>perc_A</th>
      <th>perc_C</th>
      <th>perc_G</th>
      <th>...</th>
      <th>perc_T</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_100kb</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_10kb</th>
      <th>perc_genome_covered_by_scaffolds_greater_than_1mb</th>
      <th>scaffolds_greater_than_100Mb</th>
      <th>scaffolds_greater_than_100kb</th>
      <th>scaffolds_greater_than_10Mb</th>
      <th>scaffolds_greater_than_10kb</th>
      <th>scaffolds_greater_than_1Mb</th>
      <th>total bases</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>statistic</th>
      <td>32220929</td>
      <td>8340160</td>
      <td>85027298</td>
      <td>7.324693e+06</td>
      <td>296338</td>
      <td>10129</td>
      <td>223</td>
      <td>27.049532</td>
      <td>20.123188</td>
      <td>20.129348</td>
      <td>...</td>
      <td>27.050599</td>
      <td>99.830771</td>
      <td>100.0</td>
      <td>98.841232</td>
      <td>0</td>
      <td>133</td>
      <td>45</td>
      <td>223</td>
      <td>90</td>
      <td>1633406540</td>
    </tr>
  </tbody>
</table>
<p>1 rows × 21 columns</p>
</div></div>
</div>
<p>Here I show the basic statistics gathered by the fasta class</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [50]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF</span><span class="p">[</span><span class="s1">&#39;non_n_GC%&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF</span><span class="p">[[</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">/</span> <span class="n">statDF</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">*</span> <span class="mf">100.0</span>
<span class="n">statDF</span><span class="p">[</span><span class="s1">&#39;non_n_AT%&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">/</span> <span class="n">statDF</span><span class="p">[[</span><span class="s1">&#39;perc_A&#39;</span><span class="p">,</span><span class="s1">&#39;perc_G&#39;</span><span class="p">,</span><span class="s1">&#39;perc_C&#39;</span><span class="p">,</span><span class="s1">&#39;perc_T&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">*</span> <span class="mf">100.0</span>

<span class="n">statDF</span> <span class="o">=</span> <span class="n">statDF</span><span class="o">.</span><span class="n">T</span>
<span class="n">statDF</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">statDF</span><span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;</span><span class="si">%.5f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">x</span><span class="p">)</span>
<span class="n">statDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[50]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>statistic</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>N50</th>
      <td>32220929.00000</td>
    </tr>
    <tr>
      <th>N90</th>
      <td>8340160.00000</td>
    </tr>
    <tr>
      <th>max_scaffold</th>
      <td>85027298.00000</td>
    </tr>
    <tr>
      <th>mean_scaffold</th>
      <td>7324693.00448</td>
    </tr>
    <tr>
      <th>median_scaffold</th>
      <td>296338.00000</td>
    </tr>
    <tr>
      <th>min_scaffold</th>
      <td>10129.00000</td>
    </tr>
    <tr>
      <th>number of scaffolds</th>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>perc_A</th>
      <td>27.04953</td>
    </tr>
    <tr>
      <th>perc_C</th>
      <td>20.12319</td>
    </tr>
    <tr>
      <th>perc_G</th>
      <td>20.12935</td>
    </tr>
    <tr>
      <th>perc_N</th>
      <td>5.64733</td>
    </tr>
    <tr>
      <th>perc_T</th>
      <td>27.05060</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_100kb</th>
      <td>99.83077</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_10kb</th>
      <td>100.00000</td>
    </tr>
    <tr>
      <th>perc_genome_covered_by_scaffolds_greater_than_1mb</th>
      <td>98.84123</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_100Mb</th>
      <td>0.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_100kb</th>
      <td>133.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_10Mb</th>
      <td>45.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_10kb</th>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>scaffolds_greater_than_1Mb</th>
      <td>90.00000</td>
    </tr>
    <tr>
      <th>total bases</th>
      <td>1633406540.00000</td>
    </tr>
    <tr>
      <th>non_n_GC%</th>
      <td>42.66179</td>
    </tr>
    <tr>
      <th>non_n_AT%</th>
      <td>57.33821</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [51]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF</span> <span class="o">=</span> <span class="n">statDF</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [52]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;statistic&#39;</span><span class="p">,</span> <span class="s1">&#39;value&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [53]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">statDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[53]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>statistic</th>
      <th>value</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>N50</td>
      <td>32220929.00000</td>
    </tr>
    <tr>
      <th>1</th>
      <td>N90</td>
      <td>8340160.00000</td>
    </tr>
    <tr>
      <th>2</th>
      <td>max_scaffold</td>
      <td>85027298.00000</td>
    </tr>
    <tr>
      <th>3</th>
      <td>mean_scaffold</td>
      <td>7324693.00448</td>
    </tr>
    <tr>
      <th>4</th>
      <td>median_scaffold</td>
      <td>296338.00000</td>
    </tr>
    <tr>
      <th>5</th>
      <td>min_scaffold</td>
      <td>10129.00000</td>
    </tr>
    <tr>
      <th>6</th>
      <td>number of scaffolds</td>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>7</th>
      <td>perc_A</td>
      <td>27.04953</td>
    </tr>
    <tr>
      <th>8</th>
      <td>perc_C</td>
      <td>20.12319</td>
    </tr>
    <tr>
      <th>9</th>
      <td>perc_G</td>
      <td>20.12935</td>
    </tr>
    <tr>
      <th>10</th>
      <td>perc_N</td>
      <td>5.64733</td>
    </tr>
    <tr>
      <th>11</th>
      <td>perc_T</td>
      <td>27.05060</td>
    </tr>
    <tr>
      <th>12</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_100kb</td>
      <td>99.83077</td>
    </tr>
    <tr>
      <th>13</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_10kb</td>
      <td>100.00000</td>
    </tr>
    <tr>
      <th>14</th>
      <td>perc_genome_covered_by_scaffolds_greater_than_1mb</td>
      <td>98.84123</td>
    </tr>
    <tr>
      <th>15</th>
      <td>scaffolds_greater_than_100Mb</td>
      <td>0.00000</td>
    </tr>
    <tr>
      <th>16</th>
      <td>scaffolds_greater_than_100kb</td>
      <td>133.00000</td>
    </tr>
    <tr>
      <th>17</th>
      <td>scaffolds_greater_than_10Mb</td>
      <td>45.00000</td>
    </tr>
    <tr>
      <th>18</th>
      <td>scaffolds_greater_than_10kb</td>
      <td>223.00000</td>
    </tr>
    <tr>
      <th>19</th>
      <td>scaffolds_greater_than_1Mb</td>
      <td>90.00000</td>
    </tr>
    <tr>
      <th>20</th>
      <td>total bases</td>
      <td>1633406540.00000</td>
    </tr>
    <tr>
      <th>21</th>
      <td>non_n_GC%</td>
      <td>42.66179</td>
    </tr>
    <tr>
      <th>22</th>
      <td>non_n_AT%</td>
      <td>57.33821</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [54]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1">#statDF.to_csv(&#39;../data/supplemental_data/supplemental_table_5.csv&#39;,index=False)</span>
</pre></div>
</div>
</div>
<p>Next, I use the module to calculate N50 like scores. I then plot those
scores in the next figure.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [55]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">nxDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">dtGenome</span><span class="o">.</span><span class="n">return_nx_dist</span><span class="p">())</span><span class="o">.</span><span class="n">T</span>
<span class="n">nxDF</span><span class="o">.</span><span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;perc&#39;</span><span class="p">,</span><span class="s1">&#39;size&#39;</span><span class="p">]</span>
<span class="n">nxDF</span> <span class="o">=</span> <span class="n">nxDF</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;perc&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [56]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">ax1</span> <span class="o">=</span> <span class="n">nxDF</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="s1">&#39;perc&#39;</span><span class="p">,</span><span class="s1">&#39;size&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.6</span><span class="p">,</span><span class="mf">3.4</span><span class="p">))</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">get_major_formatter</span><span class="p">()</span><span class="o">.</span><span class="n">set_scientific</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">FuncFormatter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">p</span><span class="p">:</span> <span class="n">format</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="mi">1000000</span><span class="p">),</span> <span class="s1">&#39;,&#39;</span><span class="p">)))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Scaffold Size (MB)&#39;</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Percentage of Total Bases in Genome (%)&#39;</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">120</span><span class="p">,</span><span class="mi">20</span><span class="p">))</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span><span class="n">ymin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">ymax</span><span class="o">=</span><span class="mi">100000000</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">hlines</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="mi">32220929</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span><span class="n">xmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">xmax</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="s1">&#39;N50 = 32.22 MB&#39;</span><span class="p">,</span><span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span><span class="mi">28000000</span><span class="p">),</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;g&#39;</span><span class="p">,</span><span class="n">ymin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">ymax</span><span class="o">=</span><span class="mi">100000000</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">hlines</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="mi">8340160</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;g&#39;</span><span class="p">,</span><span class="n">xmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">xmax</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="s1">&#39;N90 = 8.34 MB&#39;</span><span class="p">,</span><span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">55</span><span class="p">,</span><span class="mi">4500000</span><span class="p">),</span><span class="n">fontsize</span> <span class="o">=</span> <span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">90000000</span><span class="p">)</span>


<span class="n">ax1</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_bottom</span><span class="p">()</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_left</span><span class="p">()</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure3A.pdf&#39;, bbox_inches=&#39;tight&#39;)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_84_0.png" src="_images/marmorata_genome_paper_figures_3_84_0.png" />
</div>
</div>
<p>Here I load in the sizes of each scaffold in the genome. These are used
multiple times later in analysis.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [57]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">scaffoldSizes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;/n/projects/dut/a_marmorata/dovetail_genome_delivery/data/fastas/scaffold_sizes.clean.tsv&#39;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;scaffold&#39;</span><span class="p">,</span><span class="s1">&#39;scaffold_size&#39;</span><span class="p">])</span>
<span class="n">scaffoldSizes</span> <span class="o">=</span> <span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;scaffold_size&#39;</span><span class="p">,</span><span class="n">ascending</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">orderedScaffolds</span> <span class="o">=</span> <span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
<span class="n">scaffoldSizeDict</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold</span><span class="p">,</span><span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold_size</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Isochore-Analysis">
<h2>Isochore Analysis<a class="headerlink" href="#Isochore-Analysis" title="Permalink to this headline">¶</a></h2>
<p>Full details on the ischore analysis are included in a separate
notebook, “../../marmorata_isochores” is the location of the project</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [58]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1">#See notebook at ../../marmorata_isochores for full details on analysis</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [59]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">isochoreFiles</span> <span class="o">=</span> <span class="n">glob</span><span class="o">.</span><span class="n">glob</span><span class="p">(</span><span class="s1">&#39;../../marmorata_isochores/data/*window.tsv&#39;</span><span class="p">)</span>
<span class="n">isochoreFiles</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[59]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>[&#39;../../marmorata_isochores/data/Gallus_gallus.Gallus_gallus-5.0.dna.toplevel.gc_5000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/Pelodiscus_sinensis.PelSin_1.0.dna.toplevel.gc_10000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/Danio_rerio.GRCz10.dna.toplevel.gc_80000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/tigris_scaffolds_filt_10000.gc_160000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/Pelodiscus_sinensis.PelSin_1.0.dna.toplevel.gc_160000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/Danio_rerio.GRCz10.dna.toplevel.gc_320000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/Homo_sapiens.GRCh38.dna.primary_assembly.gc_20000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/GCA_000186305.2_Python_molurus_bivittatus-5.0.2_genomic.gc_160000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/GCA_000186305.2_Python_molurus_bivittatus-5.0.2_genomic.gc_40000_scaffold_window.tsv&#39;,
 &#39;../../marmorata_isochores/data/GCA_000186305.2_Python_molurus_bivittatus-5.0.2_genomic.gc_80000_scaffold_window.tsv&#39;]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [60]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">list_std</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">isochorePathDict</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">path</span> <span class="ow">in</span> <span class="n">isochoreFiles</span><span class="p">:</span>
    <span class="n">sample</span> <span class="o">=</span> <span class="n">path</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;/&#39;</span><span class="p">)[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;.&#39;</span><span class="p">)[:</span><span class="mi">2</span><span class="p">]</span>
    <span class="n">window_size</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">path</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">)[</span><span class="o">-</span><span class="mi">3</span><span class="p">])</span>
    <span class="k">if</span> <span class="s1">&#39;Python&#39;</span> <span class="ow">in</span> <span class="n">sample</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span>
        <span class="n">sample</span> <span class="o">=</span> <span class="n">sample</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>

    <span class="k">elif</span> <span class="s1">&#39;tigris&#39;</span> <span class="ow">in</span> <span class="n">sample</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span>
        <span class="n">sample</span> <span class="o">=</span> <span class="s1">&#39;Aspidoscelis_marmorata&#39;</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">sample</span> <span class="o">=</span> <span class="n">sample</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">sample</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;2_&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;-5&#39;</span><span class="p">,</span><span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">,</span> <span class="s1">&#39; &#39;</span><span class="p">)</span>
    <span class="n">isochorePathDict</span><span class="p">[(</span><span class="n">name</span><span class="p">,</span><span class="n">window_size</span><span class="p">)]</span><span class="o">=</span><span class="n">path</span>
    <span class="n">windows_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;scaffold&#39;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">,</span> <span class="s1">&#39;perc_gc&#39;</span><span class="p">])</span>
    <span class="n">windows_df</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">amount_data</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">windows_df</span><span class="p">)</span> <span class="o">*</span> <span class="n">window_size</span>
    <span class="n">std</span> <span class="o">=</span> <span class="n">windows_df</span><span class="o">.</span><span class="n">perc_gc</span><span class="o">.</span><span class="n">std</span><span class="p">()</span>
    <span class="n">dict_data</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;organism&#39;</span><span class="p">:</span><span class="n">sample</span><span class="p">,</span> <span class="s1">&#39;window_size&#39;</span><span class="p">:</span><span class="n">window_size</span><span class="p">,</span> <span class="s1">&#39;bases_analyzed&#39;</span><span class="p">:</span><span class="n">amount_data</span><span class="p">,</span> <span class="s1">&#39;std_dev&#39;</span><span class="p">:</span><span class="n">std</span><span class="p">}</span>
    <span class="n">list_std</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dict_data</span><span class="p">)</span>


<span class="n">gcSTD</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">list_std</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Here I plot the standard deviation of GC content across various genomes
with different window sizes. The data shows that Aspidoscelis marmoratus
has similar isochore structure to that of the python, turtle, and mouse.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [61]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.4</span><span class="p">,</span> <span class="mf">3.4</span><span class="p">))</span>
<span class="n">ax2</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">organism_map</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Gallus_gallus&#39;</span><span class="p">:</span> <span class="s1">&#39;Chicken&#39;</span><span class="p">,</span> <span class="s1">&#39;2_Python_molurus_bivittatus-5&#39;</span><span class="p">:</span> <span class="s1">&#39;Burmese python&#39;</span><span class="p">,</span> <span class="s1">&#39;Aspidoscelis_marmorata&#39;</span><span class="p">:</span> <span class="s1">&#39;Aspidoscelis marmoratus&#39;</span><span class="p">,</span>
                <span class="s1">&#39;Pelodiscus_sinensis&#39;</span><span class="p">:</span> <span class="s1">&#39;Chinese softshell turtle&#39;</span><span class="p">,</span> <span class="s1">&#39;Homo_sapiens&#39;</span><span class="p">:</span> <span class="s1">&#39;Human&#39;</span><span class="p">,</span> <span class="s1">&#39;Canis_familiaris&#39;</span><span class="p">:</span> <span class="s1">&#39;Dog&#39;</span><span class="p">,</span> <span class="s1">&#39;Mus_musculus&#39;</span><span class="p">:</span> <span class="s1">&#39;Mouse&#39;</span><span class="p">,</span>
                <span class="s1">&#39;Danio_rerio&#39;</span><span class="p">:</span> <span class="s1">&#39;Zebrafish&#39;</span><span class="p">,</span> <span class="s1">&#39;Anolis_carolinensis&#39;</span><span class="p">:</span> <span class="s1">&#39;Carolina anole lizard&#39;</span><span class="p">}</span>

<span class="n">markers</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;--o&#39;</span><span class="p">,</span> <span class="s1">&#39;--v&#39;</span><span class="p">,</span> <span class="s1">&#39;--^&#39;</span><span class="p">,</span> <span class="s1">&#39;--&lt;&#39;</span><span class="p">,</span> <span class="s1">&#39;--&gt;&#39;</span><span class="p">,</span> <span class="s1">&#39;--8&#39;</span><span class="p">,</span> <span class="s1">&#39;--s&#39;</span><span class="p">,</span> <span class="s1">&#39;--p&#39;</span><span class="p">,</span> <span class="s1">&#39;--h&#39;</span><span class="p">,</span> <span class="s1">&#39;--H&#39;</span><span class="p">,</span> <span class="s1">&#39;--D&#39;</span><span class="p">,</span> <span class="s1">&#39;--d&#39;</span><span class="p">]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">organism</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">gcSTD</span><span class="o">.</span><span class="n">organism</span><span class="o">.</span><span class="n">unique</span><span class="p">()):</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">organism_map</span><span class="p">[</span><span class="n">organism</span><span class="p">]</span>
    <span class="n">gc_dist</span> <span class="o">=</span> <span class="n">gcSTD</span><span class="p">[</span><span class="n">gcSTD</span><span class="o">.</span><span class="n">organism</span> <span class="o">==</span><span class="n">organism</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">gc_dist</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;window_size&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;window_size&#39;</span><span class="p">],</span>
            <span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;std_dev&#39;</span><span class="p">],</span>
            <span class="n">markers</span><span class="p">[</span><span class="n">i</span><span class="p">],</span>
            <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">,</span>
            <span class="n">markersize</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
            <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s2">&quot;log&quot;</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="mi">10</span><span class="p">]</span> <span class="o">+</span> <span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;window_size&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">matplotlib</span><span class="o">.</span><span class="n">ticker</span><span class="o">.</span><span class="n">ScalarFormatter</span><span class="p">())</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">2770</span><span class="p">,</span><span class="mi">330000</span><span class="p">)</span>
<span class="n">xtick_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="mf">1000.0</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">xtick_labels</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax2</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">% G</span><span class="s1">C Standard Deviation&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;log(Window Size(kb))&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.7</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_bottom</span><span class="p">()</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_left</span><span class="p">()</span>

<span class="c1">#fig.savefig(&#39;../fig2/Figure3B.pdf&#39;, bbox_inches=&#39;tight&#39;)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_93_0.png" src="_images/marmorata_genome_paper_figures_3_93_0.png" />
</div>
</div>
</div>
<div class="section" id="Busco-Analysis-(Done-by-Rutendo-Sigauke)">
<h2>Busco Analysis (Done by Rutendo Sigauke)<a class="headerlink" href="#Busco-Analysis-(Done-by-Rutendo-Sigauke)" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [62]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/busco_analysis/figures/busco_summary_figure.pdf ../fig2/SupplementalFigure1A.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [63]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/n/projects/rfs/lizard/marmorata/busco_analysis/figures/busco_summary_figure.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">900</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">900</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[63]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_96_0.png" src="_images/marmorata_genome_paper_figures_3_96_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [64]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/busco_analysis/figures/phylogeny_1536BUSCOs_Gblocks_100bootstrap_Arial_Common_Name.pdf ../fig2/SupplementalFigure1B.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [65]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/n/projects/rfs/lizard/marmorata/busco_analysis/figures/phylogeny_1536BUSCOs_Gblocks_100bootstrap_Arial_Common_Name.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">900</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">900</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[65]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_98_0.png" src="_images/marmorata_genome_paper_figures_3_98_0.png" />
</div>
</div>
</div>
<div class="section" id="Synteny-Analysis-(Done-by-Rutendo-Sigauke)">
<h2>Synteny Analysis (Done by Rutendo Sigauke)<a class="headerlink" href="#Synteny-Analysis-(Done-by-Rutendo-Sigauke)" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [66]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/synteny/figures/Marmorata_Anole_synteny_99score_allScaff_Arial_common_name.pdf ../fig2/SupplementalFigure1C.pdf
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [67]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/n/projects/rfs/lizard/marmorata/synteny/figures/Marmorata_Anole_synteny_99score_allScaff_Arial_common_name.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[67]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_102_0.png" src="_images/marmorata_genome_paper_figures_3_102_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [68]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/synteny/figures/Marmorata_Chicken_synteny_99score_allScaff_Arial_common_names.pdf ../fig2/SupplementalFigure1D.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [69]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/n/projects/rfs/lizard/marmorata/synteny/figures/Marmorata_Chicken_synteny_99score_allScaff_Arial_common_names.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[69]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_104_0.png" src="_images/marmorata_genome_paper_figures_3_104_0.png" />
</div>
</div>
</div>
<div class="section" id="Repeat-Analysis-(Done-by-Rutendo-Sigauke)">
<h2>Repeat Analysis (Done by Rutendo Sigauke)<a class="headerlink" href="#Repeat-Analysis-(Done-by-Rutendo-Sigauke)" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [70]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/repeats/analysis/figures/repeats_genomes_summary_common_names_Ariel.pdf ../fig2/SupplementalFigure2.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [71]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/n/projects/rfs/lizard/marmorata/repeats/analysis/figures/repeats_genomes_summary_common_names_Ariel.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[71]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_107_0.png" src="_images/marmorata_genome_paper_figures_3_107_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [72]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /n/projects/rfs/lizard/marmorata/repeats/repeat_masker/repeat_summaries/RepeatFamilySummariesPercentOfGenome.tsv ../data/supplemental_data/supplemental_table_6.tsv
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [73]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_6.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">tail</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[73]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>repeats</th>
      <th>marmorata</th>
      <th>anole</th>
      <th>xenopus</th>
      <th>gallus</th>
      <th>python</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>67</th>
      <td>SINE/tRNA-Sauria-L2</td>
      <td>0.000000</td>
      <td>1.134833</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.011473</td>
    </tr>
    <tr>
      <th>68</th>
      <td>SINE/tRNA-Sauria-RTE</td>
      <td>0.000000</td>
      <td>0.102815</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.099582</td>
    </tr>
    <tr>
      <th>69</th>
      <td>SINE/U</td>
      <td>0.000000</td>
      <td>0.000624</td>
      <td>0.011249</td>
      <td>0.000000</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>70</th>
      <td>snRNA</td>
      <td>0.004796</td>
      <td>0.000000</td>
      <td>0.009599</td>
      <td>0.000000</td>
      <td>0.001264</td>
    </tr>
    <tr>
      <th>71</th>
      <td>Unknown</td>
      <td>20.640800</td>
      <td>7.684774</td>
      <td>5.144897</td>
      <td>0.531585</td>
      <td>8.898379</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</div>
<div class="section" id="Load-Annotations">
<h2>Load Annotations<a class="headerlink" href="#Load-Annotations" title="Permalink to this headline">¶</a></h2>
<p>The following python module can be used to manipulate and plot the
contained in a GFF3 formatted file.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [74]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/gff3_plotting.py</span>
</pre></div>
</div>
</div>
<p>Here I load in the genome annotations, these are used later in the hox
gene analysis and vmnr analysis.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [75]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">gffMarmAbInPath</span><span class="o">=</span><span class="s1">&#39;../data/augustus.renamed.putative_function.iprscan.gff3&#39;</span>
<span class="n">gffMarmPath</span> <span class="o">=</span> <span class="s1">&#39;../data/a_tigris_1.renamed.putative_function.iprscan.wintrons.gff&#39;</span>
<span class="n">gffMarmDF</span><span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">gff3_to_attribute_df</span><span class="p">(</span><span class="n">gff_path</span><span class="o">=</span><span class="n">gffMarmPath</span><span class="p">,</span><span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_symbol</span><span class="p">(</span><span class="n">gffMarmDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_id</span><span class="p">(</span><span class="n">gffMarmDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_name</span><span class="p">(</span><span class="n">gffMarmDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmDF</span><span class="p">[</span><span class="s1">&#39;count_column&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">gffMarmDF</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;seqid&#39;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">gffMarmDF</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;MAKER2&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
691299 691299
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [76]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">gffMarmAbInDF</span><span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">gff3_to_attribute_df</span><span class="p">(</span><span class="n">gff_path</span><span class="o">=</span><span class="n">gffMarmAbInPath</span><span class="p">,</span><span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmAbInDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_symbol</span><span class="p">(</span><span class="n">gffMarmAbInDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmAbInDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_id</span><span class="p">(</span><span class="n">gffMarmAbInDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmAbInDF</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">add_gene_name</span><span class="p">(</span><span class="n">gffMarmAbInDF</span><span class="p">,</span> <span class="n">ensembl</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">gffMarmAbInDF</span><span class="p">[</span><span class="s1">&#39;count_column&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">gffMarmAbInDF</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;seqid&#39;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">gffMarmAbInDF</span><span class="p">[</span><span class="s1">&#39;source&#39;</span><span class="p">]</span><span class="o">=</span><span class="s1">&#39;BRAKER1&#39;</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1737575 1737575
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [77]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span>
    <span class="p">[</span>
        <span class="n">gffMarmAbInDF</span><span class="p">[[</span><span class="s1">&#39;source&#39;</span><span class="p">,</span><span class="s1">&#39;feature&#39;</span><span class="p">]],</span>
        <span class="n">gffMarmDF</span><span class="p">[[</span><span class="s1">&#39;source&#39;</span><span class="p">,</span><span class="s1">&#39;feature&#39;</span><span class="p">]]</span>
    <span class="p">]</span>
     <span class="p">)</span>
<span class="n">supplementalAnnotDF</span><span class="p">[</span><span class="s1">&#39;count&#39;</span><span class="p">]</span> <span class="o">=</span><span class="mi">1</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [78]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;source&#39;</span><span class="p">,</span><span class="s1">&#39;feature&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">supplementalAnnotDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[78]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>source</th>
      <th>feature</th>
      <th>count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>BRAKER1</td>
      <td>CDS</td>
      <td>424058</td>
    </tr>
    <tr>
      <th>1</th>
      <td>BRAKER1</td>
      <td>exon</td>
      <td>424058</td>
    </tr>
    <tr>
      <th>2</th>
      <td>BRAKER1</td>
      <td>gene</td>
      <td>152361</td>
    </tr>
    <tr>
      <th>3</th>
      <td>BRAKER1</td>
      <td>intron</td>
      <td>267602</td>
    </tr>
    <tr>
      <th>4</th>
      <td>BRAKER1</td>
      <td>mRNA</td>
      <td>156525</td>
    </tr>
    <tr>
      <th>5</th>
      <td>BRAKER1</td>
      <td>start_codon</td>
      <td>156457</td>
    </tr>
    <tr>
      <th>6</th>
      <td>BRAKER1</td>
      <td>stop_codon</td>
      <td>156514</td>
    </tr>
    <tr>
      <th>7</th>
      <td>MAKER2</td>
      <td>CDS</td>
      <td>222450</td>
    </tr>
    <tr>
      <th>8</th>
      <td>MAKER2</td>
      <td>exon</td>
      <td>164669</td>
    </tr>
    <tr>
      <th>9</th>
      <td>MAKER2</td>
      <td>five_prime_UTR</td>
      <td>22402</td>
    </tr>
    <tr>
      <th>10</th>
      <td>MAKER2</td>
      <td>gene</td>
      <td>25856</td>
    </tr>
    <tr>
      <th>11</th>
      <td>MAKER2</td>
      <td>intron</td>
      <td>189732</td>
    </tr>
    <tr>
      <th>12</th>
      <td>MAKER2</td>
      <td>mRNA</td>
      <td>44461</td>
    </tr>
    <tr>
      <th>13</th>
      <td>MAKER2</td>
      <td>three_prime_UTR</td>
      <td>21729</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [79]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">T</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [80]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
<span class="n">supplementalAnnotDF</span> <span class="o">=</span> <span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">reindex</span><span class="p">(</span><span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="mi">1</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [81]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[81]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>feature</th>
      <th>CDS</th>
      <th>exon</th>
      <th>gene</th>
      <th>intron</th>
      <th>mRNA</th>
      <th>start_codon</th>
      <th>stop_codon</th>
      <th>CDS</th>
      <th>exon</th>
      <th>five_prime_UTR</th>
      <th>gene</th>
      <th>intron</th>
      <th>mRNA</th>
      <th>three_prime_UTR</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>source</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>BRAKER1</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
      <td>MAKER2</td>
    </tr>
    <tr>
      <th>2</th>
      <td>count</td>
      <td>424058</td>
      <td>424058</td>
      <td>152361</td>
      <td>267602</td>
      <td>156525</td>
      <td>156457</td>
      <td>156514</td>
      <td>222450</td>
      <td>164669</td>
      <td>22402</td>
      <td>25856</td>
      <td>189732</td>
      <td>44461</td>
      <td>21729</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [82]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">supplementalAnnotDF</span><span class="o">.</span><span class="n">T</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../data/supplemental_data/supplemental_table_7.csv&#39;</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="s1">&#39;False&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="HOX-Gene-Analysis">
<h2>HOX Gene Analysis<a class="headerlink" href="#HOX-Gene-Analysis" title="Permalink to this headline">¶</a></h2>
<p>Here I link to the braker ab initio gene predictions</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [83]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#ln -s ../../dovetail_braker_annotation/data/augustus.renamed.putative_function.iprscan.gff3</span>
</pre></div>
</div>
</div>
<p>Next, I downloaded annotation sets for human, mouse, and anolis. I was
originally going to compare annotations directly in the GFF file, but I
decided not too because the annotations for mouse and human have far
more evidence than the annotations for anolis.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [84]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/homo_sapiens/Homo_sapiens.GRCh38.87.abinitio.gff3.gz</span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/homo_sapiens/Homo_sapiens.GRCh38.87.chr.gff3.gz</span>
<span class="c1">#gunzip Homo_sapiens.GRCh38.87.abinitio.gff3.gz</span>
<span class="c1">#gunzip Homo_sapiens.GRCh38.87.chr.gff3.gz</span>

</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [85]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/mus_musculus/Mus_musculus.GRCm38.87.chr.gff3.gz</span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/mus_musculus/Mus_musculus.GRCm38.87.abinitio.gff3.gz</span>
<span class="c1">#gunzip Mus_musculus.GRCm38.87.chr.gff3.gz</span>
<span class="c1">#gunzip Mus_musculus.GRCm38.87.abinitio.gff3.gz</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [86]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/anolis_carolinensis/Anolis_carolinensis.AnoCar2.0.87.chr.gff3.gz</span>
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/anolis_carolinensis/Anolis_carolinensis.AnoCar2.0.87.abinitio.gff3.gz</span>
<span class="c1">#gunzip Anolis_carolinensis.AnoCar2.0.87.chr.gff3.gz</span>
<span class="c1">#gunzip Anolis_carolinensis.AnoCar2.0.87.abinitio.gff3.gz</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [87]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#wget ftp://ftp.ensembl.org/pub/release-87/gff3/anolis_carolinensis/Anolis_carolinensis.AnoCar2.0.87.gff3.gz</span>
<span class="c1">#gunzip Anolis_carolinensis.AnoCar2.0.87.gff3.gz</span>
</pre></div>
</div>
</div>
<p>Here I am searching for the HOX gene clusters and plotting them using
the python script loaded into this notebook (gff3).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [88]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;a&#39;</span><span class="p">,</span><span class="s1">&#39;b&#39;</span><span class="p">,</span><span class="s1">&#39;c&#39;</span><span class="p">,</span><span class="s1">&#39;d&#39;</span><span class="p">]:</span>
    <span class="c1">#Aspidoscelis marmorata</span>
    <span class="n">regex</span> <span class="o">=</span> <span class="s1">&#39;hox</span><span class="si">%s</span><span class="s1">|evx&#39;</span><span class="o">%</span><span class="k">c</span>
    <span class="n">organism</span> <span class="o">=</span> <span class="s1">&#39;Aspidoscelis marmoratus&#39;</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">gff3</span><span class="o">.</span><span class="n">plot_single_cluster_series</span><span class="p">(</span><span class="n">gff3_paths</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;../data/augustus.renamed.putative_function.iprscan.gff3&#39;</span><span class="p">],</span> <span class="n">subset_regex</span><span class="o">=</span><span class="n">regex</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="n">organism</span><span class="p">,</span> <span class="n">path_targ</span><span class="o">=</span><span class="s1">&#39;augustus&#39;</span><span class="p">)</span>
    <span class="c1">#fig.savefig(&#39;../fig/%s_hox-%s.pdf&#39;%(organism, c), bbox_inches=&#39;tight&#39;, pad_inches=0.2)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1737575 1737575
1737575
165
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_131_1.png" src="_images/marmorata_genome_paper_figures_3_131_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1737575 1737575
1737575
171
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_131_3.png" src="_images/marmorata_genome_paper_figures_3_131_3.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1737575 1737575
1737575
168
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_131_5.png" src="_images/marmorata_genome_paper_figures_3_131_5.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
1737575 1737575
1737575
200
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_131_7.png" src="_images/marmorata_genome_paper_figures_3_131_7.png" />
</div>
</div>
<p>Here I subset out the scaffolds which contain the hox gene clusters.
This was done in case they were needed for supplemental. I also
extracted out the annotations for just those scaffolds from the GFF3
braker gene predictions.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [89]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">hoxScaffolds</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;&gt;Scpiz6a_1&#39;</span><span class="p">,</span><span class="s1">&#39;&gt;Scpiz6a_37&#39;</span><span class="p">,</span><span class="s1">&#39;&gt;Scpiz6a_86&#39;</span><span class="p">,</span><span class="s1">&#39;&gt;Scpiz6a_30.1&#39;</span><span class="p">]</span>
<span class="n">hoxGenomeDict</span> <span class="o">=</span> <span class="p">{</span><span class="n">k</span><span class="p">:</span><span class="n">v</span> <span class="k">for</span> <span class="n">k</span><span class="p">,</span><span class="n">v</span> <span class="ow">in</span> <span class="n">dtGenome</span><span class="o">.</span><span class="n">fasta_dict</span><span class="o">.</span><span class="n">items</span><span class="p">()</span> <span class="k">if</span> <span class="n">k</span> <span class="ow">in</span> <span class="n">hoxScaffolds</span><span class="p">}</span>
<span class="n">fa</span><span class="o">.</span><span class="n">write_two_line_fasta</span><span class="p">(</span><span class="s1">&#39;../data/hox_scaffolds.fa&#39;</span><span class="p">,</span> <span class="nb">list</span><span class="p">(</span><span class="n">hoxGenomeDict</span><span class="o">.</span><span class="n">items</span><span class="p">()))</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [90]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">augHoxGFFDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/augustus.renamed.putative_function.iprscan.gff3&#39;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="bp">None</span><span class="p">)</span>
<span class="n">augHoxGFFDF</span> <span class="o">=</span> <span class="n">augHoxGFFDF</span><span class="p">[</span><span class="n">augHoxGFFDF</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="n">s</span><span class="o">.</span><span class="n">strip</span><span class="p">(</span><span class="s1">&#39;&gt;&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="n">hoxScaffolds</span><span class="p">])]</span>
<span class="n">augHoxGFFDF</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../data/aug_hox_annotations.gff3&#39;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">augHoxGFFDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[90]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>0</th>
      <th>1</th>
      <th>2</th>
      <th>3</th>
      <th>4</th>
      <th>5</th>
      <th>6</th>
      <th>7</th>
      <th>8</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>348850</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>3902</td>
      <td>4216</td>
      <td>0.59</td>
      <td>+</td>
      <td>.</td>
      <td>ID=ATIG_ab_00023958;Name=;Alias=g31007;Note=Similar to NSMF: NMDA receptor synaptonuclear signaling and neuronal migration factor (Homo sapiens);</td>
    </tr>
    <tr>
      <th>348851</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>mRNA</td>
      <td>3902</td>
      <td>4216</td>
      <td>0.59</td>
      <td>+</td>
      <td>.</td>
      <td>ID=ATIG_ab_00023958-RA;Parent=ATIG_ab_00023958;Name=;Alias=g31007.t1;Note=Similar to NSMF: NMDA receptor synaptonuclear signaling and neuronal migration factor (Homo sapiens);</td>
    </tr>
    <tr>
      <th>348852</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>start_codon</td>
      <td>3902</td>
      <td>3904</td>
      <td>.</td>
      <td>+</td>
      <td>0</td>
      <td>Parent=ATIG_ab_00023958-RA;</td>
    </tr>
    <tr>
      <th>348853</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>CDS</td>
      <td>3902</td>
      <td>4216</td>
      <td>0.59</td>
      <td>+</td>
      <td>0</td>
      <td>ID=g31007.t1.CDS1;Parent=ATIG_ab_00023958-RA;</td>
    </tr>
    <tr>
      <th>348854</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>exon</td>
      <td>3902</td>
      <td>4216</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>ID=g31007.t1.exon1;Parent=ATIG_ab_00023958-RA;</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Because the above figures are a bit messy looking I decided to make a
simple schematic for the text. I start with pulling out all the gene
coordinates for the above figures.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [91]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#awk &#39;$3==&quot;gene&quot;&#39; augustus.renamed.putative_function.iprscan.gff3 | grep -E -i &#39; hox|evx&#39; | awk  &#39;$1==&quot;Scpiz6a_86&quot; || $1==&quot;Scpiz6a_30.1&quot; || $1==&quot;Scpiz6a_37&quot; || $1==&quot;Scpiz6a_1&quot;&#39; | sed &#39;s/Similar to /\t/&#39; | cut -f -5,7,10 | tr &#39;:&#39; &#39;\t&#39; | cut -f -7 &gt; hox_gene_coords.tsv</span>
</pre></div>
</div>
</div>
<p>I also independently found HOXB13 in the MAKER annotations by manually
blasting a nearby model. I next added the coordinates from that model to
the file I generated above.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [92]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#grep ATIG_00008349 a_tigris_1.renamed.putative_function.iprscan.wintrons.gff  | awk &#39;$3==&quot;gene&quot;&#39; | cut -f -5,7 | tr &#39;\n&#39; &#39;~&#39; | sed &quot;s/~/@Hoxb13~/&quot; | tr &#39;~&#39; &#39;\n&#39; | tr &#39;@&#39; &#39;\t&#39; &gt;&gt; hox_gene_coords.tsv</span>
</pre></div>
</div>
</div>
<p>I found later on that hoxc3 is present in the anolis lizard
(<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665779/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2665779/</a>). I couldn’t find
a copy of the sequence mentioned in the paper, but I was able to find a
sequence for Xenopus HOXC3. I generated the following curl command
directly from the ensemble website.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [93]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#curl --header &#39;Host: www.ensembl.org&#39; \</span>
<span class="c1">#--header &#39;User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:52.0) Gecko/20100101 Firefox/52.0&#39; \</span>
<span class="c1">#--header &#39;Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8&#39; \</span>
<span class="c1">#--header &#39;Accept-Language: en-US,en;q=0.5&#39; \</span>
<span class="c1">#--header &#39;Referer: http://www.ensembl.org/Xenopus_tropicalis/Gene/Sequence?db=core;g=ENSXETG00000025181;r=GL172862.1:552422-558412;t=ENSXETT00000053914&#39; \</span>
<span class="c1">#--header &#39;Cookie: _ga=GA1.2.780793432.1475001731; ENSEMBL_WIDTH=2300; DYNAMIC_WIDTH=1; _pk_ref..c10e=%5B%22%22%2C%22%22%2C1481326908%2C%22http%3A%2F%2Fuseast.ensembl.org%2FAnolis_carolinensis%2FInfo%2FIndex%22%5D; _pk_id..c10e=6f2a23a1cab0d268.1481326908.1.1481327562.1481326908.; ENSEMBL_HINX_SESSION=0ea132a2ddae24e00aa90673589a5dc90ab37997b5a94c19eec2634e; ENSEMBL_GENOVERSE_SCROLL=0; _gid=GA1.2.2055248845.1494971795; ENSEMBL_REGION_PAN=1; ENSEMBL_SEARCH=ensembl_all; _gali=KDbFOwdW_3; _gat=1&#39; \</span>
<span class="c1">#--header &#39;DNT: 1&#39; \</span>
<span class="c1">#--header &#39;Connection: keep-alive&#39; \</span>
<span class="c1">#--header &#39;Upgrade-Insecure-Requests: 1&#39; &#39;http://www.ensembl.org/Xenopus_tropicalis/Download/DataExport?compression=;db=core;file=temporary/2017_05_16/session_295535549/HAUljTwe/Xenopus_tropicalis_hoxc3_sequence.fa;filename=Xenopus_tropicalis_hoxc3_sequence.fa;format=FASTA;g=ENSXETG00000025181;r=GL172862.1:552422-558412;t=ENSXETT00000053914&#39; \</span>
<span class="c1">#-o &#39;Xenopus_tropicalis_hoxc3_sequence.fa&#39; \</span>
<span class="c1">#-L</span>
cat ../data/Xenopus_tropicalis_hoxc3_sequence.fa
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&gt;hoxc3-201 peptide: ENSXETP00000053914 pep:KNOWN_protein_coding
MPKSLFYDNSASFGGCGFQGNNGMGYLGQQDYPSEDDYQPPFCLPPDTANGSASHKGEHS
IKGIDFHLSEVSEQAQQPKSPNSDSPLPKSASTQSCTSKKSTGPVSSDVTSPNKKSKGSN
MPKQIFPWMKETRQNSKQKKQAPPPAEEACKVREKNTGSFSASKRARTAYTNSQLVELEK
EFHFNRYLCRPRRLEMAKLLNLSERQIKIWFQNRRMKFKKDHKGKGGGGSPGGLSPSSSP
SLMPYSGNLPLDGDCGYEVPMATGAYNKSPGNMYGLTAYSAPLFEGPSAQKRYGPQSLAP
EYDPHSMQGDNNYDTSGLPNGQGYLGNYLENGSESCSMFSLPHPSSESMDYSCAAQTPSK
HHLGPCDPHPTYTDLHIHPVPQACSQEPPVLTHL

</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [94]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="c1">#exonerate --model protein2genome \</span>
<span class="c1">#--showvulgar no \</span>
<span class="c1">#--showalignment no \</span>
<span class="c1">#--showtargetgff yes \</span>
<span class="c1">#--maxintron 50000 \</span>
<span class="c1">#--ryo &quot;&gt;%qi length=%ql alnlen=%qaln&gt;%ti length=%tl alnlen=%taln&quot; \</span>
<span class="c1">#--query ../data/Xenopus_tropicalis_hoxc3_sequence.fa \</span>
<span class="c1">#--target ../data/tigris_scaffolds_filt_10000.fa \</span>
<span class="c1">#&gt; Xenopus_tropicalis_hoxc3_to_marmoratus_genome.out &amp;</span>
<span class="c1">#mv Xenopus_tropicalis_hoxc3_to_marmoratus_genome.out ../data</span>

</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [95]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#grep -v &#39;^#&#39; Xenopus_tropicalis_hoxc3_to_marmoratus_genome.out | grep -v &#39;^&gt;&#39; | grep -v &#39;bonsai.sgc.loc&#39; | grep -v &#39;^Command line:&#39; &gt; Xenopus_tropicalis_hoxc3_to_marmoratus_genome.gff</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [96]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">xenopusHoxcHits</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/Xenopus_tropicalis_hoxc3_to_marmoratus_genome.gff&#39;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;seqid&#39;</span><span class="p">,</span> <span class="s1">&#39;source&#39;</span><span class="p">,</span> <span class="s1">&#39;feature&#39;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">,</span> <span class="s1">&#39;end&#39;</span><span class="p">,</span> <span class="s1">&#39;score&#39;</span><span class="p">,</span> <span class="s1">&#39;strand&#39;</span><span class="p">,</span> <span class="s1">&#39;phase&#39;</span><span class="p">,</span> <span class="s1">&#39;attributes&#39;</span><span class="p">])</span>
<span class="n">xenopusHoxcHits</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[96]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>feature</th>
      <th>start</th>
      <th>end</th>
      <th>score</th>
      <th>strand</th>
      <th>phase</th>
      <th>attributes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>19691975</td>
      <td>19692514</td>
      <td>198</td>
      <td>+</td>
      <td>.</td>
      <td>gene_id 1 ; sequence hoxc3-201 ; gene_orientation + ; identity 35.40 ; similarity 50.31</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>exonerate:protein2genome:local</td>
      <td>cds</td>
      <td>19691975</td>
      <td>19692248</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>exonerate:protein2genome:local</td>
      <td>exon</td>
      <td>19691975</td>
      <td>19692248</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>insertions 0 ; deletions 0 ; identity 43.96 ; similarity 57.14</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>exonerate:protein2genome:local</td>
      <td>splice5</td>
      <td>19692249</td>
      <td>19692250</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>intron_id 1 ; splice_site "TC"</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>exonerate:protein2genome:local</td>
      <td>intron</td>
      <td>19692249</td>
      <td>19692302</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>intron_id 1</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>The second best hit appears in the right area.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [97]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">xenopusHoxcHits</span><span class="p">[(</span><span class="n">xenopusHoxcHits</span><span class="o">.</span><span class="n">feature</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span><span class="p">(</span><span class="n">xenopusHoxcHits</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="s1">&#39;Scpiz6a_37&#39;</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;score&#39;</span><span class="p">,</span><span class="n">ascending</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[97]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>feature</th>
      <th>start</th>
      <th>end</th>
      <th>score</th>
      <th>strand</th>
      <th>phase</th>
      <th>attributes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>391</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9403550</td>
      <td>9404943</td>
      <td>346</td>
      <td>-</td>
      <td>.</td>
      <td>gene_id 1 ; sequence hoxc3-201 ; gene_orientation + ; identity 60.80 ; similarity 72.00</td>
    </tr>
    <tr>
      <th>400</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9473492</td>
      <td>9474480</td>
      <td>243</td>
      <td>-</td>
      <td>.</td>
      <td>gene_id 2 ; sequence hoxc3-201 ; gene_orientation + ; identity 40.85 ; similarity 56.34</td>
    </tr>
    <tr>
      <th>409</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9500001</td>
      <td>9500189</td>
      <td>228</td>
      <td>-</td>
      <td>.</td>
      <td>gene_id 3 ; sequence hoxc3-201 ; gene_orientation . ; identity 71.43 ; similarity 82.54</td>
    </tr>
    <tr>
      <th>413</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9504971</td>
      <td>9505144</td>
      <td>205</td>
      <td>-</td>
      <td>.</td>
      <td>gene_id 4 ; sequence hoxc3-201 ; gene_orientation . ; identity 67.24 ; similarity 81.03</td>
    </tr>
    <tr>
      <th>417</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9535775</td>
      <td>9535966</td>
      <td>203</td>
      <td>-</td>
      <td>.</td>
      <td>gene_id 5 ; sequence hoxc3-201 ; gene_orientation . ; identity 60.94 ; similarity 78.12</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [98]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data
<span class="c1">#cat Xenopus_tropicalis_hoxc3_to_marmoratus_genome.gff | awk &#39;$1==&quot;Scpiz6a_37&quot; &amp;&amp; $4==9403550 &amp;&amp; $3==&quot;gene&quot;&#39; | cut -f -5,7 | tr &#39;\n&#39; &#39;~&#39; | sed &quot;s/~/@Hoxc3~/&quot; | tr &#39;~&#39; &#39;\n&#39; | tr &#39;@&#39; &#39;\t&#39; &gt;&gt; hox_gene_coords.tsv</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [99]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">hoxExonCoords</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/hox_gene_coords.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;seqid&#39;</span><span class="p">,</span><span class="s1">&#39;source&#39;</span><span class="p">,</span><span class="s1">&#39;type&#39;</span><span class="p">,</span><span class="s1">&#39;start&#39;</span><span class="p">,</span><span class="s1">&#39;stop&#39;</span><span class="p">,</span><span class="s1">&#39;strand&#39;</span><span class="p">,</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
<span class="n">hoxExonCoords</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[99]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>type</th>
      <th>start</th>
      <th>stop</th>
      <th>strand</th>
      <th>name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9282120</td>
      <td>9288133</td>
      <td>-</td>
      <td>hoxa1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9474154</td>
      <td>9474759</td>
      <td>-</td>
      <td>HOXC4</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9499983</td>
      <td>9501230</td>
      <td>-</td>
      <td>Hoxc5</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9504857</td>
      <td>9506031</td>
      <td>-</td>
      <td>hoxc6</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9526250</td>
      <td>9528579</td>
      <td>-</td>
      <td>HOXC8</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9535748</td>
      <td>9538860</td>
      <td>-</td>
      <td>Hoxc9</td>
    </tr>
    <tr>
      <th>6</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9550771</td>
      <td>9554871</td>
      <td>-</td>
      <td>HOXC10</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9562146</td>
      <td>9568818</td>
      <td>-</td>
      <td>hoxc11a</td>
    </tr>
    <tr>
      <th>8</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9589583</td>
      <td>9593356</td>
      <td>-</td>
      <td>hoxc12a</td>
    </tr>
    <tr>
      <th>9</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9593673</td>
      <td>9594302</td>
      <td>-</td>
      <td>HOXC12</td>
    </tr>
    <tr>
      <th>10</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9606524</td>
      <td>9610151</td>
      <td>-</td>
      <td>hoxc13a</td>
    </tr>
    <tr>
      <th>11</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>18770207</td>
      <td>18772219</td>
      <td>-</td>
      <td>hoxM</td>
    </tr>
    <tr>
      <th>12</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>67494760</td>
      <td>67495389</td>
      <td>+</td>
      <td>HOXA4</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12112697</td>
      <td>12116784</td>
      <td>-</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>14</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12118635</td>
      <td>12119576</td>
      <td>-</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>15</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12199764</td>
      <td>12201101</td>
      <td>+</td>
      <td>HOXA13</td>
    </tr>
    <tr>
      <th>16</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12219716</td>
      <td>12221513</td>
      <td>+</td>
      <td>HOXA11</td>
    </tr>
    <tr>
      <th>17</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12233452</td>
      <td>12234549</td>
      <td>+</td>
      <td>HOXA10</td>
    </tr>
    <tr>
      <th>18</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12235671</td>
      <td>12236180</td>
      <td>+</td>
      <td>Hoxa10</td>
    </tr>
    <tr>
      <th>19</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12243941</td>
      <td>12246059</td>
      <td>+</td>
      <td>HOXA9</td>
    </tr>
    <tr>
      <th>20</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12254353</td>
      <td>12254898</td>
      <td>+</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>21</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12255141</td>
      <td>12256495</td>
      <td>+</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>22</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12263757</td>
      <td>12265990</td>
      <td>+</td>
      <td>HOXA6</td>
    </tr>
    <tr>
      <th>23</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12267708</td>
      <td>12269129</td>
      <td>+</td>
      <td>HOXA5</td>
    </tr>
    <tr>
      <th>24</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12282223</td>
      <td>12283761</td>
      <td>+</td>
      <td>HOXA4</td>
    </tr>
    <tr>
      <th>25</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12303335</td>
      <td>12306741</td>
      <td>+</td>
      <td>HOXA3</td>
    </tr>
    <tr>
      <th>26</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12312194</td>
      <td>12314293</td>
      <td>+</td>
      <td>HOXA2</td>
    </tr>
    <tr>
      <th>27</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12319290</td>
      <td>12323555</td>
      <td>+</td>
      <td>Hoxa1</td>
    </tr>
    <tr>
      <th>28</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>45958042</td>
      <td>45958654</td>
      <td>+</td>
      <td>HOX3</td>
    </tr>
    <tr>
      <th>29</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>13989488</td>
      <td>13990662</td>
      <td>+</td>
      <td>hoxU</td>
    </tr>
    <tr>
      <th>30</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>17298735</td>
      <td>17299199</td>
      <td>+</td>
      <td>hoxM</td>
    </tr>
    <tr>
      <th>31</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25629515</td>
      <td>25631702</td>
      <td>-</td>
      <td>Hoxd1</td>
    </tr>
    <tr>
      <th>32</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25663228</td>
      <td>25667221</td>
      <td>-</td>
      <td>hoxd3a</td>
    </tr>
    <tr>
      <th>33</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25692499</td>
      <td>25693667</td>
      <td>-</td>
      <td>HOXD4</td>
    </tr>
    <tr>
      <th>34</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25722293</td>
      <td>25724423</td>
      <td>-</td>
      <td>Hoxd8</td>
    </tr>
    <tr>
      <th>35</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25730347</td>
      <td>25732122</td>
      <td>-</td>
      <td>Hoxd9</td>
    </tr>
    <tr>
      <th>36</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25732652</td>
      <td>25739472</td>
      <td>-</td>
      <td>Hoxd10</td>
    </tr>
    <tr>
      <th>37</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25749545</td>
      <td>25750504</td>
      <td>-</td>
      <td>HOXD11</td>
    </tr>
    <tr>
      <th>38</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25758704</td>
      <td>25759860</td>
      <td>-</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>39</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25760003</td>
      <td>25760716</td>
      <td>-</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>40</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25770564</td>
      <td>25772996</td>
      <td>-</td>
      <td>HOXD13</td>
    </tr>
    <tr>
      <th>41</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25786550</td>
      <td>25789410</td>
      <td>+</td>
      <td>Evx2</td>
    </tr>
    <tr>
      <th>42</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6207845</td>
      <td>6211976</td>
      <td>+</td>
      <td>HOXB9</td>
    </tr>
    <tr>
      <th>43</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6217173</td>
      <td>6222005</td>
      <td>+</td>
      <td>HOXB8</td>
    </tr>
    <tr>
      <th>44</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6226105</td>
      <td>6226683</td>
      <td>+</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>45</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6227951</td>
      <td>6232453</td>
      <td>+</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>46</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6243660</td>
      <td>6246408</td>
      <td>+</td>
      <td>Hoxb6</td>
    </tr>
    <tr>
      <th>47</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6249716</td>
      <td>6250567</td>
      <td>+</td>
      <td>hoxb5a</td>
    </tr>
    <tr>
      <th>48</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6252113</td>
      <td>6263275</td>
      <td>+</td>
      <td>HOXB5</td>
    </tr>
    <tr>
      <th>49</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6272733</td>
      <td>6275177</td>
      <td>+</td>
      <td>HOXB4</td>
    </tr>
    <tr>
      <th>50</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6308863</td>
      <td>6317043</td>
      <td>+</td>
      <td>hoxb3a</td>
    </tr>
    <tr>
      <th>51</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6328011</td>
      <td>6330803</td>
      <td>+</td>
      <td>HOXB2</td>
    </tr>
    <tr>
      <th>52</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6356743</td>
      <td>6360890</td>
      <td>+</td>
      <td>HOXB1</td>
    </tr>
    <tr>
      <th>53</th>
      <td>Scpiz6a_86</td>
      <td>maker</td>
      <td>gene</td>
      <td>6098404</td>
      <td>6105626</td>
      <td>+</td>
      <td>Hoxb13</td>
    </tr>
    <tr>
      <th>54</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9403550</td>
      <td>9404943</td>
      <td>-</td>
      <td>Hoxc3</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Here I drop annotations that I accidentally picked up a</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [100]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">hoxExonCoords</span> <span class="o">=</span> <span class="n">hoxExonCoords</span><span class="p">[</span><span class="o">~</span><span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">name</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;hoxU&#39;</span><span class="p">,</span><span class="s1">&#39;hoxM&#39;</span><span class="p">,</span><span class="s1">&#39;HOX3&#39;</span><span class="p">])]</span>
<span class="n">hoxExonCoords</span><span class="p">[</span><span class="s1">&#39;factor_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">name</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">upper</span><span class="p">()[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="k">if</span> <span class="n">x</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="ow">not</span> <span class="ow">in</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">s</span><span class="p">)</span> <span class="k">for</span> <span class="n">s</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">20</span><span class="p">)]</span> <span class="k">else</span> <span class="n">x</span><span class="o">.</span><span class="n">upper</span><span class="p">())</span>
<span class="n">hoxExonCoords</span> <span class="o">=</span> <span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;seqid&#39;</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">hoxExonCoords</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="stderr output_area docutils container">
<div class="highlight"><pre>
/home/dut/anaconda3/envs/anaconda2/lib/python2.7/site-packages/ipykernel/__main__.py:2: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from ipykernel import kernelapp as app
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[100]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>type</th>
      <th>start</th>
      <th>stop</th>
      <th>strand</th>
      <th>name</th>
      <th>factor_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25629515</td>
      <td>25631702</td>
      <td>-</td>
      <td>Hoxd1</td>
      <td>HOXD1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25663228</td>
      <td>25667221</td>
      <td>-</td>
      <td>hoxd3a</td>
      <td>HOXD3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25692499</td>
      <td>25693667</td>
      <td>-</td>
      <td>HOXD4</td>
      <td>HOXD4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25722293</td>
      <td>25724423</td>
      <td>-</td>
      <td>Hoxd8</td>
      <td>HOXD8</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25730347</td>
      <td>25732122</td>
      <td>-</td>
      <td>Hoxd9</td>
      <td>HOXD9</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25732652</td>
      <td>25739472</td>
      <td>-</td>
      <td>Hoxd10</td>
      <td>HOXD10</td>
    </tr>
    <tr>
      <th>6</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25749545</td>
      <td>25750504</td>
      <td>-</td>
      <td>HOXD11</td>
      <td>HOXD11</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25758704</td>
      <td>25759860</td>
      <td>-</td>
      <td>HOXD12</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>8</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25760003</td>
      <td>25760716</td>
      <td>-</td>
      <td>HOXD12</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>9</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25770564</td>
      <td>25772996</td>
      <td>-</td>
      <td>HOXD13</td>
      <td>HOXD13</td>
    </tr>
    <tr>
      <th>10</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25786550</td>
      <td>25789410</td>
      <td>+</td>
      <td>Evx2</td>
      <td>EVX2</td>
    </tr>
    <tr>
      <th>11</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12112697</td>
      <td>12116784</td>
      <td>-</td>
      <td>EVX1</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>12</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12118635</td>
      <td>12119576</td>
      <td>-</td>
      <td>EVX1</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12199764</td>
      <td>12201101</td>
      <td>+</td>
      <td>HOXA13</td>
      <td>HOXA13</td>
    </tr>
    <tr>
      <th>14</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12219716</td>
      <td>12221513</td>
      <td>+</td>
      <td>HOXA11</td>
      <td>HOXA11</td>
    </tr>
    <tr>
      <th>15</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12233452</td>
      <td>12234549</td>
      <td>+</td>
      <td>HOXA10</td>
      <td>HOXA10</td>
    </tr>
    <tr>
      <th>16</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12235671</td>
      <td>12236180</td>
      <td>+</td>
      <td>Hoxa10</td>
      <td>HOXA10</td>
    </tr>
    <tr>
      <th>17</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12243941</td>
      <td>12246059</td>
      <td>+</td>
      <td>HOXA9</td>
      <td>HOXA9</td>
    </tr>
    <tr>
      <th>18</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12254353</td>
      <td>12254898</td>
      <td>+</td>
      <td>HOXA7</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>19</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12255141</td>
      <td>12256495</td>
      <td>+</td>
      <td>HOXA7</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>20</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12263757</td>
      <td>12265990</td>
      <td>+</td>
      <td>HOXA6</td>
      <td>HOXA6</td>
    </tr>
    <tr>
      <th>21</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12267708</td>
      <td>12269129</td>
      <td>+</td>
      <td>HOXA5</td>
      <td>HOXA5</td>
    </tr>
    <tr>
      <th>22</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12282223</td>
      <td>12283761</td>
      <td>+</td>
      <td>HOXA4</td>
      <td>HOXA4</td>
    </tr>
    <tr>
      <th>23</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12303335</td>
      <td>12306741</td>
      <td>+</td>
      <td>HOXA3</td>
      <td>HOXA3</td>
    </tr>
    <tr>
      <th>24</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12312194</td>
      <td>12314293</td>
      <td>+</td>
      <td>HOXA2</td>
      <td>HOXA2</td>
    </tr>
    <tr>
      <th>25</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12319290</td>
      <td>12323555</td>
      <td>+</td>
      <td>Hoxa1</td>
      <td>HOXA1</td>
    </tr>
    <tr>
      <th>26</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9282120</td>
      <td>9288133</td>
      <td>-</td>
      <td>hoxa1</td>
      <td>HOXA1</td>
    </tr>
    <tr>
      <th>27</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9403550</td>
      <td>9404943</td>
      <td>-</td>
      <td>Hoxc3</td>
      <td>HOXC3</td>
    </tr>
    <tr>
      <th>28</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9474154</td>
      <td>9474759</td>
      <td>-</td>
      <td>HOXC4</td>
      <td>HOXC4</td>
    </tr>
    <tr>
      <th>29</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9499983</td>
      <td>9501230</td>
      <td>-</td>
      <td>Hoxc5</td>
      <td>HOXC5</td>
    </tr>
    <tr>
      <th>30</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9504857</td>
      <td>9506031</td>
      <td>-</td>
      <td>hoxc6</td>
      <td>HOXC6</td>
    </tr>
    <tr>
      <th>31</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9526250</td>
      <td>9528579</td>
      <td>-</td>
      <td>HOXC8</td>
      <td>HOXC8</td>
    </tr>
    <tr>
      <th>32</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9535748</td>
      <td>9538860</td>
      <td>-</td>
      <td>Hoxc9</td>
      <td>HOXC9</td>
    </tr>
    <tr>
      <th>33</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9550771</td>
      <td>9554871</td>
      <td>-</td>
      <td>HOXC10</td>
      <td>HOXC10</td>
    </tr>
    <tr>
      <th>34</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9562146</td>
      <td>9568818</td>
      <td>-</td>
      <td>hoxc11a</td>
      <td>HOXC11</td>
    </tr>
    <tr>
      <th>35</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9589583</td>
      <td>9593356</td>
      <td>-</td>
      <td>hoxc12a</td>
      <td>HOXC12</td>
    </tr>
    <tr>
      <th>36</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9593673</td>
      <td>9594302</td>
      <td>-</td>
      <td>HOXC12</td>
      <td>HOXC12</td>
    </tr>
    <tr>
      <th>37</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9606524</td>
      <td>9610151</td>
      <td>-</td>
      <td>hoxc13a</td>
      <td>HOXC13</td>
    </tr>
    <tr>
      <th>38</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>67494760</td>
      <td>67495389</td>
      <td>+</td>
      <td>HOXA4</td>
      <td>HOXA4</td>
    </tr>
    <tr>
      <th>39</th>
      <td>Scpiz6a_86</td>
      <td>maker</td>
      <td>gene</td>
      <td>6098404</td>
      <td>6105626</td>
      <td>+</td>
      <td>Hoxb13</td>
      <td>HOXB13</td>
    </tr>
    <tr>
      <th>40</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6207845</td>
      <td>6211976</td>
      <td>+</td>
      <td>HOXB9</td>
      <td>HOXB9</td>
    </tr>
    <tr>
      <th>41</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6217173</td>
      <td>6222005</td>
      <td>+</td>
      <td>HOXB8</td>
      <td>HOXB8</td>
    </tr>
    <tr>
      <th>42</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6226105</td>
      <td>6226683</td>
      <td>+</td>
      <td>HOXB7</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>43</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6227951</td>
      <td>6232453</td>
      <td>+</td>
      <td>HOXB7</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>44</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6243660</td>
      <td>6246408</td>
      <td>+</td>
      <td>Hoxb6</td>
      <td>HOXB6</td>
    </tr>
    <tr>
      <th>45</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6249716</td>
      <td>6250567</td>
      <td>+</td>
      <td>hoxb5a</td>
      <td>HOXB5</td>
    </tr>
    <tr>
      <th>46</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6252113</td>
      <td>6263275</td>
      <td>+</td>
      <td>HOXB5</td>
      <td>HOXB5</td>
    </tr>
    <tr>
      <th>47</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6272733</td>
      <td>6275177</td>
      <td>+</td>
      <td>HOXB4</td>
      <td>HOXB4</td>
    </tr>
    <tr>
      <th>48</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6308863</td>
      <td>6317043</td>
      <td>+</td>
      <td>hoxb3a</td>
      <td>HOXB3</td>
    </tr>
    <tr>
      <th>49</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6328011</td>
      <td>6330803</td>
      <td>+</td>
      <td>HOXB2</td>
      <td>HOXB2</td>
    </tr>
    <tr>
      <th>50</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6356743</td>
      <td>6360890</td>
      <td>+</td>
      <td>HOXB1</td>
      <td>HOXB1</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>There are still certain HOX gene homologies present on scaffolds outside
of the clusters.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [101]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">hoxExonCoords</span> <span class="o">=</span> <span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">index</span><span class="p">[[</span><span class="mi">38</span><span class="p">,</span><span class="mi">26</span><span class="p">]])</span>
<span class="n">hoxExonCoords</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[101]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>type</th>
      <th>start</th>
      <th>stop</th>
      <th>strand</th>
      <th>name</th>
      <th>factor_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25629515</td>
      <td>25631702</td>
      <td>-</td>
      <td>Hoxd1</td>
      <td>HOXD1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25663228</td>
      <td>25667221</td>
      <td>-</td>
      <td>hoxd3a</td>
      <td>HOXD3</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25692499</td>
      <td>25693667</td>
      <td>-</td>
      <td>HOXD4</td>
      <td>HOXD4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25722293</td>
      <td>25724423</td>
      <td>-</td>
      <td>Hoxd8</td>
      <td>HOXD8</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25730347</td>
      <td>25732122</td>
      <td>-</td>
      <td>Hoxd9</td>
      <td>HOXD9</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25732652</td>
      <td>25739472</td>
      <td>-</td>
      <td>Hoxd10</td>
      <td>HOXD10</td>
    </tr>
    <tr>
      <th>6</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25749545</td>
      <td>25750504</td>
      <td>-</td>
      <td>HOXD11</td>
      <td>HOXD11</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25758704</td>
      <td>25759860</td>
      <td>-</td>
      <td>HOXD12</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>8</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25760003</td>
      <td>25760716</td>
      <td>-</td>
      <td>HOXD12</td>
      <td>HOXD12</td>
    </tr>
    <tr>
      <th>9</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25770564</td>
      <td>25772996</td>
      <td>-</td>
      <td>HOXD13</td>
      <td>HOXD13</td>
    </tr>
    <tr>
      <th>10</th>
      <td>Scpiz6a_1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>25786550</td>
      <td>25789410</td>
      <td>+</td>
      <td>Evx2</td>
      <td>EVX2</td>
    </tr>
    <tr>
      <th>11</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12112697</td>
      <td>12116784</td>
      <td>-</td>
      <td>EVX1</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>12</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12118635</td>
      <td>12119576</td>
      <td>-</td>
      <td>EVX1</td>
      <td>EVX1</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12199764</td>
      <td>12201101</td>
      <td>+</td>
      <td>HOXA13</td>
      <td>HOXA13</td>
    </tr>
    <tr>
      <th>14</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12219716</td>
      <td>12221513</td>
      <td>+</td>
      <td>HOXA11</td>
      <td>HOXA11</td>
    </tr>
    <tr>
      <th>15</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12233452</td>
      <td>12234549</td>
      <td>+</td>
      <td>HOXA10</td>
      <td>HOXA10</td>
    </tr>
    <tr>
      <th>16</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12235671</td>
      <td>12236180</td>
      <td>+</td>
      <td>Hoxa10</td>
      <td>HOXA10</td>
    </tr>
    <tr>
      <th>17</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12243941</td>
      <td>12246059</td>
      <td>+</td>
      <td>HOXA9</td>
      <td>HOXA9</td>
    </tr>
    <tr>
      <th>18</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12254353</td>
      <td>12254898</td>
      <td>+</td>
      <td>HOXA7</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>19</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12255141</td>
      <td>12256495</td>
      <td>+</td>
      <td>HOXA7</td>
      <td>HOXA7</td>
    </tr>
    <tr>
      <th>20</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12263757</td>
      <td>12265990</td>
      <td>+</td>
      <td>HOXA6</td>
      <td>HOXA6</td>
    </tr>
    <tr>
      <th>21</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12267708</td>
      <td>12269129</td>
      <td>+</td>
      <td>HOXA5</td>
      <td>HOXA5</td>
    </tr>
    <tr>
      <th>22</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12282223</td>
      <td>12283761</td>
      <td>+</td>
      <td>HOXA4</td>
      <td>HOXA4</td>
    </tr>
    <tr>
      <th>23</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12303335</td>
      <td>12306741</td>
      <td>+</td>
      <td>HOXA3</td>
      <td>HOXA3</td>
    </tr>
    <tr>
      <th>24</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12312194</td>
      <td>12314293</td>
      <td>+</td>
      <td>HOXA2</td>
      <td>HOXA2</td>
    </tr>
    <tr>
      <th>25</th>
      <td>Scpiz6a_30.1</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>12319290</td>
      <td>12323555</td>
      <td>+</td>
      <td>Hoxa1</td>
      <td>HOXA1</td>
    </tr>
    <tr>
      <th>27</th>
      <td>Scpiz6a_37</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>9403550</td>
      <td>9404943</td>
      <td>-</td>
      <td>Hoxc3</td>
      <td>HOXC3</td>
    </tr>
    <tr>
      <th>28</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9474154</td>
      <td>9474759</td>
      <td>-</td>
      <td>HOXC4</td>
      <td>HOXC4</td>
    </tr>
    <tr>
      <th>29</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9499983</td>
      <td>9501230</td>
      <td>-</td>
      <td>Hoxc5</td>
      <td>HOXC5</td>
    </tr>
    <tr>
      <th>30</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9504857</td>
      <td>9506031</td>
      <td>-</td>
      <td>hoxc6</td>
      <td>HOXC6</td>
    </tr>
    <tr>
      <th>31</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9526250</td>
      <td>9528579</td>
      <td>-</td>
      <td>HOXC8</td>
      <td>HOXC8</td>
    </tr>
    <tr>
      <th>32</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9535748</td>
      <td>9538860</td>
      <td>-</td>
      <td>Hoxc9</td>
      <td>HOXC9</td>
    </tr>
    <tr>
      <th>33</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9550771</td>
      <td>9554871</td>
      <td>-</td>
      <td>HOXC10</td>
      <td>HOXC10</td>
    </tr>
    <tr>
      <th>34</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9562146</td>
      <td>9568818</td>
      <td>-</td>
      <td>hoxc11a</td>
      <td>HOXC11</td>
    </tr>
    <tr>
      <th>35</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9589583</td>
      <td>9593356</td>
      <td>-</td>
      <td>hoxc12a</td>
      <td>HOXC12</td>
    </tr>
    <tr>
      <th>36</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9593673</td>
      <td>9594302</td>
      <td>-</td>
      <td>HOXC12</td>
      <td>HOXC12</td>
    </tr>
    <tr>
      <th>37</th>
      <td>Scpiz6a_37</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>9606524</td>
      <td>9610151</td>
      <td>-</td>
      <td>hoxc13a</td>
      <td>HOXC13</td>
    </tr>
    <tr>
      <th>39</th>
      <td>Scpiz6a_86</td>
      <td>maker</td>
      <td>gene</td>
      <td>6098404</td>
      <td>6105626</td>
      <td>+</td>
      <td>Hoxb13</td>
      <td>HOXB13</td>
    </tr>
    <tr>
      <th>40</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6207845</td>
      <td>6211976</td>
      <td>+</td>
      <td>HOXB9</td>
      <td>HOXB9</td>
    </tr>
    <tr>
      <th>41</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6217173</td>
      <td>6222005</td>
      <td>+</td>
      <td>HOXB8</td>
      <td>HOXB8</td>
    </tr>
    <tr>
      <th>42</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6226105</td>
      <td>6226683</td>
      <td>+</td>
      <td>HOXB7</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>43</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6227951</td>
      <td>6232453</td>
      <td>+</td>
      <td>HOXB7</td>
      <td>HOXB7</td>
    </tr>
    <tr>
      <th>44</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6243660</td>
      <td>6246408</td>
      <td>+</td>
      <td>Hoxb6</td>
      <td>HOXB6</td>
    </tr>
    <tr>
      <th>45</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6249716</td>
      <td>6250567</td>
      <td>+</td>
      <td>hoxb5a</td>
      <td>HOXB5</td>
    </tr>
    <tr>
      <th>46</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6252113</td>
      <td>6263275</td>
      <td>+</td>
      <td>HOXB5</td>
      <td>HOXB5</td>
    </tr>
    <tr>
      <th>47</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6272733</td>
      <td>6275177</td>
      <td>+</td>
      <td>HOXB4</td>
      <td>HOXB4</td>
    </tr>
    <tr>
      <th>48</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6308863</td>
      <td>6317043</td>
      <td>+</td>
      <td>hoxb3a</td>
      <td>HOXB3</td>
    </tr>
    <tr>
      <th>49</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6328011</td>
      <td>6330803</td>
      <td>+</td>
      <td>HOXB2</td>
      <td>HOXB2</td>
    </tr>
    <tr>
      <th>50</th>
      <td>Scpiz6a_86</td>
      <td>AUGUSTUS</td>
      <td>gene</td>
      <td>6356743</td>
      <td>6360890</td>
      <td>+</td>
      <td>HOXB1</td>
      <td>HOXB1</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Next I collapse all of the coordinates for similar gene models into a
single range of evidence for each HOX gene.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [102]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">collapsedCoordsList</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">hox_factor</span> <span class="ow">in</span> <span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">factor_name</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">hox_df</span> <span class="o">=</span> <span class="n">hoxExonCoords</span><span class="p">[</span><span class="n">hoxExonCoords</span><span class="o">.</span><span class="n">factor_name</span> <span class="o">==</span> <span class="n">hox_factor</span><span class="p">]</span>
    <span class="n">start</span> <span class="o">=</span> <span class="n">hox_df</span><span class="o">.</span><span class="n">start</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
    <span class="n">stop</span> <span class="o">=</span> <span class="n">hox_df</span><span class="o">.</span><span class="n">stop</span><span class="o">.</span><span class="n">max</span><span class="p">()</span>
    <span class="n">data_dict</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;seqid&#39;</span><span class="p">:</span><span class="n">hox_df</span><span class="o">.</span><span class="n">seqid</span><span class="o">.</span><span class="n">unique</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span> <span class="s1">&#39;gene&#39;</span><span class="p">:</span><span class="n">hox_factor</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">:</span><span class="n">start</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">:</span><span class="n">stop</span><span class="p">,</span> <span class="s1">&#39;strand&#39;</span><span class="p">:</span><span class="n">hox_df</span><span class="o">.</span><span class="n">strand</span><span class="o">.</span><span class="n">unique</span><span class="p">()[</span><span class="mi">0</span><span class="p">]}</span>
    <span class="n">collapsedCoordsList</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">data_dict</span><span class="p">)</span>

<span class="n">hoxGeneCoordsDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">collapsedCoordsList</span><span class="p">)</span>
<span class="n">hoxGeneCoordsDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[102]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene</th>
      <th>seqid</th>
      <th>start</th>
      <th>stop</th>
      <th>strand</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>HOXD1</td>
      <td>Scpiz6a_1</td>
      <td>25629515</td>
      <td>25631702</td>
      <td>-</td>
    </tr>
    <tr>
      <th>1</th>
      <td>HOXD3</td>
      <td>Scpiz6a_1</td>
      <td>25663228</td>
      <td>25667221</td>
      <td>-</td>
    </tr>
    <tr>
      <th>2</th>
      <td>HOXD4</td>
      <td>Scpiz6a_1</td>
      <td>25692499</td>
      <td>25693667</td>
      <td>-</td>
    </tr>
    <tr>
      <th>3</th>
      <td>HOXD8</td>
      <td>Scpiz6a_1</td>
      <td>25722293</td>
      <td>25724423</td>
      <td>-</td>
    </tr>
    <tr>
      <th>4</th>
      <td>HOXD9</td>
      <td>Scpiz6a_1</td>
      <td>25730347</td>
      <td>25732122</td>
      <td>-</td>
    </tr>
    <tr>
      <th>5</th>
      <td>HOXD10</td>
      <td>Scpiz6a_1</td>
      <td>25732652</td>
      <td>25739472</td>
      <td>-</td>
    </tr>
    <tr>
      <th>6</th>
      <td>HOXD11</td>
      <td>Scpiz6a_1</td>
      <td>25749545</td>
      <td>25750504</td>
      <td>-</td>
    </tr>
    <tr>
      <th>7</th>
      <td>HOXD12</td>
      <td>Scpiz6a_1</td>
      <td>25758704</td>
      <td>25760716</td>
      <td>-</td>
    </tr>
    <tr>
      <th>8</th>
      <td>HOXD13</td>
      <td>Scpiz6a_1</td>
      <td>25770564</td>
      <td>25772996</td>
      <td>-</td>
    </tr>
    <tr>
      <th>9</th>
      <td>EVX2</td>
      <td>Scpiz6a_1</td>
      <td>25786550</td>
      <td>25789410</td>
      <td>+</td>
    </tr>
    <tr>
      <th>10</th>
      <td>EVX1</td>
      <td>Scpiz6a_30.1</td>
      <td>12112697</td>
      <td>12119576</td>
      <td>-</td>
    </tr>
    <tr>
      <th>11</th>
      <td>HOXA13</td>
      <td>Scpiz6a_30.1</td>
      <td>12199764</td>
      <td>12201101</td>
      <td>+</td>
    </tr>
    <tr>
      <th>12</th>
      <td>HOXA11</td>
      <td>Scpiz6a_30.1</td>
      <td>12219716</td>
      <td>12221513</td>
      <td>+</td>
    </tr>
    <tr>
      <th>13</th>
      <td>HOXA10</td>
      <td>Scpiz6a_30.1</td>
      <td>12233452</td>
      <td>12236180</td>
      <td>+</td>
    </tr>
    <tr>
      <th>14</th>
      <td>HOXA9</td>
      <td>Scpiz6a_30.1</td>
      <td>12243941</td>
      <td>12246059</td>
      <td>+</td>
    </tr>
    <tr>
      <th>15</th>
      <td>HOXA7</td>
      <td>Scpiz6a_30.1</td>
      <td>12254353</td>
      <td>12256495</td>
      <td>+</td>
    </tr>
    <tr>
      <th>16</th>
      <td>HOXA6</td>
      <td>Scpiz6a_30.1</td>
      <td>12263757</td>
      <td>12265990</td>
      <td>+</td>
    </tr>
    <tr>
      <th>17</th>
      <td>HOXA5</td>
      <td>Scpiz6a_30.1</td>
      <td>12267708</td>
      <td>12269129</td>
      <td>+</td>
    </tr>
    <tr>
      <th>18</th>
      <td>HOXA4</td>
      <td>Scpiz6a_30.1</td>
      <td>12282223</td>
      <td>12283761</td>
      <td>+</td>
    </tr>
    <tr>
      <th>19</th>
      <td>HOXA3</td>
      <td>Scpiz6a_30.1</td>
      <td>12303335</td>
      <td>12306741</td>
      <td>+</td>
    </tr>
    <tr>
      <th>20</th>
      <td>HOXA2</td>
      <td>Scpiz6a_30.1</td>
      <td>12312194</td>
      <td>12314293</td>
      <td>+</td>
    </tr>
    <tr>
      <th>21</th>
      <td>HOXA1</td>
      <td>Scpiz6a_30.1</td>
      <td>12319290</td>
      <td>12323555</td>
      <td>+</td>
    </tr>
    <tr>
      <th>22</th>
      <td>HOXC3</td>
      <td>Scpiz6a_37</td>
      <td>9403550</td>
      <td>9404943</td>
      <td>-</td>
    </tr>
    <tr>
      <th>23</th>
      <td>HOXC4</td>
      <td>Scpiz6a_37</td>
      <td>9474154</td>
      <td>9474759</td>
      <td>-</td>
    </tr>
    <tr>
      <th>24</th>
      <td>HOXC5</td>
      <td>Scpiz6a_37</td>
      <td>9499983</td>
      <td>9501230</td>
      <td>-</td>
    </tr>
    <tr>
      <th>25</th>
      <td>HOXC6</td>
      <td>Scpiz6a_37</td>
      <td>9504857</td>
      <td>9506031</td>
      <td>-</td>
    </tr>
    <tr>
      <th>26</th>
      <td>HOXC8</td>
      <td>Scpiz6a_37</td>
      <td>9526250</td>
      <td>9528579</td>
      <td>-</td>
    </tr>
    <tr>
      <th>27</th>
      <td>HOXC9</td>
      <td>Scpiz6a_37</td>
      <td>9535748</td>
      <td>9538860</td>
      <td>-</td>
    </tr>
    <tr>
      <th>28</th>
      <td>HOXC10</td>
      <td>Scpiz6a_37</td>
      <td>9550771</td>
      <td>9554871</td>
      <td>-</td>
    </tr>
    <tr>
      <th>29</th>
      <td>HOXC11</td>
      <td>Scpiz6a_37</td>
      <td>9562146</td>
      <td>9568818</td>
      <td>-</td>
    </tr>
    <tr>
      <th>30</th>
      <td>HOXC12</td>
      <td>Scpiz6a_37</td>
      <td>9589583</td>
      <td>9594302</td>
      <td>-</td>
    </tr>
    <tr>
      <th>31</th>
      <td>HOXC13</td>
      <td>Scpiz6a_37</td>
      <td>9606524</td>
      <td>9610151</td>
      <td>-</td>
    </tr>
    <tr>
      <th>32</th>
      <td>HOXB13</td>
      <td>Scpiz6a_86</td>
      <td>6098404</td>
      <td>6105626</td>
      <td>+</td>
    </tr>
    <tr>
      <th>33</th>
      <td>HOXB9</td>
      <td>Scpiz6a_86</td>
      <td>6207845</td>
      <td>6211976</td>
      <td>+</td>
    </tr>
    <tr>
      <th>34</th>
      <td>HOXB8</td>
      <td>Scpiz6a_86</td>
      <td>6217173</td>
      <td>6222005</td>
      <td>+</td>
    </tr>
    <tr>
      <th>35</th>
      <td>HOXB7</td>
      <td>Scpiz6a_86</td>
      <td>6226105</td>
      <td>6232453</td>
      <td>+</td>
    </tr>
    <tr>
      <th>36</th>
      <td>HOXB6</td>
      <td>Scpiz6a_86</td>
      <td>6243660</td>
      <td>6246408</td>
      <td>+</td>
    </tr>
    <tr>
      <th>37</th>
      <td>HOXB5</td>
      <td>Scpiz6a_86</td>
      <td>6249716</td>
      <td>6263275</td>
      <td>+</td>
    </tr>
    <tr>
      <th>38</th>
      <td>HOXB4</td>
      <td>Scpiz6a_86</td>
      <td>6272733</td>
      <td>6275177</td>
      <td>+</td>
    </tr>
    <tr>
      <th>39</th>
      <td>HOXB3</td>
      <td>Scpiz6a_86</td>
      <td>6308863</td>
      <td>6317043</td>
      <td>+</td>
    </tr>
    <tr>
      <th>40</th>
      <td>HOXB2</td>
      <td>Scpiz6a_86</td>
      <td>6328011</td>
      <td>6330803</td>
      <td>+</td>
    </tr>
    <tr>
      <th>41</th>
      <td>HOXB1</td>
      <td>Scpiz6a_86</td>
      <td>6356743</td>
      <td>6360890</td>
      <td>+</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>I then need to normalize each interval so that they start at 0.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [103]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">scaffold</span> <span class="ow">in</span> <span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">min_start</span> <span class="o">=</span> <span class="n">hoxGeneCoordsDF</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span><span class="o">.</span><span class="n">start</span><span class="o">.</span><span class="n">min</span><span class="p">()</span>
    <span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">,</span> <span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">hoxGeneCoordsDF</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span><span class="o">.</span><span class="n">start</span> <span class="o">-</span> <span class="n">min_start</span>
    <span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">,</span> <span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">hoxGeneCoordsDF</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span><span class="o">.</span><span class="n">stop</span> <span class="o">-</span> <span class="n">min_start</span>
<span class="n">hoxGeneCoordsDF</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[103]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene</th>
      <th>seqid</th>
      <th>start</th>
      <th>stop</th>
      <th>strand</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>HOXD1</td>
      <td>Scpiz6a_1</td>
      <td>0</td>
      <td>2187</td>
      <td>-</td>
    </tr>
    <tr>
      <th>1</th>
      <td>HOXD3</td>
      <td>Scpiz6a_1</td>
      <td>33713</td>
      <td>37706</td>
      <td>-</td>
    </tr>
    <tr>
      <th>2</th>
      <td>HOXD4</td>
      <td>Scpiz6a_1</td>
      <td>62984</td>
      <td>64152</td>
      <td>-</td>
    </tr>
    <tr>
      <th>3</th>
      <td>HOXD8</td>
      <td>Scpiz6a_1</td>
      <td>92778</td>
      <td>94908</td>
      <td>-</td>
    </tr>
    <tr>
      <th>4</th>
      <td>HOXD9</td>
      <td>Scpiz6a_1</td>
      <td>100832</td>
      <td>102607</td>
      <td>-</td>
    </tr>
    <tr>
      <th>5</th>
      <td>HOXD10</td>
      <td>Scpiz6a_1</td>
      <td>103137</td>
      <td>109957</td>
      <td>-</td>
    </tr>
    <tr>
      <th>6</th>
      <td>HOXD11</td>
      <td>Scpiz6a_1</td>
      <td>120030</td>
      <td>120989</td>
      <td>-</td>
    </tr>
    <tr>
      <th>7</th>
      <td>HOXD12</td>
      <td>Scpiz6a_1</td>
      <td>129189</td>
      <td>131201</td>
      <td>-</td>
    </tr>
    <tr>
      <th>8</th>
      <td>HOXD13</td>
      <td>Scpiz6a_1</td>
      <td>141049</td>
      <td>143481</td>
      <td>-</td>
    </tr>
    <tr>
      <th>9</th>
      <td>EVX2</td>
      <td>Scpiz6a_1</td>
      <td>157035</td>
      <td>159895</td>
      <td>+</td>
    </tr>
    <tr>
      <th>10</th>
      <td>EVX1</td>
      <td>Scpiz6a_30.1</td>
      <td>0</td>
      <td>6879</td>
      <td>-</td>
    </tr>
    <tr>
      <th>11</th>
      <td>HOXA13</td>
      <td>Scpiz6a_30.1</td>
      <td>87067</td>
      <td>88404</td>
      <td>+</td>
    </tr>
    <tr>
      <th>12</th>
      <td>HOXA11</td>
      <td>Scpiz6a_30.1</td>
      <td>107019</td>
      <td>108816</td>
      <td>+</td>
    </tr>
    <tr>
      <th>13</th>
      <td>HOXA10</td>
      <td>Scpiz6a_30.1</td>
      <td>120755</td>
      <td>123483</td>
      <td>+</td>
    </tr>
    <tr>
      <th>14</th>
      <td>HOXA9</td>
      <td>Scpiz6a_30.1</td>
      <td>131244</td>
      <td>133362</td>
      <td>+</td>
    </tr>
    <tr>
      <th>15</th>
      <td>HOXA7</td>
      <td>Scpiz6a_30.1</td>
      <td>141656</td>
      <td>143798</td>
      <td>+</td>
    </tr>
    <tr>
      <th>16</th>
      <td>HOXA6</td>
      <td>Scpiz6a_30.1</td>
      <td>151060</td>
      <td>153293</td>
      <td>+</td>
    </tr>
    <tr>
      <th>17</th>
      <td>HOXA5</td>
      <td>Scpiz6a_30.1</td>
      <td>155011</td>
      <td>156432</td>
      <td>+</td>
    </tr>
    <tr>
      <th>18</th>
      <td>HOXA4</td>
      <td>Scpiz6a_30.1</td>
      <td>169526</td>
      <td>171064</td>
      <td>+</td>
    </tr>
    <tr>
      <th>19</th>
      <td>HOXA3</td>
      <td>Scpiz6a_30.1</td>
      <td>190638</td>
      <td>194044</td>
      <td>+</td>
    </tr>
    <tr>
      <th>20</th>
      <td>HOXA2</td>
      <td>Scpiz6a_30.1</td>
      <td>199497</td>
      <td>201596</td>
      <td>+</td>
    </tr>
    <tr>
      <th>21</th>
      <td>HOXA1</td>
      <td>Scpiz6a_30.1</td>
      <td>206593</td>
      <td>210858</td>
      <td>+</td>
    </tr>
    <tr>
      <th>22</th>
      <td>HOXC3</td>
      <td>Scpiz6a_37</td>
      <td>0</td>
      <td>1393</td>
      <td>-</td>
    </tr>
    <tr>
      <th>23</th>
      <td>HOXC4</td>
      <td>Scpiz6a_37</td>
      <td>70604</td>
      <td>71209</td>
      <td>-</td>
    </tr>
    <tr>
      <th>24</th>
      <td>HOXC5</td>
      <td>Scpiz6a_37</td>
      <td>96433</td>
      <td>97680</td>
      <td>-</td>
    </tr>
    <tr>
      <th>25</th>
      <td>HOXC6</td>
      <td>Scpiz6a_37</td>
      <td>101307</td>
      <td>102481</td>
      <td>-</td>
    </tr>
    <tr>
      <th>26</th>
      <td>HOXC8</td>
      <td>Scpiz6a_37</td>
      <td>122700</td>
      <td>125029</td>
      <td>-</td>
    </tr>
    <tr>
      <th>27</th>
      <td>HOXC9</td>
      <td>Scpiz6a_37</td>
      <td>132198</td>
      <td>135310</td>
      <td>-</td>
    </tr>
    <tr>
      <th>28</th>
      <td>HOXC10</td>
      <td>Scpiz6a_37</td>
      <td>147221</td>
      <td>151321</td>
      <td>-</td>
    </tr>
    <tr>
      <th>29</th>
      <td>HOXC11</td>
      <td>Scpiz6a_37</td>
      <td>158596</td>
      <td>165268</td>
      <td>-</td>
    </tr>
    <tr>
      <th>30</th>
      <td>HOXC12</td>
      <td>Scpiz6a_37</td>
      <td>186033</td>
      <td>190752</td>
      <td>-</td>
    </tr>
    <tr>
      <th>31</th>
      <td>HOXC13</td>
      <td>Scpiz6a_37</td>
      <td>202974</td>
      <td>206601</td>
      <td>-</td>
    </tr>
    <tr>
      <th>32</th>
      <td>HOXB13</td>
      <td>Scpiz6a_86</td>
      <td>0</td>
      <td>7222</td>
      <td>+</td>
    </tr>
    <tr>
      <th>33</th>
      <td>HOXB9</td>
      <td>Scpiz6a_86</td>
      <td>109441</td>
      <td>113572</td>
      <td>+</td>
    </tr>
    <tr>
      <th>34</th>
      <td>HOXB8</td>
      <td>Scpiz6a_86</td>
      <td>118769</td>
      <td>123601</td>
      <td>+</td>
    </tr>
    <tr>
      <th>35</th>
      <td>HOXB7</td>
      <td>Scpiz6a_86</td>
      <td>127701</td>
      <td>134049</td>
      <td>+</td>
    </tr>
    <tr>
      <th>36</th>
      <td>HOXB6</td>
      <td>Scpiz6a_86</td>
      <td>145256</td>
      <td>148004</td>
      <td>+</td>
    </tr>
    <tr>
      <th>37</th>
      <td>HOXB5</td>
      <td>Scpiz6a_86</td>
      <td>151312</td>
      <td>164871</td>
      <td>+</td>
    </tr>
    <tr>
      <th>38</th>
      <td>HOXB4</td>
      <td>Scpiz6a_86</td>
      <td>174329</td>
      <td>176773</td>
      <td>+</td>
    </tr>
    <tr>
      <th>39</th>
      <td>HOXB3</td>
      <td>Scpiz6a_86</td>
      <td>210459</td>
      <td>218639</td>
      <td>+</td>
    </tr>
    <tr>
      <th>40</th>
      <td>HOXB2</td>
      <td>Scpiz6a_86</td>
      <td>229607</td>
      <td>232399</td>
      <td>+</td>
    </tr>
    <tr>
      <th>41</th>
      <td>HOXB1</td>
      <td>Scpiz6a_86</td>
      <td>258339</td>
      <td>262486</td>
      <td>+</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [104]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">coord_to_square</span><span class="p">(</span><span class="n">coord</span><span class="p">,</span> <span class="n">level</span><span class="p">,</span> <span class="n">color</span><span class="p">,</span> <span class="n">width</span><span class="p">):</span>
    <span class="n">start</span> <span class="o">=</span> <span class="n">coord</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
    <span class="n">end</span> <span class="o">=</span> <span class="n">coord</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">polygon</span> <span class="o">=</span> <span class="n">PolygonPatch</span><span class="p">(</span>
        <span class="n">Polygon</span><span class="p">(</span>
            <span class="p">[(</span><span class="n">start</span><span class="p">,</span> <span class="n">level</span> <span class="o">-</span> <span class="n">width</span><span class="p">),</span>
             <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">level</span> <span class="o">+</span> <span class="n">width</span><span class="p">),</span>
             <span class="p">(</span><span class="n">end</span><span class="p">,</span> <span class="n">level</span> <span class="o">+</span> <span class="n">width</span><span class="p">),</span>
             <span class="p">(</span><span class="n">end</span><span class="p">,</span> <span class="n">level</span> <span class="o">-</span> <span class="n">width</span><span class="p">)]</span>
        <span class="p">),</span> <span class="n">color</span><span class="o">=</span><span class="n">color</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">polygon</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [105]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">x_buffer</span> <span class="o">=</span> <span class="mi">10000</span>
<span class="n">num_clusters</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>


<span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">style</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">3.5</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="n">num_clusters</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">base_cluster_length</span> <span class="o">=</span> <span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">stop</span><span class="o">.</span><span class="n">max</span><span class="p">()</span> <span class="o">+</span> <span class="n">x_buffer</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">scaffold</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="s1">&#39;Scpiz6a_30.1&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_86&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_37&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_1&#39;</span><span class="p">]):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">)</span>
    <span class="c1">#base_cluster_length = hoxGeneCoordsDF[hoxGeneCoordsDF.seqid == scaffold].stop.max() + x_buffer</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="n">x_buffer</span><span class="p">,</span> <span class="n">base_cluster_length</span> <span class="o">+</span> <span class="n">x_buffer</span><span class="p">)</span>
    <span class="n">cluster_patch</span> <span class="o">=</span> <span class="n">coord_to_square</span><span class="p">(</span><span class="n">coord</span> <span class="o">=</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="n">base_cluster_length</span><span class="p">),</span> <span class="n">level</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;lightgrey&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.15</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">cluster_patch</span><span class="p">)</span>
    <span class="n">counter</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">hoxGeneCoordsDF</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;+&#39;</span><span class="p">:</span>
            <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;firebrick&#39;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;midnightblue&#39;</span>
        <span class="n">evidence_square</span> <span class="o">=</span> <span class="n">coord_to_square</span><span class="p">(</span><span class="n">coord</span> <span class="o">=</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;stop&#39;</span><span class="p">]),</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.45</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">evidence_square</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">counter</span> <span class="o">%</span> <span class="mi">2</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">text_div</span> <span class="o">=</span> <span class="mf">15.0</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">text_div</span> <span class="o">=</span> <span class="mf">1.3</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;gene&#39;</span><span class="p">],</span> <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">((</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">])</span> <span class="o">/</span> <span class="n">text_div</span><span class="p">),</span>
                                   <span class="mf">4.0</span><span class="p">),</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">)</span>
        <span class="n">counter</span><span class="o">+=</span><span class="mi">1</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
    <span class="n">xticks</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">True</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">280000</span><span class="p">,</span><span class="mi">20000</span><span class="p">))</span>
<span class="n">x1kb_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="mi">1000</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">x1kb_labels</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;scale (KB)&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/SupplementalFigure3.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_157_0.png" src="_images/marmorata_genome_paper_figures_3_157_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [106]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
<span class="p">[</span><span class="s1">&#39;Scpiz6a_30.1&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_86&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_37&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_1&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[106]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>[&#39;Scpiz6a_30.1&#39;, &#39;Scpiz6a_86&#39;, &#39;Scpiz6a_37&#39;, &#39;Scpiz6a_1&#39;]
</pre></div>
</div>
</div>
</div>
<div class="section" id="GATK">
<h2>GATK<a class="headerlink" href="#GATK" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Running-GATK">
<h3>Running GATK<a class="headerlink" href="#Running-GATK" title="Permalink to this headline">¶</a></h3>
<p>The following python command line tool was used organize and run the
gatk pipeline on our computational server. The tool outputs shell
scripts that can be run directly from the commandline. In hindsight
making this tool was a bit of a waste of time, given that GATK has since
provided their own framework for running GATK</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/gatk6_pipeline.py</span>
<span class="c1">#!/usr/bin/env python</span>
<span class="c1">#Author: Duncan Tormey</span>
<span class="c1">#Email: dut@stowers.org or duncantormey@gmail.com</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">datetime</span>
<span class="kn">import</span> <span class="nn">argparse</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>
<span class="n">sys</span><span class="o">.</span><span class="n">path</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;/home/dut/bin/python_scripts/&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="n">sys</span><span class="o">.</span><span class="n">path</span>
<span class="kn">import</span> <span class="nn">lims_classes</span> <span class="kn">as</span> <span class="nn">lims</span>


<span class="k">def</span> <span class="nf">make_script_fh</span><span class="p">(</span><span class="n">base_name</span><span class="p">):</span>
    <span class="n">date</span> <span class="o">=</span> <span class="n">datetime</span><span class="o">.</span><span class="n">datetime</span><span class="o">.</span><span class="n">today</span><span class="p">()</span>
    <span class="n">date</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">date</span><span class="p">)</span>
    <span class="n">date</span> <span class="o">=</span> <span class="n">date</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span>
    <span class="n">name</span> <span class="o">=</span> <span class="s1">&#39;./</span><span class="si">%s</span><span class="s1">_</span><span class="si">%s</span><span class="s1">.sh&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">base_name</span><span class="p">,</span> <span class="n">date</span><span class="p">)</span>
    <span class="n">fh</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">fh</span><span class="p">,</span> <span class="n">name</span>


<span class="k">def</span> <span class="nf">make_output_dir</span><span class="p">(</span><span class="n">directory_name</span><span class="p">):</span>
    <span class="n">working_directory</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">getcwd</span><span class="p">()</span>
    <span class="n">output_directory</span> <span class="o">=</span> <span class="n">working_directory</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">directory_name</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">output_directory</span><span class="p">):</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">output_directory</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">output_directory</span>


<span class="k">def</span> <span class="nf">make_proc_capture</span><span class="p">(</span><span class="n">i</span><span class="p">):</span>
    <span class="n">proc_capture</span> <span class="o">=</span> <span class="s1">&#39;proc</span><span class="si">%s</span><span class="s1">=$!</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="n">proc</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\&quot;</span><span class="s1">$proc</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span>


<span class="k">def</span> <span class="nf">prep_reference_commands</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">,</span>  <span class="n">picard_tools_path</span><span class="p">,</span> <span class="n">write_prep_ref</span><span class="p">):</span>
    <span class="n">output_dir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="s2">&quot;reference&quot;</span><span class="p">)</span>
    <span class="n">fasta_name</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">)</span>
    <span class="n">fasta_ref_path</span> <span class="o">=</span> <span class="n">output_dir</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">fasta_name</span>
    <span class="n">fasta_dict_path</span> <span class="o">=</span> <span class="n">output_dir</span> <span class="o">+</span> <span class="s2">&quot;/&quot;</span> <span class="o">+</span> <span class="n">fasta_name</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s2">&quot;.fa&quot;</span><span class="p">,</span> <span class="s2">&quot;.dict&quot;</span><span class="p">)</span>
    <span class="n">create_seq_dict</span> <span class="o">=</span> <span class="n">picard_tools_path</span> <span class="o">+</span> <span class="s1">&#39;CreateSequenceDictionary.jar&#39;</span>
    <span class="n">sym_link</span> <span class="o">=</span> <span class="s2">&quot;ln -s </span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">/</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">fasta_path</span><span class="p">,</span> <span class="n">output_dir</span><span class="p">)</span>
    <span class="n">bwa_index</span> <span class="o">=</span> <span class="s2">&quot;bwa index -a bwtsw </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">fasta_ref_path</span>
    <span class="n">samtools_faidx</span> <span class="o">=</span> <span class="s2">&quot;samtools faidx </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">fasta_ref_path</span>
    <span class="n">sequence_dir</span> <span class="o">=</span> <span class="s2">&quot;java -jar </span><span class="si">%s</span><span class="s2"> REFERENCE=</span><span class="si">%s</span><span class="s2"> OUTPUT=</span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">create_seq_dict</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">fasta_dict_path</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">write_prep_ref</span><span class="p">:</span>
        <span class="n">fh</span><span class="p">,</span> <span class="n">scriptname</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="s2">&quot;gatk_0_prep_ref&quot;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;#!/bin/bash</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">sym_link</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">bwa_index</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">samtools_faidx</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">sequence_dir</span><span class="p">)</span>

        <span class="n">fh</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

    <span class="k">return</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">scriptname</span>


<span class="k">def</span> <span class="nf">ret_down_sample_read_pairs</span><span class="p">(</span><span class="n">pairs</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">dwn_frac</span><span class="p">,</span><span class="n">outdir</span><span class="p">):</span>
    <span class="c1">#print(pairs)</span>
    <span class="n">dwn_frac</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">dwn_frac</span><span class="p">)</span>
    <span class="n">r_pair</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">path</span>
    <span class="n">f_pair</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">path</span>
    <span class="n">seed</span><span class="o">=</span><span class="nb">sum</span><span class="p">([</span><span class="nb">ord</span><span class="p">(</span><span class="n">a</span><span class="p">)</span> <span class="k">for</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">sample</span><span class="p">])</span>
    <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">path</span> <span class="o">=</span> <span class="n">outdir</span> <span class="o">+</span> <span class="s1">&#39;/&#39;</span> <span class="o">+</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">r_pair</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.fq.gz&#39;</span><span class="p">,</span> <span class="s1">&#39;.</span><span class="si">%s</span><span class="s1">ds.fq&#39;</span><span class="o">%</span><span class="k">str</span>(dwn_frac)).replace(&#39;.fastq.gz&#39;, &#39;.%sds.fastq&#39;%str(dwn_frac))
    <span class="n">pairs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">path</span> <span class="o">=</span> <span class="n">outdir</span> <span class="o">+</span> <span class="s1">&#39;/&#39;</span> <span class="o">+</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">f_pair</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.fq.gz&#39;</span><span class="p">,</span> <span class="s1">&#39;.</span><span class="si">%s</span><span class="s1">ds.fq&#39;</span><span class="o">%</span><span class="k">str</span>(dwn_frac)).replace(&#39;.fastq.gz&#39;, &#39;.%sds.fastq&#39;%str(dwn_frac))
    <span class="n">dwn_sample_cmd</span> <span class="o">=</span> <span class="s1">&#39;seqtk sample -s</span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> &gt; </span><span class="si">%s</span><span class="s1">; seqtk sample -s</span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> &gt; </span><span class="si">%s</span><span class="s1"> &amp;</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">seed</span><span class="p">,</span> <span class="n">r_pair</span><span class="p">,</span> <span class="n">dwn_frac</span><span class="p">,</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">path</span><span class="p">,</span>
                                                                                         <span class="n">seed</span><span class="p">,</span> <span class="n">f_pair</span><span class="p">,</span> <span class="n">dwn_frac</span><span class="p">,</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">path</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">dwn_sample_cmd</span><span class="p">,</span> <span class="n">pairs</span>

<span class="k">def</span> <span class="nf">down_sample_cmds</span><span class="p">(</span><span class="n">lane_pairs</span><span class="p">,</span> <span class="n">down_sample</span><span class="p">,</span> <span class="n">write</span><span class="p">,</span> <span class="n">write_prefix</span><span class="p">,</span> <span class="n">outdir</span><span class="p">):</span>
    <span class="k">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n\n</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="k">print</span><span class="p">(</span><span class="n">lane_pairs</span><span class="p">)</span>
    <span class="k">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n\n</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="n">outdir</span><span class="p">)</span>
    <span class="n">down_samples</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">down_sample</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="p">,</span> <span class="n">scriptname</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="n">write_prefix</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;#!/bin/bash</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">n</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">all_procs</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">lane_pairs</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">down_samples</span><span class="p">:</span>
            <span class="k">for</span>  <span class="n">lane</span><span class="p">,</span> <span class="n">pairs</span> <span class="ow">in</span> <span class="n">lane_pairs</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
                <span class="k">print</span><span class="p">(</span><span class="n">sample</span><span class="p">)</span>
                <span class="k">print</span><span class="p">(</span><span class="n">lane</span><span class="p">)</span>
                <span class="k">print</span><span class="p">(</span><span class="n">pairs</span><span class="p">)</span>
                <span class="n">cmd</span><span class="p">,</span> <span class="n">pairs</span> <span class="o">=</span> <span class="n">ret_down_sample_read_pairs</span><span class="p">(</span><span class="n">pairs</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">down_samples</span><span class="p">[</span><span class="n">sample</span><span class="p">],</span> <span class="n">outdir</span><span class="p">)</span>

                <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">n</span><span class="p">)</span>
                <span class="n">all_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
                <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">cmd</span><span class="p">)</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>

                <span class="n">n</span><span class="o">+=</span><span class="mi">1</span>
                <span class="n">lane_pairs</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="n">lane</span><span class="p">]</span> <span class="o">=</span> <span class="n">pairs</span>
    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">all_procs</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">lane_pairs</span><span class="p">,</span> <span class="n">scriptname</span>

<span class="k">def</span> <span class="nf">ret_bwa_align_sort_cmd</span><span class="p">(</span><span class="n">pairs</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">):</span>
    <span class="n">cpus</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">cpus</span><span class="p">)</span>
    <span class="n">r_pair</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">path</span>
    <span class="n">f_pair</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">path</span>
    <span class="n">lane_id</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">lane_id</span>
    <span class="n">read_group</span> <span class="o">=</span> <span class="n">pairs</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">read_group</span>

    <span class="n">output_bam</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.bam&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">lane_id</span><span class="p">)</span>
    <span class="n">bwa_command</span> <span class="o">=</span> <span class="s2">&quot;bwa mem -M -R </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s2"> -t </span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">read_group</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">r_pair</span><span class="p">,</span> <span class="n">f_pair</span><span class="p">)</span>
    <span class="n">sam_to_sorted_bam</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;samtools view -Sb -@ </span><span class="si">%s</span><span class="s1"> - | &#39;</span>
                         <span class="s1">&#39;samtools sort -o -@ </span><span class="si">%s</span><span class="s1"> - </span><span class="si">%s</span><span class="s1"> &gt; </span><span class="si">%s</span><span class="s1">&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">cpus</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">lane_id</span><span class="p">,</span> <span class="n">output_bam</span><span class="p">)</span>

    <span class="n">align_sort_cmd</span> <span class="o">=</span> <span class="s2">&quot;(</span><span class="si">%s</span><span class="s2"> | </span><span class="si">%s</span><span class="s2">) &amp;</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">bwa_command</span><span class="p">,</span> <span class="n">sam_to_sorted_bam</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">align_sort_cmd</span><span class="p">,</span> <span class="n">output_bam</span>


<span class="k">def</span> <span class="nf">ret_dedup_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span>  <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">picard_tools_path</span><span class="p">):</span>
    <span class="n">dedup</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/MarkDuplicates.jar&#39;</span> <span class="o">%</span> <span class="n">picard_tools_path</span>
    <span class="n">dedup_basename</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">sample_input</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>
    <span class="n">dedup_metric</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">_metrics.txt&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">dedup_basename</span><span class="p">)</span>
    <span class="n">dedup_output</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.dedup.bam&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">dedup_basename</span><span class="p">)</span>
    <span class="n">dedup_command</span> <span class="o">=</span> <span class="s2">&quot;java -Xmx4g -jar </span><span class="si">%s</span><span class="s2"> I=</span><span class="si">%s</span><span class="s2"> O=</span><span class="si">%s</span><span class="s2"> M=</span><span class="si">%s</span><span class="s2"> &amp;</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">dedup</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">dedup_output</span><span class="p">,</span> <span class="n">dedup_metric</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">dedup_command</span><span class="p">,</span> <span class="n">dedup_output</span>


<span class="k">def</span> <span class="nf">ret_merge_bams_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span>  <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">tempdir</span><span class="p">):</span>
    <span class="n">cpus</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">cpus</span><span class="p">)</span>
    <span class="n">bam_files</span> <span class="o">=</span> <span class="s1">&#39; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">sample_input</span><span class="p">)</span>
    <span class="n">temp_file</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.temp&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">tempdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">)</span>
    <span class="n">out_bam</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.merged.bam&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">)</span>
    <span class="n">merge_cmd</span> <span class="o">=</span> <span class="p">(</span><span class="s1">&#39;(samtools merge -@ </span><span class="si">%s</span><span class="s1"> - </span><span class="si">%s</span><span class="s1"> | samtools sort &#39;</span>
                 <span class="s1">&#39;- -m 10G -@ </span><span class="si">%s</span><span class="s1"> -T </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">; samtools index </span><span class="si">%s</span><span class="s1">) &amp;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span> <span class="o">%</span> <span class="p">(</span>
                     <span class="n">cpus</span><span class="p">,</span> <span class="n">bam_files</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">temp_file</span><span class="p">,</span> <span class="n">out_bam</span><span class="p">,</span> <span class="n">out_bam</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">merge_cmd</span><span class="p">,</span> <span class="n">out_bam</span>


<span class="k">def</span> <span class="nf">ret_index_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">):</span>
    <span class="n">index_cmd</span> <span class="o">=</span> <span class="s1">&#39;samtools index -b </span><span class="si">%s</span><span class="s1"> &amp;</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">sample_input</span>

    <span class="k">return</span> <span class="n">index_cmd</span><span class="p">,</span> <span class="n">sample_input</span>


<span class="k">def</span> <span class="nf">ret_realign_indels_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">):</span>
    <span class="n">realign_basename</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">sample_input</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>
    <span class="n">realign_list</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">_target_intervals.list&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">realign_basename</span><span class="p">)</span>
    <span class="n">out_bam</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.realigned.bam&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">realign_basename</span><span class="p">)</span>
    <span class="n">targets_cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T RealignerTargetCreator -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1"> -nt </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">realign_list</span><span class="p">,</span> <span class="n">cpus</span><span class="p">)</span>

    <span class="n">raln_cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T IndelRealigner -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> -targetIntervals </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">realign_list</span><span class="p">,</span> <span class="n">out_bam</span><span class="p">)</span>

    <span class="n">target_realign_cmd</span> <span class="o">=</span> <span class="s1">&#39;(</span><span class="si">%s</span><span class="s1">; </span><span class="si">%s</span><span class="s1">) &amp;</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">targets_cmd</span><span class="p">,</span> <span class="n">raln_cmd</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">target_realign_cmd</span><span class="p">,</span> <span class="n">out_bam</span>


<span class="k">def</span> <span class="nf">ret_haplotype_caller_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span>
                             <span class="n">gatk_path</span><span class="p">,</span> <span class="n">gatk_args</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span> <span class="n">param_string</span><span class="o">=</span><span class="bp">None</span><span class="p">):</span>
    <span class="n">out_vars</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.</span><span class="si">%s</span><span class="s1">_raw_var.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="k">if</span> <span class="n">param_string</span><span class="p">:</span>
        <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T HaplotypeCaller </span><span class="si">%s</span><span class="s1"> -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
            <span class="n">gatk_path</span><span class="p">,</span> <span class="n">param_string</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span>
            <span class="n">sample_input</span><span class="p">,</span> <span class="n">gatk_args</span><span class="p">,</span> <span class="n">out_vars</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T HaplotypeCaller -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
            <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">gatk_args</span><span class="p">,</span> <span class="n">out_vars</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_vars</span>


<span class="k">def</span> <span class="nf">ret_extract_snps_cmd</span><span class="p">(</span><span class="n">input_vars</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span>
                         <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_snps</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.</span><span class="si">%s</span><span class="s1">_raw_snps.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T SelectVariants -R </span><span class="si">%s</span><span class="s1"> -V </span><span class="si">%s</span><span class="s1"> -selectType SNP -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">input_vars</span><span class="p">,</span> <span class="n">out_snps</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_snps</span>


<span class="k">def</span> <span class="nf">ret_filter_snps_cmd</span><span class="p">(</span><span class="n">input_snps</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">snp_filter</span><span class="p">,</span>
                        <span class="n">snp_filter_name</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_snps</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.</span><span class="si">%s</span><span class="s1">_filt_snps.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T VariantFiltration -R </span><span class="si">%s</span><span class="s1"> -V </span><span class="si">%s</span><span class="s1"> --filterExpression </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s1"> --filterName </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span> <span class="n">snp_filter</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="p">,</span> <span class="n">out_snps</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_snps</span>


<span class="k">def</span> <span class="nf">ret_extract_indels_cmd</span><span class="p">(</span><span class="n">input_vars</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span>
                           <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_indels</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.</span><span class="si">%s</span><span class="s1">_raw_indels.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T SelectVariants -R </span><span class="si">%s</span><span class="s1"> -V </span><span class="si">%s</span><span class="s1"> -selectType INDEL -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">input_vars</span><span class="p">,</span> <span class="n">out_indels</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_indels</span>


<span class="k">def</span> <span class="nf">ret_filter_indels_cmd</span><span class="p">(</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">indel_filter</span><span class="p">,</span>
                          <span class="n">indel_filter_name</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_indels</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">.</span><span class="si">%s</span><span class="s1">_filt_indels.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T VariantFiltration -R </span><span class="si">%s</span><span class="s1"> -V </span><span class="si">%s</span><span class="s1"> --filterExpression </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s1"> --filterName </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">input_indels</span><span class="p">,</span> <span class="n">indel_filter</span><span class="p">,</span>
        <span class="n">indel_filter_name</span><span class="p">,</span> <span class="n">out_indels</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_indels</span>


<span class="k">def</span> <span class="nf">ret_combine_snps_cmd</span><span class="p">(</span><span class="n">all_snps</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_snps</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/all_snps_</span><span class="si">%s</span><span class="s1">.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -R </span><span class="si">%s</span><span class="s1"> -T CombineVariants --variant </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1"> --excludeNonVariants --minimumN 1&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="s1">&#39; --variant &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">all_snps</span><span class="p">),</span> <span class="n">out_snps</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_snps</span>


<span class="k">def</span> <span class="nf">ret_combine_indels_cmd</span><span class="p">(</span><span class="n">all_indels</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_indels</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/all_indels_</span><span class="si">%s</span><span class="s1">.vcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -R </span><span class="si">%s</span><span class="s1"> -T CombineVariants --variant </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1"> --excludeNonVariants --minimumN 1&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="s1">&#39; --variant &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">all_indels</span><span class="p">),</span> <span class="n">out_indels</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_indels</span>


<span class="k">def</span> <span class="nf">ret_vcf_tools_snp_cmd</span><span class="p">(</span><span class="n">input_snps</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="p">):</span>
    <span class="n">out_snps</span> <span class="o">=</span> <span class="n">input_snps</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.vcf&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>

    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;vcftools --vcf </span><span class="si">%s</span><span class="s1"> --remove-filtered LowQual --remove-filtered </span><span class="si">%s</span><span class="s1"> --recode --recode-INFO-all --out </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">input_snps</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="p">,</span> <span class="n">out_snps</span><span class="p">)</span>

    <span class="n">out_snps</span> <span class="o">=</span> <span class="n">out_snps</span> <span class="o">+</span> <span class="s1">&#39;.recode.vcf&#39;</span>
    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_snps</span>


<span class="k">def</span> <span class="nf">ret_vcf_tools_indel_cmd</span><span class="p">(</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="p">):</span>
    <span class="n">out_indels</span> <span class="o">=</span> <span class="n">input_indels</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.vcf&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;vcftools --vcf </span><span class="si">%s</span><span class="s1"> --remove-filtered LowQual --remove-filtered </span><span class="si">%s</span><span class="s1"> --recode --recode-INFO-all --out </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">input_indels</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="p">,</span> <span class="n">out_indels</span><span class="p">)</span>

    <span class="n">out_indels</span> <span class="o">=</span> <span class="n">out_indels</span> <span class="o">+</span> <span class="s1">&#39;.recode.vcf&#39;</span>
    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_indels</span>


<span class="k">def</span> <span class="nf">ret_first_recal_cmd</span><span class="p">(</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span>
                        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">before_table</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">_</span><span class="si">%s</span><span class="s1">.before_table&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T BaseRecalibrator -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> -knownSites </span><span class="si">%s</span><span class="s1"> -knownSites </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">input_indels</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span>
        <span class="n">before_table</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">before_table</span>


<span class="k">def</span> <span class="nf">ret_print_reads_cmd</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">recal_table</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span>
                        <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">out_bam</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">sample_input</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span>
        <span class="s1">&#39;.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;.recal_</span><span class="si">%s</span><span class="s1">.bam&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">out_bam</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">out_bam</span><span class="p">)</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T PrintReads -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> -BQSR </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">recal_table</span><span class="p">,</span> <span class="n">out_bam</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">out_bam</span>


<span class="k">def</span> <span class="nf">ret_second_recal_cmd</span><span class="p">(</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">before_table</span><span class="p">,</span>
                         <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">after_table</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">_</span><span class="si">%s</span><span class="s1">.after_table&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T BaseRecalibrator -R </span><span class="si">%s</span><span class="s1"> -I </span><span class="si">%s</span><span class="s1"> -knownSites </span><span class="si">%s</span><span class="s1"> -knownSites </span><span class="si">%s</span><span class="s1"> -BQSR </span><span class="si">%s</span><span class="s1"> -o </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">sample_input</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span> <span class="n">input_indels</span><span class="p">,</span>
        <span class="n">before_table</span><span class="p">,</span> <span class="n">after_table</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">after_table</span>


<span class="k">def</span> <span class="nf">ret_recal_plot_cmd</span><span class="p">(</span><span class="n">before_table</span><span class="p">,</span> <span class="n">after_table</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span>
                       <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">recal_plot</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/</span><span class="si">%s</span><span class="s1">_</span><span class="si">%s</span><span class="s1">.plots&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T AnalyzeCovariates -R </span><span class="si">%s</span><span class="s1"> -before </span><span class="si">%s</span><span class="s1"> -after </span><span class="si">%s</span><span class="s1"> -plots </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">before_table</span><span class="p">,</span> <span class="n">after_table</span><span class="p">,</span> <span class="n">recal_plot</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span>


<span class="k">def</span> <span class="nf">ret_genotype_gvcf_command</span><span class="p">(</span><span class="n">sample_inputs</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">outdir</span><span class="p">):</span>
    <span class="n">all_inputs</span> <span class="o">=</span> <span class="s1">&#39;--variant </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="s1">&#39; --variant &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
        <span class="p">[</span><span class="n">item</span> <span class="k">for</span> <span class="n">sublist</span> <span class="ow">in</span> <span class="n">sample_inputs</span><span class="o">.</span><span class="n">values</span><span class="p">()</span> <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">sublist</span><span class="p">])</span>
    <span class="n">output_gvcf</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1">/jg_</span><span class="si">%s</span><span class="s1">.gvcf&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">outdir</span><span class="p">,</span> <span class="s1">&#39;_&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">sample_inputs</span><span class="o">.</span><span class="n">keys</span><span class="p">()))</span>
    <span class="n">cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> -T GenotypeGVCFs -nt </span><span class="si">%s</span><span class="s1"> -ploidy 2 -R </span><span class="si">%s</span><span class="s1"> </span><span class="si">%s</span><span class="s1"> -allSites -o </span><span class="si">%s</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">gatk_path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">all_inputs</span><span class="p">,</span> <span class="n">output_gvcf</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">cmd</span><span class="p">,</span> <span class="n">output_gvcf</span>


<span class="k">def</span> <span class="nf">ret_hc_call_cmds</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span> <span class="n">param_string</span><span class="o">=</span><span class="bp">None</span><span class="p">):</span>
    <span class="n">gatk_args</span> <span class="o">=</span> <span class="s2">&quot; -stand_call_conf 30 -stand_emit_conf 30 -mbq 17 &quot;</span>
    <span class="n">snp_filter</span> <span class="o">=</span> <span class="s1">&#39;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#39;</span>
    <span class="n">indel_filter</span> <span class="o">=</span> <span class="s1">&#39;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#39;</span>
    <span class="n">indel_filter_name</span> <span class="o">=</span> <span class="s1">&#39;default_indel_filter&#39;</span>
    <span class="n">snp_filter_name</span> <span class="o">=</span> <span class="s1">&#39;default_snp_filter&#39;</span>
    <span class="n">haplotype_caller_cmd</span><span class="p">,</span> <span class="n">out_vars</span> <span class="o">=</span> <span class="n">ret_haplotype_caller_cmd</span><span class="p">(</span>
        <span class="n">sample_input</span><span class="o">=</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">gatk_args</span><span class="o">=</span><span class="n">gatk_args</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">,</span> <span class="n">param_string</span><span class="o">=</span><span class="n">param_string</span><span class="p">)</span>

    <span class="n">extract_snps_cmd</span><span class="p">,</span> <span class="n">raw_snps</span> <span class="o">=</span> <span class="n">ret_extract_snps_cmd</span><span class="p">(</span>
        <span class="n">input_vars</span><span class="o">=</span><span class="n">out_vars</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">filter_snps_cmd</span><span class="p">,</span> <span class="n">out_snps</span> <span class="o">=</span> <span class="n">ret_filter_snps_cmd</span><span class="p">(</span>
        <span class="n">input_snps</span><span class="o">=</span><span class="n">raw_snps</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">snp_filter</span><span class="o">=</span><span class="n">snp_filter</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="o">=</span><span class="n">snp_filter_name</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">extract_indels_cmd</span><span class="p">,</span> <span class="n">raw_indels</span> <span class="o">=</span> <span class="n">ret_extract_indels_cmd</span><span class="p">(</span>
        <span class="n">input_vars</span><span class="o">=</span><span class="n">out_vars</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">filter_indels_cmd</span><span class="p">,</span> <span class="n">out_indels</span> <span class="o">=</span> <span class="n">ret_filter_indels_cmd</span><span class="p">(</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">raw_indels</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">indel_filter</span><span class="o">=</span><span class="n">indel_filter</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="o">=</span><span class="n">indel_filter_name</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">hc_commands</span> <span class="o">=</span> <span class="s1">&#39;(</span><span class="si">%s</span><span class="s1">) &amp;</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="s1">&#39;; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
        <span class="p">[</span><span class="n">haplotype_caller_cmd</span><span class="p">,</span> <span class="n">extract_snps_cmd</span><span class="p">,</span> <span class="n">filter_snps_cmd</span><span class="p">,</span>
         <span class="n">extract_indels_cmd</span><span class="p">,</span> <span class="n">filter_indels_cmd</span><span class="p">])</span>

    <span class="k">return</span> <span class="n">hc_commands</span><span class="p">,</span> <span class="n">out_snps</span><span class="p">,</span> <span class="n">out_indels</span>


<span class="k">def</span> <span class="nf">ret_recode_indels_and_snps_cmd</span><span class="p">(</span><span class="n">all_snps</span><span class="p">,</span> <span class="n">all_indels</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="p">,</span>
                                   <span class="n">indel_filter_name</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span>
                                   <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>

    <span class="n">combine_snps_cmd</span><span class="p">,</span> <span class="n">out_snps</span> <span class="o">=</span> <span class="n">ret_combine_snps_cmd</span><span class="p">(</span>
        <span class="n">all_snps</span><span class="o">=</span><span class="n">all_snps</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">combine_indels_cmd</span><span class="p">,</span> <span class="n">out_indels</span> <span class="o">=</span> <span class="n">ret_combine_indels_cmd</span><span class="p">(</span>
        <span class="n">all_indels</span><span class="o">=</span><span class="n">all_indels</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">vcf_tools_snp_cmd</span><span class="p">,</span> <span class="n">recode_snps</span> <span class="o">=</span> <span class="n">ret_vcf_tools_snp_cmd</span><span class="p">(</span>
        <span class="n">input_snps</span><span class="o">=</span><span class="n">out_snps</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="o">=</span><span class="n">snp_filter_name</span><span class="p">)</span>

    <span class="n">vcf_tools_indel_cmd</span><span class="p">,</span> <span class="n">recode_indels</span> <span class="o">=</span> <span class="n">ret_vcf_tools_indel_cmd</span><span class="p">(</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">out_indels</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="o">=</span><span class="n">indel_filter_name</span><span class="p">)</span>

    <span class="n">combine_recode_cmd</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%s</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
        <span class="p">[</span><span class="n">combine_snps_cmd</span><span class="p">,</span> <span class="n">combine_indels_cmd</span><span class="p">,</span> <span class="n">vcf_tools_snp_cmd</span><span class="p">,</span>
         <span class="n">vcf_tools_indel_cmd</span><span class="p">])</span>

    <span class="k">return</span> <span class="n">combine_recode_cmd</span><span class="p">,</span> <span class="n">recode_snps</span><span class="p">,</span> <span class="n">recode_indels</span>


<span class="k">def</span> <span class="nf">ret_recalibration_cmds</span><span class="p">(</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="n">input_snps</span><span class="p">,</span>
                           <span class="n">input_indels</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="p">):</span>
    <span class="n">first_recal_cmd</span><span class="p">,</span> <span class="n">before_table</span> <span class="o">=</span> <span class="n">ret_first_recal_cmd</span><span class="p">(</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">input_snps</span><span class="o">=</span><span class="n">input_snps</span><span class="p">,</span> <span class="n">sample_input</span><span class="o">=</span><span class="n">sample_input</span><span class="p">,</span>
        <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">print_reads_cmd</span><span class="p">,</span> <span class="n">out_bam</span> <span class="o">=</span> <span class="n">ret_print_reads_cmd</span><span class="p">(</span>
        <span class="n">sample_input</span><span class="o">=</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">recal_table</span><span class="o">=</span><span class="n">before_table</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
        <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">second_recal_cmd</span><span class="p">,</span> <span class="n">after_table</span> <span class="o">=</span> <span class="n">ret_second_recal_cmd</span><span class="p">(</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">input_indels</span><span class="p">,</span> <span class="n">input_snps</span><span class="o">=</span><span class="n">input_snps</span><span class="p">,</span> <span class="n">sample_input</span><span class="o">=</span><span class="n">sample_input</span><span class="p">,</span>
        <span class="n">before_table</span><span class="o">=</span><span class="n">before_table</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span>
        <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">recal_plot_cmd</span> <span class="o">=</span> <span class="n">ret_recal_plot_cmd</span><span class="p">(</span>
        <span class="n">before_table</span><span class="o">=</span><span class="n">before_table</span><span class="p">,</span> <span class="n">after_table</span><span class="o">=</span><span class="n">after_table</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

    <span class="n">recal_cmd</span> <span class="o">=</span> <span class="s1">&#39;(</span><span class="si">%s</span><span class="s1">) &amp;</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="s1">&#39;; &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span>
        <span class="p">[</span><span class="n">first_recal_cmd</span><span class="p">,</span> <span class="n">print_reads_cmd</span><span class="p">,</span> <span class="n">second_recal_cmd</span><span class="p">,</span> <span class="n">recal_plot_cmd</span><span class="p">])</span>

    <span class="k">return</span> <span class="n">recal_cmd</span><span class="p">,</span> <span class="n">out_bam</span>


<span class="k">def</span> <span class="nf">final_variant_calling</span><span class="p">(</span><span class="n">input_dict</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">write</span><span class="p">,</span>
                          <span class="n">write_prefix</span><span class="p">,</span> <span class="n">hcgvcfs</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="p">,</span> <span class="n">scriptname</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="n">write_prefix</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;#!/bin/bash</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">gatk_args</span> <span class="o">=</span> <span class="s2">&quot; -stand_call_conf 30 -stand_emit_conf 30 -mbq 17 &quot;</span>
    <span class="n">param_string</span> <span class="o">=</span> <span class="s1">&#39;--variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF&#39;</span>
    <span class="n">hc_outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="s1">&#39;final_variant_calling/hc_variant_calling&#39;</span><span class="p">)</span>
    <span class="n">jg_outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="s1">&#39;final_variant_calling/joint_genotypes&#39;</span><span class="p">)</span>
    <span class="n">variant_dict</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
    <span class="n">snp_filter</span> <span class="o">=</span> <span class="s1">&#39;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#39;</span>
    <span class="n">indel_filter</span> <span class="o">=</span> <span class="s1">&#39;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#39;</span>
    <span class="n">indel_filter_name</span> <span class="o">=</span> <span class="s1">&#39;default_indel_filter&#39;</span>
    <span class="n">snp_filter_name</span> <span class="o">=</span> <span class="s1">&#39;default_snp_filter&#39;</span>
    <span class="n">p</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">input_dict</span><span class="p">:</span>
        <span class="n">cmd</span><span class="p">,</span> <span class="n">out_var</span> <span class="o">=</span> <span class="n">ret_haplotype_caller_cmd</span><span class="p">(</span>
            <span class="n">sample_input</span><span class="o">=</span><span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span>
            <span class="n">outdir</span><span class="o">=</span><span class="n">hc_outdir</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
            <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">gatk_args</span><span class="o">=</span><span class="n">gatk_args</span><span class="p">,</span>
            <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">,</span> <span class="n">param_string</span><span class="o">=</span><span class="n">param_string</span><span class="p">)</span>
        <span class="n">variant_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">out_var</span><span class="p">)</span>
        <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
        <span class="n">job_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
            <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">cmd</span> <span class="o">+</span> <span class="s1">&#39; &amp;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
            <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>
        <span class="n">p</span> <span class="o">+=</span> <span class="mi">1</span>

    <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">job_procs</span><span class="p">)</span>
    <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">hcgvcfs</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">sample</span><span class="p">,</span> <span class="n">gvcf</span> <span class="ow">in</span> <span class="n">hcgvcfs</span><span class="p">:</span>
            <span class="n">variant_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">gvcf</span><span class="p">)</span>

    <span class="n">genotype_cmd</span><span class="p">,</span> <span class="n">out_gvcf</span> <span class="o">=</span> <span class="n">ret_genotype_gvcf_command</span><span class="p">(</span>
        <span class="n">sample_inputs</span><span class="o">=</span><span class="n">variant_dict</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
        <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">cpus</span><span class="o">=</span><span class="n">cpus</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">jg_outdir</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">genotype_cmd</span><span class="p">)</span>
    <span class="n">sample</span> <span class="o">=</span> <span class="s1">&#39;_&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">input_dict</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span>
    <span class="n">extract_snps_cmd</span><span class="p">,</span> <span class="n">raw_snps</span> <span class="o">=</span> <span class="n">ret_extract_snps_cmd</span><span class="p">(</span>
        <span class="n">input_vars</span><span class="o">=</span><span class="n">out_gvcf</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">jg_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">)</span>

    <span class="n">filter_snps_cmd</span><span class="p">,</span> <span class="n">out_snps</span> <span class="o">=</span> <span class="n">ret_filter_snps_cmd</span><span class="p">(</span>
        <span class="n">input_snps</span><span class="o">=</span><span class="n">raw_snps</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">snp_filter</span><span class="o">=</span><span class="n">snp_filter</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="o">=</span><span class="n">snp_filter_name</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">jg_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">)</span>

    <span class="n">extract_indels_cmd</span><span class="p">,</span> <span class="n">raw_indels</span> <span class="o">=</span> <span class="n">ret_extract_indels_cmd</span><span class="p">(</span>
        <span class="n">input_vars</span><span class="o">=</span><span class="n">out_gvcf</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">jg_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">)</span>

    <span class="n">filter_indels_cmd</span><span class="p">,</span> <span class="n">out_indels</span> <span class="o">=</span> <span class="n">ret_filter_indels_cmd</span><span class="p">(</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">raw_indels</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
        <span class="n">indel_filter</span><span class="o">=</span><span class="n">indel_filter</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="o">=</span><span class="n">indel_filter_name</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">jg_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">)</span>

    <span class="n">remove_nocall_cmd</span> <span class="o">=</span> <span class="s1">&#39;cat </span><span class="si">%s</span><span class="s1"> | grep -v </span><span class="se">\&#39;</span><span class="s1">\./\.</span><span class="se">\&#39;</span><span class="s1">  &gt; </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">out_gvcf</span><span class="p">,</span> <span class="n">out_gvcf</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.gvcf&#39;</span><span class="p">,</span> <span class="s1">&#39;.no_call_removed.gvcf&#39;</span><span class="p">))</span>
    <span class="n">badsnps_cmd</span> <span class="o">=</span> <span class="s1">&#39;cat </span><span class="si">%s</span><span class="s1"> | grep </span><span class="se">\&#39;</span><span class="si">%s</span><span class="se">\&#39;</span><span class="s1"> | cut -f 1,2 &gt; </span><span class="si">%s</span><span class="s1">/bad_snps.tsv&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">out_snps</span><span class="p">,</span> <span class="n">snp_filter_name</span><span class="p">,</span> <span class="n">jg_outdir</span><span class="p">)</span>
    <span class="n">badindels_cmd</span> <span class="o">=</span> <span class="s1">&#39;cat </span><span class="si">%s</span><span class="s1">   | grep </span><span class="se">\&#39;</span><span class="si">%s</span><span class="se">\&#39;</span><span class="s1"> | cut -f 1,2 &gt; </span><span class="si">%s</span><span class="s1">/bad_indels.tsv&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">out_indels</span><span class="p">,</span> <span class="n">indel_filter_name</span><span class="p">,</span> <span class="n">jg_outdir</span><span class="p">)</span>
    <span class="n">bad_positions_cmd</span> <span class="o">=</span> <span class="s1">&#39;cat </span><span class="si">%s</span><span class="s1">/bad_indels.tsv </span><span class="si">%s</span><span class="s1">/bad_snps.tsv &gt; </span><span class="si">%s</span><span class="s1">/bad_positions.tsv&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">jg_outdir</span><span class="p">,</span> <span class="n">jg_outdir</span><span class="p">,</span> <span class="n">jg_outdir</span><span class="p">)</span>
    <span class="n">remove_bad_positions</span> <span class="o">=</span> <span class="s1">&#39;vcftools --vcf </span><span class="si">%s</span><span class="s1"> --recode --out </span><span class="si">%s</span><span class="s1"> --exclude-positions </span><span class="si">%s</span><span class="s1">/bad_positions.tsv&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">out_gvcf</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.gvcf&#39;</span><span class="p">,</span> <span class="s1">&#39;.no_call_removed.gvcf&#39;</span><span class="p">),</span> <span class="n">out_gvcf</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.gvcf&#39;</span><span class="p">,</span> <span class="s1">&#39;.no_call_removed.filt&#39;</span><span class="p">),</span> <span class="n">jg_outdir</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">extract_snps_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">filter_snps_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">extract_indels_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">filter_indels_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">remove_nocall_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">badsnps_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">badindels_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">bad_positions_cmd</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">remove_bad_positions</span> <span class="o">+</span> <span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>

        <span class="n">fh</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

    <span class="k">return</span> <span class="n">out_gvcf</span><span class="p">,</span> <span class="n">scriptname</span>


<span class="k">def</span> <span class="nf">training_commands</span><span class="p">(</span><span class="n">input_dict</span><span class="p">,</span> <span class="n">rounds</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="p">,</span> <span class="n">write</span><span class="p">,</span>
                      <span class="n">write_prefix</span><span class="p">):</span>
    <span class="n">variant_outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="s1">&#39;training/variant_calling&#39;</span><span class="p">)</span>
    <span class="n">recal_outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="s1">&#39;training/recal_bams&#39;</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="p">,</span> <span class="n">scriptname</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="n">write_prefix</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;#!/bin/bash</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">p</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">print</span><span class="p">(</span><span class="n">rounds</span><span class="p">,</span> <span class="nb">range</span><span class="p">(</span><span class="n">rounds</span><span class="p">))</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">rounds</span><span class="p">):</span>
        <span class="n">all_snps</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="n">all_indels</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">bam_dict</span> <span class="o">=</span> <span class="n">input_dict</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">bam_dict</span> <span class="o">=</span> <span class="n">recal_bams</span>
        <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">bam_dict</span><span class="p">:</span>
            <span class="n">hc_cmd</span><span class="p">,</span> <span class="n">snps</span><span class="p">,</span> <span class="n">indels</span> <span class="o">=</span> <span class="n">ret_hc_call_cmds</span><span class="p">(</span>
                <span class="n">sample_input</span><span class="o">=</span><span class="n">bam_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span>
                <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
                <span class="n">outdir</span><span class="o">=</span><span class="n">variant_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

            <span class="n">all_snps</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">snps</span><span class="p">)</span>
            <span class="n">all_indels</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">indels</span><span class="p">)</span>
            <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
            <span class="n">job_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">hc_cmd</span><span class="p">)</span>
                <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>
            <span class="n">p</span> <span class="o">+=</span> <span class="mi">1</span>
        <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">job_procs</span><span class="p">)</span>
        <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
            <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

        <span class="n">recode_cmd</span><span class="p">,</span> <span class="n">recode_snps</span><span class="p">,</span> <span class="n">recode_indels</span> <span class="o">=</span> <span class="n">ret_recode_indels_and_snps_cmd</span><span class="p">(</span>
            <span class="n">all_snps</span><span class="o">=</span><span class="n">all_snps</span><span class="p">,</span> <span class="n">all_indels</span><span class="o">=</span><span class="n">all_indels</span><span class="p">,</span>
            <span class="n">snp_filter_name</span><span class="o">=</span><span class="s1">&#39;default_snp_filter&#39;</span><span class="p">,</span>
            <span class="n">indel_filter_name</span><span class="o">=</span><span class="s1">&#39;default_indel_filter&#39;</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span>
            <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">variant_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
            <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">recode_cmd</span><span class="p">)</span>
        <span class="n">recal_bams</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">&lt;</span> <span class="n">rounds</span> <span class="o">-</span> <span class="mi">1</span><span class="p">:</span>
            <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">bam_dict</span><span class="p">:</span>
                <span class="n">recal_cmd</span><span class="p">,</span> <span class="n">recal_bam</span> <span class="o">=</span> <span class="n">ret_recalibration_cmds</span><span class="p">(</span>
                    <span class="n">input_snps</span><span class="o">=</span><span class="n">recode_snps</span><span class="p">,</span> <span class="n">input_indels</span><span class="o">=</span><span class="n">recode_indels</span><span class="p">,</span>
                    <span class="n">sample_input</span><span class="o">=</span><span class="n">bam_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span>
                    <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk_path</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
                    <span class="n">outdir</span><span class="o">=</span><span class="n">recal_outdir</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>

                <span class="n">recal_bams</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">recal_bam</span><span class="p">)</span>
                <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
                <span class="n">job_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
                <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">recal_cmd</span><span class="p">)</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>
                <span class="n">p</span> <span class="o">+=</span> <span class="mi">1</span>
            <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">job_procs</span><span class="p">)</span>
            <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
            <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>
    <span class="n">fh</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

    <span class="k">return</span> <span class="n">recode_snps</span><span class="p">,</span> <span class="n">recode_indels</span><span class="p">,</span> <span class="n">scriptname</span>


<span class="k">def</span> <span class="nf">parallel_cmd_loop</span><span class="p">(</span><span class="n">input_dict</span><span class="p">,</span> <span class="n">nested</span><span class="p">,</span> <span class="n">write</span><span class="p">,</span> <span class="n">write_prefix</span><span class="p">,</span>
                      <span class="n">cmd_function</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">):</span>
    <span class="n">out_dict</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
    <span class="n">outdir</span> <span class="o">=</span> <span class="n">make_output_dir</span><span class="p">(</span><span class="n">outdir</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">fh</span><span class="p">,</span> <span class="n">scriptname</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="n">write_prefix</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s2">&quot;#!/bin/bash</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">n</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="n">all_procs</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">out_dict</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">input_dict</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">nested</span><span class="p">:</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">nested_inputs</span> <span class="o">=</span> <span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">()</span>
            <span class="k">except</span> <span class="ne">AttributeError</span><span class="p">:</span>
                <span class="n">nested_inputs</span> <span class="o">=</span> <span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span>

            <span class="k">for</span> <span class="n">nested_input</span> <span class="ow">in</span> <span class="n">nested_inputs</span><span class="p">:</span>
                <span class="n">cmd</span><span class="p">,</span> <span class="n">output</span> <span class="o">=</span> <span class="n">cmd_function</span><span class="p">(</span>
                    <span class="n">nested_input</span><span class="p">,</span> <span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>
                <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">n</span><span class="p">)</span>
                <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">cmd</span><span class="p">)</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>

                <span class="n">job_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
                <span class="n">all_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>

                <span class="k">if</span> <span class="n">n</span> <span class="o">%</span> <span class="mi">3</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                    <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">job_procs</span><span class="p">)</span>
                    <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
                    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

                <span class="n">out_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">output</span><span class="p">)</span>
                <span class="c1"># print(out_dict)</span>
                <span class="n">n</span> <span class="o">+=</span> <span class="mi">1</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c1"># print(input_dict[sample])</span>
            <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">])</span> <span class="o">==</span> <span class="nb">list</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">])</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
                <span class="n">sample_input</span> <span class="o">=</span> <span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">sample_input</span> <span class="o">=</span> <span class="n">input_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span>

            <span class="n">cmd</span><span class="p">,</span> <span class="n">output</span> <span class="o">=</span> <span class="n">cmd_function</span><span class="p">(</span>
                <span class="n">sample_input</span><span class="o">=</span><span class="n">sample_input</span><span class="p">,</span> <span class="n">sample</span><span class="o">=</span><span class="n">sample</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="n">outdir</span><span class="p">,</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>
            <span class="c1"># print(output)</span>
            <span class="n">proc_capture</span><span class="p">,</span> <span class="n">proc</span> <span class="o">=</span> <span class="n">make_proc_capture</span><span class="p">(</span><span class="n">n</span><span class="p">)</span>
            <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">cmd</span><span class="p">)</span>
                <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">proc_capture</span><span class="p">)</span>

            <span class="n">job_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>
            <span class="n">all_procs</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">proc</span><span class="p">)</span>

            <span class="k">if</span> <span class="n">n</span> <span class="o">%</span> <span class="mi">3</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
                <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">job_procs</span><span class="p">)</span>
                <span class="n">job_procs</span> <span class="o">=</span> <span class="p">[]</span>
                <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
                    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

            <span class="n">out_dict</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">output</span><span class="p">)</span>

            <span class="n">n</span> <span class="o">+=</span> <span class="mi">1</span>

    <span class="k">if</span> <span class="n">write</span><span class="p">:</span>
        <span class="n">wait_command</span> <span class="o">=</span> <span class="s2">&quot;wait </span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="s2">&quot; &quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">all_procs</span><span class="p">)</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">wait_command</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">out_dict</span><span class="p">,</span> <span class="n">scriptname</span>


<span class="k">def</span> <span class="nf">run</span><span class="p">(</span><span class="n">args</span><span class="p">):</span>

    <span class="n">molng</span> <span class="o">=</span> <span class="n">lims</span><span class="o">.</span><span class="n">read_list_file</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">molngs</span><span class="p">)</span>
    <span class="n">samples</span> <span class="o">=</span> <span class="n">lims</span><span class="o">.</span><span class="n">read_list_file</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">samples</span><span class="p">)</span>
    <span class="n">flowcells</span> <span class="o">=</span> <span class="n">lims</span><span class="o">.</span><span class="n">read_list_file</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">flowcells</span><span class="p">)</span>
    <span class="n">train</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">train</span>

    <span class="n">fasta_path</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">fasta_path</span>
    <span class="n">cpus</span> <span class="o">=</span> <span class="mi">8</span>
    <span class="n">picard_tools_path</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">picard_path</span>
    <span class="n">gatk</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">gatk_execute</span>

    <span class="k">print</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">hcgvcfs</span><span class="p">)</span>

    <span class="n">tigris_report</span> <span class="o">=</span> <span class="n">lims</span><span class="o">.</span><span class="n">Indexed_sample_reports</span><span class="p">(</span><span class="n">molng</span><span class="p">,</span> <span class="n">samples</span><span class="p">)</span>
    <span class="n">tigris_report</span><span class="o">.</span><span class="n">select_flowcells</span><span class="p">(</span><span class="n">flowcells</span><span class="p">)</span>
    <span class="n">lane_pairs</span> <span class="o">=</span> <span class="n">tigris_report</span><span class="o">.</span><span class="n">ret_nested_pair_by_lane</span><span class="p">()</span>
    <span class="n">scripts</span> <span class="o">=</span> <span class="p">[]</span>


    <span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">s0</span> <span class="o">=</span> <span class="n">prep_reference_commands</span><span class="p">(</span>
        <span class="n">fasta_path</span><span class="p">,</span> <span class="n">picard_tools_path</span><span class="p">,</span> <span class="bp">True</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s0</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">args</span><span class="o">.</span><span class="n">down_sample</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">down_sample</span><span class="p">)</span>
        <span class="n">lane_pairs</span><span class="p">,</span> <span class="n">dwn_s</span> <span class="o">=</span> <span class="n">down_sample_cmds</span><span class="p">(</span><span class="n">lane_pairs</span><span class="p">,</span> <span class="n">args</span><span class="o">.</span><span class="n">down_sample</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                                            <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_down_sample&#39;</span><span class="p">,</span><span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/down_sample&#39;</span><span class="p">)</span>
        <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dwn_s</span><span class="p">)</span>

    <span class="n">lane_bams</span><span class="p">,</span> <span class="n">s1</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">lane_pairs</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_1_map_to_ref&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_bwa_align_sort_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/map_to_ref_output&#39;</span><span class="p">,</span> <span class="n">cpus</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s1</span><span class="p">)</span>

    <span class="n">dedup_lane_bams</span><span class="p">,</span> <span class="n">s2</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">lane_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_2_dedup_individual&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_dedup_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/dedup_individuals&#39;</span><span class="p">,</span>
        <span class="n">picard_tools_path</span><span class="o">=</span><span class="n">picard_tools_path</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s2</span><span class="p">)</span>

    <span class="n">sample_bams</span><span class="p">,</span> <span class="n">s3</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">dedup_lane_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_3_merge_bams&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_merge_bams_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/merged_bams&#39;</span><span class="p">,</span> <span class="n">cpus</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span> <span class="n">tempdir</span><span class="o">=</span><span class="s1">&#39;/scratch/dut&#39;</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s3</span><span class="p">)</span>

    <span class="n">deduped_sample_bams</span><span class="p">,</span> <span class="n">s4</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">sample_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_4_dedup_merged&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_dedup_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/dedup_merged&#39;</span><span class="p">,</span>
        <span class="n">picard_tools_path</span><span class="o">=</span><span class="n">picard_tools_path</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s4</span><span class="p">)</span>

    <span class="n">deduped_sample_bams</span><span class="p">,</span> <span class="n">s5</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">deduped_sample_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_5_index&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_index_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/deduped_merged&#39;</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s5</span><span class="p">)</span>

    <span class="n">realigned_sample_bams</span><span class="p">,</span> <span class="n">s6</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">deduped_sample_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_6_realign_bams&#39;</span><span class="p">,</span> <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_realign_indels_cmd</span><span class="p">,</span>
        <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;preprocessing/realigned_merged_bams&#39;</span><span class="p">,</span> <span class="n">cpus</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span>
        <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s6</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">train</span><span class="p">:</span>
        <span class="n">known_snps</span><span class="p">,</span> <span class="n">known_indels</span><span class="p">,</span> <span class="n">s7</span> <span class="o">=</span> <span class="n">training_commands</span><span class="p">(</span>
            <span class="n">input_dict</span><span class="o">=</span><span class="n">realigned_sample_bams</span><span class="p">,</span> <span class="n">rounds</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
            <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_7_training&#39;</span><span class="p">)</span>
        <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s7</span><span class="p">)</span>
    <span class="k">elif</span> <span class="n">args</span><span class="o">.</span><span class="n">known_snps</span> <span class="ow">and</span> <span class="n">args</span><span class="o">.</span><span class="n">known_indels</span><span class="p">:</span>
        <span class="n">known_snps</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">known_snps</span>
        <span class="n">known_indels</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">known_indels</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="s1">&#39;train must be set to true or known snps and indels must be provieded&#39;</span><span class="p">)</span>
        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

    <span class="n">recalibrated_bams</span><span class="p">,</span> <span class="n">s8</span> <span class="o">=</span> <span class="n">parallel_cmd_loop</span><span class="p">(</span>
        <span class="n">input_dict</span><span class="o">=</span><span class="n">realigned_sample_bams</span><span class="p">,</span> <span class="n">nested</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
        <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_8_recalibrate_bams&#39;</span><span class="p">,</span> <span class="n">outdir</span><span class="o">=</span><span class="s1">&#39;final_variant_calling/recal_bams&#39;</span><span class="p">,</span>
        <span class="n">cmd_function</span><span class="o">=</span><span class="n">ret_recalibration_cmds</span><span class="p">,</span> <span class="n">input_snps</span><span class="o">=</span><span class="n">known_snps</span><span class="p">,</span>
        <span class="n">input_indels</span><span class="o">=</span><span class="n">known_indels</span><span class="p">,</span> <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span> <span class="n">i</span><span class="o">=</span><span class="s1">&#39;final&#39;</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s8</span><span class="p">)</span>

    <span class="n">gvcf</span><span class="p">,</span> <span class="n">s9</span> <span class="o">=</span> <span class="n">final_variant_calling</span><span class="p">(</span><span class="n">input_dict</span><span class="o">=</span><span class="n">recalibrated_bams</span><span class="p">,</span> <span class="n">fasta_ref_path</span><span class="o">=</span><span class="n">fasta_ref_path</span><span class="p">,</span>
                                     <span class="n">gatk_path</span><span class="o">=</span><span class="n">gatk</span><span class="p">,</span> <span class="n">cpus</span><span class="o">=</span><span class="mi">24</span><span class="p">,</span> <span class="n">write</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                                     <span class="n">write_prefix</span><span class="o">=</span><span class="s1">&#39;gatk_9_final_variant_calling&#39;</span><span class="p">,</span> <span class="n">hcgvcfs</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">hcgvcfs</span><span class="p">)</span>
    <span class="n">scripts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">s9</span><span class="p">)</span>

    <span class="n">fh</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">make_script_fh</span><span class="p">(</span><span class="s1">&#39;master&#39;</span><span class="p">)</span>

    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;#!/bin/bash</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>

    <span class="n">commands</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;echo </span><span class="se">\&quot;</span><span class="si">%s</span><span class="se">\&quot;\n</span><span class="s1">time </span><span class="si">%s</span><span class="s1"> &amp;&gt; </span><span class="si">%s</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span>
        <span class="n">x</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;./&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.sh&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">),</span>
        <span class="n">x</span><span class="p">,</span>
        <span class="n">x</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;./&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.sh&#39;</span><span class="p">,</span> <span class="s1">&#39;.out&#39;</span><span class="p">),</span>
    <span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">scripts</span><span class="p">]</span>
    <span class="n">commands</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">commands</span><span class="p">)</span>
    <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">commands</span><span class="p">)</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">()</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-molngs&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;molngs&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to file with all molngs paths seperated by newlines&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-samples&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;samples&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to file with all sample names seperated by newlines&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-flowcells&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;flowcells&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to file with all flowcell names seperated by newlines&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span> <span class="p">(</span><span class="s1">&#39;-down_sample&#39;</span><span class="p">,</span> <span class="n">nargs</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;append&#39;</span><span class="p">,</span>
                         <span class="n">help</span><span class="o">=</span><span class="s1">&#39;pass sample name and fraction to down sample to, e.g -down_sample Atig_122 0.33&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-cpus&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;cpus&#39;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s1">&#39;8&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;number of cpus to use per sample&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-fasta_path&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;fasta_path&#39;</span><span class="p">,</span>
                        <span class="n">default</span><span class="o">=</span><span class="s1">&#39;/home/dut/projects/tigris/genome_annotation/fasta/tigris_scaffolds_filt_10000.fa&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to genome fasta&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-picard_path&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;picard_path&#39;</span><span class="p">,</span>
                        <span class="n">default</span><span class="o">=</span><span class="s1">&#39;/home/dut/bin/picard-tools-1.119/&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to picard tools&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-gatk_execute&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;gatk_execute&#39;</span><span class="p">,</span>
                        <span class="n">default</span><span class="o">=</span><span class="s1">&#39;java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;gatk executable prefix&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-train&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store_true&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;train&#39;</span><span class="p">,</span>
                        <span class="n">default</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s1">&#39;train for known indels and snps&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-known_snps&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;known_snps&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to known snps(required if training set to false)&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-known_indels&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;known_indels&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;path to known indels(required if training set to false)&#39;</span><span class="p">)</span>

    <span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-add_hcgvcf&#39;</span><span class="p">,</span><span class="n">action</span><span class="o">=</span><span class="s1">&#39;append&#39;</span><span class="p">,</span><span class="n">nargs</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;hcgvcfs&#39;</span><span class="p">,</span>
                        <span class="n">help</span><span class="o">=</span><span class="s1">&#39;additional gvcf files produced by HaplotypeCaller to be used in joint genotypeing(-add_hcgvcf sample_name path_to_file&#39;</span><span class="p">)</span>

    <span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>

    <span class="n">run</span><span class="p">(</span><span class="n">args</span><span class="p">)</span>

</pre></div>
</div>
</div>
<p>This tool outputs a series of bash scripts, meant to be run
sequentially. During the course of writting and running our samples
throught the pipeline, I found mutiple errors and had to restart steps.
Below I include the bash scripts I ended up running in sequential order
to generat our final GVCFs.</p>
<div class="section" id="GATK-Run-on-8450,-001-and-003">
<h4>GATK Run on 8450, 001 and 003<a class="headerlink" href="#GATK-Run-on-8450,-001-and-003" title="Permalink to this headline">¶</a></h4>
<p>Here are all of the bash commands run for the GATK pipline. Some of the
scripts are redundant because errors were found that merited the
re-running of commands.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [107]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="k">for</span> file in <span class="sb">`</span>ls -1 ../data/gatk6/bash/*.sh<span class="sb">`</span>
 <span class="k">do</span>
  <span class="nb">echo</span> <span class="s2">&quot;------------------------------------------- </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="s2"> -------------------------------------------&quot;</span>
  cat <span class="nv">$file</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
------------------------------------------- ../data/gatk6/bash/gatk_0_prep_ref_2016-03-31.sh -------------------------------------------
#!/bin/bash
ln -s /home/dut/projects/tigris/genome_annotation/fasta/tigris_scaffolds_filt_10000.fa /home/dut/projects/tigris/heterozygosity/gatk6/reference/
bwa index -a bwtsw /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa
samtools faidx /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa
java -jar /home/dut/bin/picard-tools-1.119/CreateSequenceDictionary.jar REFERENCE=/home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa OUTPUT=/home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.dict
------------------------------------------- ../data/gatk6/bash/gatk_1_map_to_ref_2016-03-31.sh -------------------------------------------
#!/bin/bash
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136_HF7YWADXXb_2\tSM:A_tigris8450\tPL:illumina\tLB:L13136\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136_HF7YWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HF7YWADXXb_2.bam) &amp;
proc1=$!
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136-1_HG7HJADXXb_2\tSM:A_tigris8450\tPL:illumina\tLB:L13136-1\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136-1_HG7HJADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136-1_HG7HJADXXb_2.bam) &amp;
proc2=$!
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136-1_HG7HJADXXb_1\tSM:A_tigris8450\tPL:illumina\tLB:L13136-1\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136-1_HG7HJADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136-1_HG7HJADXXb_1.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136_HBCBWADXXb_2\tSM:A_tigris8450\tPL:illumina\tLB:L13136\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136_HBCBWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HBCBWADXXb_2.bam) &amp;
proc4=$!
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136_HBCBWADXXb_1\tSM:A_tigris8450\tPL:illumina\tLB:L13136\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136_HBCBWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HBCBWADXXb_1.bam) &amp;
proc5=$!
(bwa mem -M -R &#34;@RG\tID:A_tigris8450_L13136_HF7YWADXXb_1\tSM:A_tigris8450\tPL:illumina\tLB:L13136\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_1_GTGGCC.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_2_GTGGCC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - A_tigris8450_L13136_HF7YWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HF7YWADXXb_1.bam) &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
(bwa mem -M -R &#34;@RG\tID:Atig_122_L21676_HJ2YHBCXX_2\tSM:Atig_122\tPL:illumina\tLB:L21676\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_2_1_CTCAGA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_2_2_CTCAGA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig_122_L21676_HJ2YHBCXX_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_2.bam) &amp;
proc7=$!
(bwa mem -M -R &#34;@RG\tID:Atig_122_L21676_HJ2YHBCXX_1\tSM:Atig_122\tPL:illumina\tLB:L21676\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_1_1_CTCAGA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_1_2_CTCAGA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig_122_L21676_HJ2YHBCXX_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_1.bam) &amp;
proc8=$!
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088_HF7YWADXXb_1\tSM:Atig003\tPL:illumina\tLB:L13088\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088_HF7YWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HF7YWADXXb_1.bam) &amp;
proc9=$!
wait &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34;
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088_HBCBWADXXb_1\tSM:Atig003\tPL:illumina\tLB:L13088\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088_HBCBWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HBCBWADXXb_1.bam) &amp;
proc10=$!
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088_HBCBWADXXb_2\tSM:Atig003\tPL:illumina\tLB:L13088\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088_HBCBWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HBCBWADXXb_2.bam) &amp;
proc11=$!
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088-1_HG7HJADXXb_1\tSM:Atig003\tPL:illumina\tLB:L13088-1\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088-1_HG7HJADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088-1_HG7HJADXXb_1.bam) &amp;
proc12=$!
wait &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088-1_HG7HJADXXb_2\tSM:Atig003\tPL:illumina\tLB:L13088-1\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088-1_HG7HJADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088-1_HG7HJADXXb_2.bam) &amp;
proc13=$!
(bwa mem -M -R &#34;@RG\tID:Atig003_L13088_HF7YWADXXb_2\tSM:Atig003\tPL:illumina\tLB:L13088\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_1_TAGCTT.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_2_TAGCTT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig003_L13088_HF7YWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HF7YWADXXb_2.bam) &amp;
proc14=$!
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HG7HJADXXb_2\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_2_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HG7HJADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HG7HJADXXb_2.bam) &amp;
proc15=$!
wait &#34;$proc13&#34; &#34;$proc14&#34; &#34;$proc15&#34;
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HG7HJADXXb_1\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HG7HJADXXb/s_1_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HG7HJADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HG7HJADXXb_1.bam) &amp;
proc16=$!
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HF7YWADXXb_2\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_2_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HF7YWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HF7YWADXXb_2.bam) &amp;
proc17=$!
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HBCBWADXXb_2\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_2_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HBCBWADXXb_2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HBCBWADXXb_2.bam) &amp;
proc18=$!
wait &#34;$proc16&#34; &#34;$proc17&#34; &#34;$proc18&#34;
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HBCBWADXXb_1\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HBCBWADXXb/s_1_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HBCBWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HBCBWADXXb_1.bam) &amp;
proc19=$!
(bwa mem -M -R &#34;@RG\tID:Atig001_L13087_HF7YWADXXb_1\tSM:Atig001\tPL:illumina\tLB:L13087\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_1_ATCACG.fastq.gz /n/analysis/Baumann/aan/MOLNG-1029/HF7YWADXXb/s_1_2_ATCACG.fastq.gz | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig001_L13087_HF7YWADXXb_1 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HF7YWADXXb_1.bam) &amp;
proc20=$!
------------------------------------------- ../data/gatk6/bash/gatk_2_dedup_individual_2016-03-31.sh -------------------------------------------
#!/bin/bash
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HF7YWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_2_metrics.txt &amp;
proc1=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136-1_HG7HJADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_2_metrics.txt &amp;
proc2=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136-1_HG7HJADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_1_metrics.txt &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HBCBWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_2_metrics.txt &amp;
proc4=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HBCBWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_1_metrics.txt &amp;
proc5=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/A_tigris8450_L13136_HF7YWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_1_metrics.txt &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2_metrics.txt &amp;
proc7=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1_metrics.txt &amp;
proc8=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HF7YWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_1_metrics.txt &amp;
proc9=$!
wait &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HBCBWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_1_metrics.txt &amp;
proc10=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HBCBWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_2_metrics.txt &amp;
proc11=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088-1_HG7HJADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_1_metrics.txt &amp;
proc12=$!
wait &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088-1_HG7HJADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_2_metrics.txt &amp;
proc13=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig003_L13088_HF7YWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_2_metrics.txt &amp;
proc14=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HG7HJADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_2_metrics.txt &amp;
proc15=$!
wait &#34;$proc13&#34; &#34;$proc14&#34; &#34;$proc15&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HG7HJADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_1_metrics.txt &amp;
proc16=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HF7YWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_2_metrics.txt &amp;
proc17=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HBCBWADXXb_2.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_2_metrics.txt &amp;
proc18=$!
wait &#34;$proc16&#34; &#34;$proc17&#34; &#34;$proc18&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HBCBWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_1_metrics.txt &amp;
proc19=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/map_to_ref_output/Atig001_L13087_HF7YWADXXb_1.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_1_metrics.txt &amp;
proc20=$!
------------------------------------------- ../data/gatk6/bash/gatk_3_merge_bams_2016-03-31.sh -------------------------------------------
#!/bin/bash
(samtools merge -@ 8 - /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136-1_HG7HJADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HBCBWADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450_L13136_HF7YWADXXb_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/A_tigris8450.temp -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/A_tigris8450.merged.bam; samtools index /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/A_tigris8450.merged.bam) &amp;
proc1=$!
(samtools merge -@ 8 - /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig_122.temp -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig_122.merged.bam; samtools index /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig_122.merged.bam) &amp;
proc2=$!
(samtools merge -@ 8 - /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HBCBWADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088-1_HG7HJADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003_L13088_HF7YWADXXb_2.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig003.temp -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig003.merged.bam; samtools index /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig003.merged.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(samtools merge -@ 8 - /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HG7HJADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_2.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HBCBWADXXb_1.dedup.bam /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001_L13087_HF7YWADXXb_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig001.temp -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig001.merged.bam; samtools index /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig001.merged.bam) &amp;
proc4=$!
------------------------------------------- ../data/gatk6/bash/gatk_4_dedup_merged_2016-03-31.sh -------------------------------------------
#!/bin/bash
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/A_tigris8450.merged.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450.merged.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/A_tigris8450.merged_metrics.txt &amp;
proc1=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig_122.merged.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122.merged.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig_122.merged_metrics.txt &amp;
proc2=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig003.merged.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003.merged.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig003.merged_metrics.txt &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/merged_bams/Atig001.merged.bam O=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001.merged.dedup.bam M=/home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_individuals/Atig001.merged_metrics.txt &amp;
proc4=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34;
------------------------------------------- ../data/gatk6/bash/gatk_5_index_2016-04-04.sh -------------------------------------------
#!/bin/bash
samtools index -b /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/A_tigris8450.merged.dedup.bam &amp;
proc1=$!
samtools index -b /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig_122.merged.dedup.bam &amp;
proc2=$!
samtools index -b /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig003.merged.dedup.bam &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
samtools index -b /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig001.merged.dedup.bam &amp;
proc4=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34;
------------------------------------------- ../data/gatk6/bash/gatk_6_realign_bams_2016-04-04.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/A_tigris8450.merged.dedup.bam -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/A_tigris8450.merged.dedup.bam -targetIntervals /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup_target_intervals.list -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam) &amp;
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig_122.merged.dedup.bam -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig_122.merged.dedup.bam -targetIntervals /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup_target_intervals.list -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam) &amp;
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig003.merged.dedup.bam -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig003.merged.dedup.bam -targetIntervals /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup_target_intervals.list -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig001.merged.dedup.bam -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/dedup_merged/Atig001.merged.dedup.bam -targetIntervals /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup_target_intervals.list -o /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam) &amp;
proc4=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34;
------------------------------------------- ../data/gatk6/bash/gatk_7_training_2016-04-21.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_indels.vcf) &amp;
proc0=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_indels.vcf) &amp;
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_indels.vcf) &amp;
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_indels.vcf) &amp;
proc3=$!
wait &#34;$proc0&#34; &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.plots) &amp;
proc4=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.plots) &amp;
proc5=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.plots) &amp;
proc6=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.plots) &amp;
proc7=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34; &#34;$proc7&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_indels.vcf) &amp;
proc8=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_indels.vcf) &amp;
proc9=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_indels.vcf) &amp;
proc10=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_indels.vcf) &amp;
proc11=$!
wait &#34;$proc8&#34; &#34;$proc9&#34; &#34;$proc10&#34; &#34;$proc11&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.plots) &amp;
proc12=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.plots) &amp;
proc13=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.plots) &amp;
proc14=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.plots) &amp;
proc15=$!
wait &#34;$proc12&#34; &#34;$proc13&#34; &#34;$proc14&#34; &#34;$proc15&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_indels.vcf) &amp;
proc16=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_indels.vcf) &amp;
proc17=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_indels.vcf) &amp;
proc18=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_indels.vcf) &amp;
proc19=$!
wait &#34;$proc16&#34; &#34;$proc17&#34; &#34;$proc18&#34; &#34;$proc19&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2
------------------------------------------- ../data/gatk6/bash/gatk_7_training_2016-05-05.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.plots) &amp;
proc20=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.plots) &amp;
proc21=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.plots) &amp;
proc22=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.plots) &amp;
proc23=$!
wait &#34;$proc20&#34; &#34;$proc21&#34; &#34;$proc22&#34; &#34;$proc23&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_indels.vcf) &amp;
proc24=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_indels.vcf) &amp;
proc25=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_indels.vcf) &amp;
proc26=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_indels.vcf) &amp;
proc27=$!
wait &#34;$proc24&#34; &#34;$proc25&#34; &#34;$proc26&#34; &#34;$proc27&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3
------------------------------------------- ../data/gatk6/bash/gatk_8_recalibrate_bams_2016-05-05.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.plots) &amp;
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.plots) &amp;
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.plots) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.plots) &amp;
proc4=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34;
------------------------------------------- ../data/gatk6/bash/gatk_9_final_variant_calling_2016-05-17.sh -------------------------------------------
#!/bin/bash
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf &amp;
#proc0=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf &amp;
#proc1=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf &amp;
#proc2=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf &amp;
#proc3=$!
#wait &#34;$proc0&#34; &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf -allSites -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf
------------------------------------------- ../data/gatk6/bash/gatk_9_final_variant_calling_2016-05-23.sh -------------------------------------------
#!/bin/bash
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf &amp;
#proc0=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf &amp;
#proc1=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf &amp;
#proc2=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf &amp;
#proc3=$!
#wait &#34;$proc0&#34; &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf -allSites -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_snps.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_snps.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_indels.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_indels.vcf
#cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf | grep -v &#39;\./\.&#39;  &gt; /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_snps.vcf | grep &#39;default_snp_filter&#39; | cut -f 1,2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_snps.tsv
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_indels.vcf | grep &#39;default_indel_filter&#39; | cut -f 1,2 &gt; /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_indels.tsv
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_indels.tsv /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_snps.tsv &gt; /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_positions.tsv
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf --recode --out /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.filt --exclude-positions /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_positions.tsv
------------------------------------------- ../data/gatk6/bash/master_2016-04-21.sh -------------------------------------------
#!/bin/bash
echo &#34;gatk_0_prep_ref_2016-04-21&#34;
time ./gatk_0_prep_ref_2016-04-21.sh &amp;&gt; gatk_0_prep_ref_2016-04-21.out
echo &#34;gatk_1_map_to_ref_2016-04-21&#34;
time ./gatk_1_map_to_ref_2016-04-21.sh &amp;&gt; gatk_1_map_to_ref_2016-04-21.out
echo &#34;gatk_2_dedup_individual_2016-04-21&#34;
time ./gatk_2_dedup_individual_2016-04-21.sh &amp;&gt; gatk_2_dedup_individual_2016-04-21.out
echo &#34;gatk_3_merge_bams_2016-04-21&#34;
time ./gatk_3_merge_bams_2016-04-21.sh &amp;&gt; gatk_3_merge_bams_2016-04-21.out
echo &#34;gatk_4_dedup_merged_2016-04-21&#34;
time ./gatk_4_dedup_merged_2016-04-21.sh &amp;&gt; gatk_4_dedup_merged_2016-04-21.out
echo &#34;gatk_5_index_2016-04-21&#34;
time ./gatk_5_index_2016-04-21.sh &amp;&gt; gatk_5_index_2016-04-21.out
echo &#34;gatk_6_realign_bams_2016-04-21&#34;
time ./gatk_6_realign_bams_2016-04-21.sh &amp;&gt; gatk_6_realign_bams_2016-04-21.out
echo &#34;gatk_7_training_2016-04-21&#34;
time ./gatk_7_training_2016-04-21.sh &amp;&gt; gatk_7_training_2016-04-21.out
echo &#34;gatk_8_recalibrate_bams_2016-04-21&#34;
time ./gatk_8_recalibrate_bams_2016-04-21.sh &amp;&gt; gatk_8_recalibrate_bams_2016-04-21.out
echo &#34;gatk_9_final_variant_calling_2016-04-21&#34;
time ./gatk_9_final_variant_calling_2016-04-21.sh &amp;&gt; gatk_9_final_variant_calling_2016-04-21.out
------------------------------------------- ../data/gatk6/bash/master_2016-05-05.sh -------------------------------------------
#!/bin/bash
echo &#34;gatk_0_prep_ref_2016-05-05&#34;
time ./gatk_0_prep_ref_2016-05-05.sh &amp;&gt; gatk_0_prep_ref_2016-05-05.out
echo &#34;gatk_1_map_to_ref_2016-05-05&#34;
time ./gatk_1_map_to_ref_2016-05-05.sh &amp;&gt; gatk_1_map_to_ref_2016-05-05.out
echo &#34;gatk_2_dedup_individual_2016-05-05&#34;
time ./gatk_2_dedup_individual_2016-05-05.sh &amp;&gt; gatk_2_dedup_individual_2016-05-05.out
echo &#34;gatk_3_merge_bams_2016-05-05&#34;
time ./gatk_3_merge_bams_2016-05-05.sh &amp;&gt; gatk_3_merge_bams_2016-05-05.out
echo &#34;gatk_4_dedup_merged_2016-05-05&#34;
time ./gatk_4_dedup_merged_2016-05-05.sh &amp;&gt; gatk_4_dedup_merged_2016-05-05.out
echo &#34;gatk_5_index_2016-05-05&#34;
time ./gatk_5_index_2016-05-05.sh &amp;&gt; gatk_5_index_2016-05-05.out
echo &#34;gatk_6_realign_bams_2016-05-05&#34;
time ./gatk_6_realign_bams_2016-05-05.sh &amp;&gt; gatk_6_realign_bams_2016-05-05.out
echo &#34;gatk_7_training_2016-05-05&#34;
time ./gatk_7_training_2016-05-05.sh &amp;&gt; gatk_7_training_2016-05-05.out
echo &#34;gatk_8_recalibrate_bams_2016-05-05&#34;
time ./gatk_8_recalibrate_bams_2016-05-05.sh &amp;&gt; gatk_8_recalibrate_bams_2016-05-05.out
echo &#34;gatk_9_final_variant_calling_2016-05-05&#34;
time ./gatk_9_final_variant_calling_2016-05-05.sh &amp;&gt; gatk_9_final_variant_calling_2016-05-05.out
</pre></div></div>
</div>
</div>
<div class="section" id="GATK-Run-on-122">
<h4>GATK Run on 122<a class="headerlink" href="#GATK-Run-on-122" title="Permalink to this headline">¶</a></h4>
<p>Data for the mother was obtained later. We did not re-train, but instead
ran the mother seperatately and then re-did the joint genotypeing steps.
Here are the scripts for this run.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [108]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="k">for</span> file in <span class="sb">`</span>ls -1 ../data/dwn_sample_atig_122/bash/*.sh<span class="sb">`</span>
 <span class="k">do</span>
  <span class="nb">echo</span> <span class="s2">&quot;------------------------------------------- </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="s2"> -------------------------------------------&quot;</span>
  cat <span class="nv">$file</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_0_prep_ref_2016-06-21.sh -------------------------------------------
#!/bin/bash
ln -s /home/dut/projects/tigris/genome_annotation/fasta/tigris_scaffolds_filt_10000.fa /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/
bwa index -a bwtsw /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa
samtools faidx /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa
java -jar /home/dut/bin/picard-tools-1.119/CreateSequenceDictionary.jar REFERENCE=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa OUTPUT=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.dict
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_1_map_to_ref_2016-06-21.sh -------------------------------------------
#!/bin/bash
(bwa mem -M -R &#34;@RG\tID:Atig_122_L21676_HJ2YHBCXX_2\tSM:Atig_122\tPL:illumina\tLB:L21676\tPU:2&#34; -t 8 /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_2_1_CTCAGA.0.33ds.fastq /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_2_2_CTCAGA.0.33ds.fastq | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig_122_L21676_HJ2YHBCXX_2 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_2.bam) &amp;
proc1=$!
(bwa mem -M -R &#34;@RG\tID:Atig_122_L21676_HJ2YHBCXX_1\tSM:Atig_122\tPL:illumina\tLB:L21676\tPU:1&#34; -t 8 /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_1_1_CTCAGA.0.33ds.fastq /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_1_2_CTCAGA.0.33ds.fastq | samtools view -Sb -@ 8 - | samtools sort -o -@ 8 - Atig_122_L21676_HJ2YHBCXX_1 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_1.bam) &amp;
proc2=$!
wait &#34;$proc1&#34; &#34;$proc2&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_2_dedup_individual_2016-06-21.sh -------------------------------------------
#!/bin/bash
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_2.bam O=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2.dedup.bam M=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2_metrics.txt &amp;
proc1=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/map_to_ref_output/Atig_122_L21676_HJ2YHBCXX_1.bam O=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1.dedup.bam M=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1_metrics.txt &amp;
proc2=$!
wait &#34;$proc1&#34; &#34;$proc2&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_3_merge_bams_2016-06-21.sh -------------------------------------------
#!/bin/bash
(samtools merge -@ 8 - /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_2.dedup.bam /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_individuals/Atig_122_L21676_HJ2YHBCXX_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig_122.temp -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/merged_bams/Atig_122.merged.bam; samtools index /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/merged_bams/Atig_122.merged.bam) &amp;
proc1=$!
wait &#34;$proc1&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_4_dedup_merged_2016-06-21.sh -------------------------------------------
#!/bin/bash
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/merged_bams/Atig_122.merged.bam O=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_merged/Atig_122.merged.dedup.bam M=/home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_merged/Atig_122.merged_metrics.txt &amp;
proc1=$!
wait &#34;$proc1&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_5_index_2016-06-21.sh -------------------------------------------
#!/bin/bash
samtools index -b /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_merged/Atig_122.merged.dedup.bam &amp;
proc1=$!
wait &#34;$proc1&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_6_realign_bams_2016-06-21.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_merged/Atig_122.merged.dedup.bam -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/dedup_merged/Atig_122.merged.dedup.bam -targetIntervals /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup_target_intervals.list -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam) &amp;
proc1=$!
wait &#34;$proc1&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_8_recalibrate_bams_2016-06-21.sh -------------------------------------------
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites ./../gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites ./../gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites ./../gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites ./../gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.before_table -after /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.after_table -plots /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122_final.plots) &amp;
proc1=$!
wait &#34;$proc1&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_9_final_variant_calling_2016-06-21.sh -------------------------------------------
#!/bin/bash
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf &amp;
#proc0=$!
#wait &#34;$proc0&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa --variant ../gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf --variant ../gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf --variant ../gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf -allSites -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_raw_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_filt_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_raw_indels.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_filt_indels.vcf
cat /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf | grep -v &#39;\./\.&#39;  &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf
cat /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_filt_snps.vcf | grep &#39;default_snp_filter&#39; | cut -f 1,2 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_snps.tsv
cat /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/Atig_122.final_filt_indels.vcf   | grep &#39;default_indel_filter&#39; | cut -f 1,2 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_indels.tsv
cat /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_indels.tsv /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_snps.tsv &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_positions.tsv
vcftools --vcf /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf --recode --out /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.filt --exclude-positions /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/final_variant_calling/joint_genotypes/bad_positions.tsv
------------------------------------------- ../data/dwn_sample_atig_122/bash/gatk_down_sample_2016-06-21.sh -------------------------------------------
#!/bin/bash
seqtk sample -s633 /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_2_1_CTCAGA.fastq.gz 0.33 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_2_1_CTCAGA.0.33ds.fastq; seqtk sample -s633 /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_2_2_CTCAGA.fastq.gz 0.33 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_2_2_CTCAGA.0.33ds.fastq &amp;
proc1=$!
seqtk sample -s633 /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_1_1_CTCAGA.fastq.gz 0.33 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_1_1_CTCAGA.0.33ds.fastq; seqtk sample -s633 /n/analysis/Baumann/rrs/MOLNG-1575/HJ2YHBCXX/s_1_2_CTCAGA.fastq.gz 0.33 &gt; /home/dut/projects/tigris/heterozygosity/dwn_sample_atig_122/preprocessing/down_sample/s_1_2_CTCAGA.0.33ds.fastq &amp;
proc2=$!
wait &#34;$proc1&#34; &#34;$proc2&#34;
------------------------------------------- ../data/dwn_sample_atig_122/bash/master_2016-06-21.sh -------------------------------------------
#!/bin/bash
echo &#34;gatk_0_prep_ref_2016-06-21&#34;
time ./gatk_0_prep_ref_2016-06-21.sh &amp;&gt; gatk_0_prep_ref_2016-06-21.out
echo &#34;gatk_down_sample_2016-06-21&#34;
time ./gatk_down_sample_2016-06-21.sh &amp;&gt; gatk_down_sample_2016-06-21.out
echo &#34;gatk_1_map_to_ref_2016-06-21&#34;
time ./gatk_1_map_to_ref_2016-06-21.sh &amp;&gt; gatk_1_map_to_ref_2016-06-21.out
echo &#34;gatk_2_dedup_individual_2016-06-21&#34;
time ./gatk_2_dedup_individual_2016-06-21.sh &amp;&gt; gatk_2_dedup_individual_2016-06-21.out
echo &#34;gatk_3_merge_bams_2016-06-21&#34;
time ./gatk_3_merge_bams_2016-06-21.sh &amp;&gt; gatk_3_merge_bams_2016-06-21.out
echo &#34;gatk_4_dedup_merged_2016-06-21&#34;
time ./gatk_4_dedup_merged_2016-06-21.sh &amp;&gt; gatk_4_dedup_merged_2016-06-21.out
echo &#34;gatk_5_index_2016-06-21&#34;
time ./gatk_5_index_2016-06-21.sh &amp;&gt; gatk_5_index_2016-06-21.out
echo &#34;gatk_6_realign_bams_2016-06-21&#34;
time ./gatk_6_realign_bams_2016-06-21.sh &amp;&gt; gatk_6_realign_bams_2016-06-21.out
echo &#34;gatk_8_recalibrate_bams_2016-06-21&#34;
time ./gatk_8_recalibrate_bams_2016-06-21.sh &amp;&gt; gatk_8_recalibrate_bams_2016-06-21.out
echo &#34;gatk_9_final_variant_calling_2016-06-21&#34;
time ./gatk_9_final_variant_calling_2016-06-21.sh &amp;&gt; gatk_9_final_variant_calling_2016-06-21.out
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="GATK-Initial-Results-Figure-(Generated-by-Morgan-Weichert)">
<h3>GATK Initial Results Figure (Generated by Morgan Weichert)<a class="headerlink" href="#GATK-Initial-Results-Figure-(Generated-by-Morgan-Weichert)" title="Permalink to this headline">¶</a></h3>
<p>These figures were generated by Morgan Weichert. They summarize the
heterozygous sites detected by GATK with a few hard filters applied.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [109]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp /home/msr/Projects/marmorata_genome/gatk_analysis_A_tigris8450_Atig_122_Atig003_Atig001/fig/top5_8-36x_het_sites_partheno_mom.papersize.png.pdf ../fig2/SupplementalFigure4.pdf
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [110]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">cp</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">msr</span><span class="o">/</span><span class="n">Projects</span><span class="o">/</span><span class="n">marmorata_genome</span><span class="o">/</span><span class="n">gatk_analysis_A_tigris8450_Atig_122_Atig003_Atig001</span><span class="o">/</span><span class="n">fig</span><span class="o">/</span><span class="n">top5_8</span><span class="o">-</span><span class="mi">36</span><span class="n">x_het_sites_partheno_mom</span><span class="o">.</span><span class="n">png</span> <span class="o">../</span><span class="n">fig</span><span class="o">/</span><span class="n">initial_gatk_results</span><span class="o">.</span><span class="n">png</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [111]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;/home/msr/Projects/marmorata_genome/gatk_analysis_A_tigris8450_Atig_122_Atig003_Atig001/fig/top5_8-36x_het_sites_partheno_mom.png&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[111]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_174_0.png" src="_images/marmorata_genome_paper_figures_3_174_0.png" />
</div>
</div>
</div>
<div class="section" id="Partial-GATK-pipeline-Run-on-Additional-Animals">
<h3>Partial GATK pipeline Run on Additional Animals<a class="headerlink" href="#Partial-GATK-pipeline-Run-on-Additional-Animals" title="Permalink to this headline">¶</a></h3>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [112]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">molng2139</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;/n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/Sample_Report.csv&#39;</span><span class="p">)</span>
<span class="n">molng2140</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;/n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/Sample_Report.csv&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [113]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">newAnimalSequencingDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">molng2139</span><span class="p">,</span> <span class="n">molng2140</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [114]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">newAnimalSequencingDF</span><span class="p">[</span><span class="s1">&#39;estimated_coverage&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">newAnimalSequencingDF</span><span class="p">[</span><span class="s1">&#39;TotalReads&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">newAnimalSequencingDF</span><span class="p">[</span><span class="s1">&#39;ReadLength&#39;</span><span class="p">])</span> <span class="o">/</span> <span class="mf">1633406540.0</span>
<span class="n">newAnimalSequencingDF</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;SampleName&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()[[</span><span class="s1">&#39;SampleName&#39;</span><span class="p">,</span> <span class="s1">&#39;estimated_coverage&#39;</span><span class="p">]]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[114]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>SampleName</th>
      <th>estimated_coverage</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>A.tig_12512</td>
      <td>18.999300</td>
    </tr>
    <tr>
      <th>1</th>
      <td>A.tig_12513</td>
      <td>19.207432</td>
    </tr>
    <tr>
      <th>2</th>
      <td>A.tig_9721</td>
      <td>18.722871</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Atig_4278</td>
      <td>19.743835</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Atig_6993</td>
      <td>18.510542</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Atig_9177</td>
      <td>20.961326</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [115]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
more ../data/gatk_MOLNG-2139_MOLNG-2140/*txt <span class="p">|</span> cat

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/flowcells.txt
::::::::::::::
H2LVWBCX2
H2JNNBCX2
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_command.txt
::::::::::::::
./../../bin/gatk6_pipeline.py -molngs molng.txt -samples samples.txt -flowcells flowcells.txt -new_sample_report -train
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/molng.txt
::::::::::::::
/n/analysis/Baumann/rrs/MOLNG-2139/
/n/analysis/Baumann/rrs/MOLNG-2140/
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/samples.txt
::::::::::::::
Atig_4278
Atig_6993
Atig_9177
A.tig_12512
A.tig_12513
A.tig_9721
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [116]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
more ../data/gatk_MOLNG-2139_MOLNG-2140/*sh <span class="p">|</span> cat
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_0_prep_ref_2017-11-10.sh
::::::::::::::
#!/bin/bash
ln -s /home/dut/projects/tigris/genome_annotation/fasta/tigris_scaffolds_filt_10000.fa /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/
bwa index -a bwtsw /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa
samtools faidx /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa
java -jar /home/dut/bin/picard-tools-1.119/CreateSequenceDictionary.jar REFERENCE=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa OUTPUT=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.dict
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_1_map_to_ref_2017-11-10.sh
::::::::::::::
#!/bin/bash
(bwa mem -M -R &#34;@RG\tID:Atig_4278_L30700_H2LVWBCX2_1\tSM:Atig_4278\tPL:illumina\tLB:L30700\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_1_GCGCTA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_2_GCGCTA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_4278_L30700_H2LVWBCX2_1.bam) &amp;
proc1=$!
(bwa mem -M -R &#34;@RG\tID:Atig_4278_L30700_H2LVWBCX2_2\tSM:Atig_4278\tPL:illumina\tLB:L30700\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_1_GCGCTA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_2_GCGCTA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_4278_L30700_H2LVWBCX2_2.bam) &amp;
proc2=$!
(bwa mem -M -R &#34;@RG\tID:A.tig_12512_L30701_H2JNNBCX2_1\tSM:A.tig_12512\tPL:illumina\tLB:L30701\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_1_ATTCCT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_2_ATTCCT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12512_L30701_H2JNNBCX2_1.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(bwa mem -M -R &#34;@RG\tID:A.tig_12512_L30701_H2JNNBCX2_2\tSM:A.tig_12512\tPL:illumina\tLB:L30701\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_1_ATTCCT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_2_ATTCCT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12512_L30701_H2JNNBCX2_2.bam) &amp;
proc4=$!
(bwa mem -M -R &#34;@RG\tID:A.tig_12513_L30702_H2JNNBCX2_1\tSM:A.tig_12513\tPL:illumina\tLB:L30702\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_1_CGGAAT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_2_CGGAAT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12513_L30702_H2JNNBCX2_1.bam) &amp;
proc5=$!
(bwa mem -M -R &#34;@RG\tID:A.tig_12513_L30702_H2JNNBCX2_2\tSM:A.tig_12513\tPL:illumina\tLB:L30702\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_1_CGGAAT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_2_CGGAAT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12513_L30702_H2JNNBCX2_2.bam) &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
(bwa mem -M -R &#34;@RG\tID:A.tig_9721_L30703_H2JNNBCX2_1\tSM:A.tig_9721\tPL:illumina\tLB:L30703\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_1_TCATTC.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_1_2_TCATTC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_9721_L30703_H2JNNBCX2_1.bam) &amp;
proc7=$!
(bwa mem -M -R &#34;@RG\tID:A.tig_9721_L30703_H2JNNBCX2_2\tSM:A.tig_9721\tPL:illumina\tLB:L30703\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_1_TCATTC.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2140/H2JNNBCX2/s_2_2_TCATTC.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_9721_L30703_H2JNNBCX2_2.bam) &amp;
proc8=$!
(bwa mem -M -R &#34;@RG\tID:Atig_6993_L30699_H2LVWBCX2_2\tSM:Atig_6993\tPL:illumina\tLB:L30699\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_1_CACTCA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_2_CACTCA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_6993_L30699_H2LVWBCX2_2.bam) &amp;
proc9=$!
wait &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34;
(bwa mem -M -R &#34;@RG\tID:Atig_6993_L30699_H2LVWBCX2_1\tSM:Atig_6993\tPL:illumina\tLB:L30699\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_1_CACTCA.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_2_CACTCA.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_6993_L30699_H2LVWBCX2_1.bam) &amp;
proc10=$!
(bwa mem -M -R &#34;@RG\tID:Atig_9177_L30698_H2LVWBCX2_2\tSM:Atig_9177\tPL:illumina\tLB:L30698\tPU:2&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_1_ACTGAT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_2_2_ACTGAT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_9177_L30698_H2LVWBCX2_2.bam) &amp;
proc11=$!
(bwa mem -M -R &#34;@RG\tID:Atig_9177_L30698_H2LVWBCX2_1\tSM:Atig_9177\tPL:illumina\tLB:L30698\tPU:1&#34; -t 8 /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_1_ACTGAT.fastq.gz /n/analysis/Baumann/rrs/MOLNG-2139/H2LVWBCX2/s_1_2_ACTGAT.fastq.gz | samtools view -Sb -@ 8 - | samtools sort - -@ 8 -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_9177_L30698_H2LVWBCX2_1.bam) &amp;
proc12=$!
wait &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34; &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34; &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_2_dedup_individual_2017-11-10.sh
::::::::::::::
#!/bin/bash
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_4278_L30700_H2LVWBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_1_metrics.txt &amp;
proc1=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_4278_L30700_H2LVWBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_2_metrics.txt &amp;
proc2=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12512_L30701_H2JNNBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_1_metrics.txt &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12512_L30701_H2JNNBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_2_metrics.txt &amp;
proc4=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12513_L30702_H2JNNBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_1_metrics.txt &amp;
proc5=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_12513_L30702_H2JNNBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_2_metrics.txt &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_9721_L30703_H2JNNBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_1_metrics.txt &amp;
proc7=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/A.tig_9721_L30703_H2JNNBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_2_metrics.txt &amp;
proc8=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_6993_L30699_H2LVWBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_2_metrics.txt &amp;
proc9=$!
wait &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34;
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_6993_L30699_H2LVWBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_1_metrics.txt &amp;
proc10=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_9177_L30698_H2LVWBCX2_2.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_2.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_2_metrics.txt &amp;
proc11=$!
java -Xmx4g -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/map_to_ref_output/Atig_9177_L30698_H2LVWBCX2_1.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_1.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_1_metrics.txt &amp;
proc12=$!
wait &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34; &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34; &#34;$proc10&#34; &#34;$proc11&#34; &#34;$proc12&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_3_merge_bams_2017-11-10.sh
::::::::::::::
#!/bin/bash
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_1.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_4278_L30700_H2LVWBCX2_2.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig_4278.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_4278.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_4278.merged.bam) &amp;
proc1=$!
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_1.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12512_L30701_H2JNNBCX2_2.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/A.tig_12512.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12512.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12512.merged.bam) &amp;
proc2=$!
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_1.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_12513_L30702_H2JNNBCX2_2.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/A.tig_12513.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12513.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12513.merged.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_1.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/A.tig_9721_L30703_H2JNNBCX2_2.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/A.tig_9721.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_9721.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_9721.merged.bam) &amp;
proc4=$!
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_2.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_6993_L30699_H2LVWBCX2_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig_6993.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_6993.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_6993.merged.bam) &amp;
proc5=$!
(samtools merge -@ 8 - /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_2.dedup.bam /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_individuals/Atig_9177_L30698_H2LVWBCX2_1.dedup.bam | samtools sort - -m 10G -@ 8 -T /scratch/dut/Atig_9177.temp -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_9177.merged.bam; samtools index /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_9177.merged.bam) &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_4_dedup_merged_2017-11-10.sh
::::::::::::::
#!/bin/bash
java -Xmx4g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_4278.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_4278.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_4278.merged_metrics.txt &amp;
proc1=$!
java -Xmx4g -Djava.io.tmpdir=/scratch/dut  -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12512.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12512.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12512.merged_metrics.txt &amp;
proc2=$!
java -Xmx4g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_12513.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12513.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12513.merged_metrics.txt &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
java -Xmx4g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/A.tig_9721.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_9721.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_9721.merged_metrics.txt &amp;
proc4=$!
java -Xmx4g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_6993.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_6993.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_6993.merged_metrics.txt &amp;
proc5=$!
java -Xmx4g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/picard-tools-1.119//MarkDuplicates.jar I=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/merged_bams/Atig_9177.merged.bam O=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_9177.merged.dedup.bam M=/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_9177.merged_metrics.txt &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_5_index_2017-11-10.sh
::::::::::::::
#!/bin/bash
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_4278.merged.dedup.bam &amp;
proc1=$!
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12512.merged.dedup.bam &amp;
proc2=$!
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12513.merged.dedup.bam &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_9721.merged.dedup.bam &amp;
proc4=$!
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_6993.merged.dedup.bam &amp;
proc5=$!
samtools index -b /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_9177.merged.dedup.bam &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_6_realign_bams_2017-11-10.sh
::::::::::::::
#!/bin/bash
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_4278.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_4278.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam) &amp;
proc1=$!
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12512.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12512.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam) &amp;
proc2=$!
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12513.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_12513.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_9721.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/A.tig_9721.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam) &amp;
proc4=$!
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_6993.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_6993.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam) &amp;
proc5=$!
(java -Xmx3g -Djava.io.tmpdir=/scratch/dut -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_9177.merged.dedup.bam -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup_target_intervals.list -nt 8; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T IndelRealigner -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/dedup_merged/Atig_9177.merged.dedup.bam -targetIntervals /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup_target_intervals.list -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam) &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_7_training_2017-11-10.sh
::::::::::::::
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_filt_indels.vcf) &amp;
proc0=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_filt_indels.vcf) &amp;
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_filt_indels.vcf) &amp;
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_filt_indels.vcf) &amp;
proc3=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_filt_indels.vcf) &amp;
proc4=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_filt_indels.vcf) &amp;
proc5=$!
wait &#34;$proc0&#34; &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_filt_snps.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.0_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.0_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.0_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.0_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.0_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.0_filt_indels.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_0.plots) &amp;
proc6=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_0.plots) &amp;
proc7=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_0.plots) &amp;
proc8=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_0.plots) &amp;
proc9=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_0.plots) &amp;
proc10=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_0.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_0.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_0.plots) &amp;
proc11=$!
wait &#34;$proc6&#34; &#34;$proc7&#34; &#34;$proc8&#34; &#34;$proc9&#34; &#34;$proc10&#34; &#34;$proc11&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_filt_indels.vcf) &amp;
proc12=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_filt_indels.vcf) &amp;
proc13=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_filt_indels.vcf) &amp;
proc14=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_filt_indels.vcf) &amp;
proc15=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_filt_indels.vcf) &amp;
proc16=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_filt_indels.vcf) &amp;
proc17=$!
wait &#34;$proc12&#34; &#34;$proc13&#34; &#34;$proc14&#34; &#34;$proc15&#34; &#34;$proc16&#34; &#34;$proc17&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_filt_snps.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.1_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.1_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.1_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.1_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.1_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.1_filt_indels.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_1.plots) &amp;
proc18=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_1.plots) &amp;
proc19=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_1.plots) &amp;
proc20=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_1.plots) &amp;
proc21=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_1.plots) &amp;
proc22=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_1.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_1.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_1.plots) &amp;
proc23=$!
wait &#34;$proc18&#34; &#34;$proc19&#34; &#34;$proc20&#34; &#34;$proc21&#34; &#34;$proc22&#34; &#34;$proc23&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_filt_indels.vcf) &amp;
proc24=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_filt_indels.vcf) &amp;
proc25=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_filt_indels.vcf) &amp;
proc26=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_filt_indels.vcf) &amp;
proc27=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_filt_indels.vcf) &amp;
proc28=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_filt_indels.vcf) &amp;
proc29=$!
wait &#34;$proc24&#34; &#34;$proc25&#34; &#34;$proc26&#34; &#34;$proc27&#34; &#34;$proc28&#34; &#34;$proc29&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_filt_snps.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.2_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.2_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.2_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.2_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.2_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.2_filt_indels.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278_2.plots) &amp;
proc30=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512_2.plots) &amp;
proc31=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513_2.plots) &amp;
proc32=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721_2.plots) &amp;
proc33=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993_2.plots) &amp;
proc34=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_2.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_2.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177_2.plots) &amp;
proc35=$!
wait &#34;$proc30&#34; &#34;$proc31&#34; &#34;$proc32&#34; &#34;$proc33&#34; &#34;$proc34&#34; &#34;$proc35&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_4278.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_filt_indels.vcf) &amp;
proc36=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12512.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_filt_indels.vcf) &amp;
proc37=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_12513.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_filt_indels.vcf) &amp;
proc38=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/A.tig_9721.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_filt_indels.vcf) &amp;
proc39=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_6993.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_filt_indels.vcf) &amp;
proc40=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/recal_bams/Atig_9177.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_var.vcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_var.vcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_filt_indels.vcf) &amp;
proc41=$!
wait &#34;$proc36&#34; &#34;$proc37&#34; &#34;$proc38&#34; &#34;$proc39&#34; &#34;$proc40&#34; &#34;$proc41&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_filt_snps.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_filt_snps.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_4278.3_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12512.3_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_12513.3_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/A.tig_9721.3_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_6993.3_filt_indels.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/Atig_9177.3_filt_indels.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_8_recalibrate_bams_2017-11-10.sh
::::::::::::::
#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_4278.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278_final.plots) &amp;
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12512.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512_final.plots) &amp;
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_12513.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513_final.plots) &amp;
proc3=$!
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34;
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/A.tig_9721.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721_final.plots) &amp;
proc4=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_6993.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993_final.plots) &amp;
proc5=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/Atig_9177.merged.dedup.realigned.bam -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_snps_3.recode.vcf -knownSites /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/training/variant_calling/all_indels_3.recode.vcf -BQSR /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.before_table -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -before /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.before_table -after /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.after_table -plots /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177_final.plots) &amp;
proc6=$!
wait &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
wait &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34; &#34;$proc6&#34;
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/gatk_9_final_variant_calling_2017-11-10.sh
::::::::::::::
#!/bin/bash
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_4278.final_raw_var.vcf &amp;
proc0=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12512.final_raw_var.vcf &amp;
proc1=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12513.final_raw_var.vcf &amp;
proc2=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_9721.final_raw_var.vcf &amp;
proc3=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_6993.final_raw_var.vcf &amp;
proc4=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_9177.final_raw_var.vcf &amp;
proc5=$!
wait &#34;$proc0&#34; &#34;$proc1&#34; &#34;$proc2&#34; &#34;$proc3&#34; &#34;$proc4&#34; &#34;$proc5&#34;
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_4278.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12512.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12513.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_9721.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_6993.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_9177.final_raw_var.vcf -allSites -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_snps.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0&#34; --filterName &#34;default_snp_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_indels.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_indels.vcf --filterExpression &#34;QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0&#34; --filterName &#34;default_indel_filter&#34; -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_indels.vcf
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf | grep -v &#39;\./\.&#39;  &gt; /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.gvcf
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_snps.vcf | grep &#39;default_snp_filter&#39; | cut -f 1,2 &gt; /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_snps.tsv
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_indels.vcf   | grep &#39;default_indel_filter&#39; | cut -f 1,2 &gt; /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_indels.tsv
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_indels.tsv /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_snps.tsv &gt; /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_positions.tsv
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.gvcf --recode --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.filt --exclude-positions /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_positions.tsv
::::::::::::::
../data/gatk_MOLNG-2139_MOLNG-2140/master_2017-11-10.sh
::::::::::::::
#!/bin/bash
#echo &#34;gatk_0_prep_ref_2017-11-10&#34;
#time ./gatk_0_prep_ref_2017-11-10.sh &amp;&gt; gatk_0_prep_ref_2017-11-10.out
#echo &#34;gatk_1_map_to_ref_2017-11-10&#34;
#time ./gatk_1_map_to_ref_2017-11-10.sh &amp;&gt; gatk_1_map_to_ref_2017-11-10.out
#echo &#34;gatk_2_dedup_individual_2017-11-10&#34;
#time ./gatk_2_dedup_individual_2017-11-10.sh &amp;&gt; gatk_2_dedup_individual_2017-11-10.out
#echo &#34;gatk_3_merge_bams_2017-11-10&#34;
#time ./gatk_3_merge_bams_2017-11-10.sh &amp;&gt; gatk_3_merge_bams_2017-11-10.out
echo &#34;gatk_4_dedup_merged_2017-11-10&#34;
time ./gatk_4_dedup_merged_2017-11-10.sh &amp;&gt; gatk_4_dedup_merged_2017-11-10.out2
echo &#34;gatk_5_index_2017-11-10&#34;
time ./gatk_5_index_2017-11-10.sh &amp;&gt; gatk_5_index_2017-11-10.out2
echo &#34;gatk_6_realign_bams_2017-11-10&#34;
time ./gatk_6_realign_bams_2017-11-10.sh &amp;&gt; gatk_6_realign_bams_2017-11-10.out2
#echo &#34;gatk_7_training_2017-11-10&#34;
#time ./gatk_7_training_2017-11-10.sh &amp;&gt; gatk_7_training_2017-11-10.out
#echo &#34;gatk_8_recalibrate_bams_2017-11-10&#34;
#time ./gatk_8_recalibrate_bams_2017-11-10.sh &amp;&gt; gatk_8_recalibrate_bams_2017-11-10.out
#echo &#34;gatk_9_final_variant_calling_2017-11-10&#34;
#time ./gatk_9_final_variant_calling_2017-11-10.sh &amp;&gt; gatk_9_final_variant_calling_2017-11-10.out
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [117]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data/pysam
<span class="c1"># for bam in ../gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/*.bam;</span>
<span class="c1"># do</span>
<span class="c1">#     nohup ./../../bin/pysam_profiler.py ../gatk_MOLNG-2139_MOLNG-2140/preprocessing/realigned_merged_bams/${bam} ../gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa &amp;</span>
<span class="c1"># done</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [118]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data/pysam
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; A.tig_12512.merged.dedup.realigned.prof &gt; A.tig_12512.merged.dedup.realigned.prof.not_hom &amp;</span>
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; A.tig_12513.merged.dedup.realigned.prof &gt; A.tig_12513.merged.dedup.realigned.prof.not_hom &amp;</span>
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; A.tig_9721.merged.dedup.realigned.prof &gt; A.tig_9721.merged.dedup.realigned.prof.not_hom &amp;</span>
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; Atig_4278.merged.dedup.realigned.prof &gt; Atig_4278.merged.dedup.realigned.prof.not_hom &amp;</span>
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; Atig_6993.merged.dedup.realigned.prof &gt; Atig_6993.merged.dedup.realigned.prof.not_hom &amp;</span>
<span class="c1">#awk &#39;$8!=&quot;Homozygous&quot;&#39; Atig_9177.merged.dedup.realigned.prof &gt; Atig_9177.merged.dedup.realigned.prof.not_hom &amp;</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [119]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data/pysam
<span class="k">for</span> prof in <span class="sb">`</span>ls -1 Atig_*.prof <span class="p">|</span> grep -v <span class="m">122</span><span class="p">;</span> ls -1 A.tig*.prof<span class="sb">`</span>
<span class="k">do</span>
<span class="nb">echo</span> <span class="nv">$prof</span>
<span class="c1">#./../../bin/count_profiles.py $prof &amp;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Atig_4278.merged.dedup.realigned.prof
Atig_6993.merged.dedup.realigned.prof
Atig_9177.merged.dedup.realigned.prof
A.tig_12512.merged.dedup.realigned.prof
A.tig_12513.merged.dedup.realigned.prof
A.tig_9721.merged.dedup.realigned.prof
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [120]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../data/pysam
<span class="k">for</span> prof in <span class="sb">`</span>ls -1 Atig_*.prof.not_hom <span class="p">|</span> grep -v <span class="m">122</span><span class="p">;</span> ls -1 A.tig*.prof.not_hom<span class="sb">`</span>
<span class="k">do</span>
<span class="c1">#echo $prof</span>
<span class="nb">echo</span> <span class="s2">&quot;./../../bin/scan_profile_no_lim.py </span><span class="nv">$prof</span><span class="s2"> 20 &amp;&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
./../../bin/scan_profile_no_lim.py Atig_4278.merged.dedup.realigned.prof.not_hom 20 &amp;
./../../bin/scan_profile_no_lim.py Atig_6993.merged.dedup.realigned.prof.not_hom 20 &amp;
./../../bin/scan_profile_no_lim.py Atig_9177.merged.dedup.realigned.prof.not_hom 20 &amp;
./../../bin/scan_profile_no_lim.py A.tig_12512.merged.dedup.realigned.prof.not_hom 20 &amp;
./../../bin/scan_profile_no_lim.py A.tig_12513.merged.dedup.realigned.prof.not_hom 20 &amp;
./../../bin/scan_profile_no_lim.py A.tig_9721.merged.dedup.realigned.prof.not_hom 20 &amp;
</pre></div></div>
</div>
</div>
</div>
<div class="section" id="Coverage-Accross-Profiles-and-Affect-on-Heterozygosity">
<h2>Coverage Accross Profiles and Affect on Heterozygosity<a class="headerlink" href="#Coverage-Accross-Profiles-and-Affect-on-Heterozygosity" title="Permalink to this headline">¶</a></h2>
<p>The following python script will convert an alignment bam file (indexed)
into a tsv file of profiles.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/pysam_profiler.py</span>
<span class="c1">#!/usr/bin/env python</span>
<span class="c1">#Author: Duncan Tormey</span>
<span class="c1">#Email: dut@stowers.org or duncantormey@gmail.com</span>
<span class="c1">#################################################</span>
<span class="c1"># This script takes a bam file and a reference  #</span>
<span class="c1"># genome file as input. It uses the pysam       #</span>
<span class="c1"># python package to pull out the per position   #</span>
<span class="c1"># nucleotide coverage, and determes the statust #</span>
<span class="c1"># of the position(Transition, Transversion,     #</span>
<span class="c1"># Homozygous or Unknown)                        #</span>
<span class="c1">#################################################</span>

<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">print_function</span>
<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">division</span>
<span class="kn">import</span> <span class="nn">pysam</span>
<span class="kn">import</span> <span class="nn">pysamstats</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">os</span>

<span class="k">def</span> <span class="nf">get_prof_status</span><span class="p">(</span><span class="n">prof</span><span class="p">,</span> <span class="n">bases</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;T&#39;</span><span class="p">]):</span>
    <span class="n">bases</span> <span class="o">=</span> <span class="p">[</span><span class="n">bases</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">n</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">prof</span><span class="p">)</span> <span class="k">if</span> <span class="n">n</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">bases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">bases</span> <span class="o">==</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;G&#39;</span><span class="p">]</span> <span class="ow">or</span> <span class="n">bases</span> <span class="o">==</span> <span class="p">[</span><span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;T&#39;</span><span class="p">]:</span>
            <span class="n">status</span> <span class="o">=</span> <span class="s1">&#39;Transition&#39;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">status</span> <span class="o">=</span> <span class="s1">&#39;Transversion&#39;</span>
    <span class="k">elif</span> <span class="nb">len</span><span class="p">(</span><span class="n">bases</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">status</span> <span class="o">=</span> <span class="s1">&#39;Homozygous&#39;</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">status</span> <span class="o">=</span> <span class="s1">&#39;Unknown&#39;</span>

    <span class="n">prof</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="nb">sum</span><span class="p">(</span><span class="n">prof</span><span class="p">),</span> <span class="n">status</span><span class="p">])</span>

    <span class="k">return</span>  <span class="n">prof</span>

<span class="k">def</span> <span class="nf">write_profiles</span><span class="p">(</span><span class="n">bam_path</span><span class="p">,</span> <span class="n">ref_path</span><span class="p">):</span>
    <span class="n">outfile</span> <span class="o">=</span> <span class="s1">&#39;./&#39;</span> <span class="o">+</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">bam_path</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;.prof&#39;</span><span class="p">)</span>
    <span class="n">bam</span> <span class="o">=</span> <span class="n">pysam</span><span class="o">.</span><span class="n">AlignmentFile</span><span class="p">(</span><span class="n">bam_path</span><span class="p">)</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">outfile</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;chrom</span><span class="se">\t</span><span class="s1">pos</span><span class="se">\t</span><span class="s1">A</span><span class="se">\t</span><span class="s1">C</span><span class="se">\t</span><span class="s1">G</span><span class="se">\t</span><span class="s1">T</span><span class="se">\t</span><span class="s1">cov</span><span class="se">\t</span><span class="s1">status</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">rec</span> <span class="ow">in</span> <span class="n">pysamstats</span><span class="o">.</span><span class="n">stat_variation</span><span class="p">(</span><span class="n">bam</span><span class="p">,</span> <span class="n">ref_path</span><span class="p">,</span> <span class="n">pad</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">max_depth</span><span class="o">=</span><span class="mi">1000000</span><span class="p">):</span>
            <span class="n">prof</span> <span class="o">=</span> <span class="p">[</span><span class="n">rec</span><span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">],</span> <span class="n">rec</span><span class="p">[</span><span class="s1">&#39;C&#39;</span><span class="p">],</span> <span class="n">rec</span><span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">],</span> <span class="n">rec</span><span class="p">[</span><span class="s1">&#39;T&#39;</span><span class="p">]]</span>
            <span class="n">prof</span> <span class="o">=</span> <span class="n">get_prof_status</span><span class="p">(</span><span class="n">prof</span><span class="p">)</span>
            <span class="n">prof</span><span class="p">[:</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">rec</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">],</span> <span class="n">rec</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]]</span>
            <span class="n">fh</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span><span class="n">prof</span><span class="p">)))</span>


<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s2">&quot;__main__&quot;</span><span class="p">:</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span> <span class="o">==</span> <span class="mi">3</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span>
        <span class="n">write_profiles</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">2</span><span class="p">])</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="s1">&#39;usage: pysam_profiler.py /path/to/alignment.bam /path/to/reference.fa&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<p>The following python script takes a tsv profiles file as input and
counts the unique profiles across the genome.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/count_profiles.py</span>
<span class="c1">#!/usr/bin/env python</span>
<span class="c1">#Author: Duncan Tormey</span>
<span class="c1">#Email: dut@stowers.org or duncantormey@gmail.com</span>
<span class="c1">##################################################</span>
<span class="c1"># This script takes the tsv file output by      #</span>
<span class="c1"># pysam_profiler.py and counts the occurence of #</span>
<span class="c1"># each unique profile, ignoring scaffold and    #</span>
<span class="c1"># position.                                     #</span>
<span class="c1">#################################################</span>


<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">print_function</span>
<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">division</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">Counter</span>


<span class="k">def</span> <span class="nf">count_profiles</span><span class="p">(</span><span class="n">prof_path</span><span class="p">):</span>
    <span class="n">outfile</span> <span class="o">=</span> <span class="s1">&#39;./&#39;</span> <span class="o">+</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">basename</span><span class="p">(</span><span class="n">prof_path</span><span class="p">)</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;.prof&#39;</span><span class="p">,</span> <span class="s1">&#39;.prof_counts&#39;</span><span class="p">)</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="n">Counter</span><span class="p">()</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">prof_path</span><span class="p">,</span> <span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fh</span><span class="p">:</span>
        <span class="n">fh</span><span class="o">.</span><span class="n">next</span><span class="p">()</span> <span class="c1"># skip header</span>
        <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">fh</span><span class="p">:</span>
            <span class="n">line</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()</span>
            <span class="n">prof</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">line</span><span class="p">[</span><span class="mi">2</span><span class="p">:])</span>
            <span class="n">counts</span><span class="p">[</span><span class="n">prof</span><span class="p">]</span> <span class="o">+=</span> <span class="mi">1</span>

    <span class="k">return</span> <span class="n">counts</span><span class="p">,</span> <span class="n">outfile</span>


<span class="k">def</span> <span class="nf">write_profiles</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span> <span class="n">outfile</span><span class="p">):</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">outfile</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">fho</span><span class="p">:</span>
        <span class="n">fho</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;occurence</span><span class="se">\t</span><span class="s1">A</span><span class="se">\t</span><span class="s1">C</span><span class="se">\t</span><span class="s1">G</span><span class="se">\t</span><span class="s1">T</span><span class="se">\t</span><span class="s1">cov</span><span class="se">\t</span><span class="s1">status</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">prof</span><span class="p">,</span> <span class="n">count</span> <span class="ow">in</span> <span class="n">counts</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
            <span class="n">fho</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="se">\t</span><span class="si">%s</span><span class="se">\n</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">count</span><span class="p">),</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">prof</span><span class="p">))))</span>


<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s2">&quot;__main__&quot;</span><span class="p">:</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span> <span class="o">==</span> <span class="mi">2</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span>
        <span class="n">counts</span><span class="p">,</span> <span class="n">outfile</span> <span class="o">=</span> <span class="n">count_profiles</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
        <span class="n">write_profiles</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span> <span class="n">outfile</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">print</span><span class="p">(</span><span class="s1">&#39;usage: count_profiles /path/to/profile &#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [121]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">pysamCountsPaths</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Atig001&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig001.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A_tigris8450.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig003&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig003.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_122&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_122.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_12512.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_12513.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_9721&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_9721.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_4278&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_4278.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_6993.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_9177.merged.dedup.realigned.prof_counts&#39;</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [122]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">profCountsDict</span> <span class="o">=</span> <span class="p">{</span><span class="n">a</span> <span class="p">:</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">p</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="mi">0</span><span class="p">)</span> <span class="k">for</span> <span class="n">a</span><span class="p">,</span><span class="n">p</span> <span class="ow">in</span> <span class="n">pysamCountsPaths</span><span class="o">.</span><span class="n">items</span><span class="p">()}</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [123]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">profCountsDict</span><span class="o">.</span><span class="n">keys</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[123]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>[&#39;A_tigris8450&#39;,
 &#39;Atig_122&#39;,
 &#39;Atig_4278&#39;,
 &#39;Atig003&#39;,
 &#39;Atig001&#39;,
 &#39;A.tig_12512&#39;,
 &#39;A.tig_12513&#39;,
 &#39;A.tig_9721&#39;,
 &#39;Atig_6993&#39;,
 &#39;Atig_9177&#39;]
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [124]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">covValuesDict</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">profCountsDict</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="n">profCountsDict</span><span class="p">[</span><span class="n">animal</span><span class="p">][</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">animal</span>
    <span class="n">avg_cov</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">profCountsDict</span><span class="p">[</span><span class="n">animal</span><span class="p">][</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">*</span><span class="n">profCountsDict</span><span class="p">[</span><span class="n">animal</span><span class="p">][</span><span class="s1">&#39;occurence&#39;</span><span class="p">])</span><span class="o">/</span><span class="n">profCountsDict</span><span class="p">[</span><span class="n">animal</span><span class="p">][</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
    <span class="n">profCountsDict</span><span class="p">[</span><span class="n">animal</span><span class="p">][</span><span class="s1">&#39;avg_cov&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">avg_cov</span>
    <span class="n">covValuesDict</span><span class="p">[</span><span class="n">animal</span><span class="p">]</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">round_up_to_even</span><span class="p">(</span><span class="n">avg_cov</span><span class="p">))</span>
    <span class="k">print</span><span class="p">(</span><span class="n">animal</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">)</span>


<span class="n">profCountDF</span><span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">profCountsDict</span><span class="o">.</span><span class="n">values</span><span class="p">())</span>
<span class="n">profCountDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
A_tigris8450 17.5365944457
Atig_122 18.8524151373
Atig_4278 18.9034922139
Atig003 18.3134659875
Atig001 15.9133728851
A.tig_12512 18.2086642398
A.tig_12513 18.4186990258
A.tig_9721 17.907845091
Atig_6993 17.6018239458
Atig_9177 20.0751320379
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[124]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>occurence</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>animal</th>
      <th>avg_cov</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>38</td>
      <td>306</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>307</td>
      <td>Transversion</td>
      <td>A_tigris8450</td>
      <td>17.536594</td>
    </tr>
    <tr>
      <th>1</th>
      <td>3</td>
      <td>13</td>
      <td>0</td>
      <td>199</td>
      <td>0</td>
      <td>212</td>
      <td>Transition</td>
      <td>A_tigris8450</td>
      <td>17.536594</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>0</td>
      <td>58</td>
      <td>6</td>
      <td>4</td>
      <td>68</td>
      <td>Unknown</td>
      <td>A_tigris8450</td>
      <td>17.536594</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>1</td>
      <td>2918</td>
      <td>2</td>
      <td>1</td>
      <td>2922</td>
      <td>Unknown</td>
      <td>A_tigris8450</td>
      <td>17.536594</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>4</td>
      <td>3</td>
      <td>770</td>
      <td>2</td>
      <td>779</td>
      <td>Unknown</td>
      <td>A_tigris8450</td>
      <td>17.536594</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [125]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">,</span> <span class="p">{</span><span class="s1">&#39;axes.grid&#39;</span> <span class="p">:</span> <span class="bp">False</span><span class="p">})</span>
<span class="n">animal</span><span class="o">=</span><span class="s2">&quot;A_tigris8450&quot;</span>
<span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">rc</span><span class="p">(</span><span class="s1">&#39;font&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span>
                    <span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span>
                    <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span>
                    <span class="n">legend</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                    <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                    <span class="n">edgecolor</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">1.6</span><span class="p">,</span><span class="mf">1.88</span><span class="p">)</span>
                   <span class="p">)</span>

<span class="n">ticks</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
<span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
<span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>


<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax1.set_title(&quot;Distribution of Coverage: %s&quot; % change_name(animal))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.4e8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">1.5e8</span><span class="p">,</span><span class="mf">2e7</span><span class="p">))</span>
<span class="c1">#ax1.get_yaxis().get_major_formatter().set_useOffset(False)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">majorFontSize</span><span class="p">)</span>
<span class="c1">#add fill</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>

<span class="c1">#add figure label</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">6</span><span class="p">})</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure3A1.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_194_0.png" src="_images/marmorata_genome_paper_figures_3_194_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [126]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">,</span> <span class="p">{</span><span class="s1">&#39;axes.grid&#39;</span> <span class="p">:</span> <span class="bp">False</span><span class="p">})</span>
<span class="n">animal</span><span class="o">=</span><span class="s2">&quot;Atig_122&quot;</span>
<span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">rc</span><span class="p">(</span><span class="s1">&#39;font&#39;</span><span class="p">,</span> <span class="n">size</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span>
                    <span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span>
                    <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span>
                    <span class="n">legend</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                    <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                    <span class="n">edgecolor</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">1.6</span><span class="p">,</span><span class="mf">1.88</span><span class="p">)</span>
                   <span class="p">)</span>

<span class="n">ticks</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
<span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
<span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>


<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax1.set_title(&quot;Distribution of Coverage: %s&quot; % change_name(animal))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">1.4e8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">1.5e8</span><span class="p">,</span><span class="mf">2e7</span><span class="p">))</span>
<span class="c1">#ax1.get_yaxis().get_major_formatter().set_useOffset(False)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">majorFontSize</span><span class="p">)</span>
<span class="c1">#add fill</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>

<span class="c1">#add figure label</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">6</span><span class="p">})</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure3A2.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_195_0.png" src="_images/marmorata_genome_paper_figures_3_195_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [127]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.5</span><span class="p">,</span><span class="mf">9.0</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.45</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.45</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span>
<span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">,]</span>
<span class="n">column</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">]</span>
<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;H&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">column</span><span class="p">[</span><span class="n">i</span><span class="p">]])</span>
    <span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
    <span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
    <span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
    <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
    <span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span>
                       <span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span>
                       <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span>
                       <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                       <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                       <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                       <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span>
                       <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                      <span class="p">)</span>
    <span class="n">ticks</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
    <span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
    <span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_title( &#39;Animal:&#39;%change_name(animal), fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">140000000</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">23</span><span class="p">,</span><span class="mi">102000000</span><span class="p">,</span><span class="s1">&#39;avg. cov = </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">cov_mean</span><span class="p">,</span><span class="mi">2</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">3</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;best&#39;</span><span class="p">,</span>
              <span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
              <span class="n">markerscale</span><span class="o">=</span><span class="mf">0.2</span><span class="p">,</span>
              <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.23</span><span class="p">,</span> <span class="mf">1.12</span><span class="p">,</span> <span class="n">letters</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/SupplementalFigure5.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_196_0.png" src="_images/marmorata_genome_paper_figures_3_196_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [128]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">check_equal_het</span><span class="p">(</span><span class="n">row</span><span class="p">):</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;T&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;C&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">]]</span>
    <span class="n">non_zero_counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">n</span> <span class="k">for</span> <span class="n">n</span> <span class="ow">in</span> <span class="n">counts</span> <span class="k">if</span> <span class="n">n</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">)</span><span class="o">==</span><span class="mi">2</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">))</span><span class="o">==</span><span class="mi">1</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">True</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">False</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [129]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">allele_count</span><span class="p">(</span><span class="n">row</span><span class="p">):</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;T&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;C&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">]]</span>
    <span class="n">non_zero_counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">n</span> <span class="k">for</span> <span class="n">n</span> <span class="ow">in</span> <span class="n">counts</span> <span class="k">if</span> <span class="n">n</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">]</span>
    <span class="n">non_zero_counts</span> <span class="o">=</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">)</span>
    <span class="k">return</span> <span class="nb">tuple</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [130]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">profCountDF</span><span class="p">[</span><span class="s1">&#39;allele_count&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">apply_by_multiprocessing</span><span class="p">(</span><span class="n">profCountDF</span><span class="p">,</span> <span class="n">allele_count</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [131]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">profCountDF</span><span class="p">[</span><span class="s1">&#39;equal_het&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">apply_by_multiprocessing</span><span class="p">(</span><span class="n">profCountDF</span><span class="p">,</span><span class="n">check_equal_het</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [132]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">ret_per_cov_equal_rates</span><span class="p">(</span><span class="n">prof_count_df</span><span class="p">):</span>
    <span class="n">tuple_list</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">prof_count_df</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
        <span class="k">for</span> <span class="n">coverage</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">100</span><span class="p">,</span> <span class="mi">2</span><span class="p">):</span>
            <span class="n">grouped</span> <span class="o">=</span> <span class="n">prof_count_df</span><span class="p">[(</span><span class="n">prof_count_df</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">prof_count_df</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">coverage</span><span class="p">)]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;equal_het&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">het</span> <span class="o">=</span> <span class="n">grouped</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="bp">True</span><span class="p">]</span>
            <span class="k">except</span> <span class="ne">IndexError</span><span class="p">:</span>
                <span class="n">het</span><span class="o">=</span><span class="mi">0</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">hom</span> <span class="o">=</span> <span class="n">grouped</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="bp">False</span><span class="p">]</span>
            <span class="k">except</span> <span class="ne">IndexError</span><span class="p">:</span>
                <span class="n">hom</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">rate</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">het</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">het</span> <span class="o">+</span> <span class="n">hom</span><span class="p">)</span> <span class="o">*</span> <span class="mi">10000</span>
            <span class="k">except</span> <span class="ne">ZeroDivisionError</span><span class="p">:</span>
                <span class="n">rate</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">tuple_list</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="n">animal</span><span class="p">,</span> <span class="n">coverage</span><span class="p">,</span> <span class="n">het</span><span class="p">,</span> <span class="n">hom</span><span class="p">,</span> <span class="n">het</span><span class="o">+</span><span class="n">hom</span><span class="p">,</span> <span class="n">rate</span><span class="p">))</span>

    <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">,</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;num_equal_het&#39;</span><span class="p">,</span> <span class="s1">&#39;homozygous&#39;</span><span class="p">,</span><span class="s1">&#39;total_sites&#39;</span><span class="p">,</span><span class="s1">&#39;het_per_10kb&#39;</span><span class="p">]</span>
    <span class="n">per_cov_equal_rates_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">tuple_list</span><span class="p">)</span>
    <span class="n">per_cov_equal_rates_df</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="n">columns</span>

    <span class="k">return</span> <span class="n">per_cov_equal_rates_df</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [133]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">perCovEqualRatesDF</span> <span class="o">=</span> <span class="n">ret_per_cov_equal_rates</span><span class="p">(</span><span class="n">profCountDF</span><span class="p">)</span>
<span class="n">perCovEqualRatesDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[133]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>animal</th>
      <th>cov</th>
      <th>num_equal_het</th>
      <th>homozygous</th>
      <th>total_sites</th>
      <th>het_per_10kb</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>A_tigris8450</td>
      <td>2</td>
      <td>5583</td>
      <td>1012258</td>
      <td>1017841</td>
      <td>54.851396</td>
    </tr>
    <tr>
      <th>1</th>
      <td>A_tigris8450</td>
      <td>4</td>
      <td>1263</td>
      <td>3500489</td>
      <td>3501752</td>
      <td>3.606766</td>
    </tr>
    <tr>
      <th>2</th>
      <td>A_tigris8450</td>
      <td>6</td>
      <td>654</td>
      <td>10612477</td>
      <td>10613131</td>
      <td>0.616218</td>
    </tr>
    <tr>
      <th>3</th>
      <td>A_tigris8450</td>
      <td>8</td>
      <td>670</td>
      <td>25872112</td>
      <td>25872782</td>
      <td>0.258959</td>
    </tr>
    <tr>
      <th>4</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>726</td>
      <td>49982983</td>
      <td>49983709</td>
      <td>0.145247</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [134]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1">##### sns.set_style(&quot;whitegrid&quot;, {&#39;axes.grid&#39; : True})</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
<span class="c1">#fig = plt.figure(1, figsize=(32, 19.3))</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>

<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">og_animal_ids</span><span class="p">:</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">perCovEqualRatesDF</span><span class="p">[(</span><span class="n">perCovEqualRatesDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">perCovEqualRatesDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">2</span><span class="p">)]</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span>
                    <span class="n">y</span><span class="o">=</span><span class="s1">&#39;het_per_10kb&#39;</span><span class="p">,</span>
                    <span class="n">ylim</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mi">18</span><span class="p">),</span>
                    <span class="n">xlim</span> <span class="o">=</span> <span class="p">(</span><span class="mf">3.5</span><span class="p">,</span><span class="mf">100.5</span><span class="p">),</span>
                    <span class="n">style</span><span class="o">=</span><span class="s1">&#39;--o&#39;</span><span class="p">,</span>
                    <span class="n">yticks</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span>
                    <span class="n">logy</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                    <span class="n">xticks</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">102</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span>
                    <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span>
                    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                    <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                    <span class="n">markersize</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
                    <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span>
<span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Het. Sites / 10kb (Log10 scale)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax2.set_title(&#39;Rate of Even Split Heterozygous Sites vs Coverage&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.32</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>


<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">(),</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">);</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span> <span class="p">[</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="mi">50</span><span class="p">])</span>
<span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">ScalarFormatter</span><span class="p">())</span>
<span class="c1">#ax.yaxis.set_major_formatter(ticker.FuncFormatter(lambda y,pos: (&#39;{{:.{:1d}f}}&#39;.format(int(np.maximum(-np.log10(y),0)))).format(y)))</span>

<span class="c1">#fig.savefig(&#39;../fig2/Figure6B.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_203_0.png" src="_images/marmorata_genome_paper_figures_3_203_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [135]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1">##### sns.set_style(&quot;whitegrid&quot;, {&#39;axes.grid&#39; : True})</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span><span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
<span class="c1">#fig = plt.figure(1, figsize=(32, 19.3))</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>

<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">perCovEqualRatesDF</span><span class="p">[(</span><span class="n">perCovEqualRatesDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">perCovEqualRatesDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">2</span><span class="p">)]</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span>
                    <span class="n">y</span><span class="o">=</span><span class="s1">&#39;het_per_10kb&#39;</span><span class="p">,</span>
                    <span class="n">ylim</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mi">18</span><span class="p">),</span>
                    <span class="n">xlim</span> <span class="o">=</span> <span class="p">(</span><span class="mf">3.5</span><span class="p">,</span><span class="mf">100.5</span><span class="p">),</span>
                    <span class="n">style</span><span class="o">=</span><span class="s1">&#39;--o&#39;</span><span class="p">,</span>
                    <span class="n">yticks</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span>
                    <span class="n">logy</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                    <span class="n">xticks</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">102</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span>
                    <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span>
                    <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                    <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                    <span class="n">markersize</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
                    <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                    <span class="n">alpha</span><span class="o">=</span><span class="mf">0.8</span>
<span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Het. Sites / 10kb (Log10 scale)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax2.set_title(&#39;Rate of Even Split Heterozygous Sites vs Coverage&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.32</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>


<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">(),</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">);</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span> <span class="p">[</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="mi">50</span><span class="p">])</span>
<span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">ScalarFormatter</span><span class="p">())</span>
<span class="c1">#ax.yaxis.set_major_formatter(ticker.FuncFormatter(lambda y,pos: (&#39;{{:.{:1d}f}}&#39;.format(int(np.maximum(-np.log10(y),0)))).format(y)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure3B.pdf&#39;</span><span class="p">,</span> <span class="n">format</span><span class="o">=</span><span class="s2">&quot;pdf&quot;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_204_0.png" src="_images/marmorata_genome_paper_figures_3_204_0.png" />
</div>
</div>
</div>
<div class="section" id="Confidence-in-Allele-Support-Inferred-from-Transition-Transversion-Ratio">
<h2>Confidence in Allele Support Inferred from Transition Transversion Ratio<a class="headerlink" href="#Confidence-in-Allele-Support-Inferred-from-Transition-Transversion-Ratio" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [136]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">ret_tntv_df</span><span class="p">(</span><span class="n">prof_count_df</span><span class="p">):</span>
    <span class="n">tuple_list</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">prof_count_df</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
        <span class="k">for</span> <span class="n">cov</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">26</span><span class="p">,</span><span class="mi">2</span><span class="p">):</span>
            <span class="n">temp</span> <span class="o">=</span> <span class="n">prof_count_df</span><span class="p">[(</span><span class="n">prof_count_df</span><span class="o">.</span><span class="n">status</span> <span class="o">!=</span> <span class="s1">&#39;Homozygous&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">prof_count_df</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">prof_count_df</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">cov</span><span class="p">)]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;allele_count&#39;</span><span class="p">)</span>
            <span class="n">genos</span> <span class="o">=</span> <span class="n">temp</span><span class="p">[</span><span class="s1">&#39;allele_count&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span><span class="o">.</span><span class="n">tolist</span><span class="p">()</span>
            <span class="k">for</span> <span class="n">geno</span> <span class="ow">in</span> <span class="n">genos</span><span class="p">:</span>
                <span class="n">transitions</span> <span class="o">=</span> <span class="n">temp</span><span class="p">[(</span><span class="n">temp</span><span class="o">.</span><span class="n">status</span><span class="o">==</span><span class="s1">&#39;Transition&#39;</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">temp</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">==</span> <span class="n">geno</span><span class="p">)][</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
                <span class="n">transversions</span> <span class="o">=</span> <span class="n">temp</span><span class="p">[(</span><span class="n">temp</span><span class="o">.</span><span class="n">status</span><span class="o">==</span><span class="s1">&#39;Transversion&#39;</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">temp</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">==</span> <span class="n">geno</span><span class="p">)][</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
                <span class="n">total</span> <span class="o">=</span> <span class="n">temp</span><span class="p">[</span><span class="n">temp</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">==</span> <span class="n">geno</span><span class="p">][</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
                <span class="k">try</span><span class="p">:</span>
                    <span class="n">ratio</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">transitions</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">transversions</span><span class="p">)</span>
                    <span class="n">definable</span> <span class="o">=</span> <span class="bp">True</span>
                <span class="k">except</span> <span class="ne">ZeroDivisionError</span><span class="p">:</span>
                    <span class="n">ratio</span> <span class="o">=</span> <span class="mi">0</span>
                    <span class="n">definable</span> <span class="o">=</span> <span class="bp">False</span>
                <span class="n">t</span> <span class="o">=</span> <span class="p">(</span><span class="n">animal</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">geno</span><span class="p">,</span> <span class="n">transitions</span><span class="p">,</span> <span class="n">transversions</span><span class="p">,</span> <span class="n">ratio</span><span class="p">,</span> <span class="n">definable</span><span class="p">,</span> <span class="n">total</span><span class="p">)</span>

                <span class="n">tuple_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">t</span><span class="p">)</span>



    <span class="n">tntv_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">tuple_list</span><span class="p">)</span>
    <span class="n">tntv_df</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">,</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;allele_count&#39;</span><span class="p">,</span><span class="s1">&#39;transitions&#39;</span><span class="p">,</span><span class="s1">&#39;transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;tn_tv_ratio&#39;</span><span class="p">,</span><span class="s1">&#39;definable&#39;</span><span class="p">,</span><span class="s1">&#39;total&#39;</span><span class="p">]</span>

    <span class="k">return</span> <span class="n">tntv_df</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [137]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">tntvDF</span> <span class="o">=</span> <span class="n">ret_tntv_df</span><span class="p">(</span><span class="n">profCountDF</span><span class="p">)</span>
<span class="n">tntvDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[137]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>animal</th>
      <th>cov</th>
      <th>allele_count</th>
      <th>transitions</th>
      <th>transversions</th>
      <th>tn_tv_ratio</th>
      <th>definable</th>
      <th>total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 1, 7)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>42</td>
    </tr>
    <tr>
      <th>1</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 2, 6)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>17</td>
    </tr>
    <tr>
      <th>2</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 3, 5)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>2</td>
    </tr>
    <tr>
      <th>3</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 8)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>6942</td>
    </tr>
    <tr>
      <th>4</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 2, 2, 5)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [138]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;Atig_122&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>

<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>

<span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span>
              <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span>
              <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
              <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
              <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span>
              <span class="p">)</span>

<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;R: {}</span><span class="se">\n</span><span class="s1"> N: {:,}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)[:</span><span class="mi">4</span><span class="p">],</span> <span class="n">t</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">t</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">subed</span><span class="o">.</span><span class="n">tn_tv_ratio</span><span class="p">,</span> <span class="n">subed</span><span class="o">.</span><span class="n">total</span><span class="p">)]</span>

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">27</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">4</span><span class="p">,)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="mf">0.3</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4500000</span><span class="p">,</span><span class="mi">500000</span><span class="p">));</span>
<span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="c1">#fig.savefig(&#39;../fig/Figure6D_v1.png&#39;,bbox_inches=&#39;tight&#39;, dpi=300)</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure6D.pdf&#39;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_208_0.png" src="_images/marmorata_genome_paper_figures_3_208_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [139]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;Atig_122&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>

<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
<span class="n">subed</span> <span class="o">=</span> <span class="n">subed</span><span class="p">[</span><span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">!=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="n">cov</span><span class="o">-</span><span class="mi">1</span><span class="p">)]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>

<span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span>
              <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span>
              <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;upper center&#39;</span><span class="p">,</span>
              <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
              <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span>
              <span class="p">)</span>

<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;R: {}</span><span class="se">\n</span><span class="s1"> N: {:,}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)[:</span><span class="mi">4</span><span class="p">],</span> <span class="n">t</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">t</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">subed</span><span class="o">.</span><span class="n">tn_tv_ratio</span><span class="p">,</span> <span class="n">subed</span><span class="o">.</span><span class="n">total</span><span class="p">)]</span>

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">27</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">4</span><span class="p">,)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="mf">0.3</span><span class="p">)</span>


<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">320000</span><span class="p">,</span><span class="mi">50000</span><span class="p">));</span>

<span class="c1">#fig.savefig(&#39;../fig/Figure6D_v2.png&#39;,bbox_inches=&#39;tight&#39;, dpi=300)</span>
<span class="c1">#fig.savefig(&#39;../fig/Figure6D_v2.pdf&#39;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_209_0.png" src="_images/marmorata_genome_paper_figures_3_209_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [140]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">))</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;A_tigris8450&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>

<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span> <span class="mi">2</span>
              <span class="p">)</span>
<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

<span class="c1"># Now make some labels</span>
<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;R: {}</span><span class="se">\n</span><span class="s1"> N: {:,}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)[:</span><span class="mi">4</span><span class="p">],</span> <span class="n">t</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">t</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">subed</span><span class="o">.</span><span class="n">tn_tv_ratio</span><span class="p">,</span> <span class="n">subed</span><span class="o">.</span><span class="n">total</span><span class="p">)]</span>

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">27</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">4</span><span class="p">,)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="mf">0.3</span><span class="p">)</span>


<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4500000</span><span class="p">,</span><span class="mi">500000</span><span class="p">));</span>

<span class="c1">#fig.savefig(&#39;../fig/Figure6E_v1.png&#39;,bbox_inches=&#39;tight&#39;, dpi=300)</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure6E.pdf&#39;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_210_0.png" src="_images/marmorata_genome_paper_figures_3_210_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [141]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;A_tigris8450&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>

<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
<span class="n">subed</span> <span class="o">=</span> <span class="n">subed</span><span class="p">[</span><span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">!=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="n">cov</span><span class="o">-</span><span class="mi">1</span><span class="p">)]</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span>
              <span class="p">)</span>
<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

<span class="c1"># Now make some labels</span>
<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;R: {}</span><span class="se">\n</span><span class="s1"> N: {:,}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)[:</span><span class="mi">4</span><span class="p">],</span> <span class="n">t</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">t</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">subed</span><span class="o">.</span><span class="n">tn_tv_ratio</span><span class="p">,</span> <span class="n">subed</span><span class="o">.</span><span class="n">total</span><span class="p">)]</span>

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">27</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">4</span><span class="p">,)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">0</span><span class="p">]</span><span class="o">-</span><span class="mf">0.3</span><span class="p">,</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]</span><span class="o">+</span><span class="mf">0.3</span><span class="p">)</span>


<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">320000</span><span class="p">,</span><span class="mi">50000</span><span class="p">));</span>


<span class="c1">#fig.savefig(&#39;../fig/Figure6E_v2.png&#39;,bbox_inches=&#39;tight&#39;, dpi=300)</span>
<span class="c1">#fig.savefig(&#39;../fig/Figur6E_v2.pdf&#39;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_211_0.png" src="_images/marmorata_genome_paper_figures_3_211_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [142]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.5</span><span class="p">,</span><span class="mf">9.0</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.6</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span>
<span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">3</span><span class="p">,]</span>
<span class="n">column</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">]</span>
<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;H&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]</span>



<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">column</span><span class="p">[</span><span class="n">i</span><span class="p">]])</span>
    <span class="n">cov</span> <span class="o">=</span> <span class="n">covValuesDict</span><span class="p">[</span><span class="n">animal</span><span class="p">]</span>
    <span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
    <span class="n">subed</span> <span class="o">=</span> <span class="n">subed</span><span class="p">[</span><span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span> <span class="o">!=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="n">cov</span><span class="o">-</span><span class="mi">1</span><span class="p">)]</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
    <span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1">#legend</span>
    <span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
    <span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
    <span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span>
                  <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span>
                  <span class="n">markerscale</span><span class="o">=</span><span class="mf">0.4</span><span class="p">,</span>
                  <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                  <span class="n">ncol</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
                  <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span><span class="mi">3</span>
                  <span class="p">)</span>
    <span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

    <span class="c1">#label bars</span>
    <span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

    <span class="c1"># Now make some labels</span>
    <span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;R: {}</span><span class="se">\n</span><span class="s1"> N: {:,}&#39;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">k</span><span class="p">)[:</span><span class="mi">4</span><span class="p">],</span> <span class="n">t</span><span class="p">)</span> <span class="k">for</span> <span class="n">k</span><span class="p">,</span><span class="n">t</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">subed</span><span class="o">.</span><span class="n">tn_tv_ratio</span><span class="p">,</span> <span class="n">subed</span><span class="o">.</span><span class="n">total</span><span class="p">)]</span>

    <span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
        <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">27</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">5</span><span class="p">,)</span>

    <span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites In Genome&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>


    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">400000</span><span class="p">,</span><span class="mi">50000</span><span class="p">))</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.13</span><span class="p">,</span> <span class="mf">1.22</span><span class="p">,</span>
            <span class="n">letters</span><span class="p">[</span><span class="n">i</span><span class="p">],</span>
            <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
            <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span>
            <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span>
            <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span>
            <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>


<span class="c1">#fig.savefig(&#39;../fig2/supplemental_figure_4.pdf&#39;,bbox_inches=&#39;tight&#39;,pad_inches=0.1)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_212_0.png" src="_images/marmorata_genome_paper_figures_3_212_0.png" />
</div>
</div>
</div>
<div class="section" id="General-Genotype-Support">
<h2>General Genotype Support<a class="headerlink" href="#General-Genotype-Support" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [143]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[143]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>array([&#39;A_tigris8450&#39;, &#39;Atig_122&#39;, &#39;Atig_4278&#39;, &#39;Atig003&#39;, &#39;Atig001&#39;,
       &#39;A.tig_12512&#39;, &#39;A.tig_12513&#39;, &#39;A.tig_9721&#39;, &#39;Atig_6993&#39;, &#39;Atig_9177&#39;], dtype=object)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [144]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;num_alleles&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;allele_count&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
<span class="n">tntvDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[144]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>animal</th>
      <th>cov</th>
      <th>allele_count</th>
      <th>transitions</th>
      <th>transversions</th>
      <th>tn_tv_ratio</th>
      <th>definable</th>
      <th>total</th>
      <th>num_alleles</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 1, 7)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>42</td>
      <td>4</td>
    </tr>
    <tr>
      <th>1</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 2, 6)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>17</td>
      <td>4</td>
    </tr>
    <tr>
      <th>2</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 3, 5)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>2</td>
      <td>4</td>
    </tr>
    <tr>
      <th>3</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 1, 8)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>6942</td>
      <td>3</td>
    </tr>
    <tr>
      <th>4</th>
      <td>A_tigris8450</td>
      <td>10</td>
      <td>(1, 2, 2, 5)</td>
      <td>0</td>
      <td>0</td>
      <td>0.0</td>
      <td>False</td>
      <td>1</td>
      <td>4</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [145]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;H&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]</span>
<span class="k">for</span> <span class="n">e</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">):</span>
    <span class="n">fig</span><span class="p">,(</span><span class="n">ax</span><span class="p">,</span><span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>

    <span class="n">data</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">covValuesDict</span><span class="p">[</span><span class="n">animal</span><span class="p">])</span> <span class="o">&amp;</span> \
                  <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)]</span>

    <span class="n">imp_total</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="s1">&#39;num_alleles&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">2</span><span class="p">,</span><span class="s1">&#39;total&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>

    <span class="n">temp</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">data</span><span class="o">.</span><span class="n">num_alleles</span> <span class="o">&lt;</span> <span class="mi">3</span><span class="p">][[</span><span class="s1">&#39;animal&#39;</span><span class="p">,</span> <span class="s1">&#39;total&#39;</span><span class="p">,</span> <span class="s1">&#39;allele_count&#39;</span><span class="p">]]</span>

    <span class="n">new_row</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">([{</span><span class="s1">&#39;animal&#39;</span><span class="p">:</span> <span class="n">animal</span> <span class="p">,</span><span class="s1">&#39;allele_count&#39;</span><span class="p">:</span><span class="s1">&#39;Alleles &gt; 2&#39;</span><span class="p">,</span> <span class="s1">&#39;total&#39;</span><span class="p">:</span> <span class="n">imp_total</span><span class="p">}])</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">temp</span><span class="p">,</span><span class="n">new_row</span><span class="p">])</span>



    <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;allele_count&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;total&#39;</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">title</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">])</span>
    <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;allele_count&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;total&#39;</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax2</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">1000000</span><span class="p">,</span><span class="mi">7000000</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">240000</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>


    <span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax2</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Genotype&#39;</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">total</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">v</span> <span class="o">&lt;</span> <span class="mi">200000</span><span class="p">:</span>
            <span class="n">ax2</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">i</span> <span class="p">,</span> <span class="n">v</span><span class="o">+</span><span class="mi">8000</span><span class="p">,</span>
                     <span class="s2">&quot;{:,}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">v</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                     <span class="c1">#fontweight=&#39;bold&#39;,</span>
                     <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span> <span class="mi">2</span><span class="p">,</span>
                     <span class="n">horizontalalignment</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span>
                     <span class="n">verticalalignment</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span>
                     <span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">v</span> <span class="o">&gt;</span> <span class="mi">1000000</span><span class="p">:</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">i</span> <span class="p">,</span> <span class="n">v</span><span class="o">+</span><span class="mi">80000</span><span class="p">,</span>
                    <span class="s2">&quot;{:,}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">v</span><span class="p">),</span>
                    <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span> <span class="mi">2</span><span class="p">,</span>
                    <span class="c1">#fontweight=&#39;bold&#39;,</span>
                    <span class="n">horizontalalignment</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span>
                    <span class="n">verticalalignment</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span>
                    <span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">)</span>

    <span class="n">d</span> <span class="o">=</span> <span class="o">.</span><span class="mo">011</span>  <span class="c1"># how big to make the diagonal lines in axes coordinates</span>
    <span class="c1"># arguments to pass to plot, just so we don&#39;t keep repeating them</span>
    <span class="n">kwargs</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;k&#39;</span><span class="p">,</span> <span class="n">clip_on</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="o">-</span><span class="n">d</span><span class="p">,</span> <span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="p">(</span><span class="o">-</span><span class="n">d</span><span class="p">,</span> <span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>        <span class="c1"># top-left diagonal</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="mi">1</span> <span class="o">-</span> <span class="n">d</span><span class="p">,</span> <span class="mi">1</span> <span class="o">+</span> <span class="n">d</span><span class="p">),</span> <span class="p">(</span><span class="o">-</span><span class="n">d</span><span class="p">,</span> <span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>  <span class="c1"># top-right dibagonal</span>

    <span class="n">kwargs</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">transform</span><span class="o">=</span><span class="n">ax2</span><span class="o">.</span><span class="n">transAxes</span><span class="p">)</span>  <span class="c1"># switch to the bottom axes</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="o">-</span><span class="n">d</span><span class="p">,</span> <span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">d</span><span class="p">,</span> <span class="mi">1</span> <span class="o">+</span> <span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>  <span class="c1"># bottom-left diagonal</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="mi">1</span> <span class="o">-</span> <span class="n">d</span><span class="p">,</span> <span class="mi">1</span> <span class="o">+</span> <span class="n">d</span><span class="p">),</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">d</span><span class="p">,</span> <span class="mi">1</span> <span class="o">+</span> <span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span>  <span class="c1"># bottom-right diagonal</span>




    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="p">([</span><span class="n">ax</span><span class="o">.</span><span class="n">title</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">label</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">label</span><span class="p">]</span> <span class="o">+</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">()</span> <span class="o">+</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_yticklabels</span><span class="p">()):</span>
        <span class="n">item</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="p">([</span><span class="n">ax2</span><span class="o">.</span><span class="n">title</span><span class="p">,</span> <span class="n">ax2</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">label</span><span class="p">,</span> <span class="n">ax2</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">label</span><span class="p">]</span> <span class="o">+</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">()</span> <span class="o">+</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_yticklabels</span><span class="p">()):</span>
        <span class="n">item</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_size</span><span class="p">(</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_size</span><span class="p">(</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>

<span class="c1">#     fig.savefig(&#39;../fig2/SupplementalFigure5%s.pdf&#39; % letters[e], format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>


<span class="c1">#     if animal == &#39;Atig_122&#39;:</span>
<span class="c1">#         fig.savefig(&#39;../fig2/Figure3C.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
<span class="c1">#     elif animal == &#39;A_tigris8450&#39;:</span>
<span class="c1">#         fig.savefig(&#39;../fig2/Figure3D.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>

</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_0.png" src="_images/marmorata_genome_paper_figures_3_216_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_1.png" src="_images/marmorata_genome_paper_figures_3_216_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_2.png" src="_images/marmorata_genome_paper_figures_3_216_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_3.png" src="_images/marmorata_genome_paper_figures_3_216_3.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_4.png" src="_images/marmorata_genome_paper_figures_3_216_4.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_5.png" src="_images/marmorata_genome_paper_figures_3_216_5.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_6.png" src="_images/marmorata_genome_paper_figures_3_216_6.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_7.png" src="_images/marmorata_genome_paper_figures_3_216_7.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_8.png" src="_images/marmorata_genome_paper_figures_3_216_8.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_216_9.png" src="_images/marmorata_genome_paper_figures_3_216_9.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [146]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">,</span> <span class="s1">&#39;H&#39;</span><span class="p">,</span> <span class="s1">&#39;I&#39;</span><span class="p">,</span> <span class="s1">&#39;J&#39;</span><span class="p">]</span>
<span class="k">for</span> <span class="n">e</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">animal_ids</span><span class="p">):</span>

    <span class="n">fig</span><span class="p">,(</span><span class="n">ax</span><span class="p">,</span> <span class="n">ax2</span><span class="p">)</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span>
                                 <span class="n">sharey</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
                                 <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span>
                                 <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
                                 <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mi">0</span><span class="p">}</span>
                                <span class="p">)</span>


    <span class="n">fig</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">)</span>

    <span class="n">data</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">covValuesDict</span><span class="p">[</span><span class="n">animal</span><span class="p">])</span> <span class="o">&amp;</span> \
                  <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> \
                  <span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">num_alleles</span> <span class="o">&lt;</span> <span class="mi">3</span><span class="p">)]</span>
<span class="c1">#     imp_total = data.loc[data[&#39;num_alleles&#39;] &gt; 2,&#39;total&#39;].sum()</span>

<span class="c1">#     temp = data[data.num_alleles &lt; 3][[&#39;animal&#39;, &#39;total&#39;, &#39;allele_count&#39;]]</span>

<span class="c1">#     new_row = pd.DataFrame([{&#39;animal&#39;: animal ,&#39;allele_count&#39;:&#39;Alleles &gt; 2&#39;, &#39;total&#39;: imp_total}])</span>
<span class="c1">#     data = pd.concat([temp,new_row])</span>




    <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;allele_count&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;total&#39;</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;barh&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;allele_count&#39;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s1">&#39;total&#39;</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s1">&#39;barh&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax2</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>

    <span class="n">ax2</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">1000000</span><span class="p">,</span><span class="mi">7000000</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">240000</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">grid</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>


    <span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;x&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Genotype&#39;</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">v</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">total</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">v</span> <span class="o">&lt;</span> <span class="mi">200000</span><span class="p">:</span>
            <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">v</span><span class="o">+</span><span class="mi">9000</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span>
                     <span class="s2">&quot;{:,}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">v</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                     <span class="c1">#fontweight=&#39;bold&#39;,</span>
                     <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span> <span class="mi">2</span><span class="p">,</span>
                     <span class="n">horizontalalignment</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span>
                     <span class="n">verticalalignment</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                     <span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
        <span class="k">elif</span> <span class="n">v</span> <span class="o">&gt;</span> <span class="mi">1000000</span><span class="p">:</span>
            <span class="n">ax2</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">v</span> <span class="o">+</span> <span class="mi">80000</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span>
                    <span class="s2">&quot;{:,}&quot;</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="n">v</span><span class="p">),</span>
                    <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                    <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span> <span class="o">-</span> <span class="mi">2</span><span class="p">,</span>
                    <span class="c1">#fontweight=&#39;bold&#39;,</span>
                    <span class="n">horizontalalignment</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span>
                    <span class="n">verticalalignment</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span>
                    <span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

    <span class="n">d</span> <span class="o">=</span> <span class="o">.</span><span class="mo">013</span> <span class="c1"># how big to make the diagonal lines in axes coordinates</span>
    <span class="c1"># arguments to pass plot, just so we don&#39;t keep repeating them</span>
    <span class="n">kwargs</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;lightgrey&#39;</span><span class="p">,</span> <span class="n">clip_on</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">lw</span><span class="o">=</span><span class="mf">1.0</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="mi">1</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="mi">1</span><span class="o">+</span><span class="n">d</span><span class="p">),(</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span> <span class="c1"># top-left diagonal</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="mi">1</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="mi">1</span><span class="o">+</span><span class="n">d</span><span class="p">),(</span><span class="mi">1</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="mi">1</span><span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span> <span class="c1"># bottom-left diagonal</span>

    <span class="n">kwargs</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">transform</span><span class="o">=</span><span class="n">ax2</span><span class="o">.</span><span class="n">transAxes</span><span class="p">)</span> <span class="c1"># switch to the bottom axes</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="n">d</span><span class="p">),(</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span> <span class="c1"># top-right diagonal</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">((</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="n">d</span><span class="p">),(</span><span class="mi">1</span><span class="o">-</span><span class="n">d</span><span class="p">,</span><span class="mi">1</span><span class="o">+</span><span class="n">d</span><span class="p">),</span> <span class="o">**</span><span class="n">kwargs</span><span class="p">)</span> <span class="c1"># bottom-right diagonal</span>




    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="p">([</span><span class="n">ax</span><span class="o">.</span><span class="n">title</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">label</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">label</span><span class="p">]</span> <span class="o">+</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">()</span> <span class="o">+</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_yticklabels</span><span class="p">()):</span>
        <span class="n">item</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="p">([</span><span class="n">ax2</span><span class="o">.</span><span class="n">title</span><span class="p">,</span> <span class="n">ax2</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">label</span><span class="p">,</span> <span class="n">ax2</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">label</span><span class="p">]</span> <span class="o">+</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">()</span> <span class="o">+</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_yticklabels</span><span class="p">()):</span>
        <span class="n">item</span><span class="o">.</span><span class="n">set_fontsize</span><span class="p">(</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_size</span><span class="p">(</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">yaxis</span><span class="o">.</span><span class="n">get_offset_text</span><span class="p">()</span><span class="o">.</span><span class="n">set_size</span><span class="p">(</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>

    <span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/SupplementalFigure6</span><span class="si">%s</span><span class="s1">.pdf&#39;</span> <span class="o">%</span> <span class="n">letters</span><span class="p">[</span><span class="n">e</span><span class="p">],</span> <span class="n">format</span><span class="o">=</span><span class="s2">&quot;pdf&quot;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>

<span class="c1">#     if animal == &#39;Atig_122&#39;:</span>
<span class="c1">#         fig.savefig(&#39;../fig2/Figure3C.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
<span class="c1">#     elif animal == &#39;A_tigris8450&#39;:</span>
<span class="c1">#         fig.savefig(&#39;../fig2/Figure3D.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_0.png" src="_images/marmorata_genome_paper_figures_3_217_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_1.png" src="_images/marmorata_genome_paper_figures_3_217_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_2.png" src="_images/marmorata_genome_paper_figures_3_217_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_3.png" src="_images/marmorata_genome_paper_figures_3_217_3.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_4.png" src="_images/marmorata_genome_paper_figures_3_217_4.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_5.png" src="_images/marmorata_genome_paper_figures_3_217_5.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_6.png" src="_images/marmorata_genome_paper_figures_3_217_6.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_7.png" src="_images/marmorata_genome_paper_figures_3_217_7.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_8.png" src="_images/marmorata_genome_paper_figures_3_217_8.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_217_9.png" src="_images/marmorata_genome_paper_figures_3_217_9.png" />
</div>
</div>
</div>
<div class="section" id="Distribution-of-Genome-Average-Even-Heterozygous-Sites">
<h2>Distribution of Genome Average Even Heterozygous Sites<a class="headerlink" href="#Distribution-of-Genome-Average-Even-Heterozygous-Sites" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [147]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">covValuesDict</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[147]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>{&#39;A.tig_12512&#39;: 20,
 &#39;A.tig_12513&#39;: 20,
 &#39;A.tig_9721&#39;: 18,
 &#39;A_tigris8450&#39;: 18,
 &#39;Atig001&#39;: 16,
 &#39;Atig003&#39;: 20,
 &#39;Atig_122&#39;: 20,
 &#39;Atig_4278&#39;: 20,
 &#39;Atig_6993&#39;: 18,
 &#39;Atig_9177&#39;: 22}
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [214]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash

<span class="c1"># awk &#39;$7==16&#39; ../data/pysam/Atig001.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig001.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==20&#39; ../data/pysam/Atig003.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig003.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==20&#39; ../data/pysam/Atig_122.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig_122.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==20&#39; ../data/pysam/A.tig_12512.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/A.tig_12512.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==20&#39; ../data/pysam/A.tig_12513.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/A.tig_12513.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==20&#39; ../data/pysam/Atig_4278.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig_4278.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==18&#39; ../data/pysam/Atig_6993.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig_6993.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==22&#39; ../data/pysam/Atig_9177.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/Atig_9177.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==18&#39; ../data/pysam/A.tig_9721.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/A.tig_9721.merged.dedup.realigned.prof.avg_cov &amp;</span>
<span class="c1"># awk &#39;$7==18&#39; ../data/pysam/A_tigris8450.merged.dedup.realigned.prof.not_hom &gt; ../data/pysam/A_tigris8450.merged.dedup.realigned.prof.avg_cov &amp;</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [223]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
head ../data/pysam/Atig001.merged.dedup.realigned.prof.not_hom
head ../data/pysam/Atig001.merged.dedup.realigned.prof.avg_cov
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
chrom   pos     A       C       G       T       cov     status
Scpiz6a_49      6       0       49      7       37      93      Unknown
Scpiz6a_49      10      0       86      3       7       96      Unknown
Scpiz6a_49      11      1       47      51      0       99      Unknown
Scpiz6a_49      12      0       3       0       115     118     Transition
Scpiz6a_49      16      3       0       131     0       134     Transition
Scpiz6a_49      17      0       54      80      0       134     Transversion
Scpiz6a_49      18      6       8       0       146     160     Unknown
Scpiz6a_49      23      4       0       177     0       181     Transition
Scpiz6a_49      24      0       176     0       6       182     Transition
Scpiz6a_49      557     10      0       6       0       16      Transition
Scpiz6a_49      2067    0       0       14      2       16      Transversion
Scpiz6a_49      2875    15      0       1       0       16      Transition
Scpiz6a_49      3485    0       1       0       15      16      Transition
Scpiz6a_49      3488    0       1       15      0       16      Transversion
Scpiz6a_49      3489    0       0       1       15      16      Transversion
Scpiz6a_49      3491    0       0       15      1       16      Transversion
Scpiz6a_49      3521    15      0       0       1       16      Transversion
Scpiz6a_49      4234    1       0       15      0       16      Transition
Scpiz6a_49      4235    1       15      0       0       16      Transversion
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [321]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">df_list</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../data/pysam/</span><span class="si">%s</span><span class="s1">.merged.dedup.realigned.prof.avg_cov&#39;</span> <span class="o">%</span> <span class="n">animal</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">,</span><span class="s1">&#39;pos&#39;</span><span class="p">,</span><span class="s1">&#39;A&#39;</span><span class="p">,</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">,</span><span class="s1">&#39;T&#39;</span><span class="p">,</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;status&#39;</span><span class="p">])</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">animal</span>
    <span class="n">df_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="n">avgCovHetSitesDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">df_list</span><span class="p">,</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">avgCovHetSitesDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[321]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>pos</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>animal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>651</td>
      <td>19</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>2292</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>3000</td>
      <td>19</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>5047</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>5864</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [322]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">avgCovHetSitesDF</span><span class="p">[</span><span class="s1">&#39;allele_count&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">apply_by_multiprocessing</span><span class="p">(</span><span class="n">avgCovHetSitesDF</span><span class="p">,</span> <span class="n">allele_count</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>
<span class="n">avgCovHetSitesDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[322]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>pos</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>animal</th>
      <th>allele_count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>651</td>
      <td>19</td>
      <td>1</td>
      <td>0</td>
      <td>0</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(1, 19)</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>2292</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(1, 19)</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>3000</td>
      <td>19</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(1, 19)</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>5047</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(1, 19)</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>5864</td>
      <td>0</td>
      <td>1</td>
      <td>0</td>
      <td>19</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(1, 19)</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [323]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">avgCovHetSitesDF</span><span class="p">[</span><span class="s1">&#39;equal&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">avgCovHetSitesDF</span><span class="p">[</span><span class="s1">&#39;allele_count&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">==</span><span class="mi">2</span><span class="p">)</span>
<span class="n">avgCovEqualHetSitesDF</span> <span class="o">=</span> <span class="n">avgCovHetSitesDF</span><span class="p">[</span><span class="n">avgCovHetSitesDF</span><span class="p">[</span><span class="s1">&#39;equal&#39;</span><span class="p">]</span><span class="o">==</span><span class="bp">True</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[323]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>pos</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>animal</th>
      <th>allele_count</th>
      <th>equal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>357652</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>533563</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>533564</td>
      <td>10</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>1076362</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>1258626</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [324]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">scaffoldSizes</span><span class="p">[</span><span class="s1">&#39;genome_scale&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold_size</span><span class="o">.</span><span class="n">cumsum</span><span class="p">()</span>
<span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">genome_scale</span> <span class="o">=</span> <span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">genome_scale</span><span class="o">.</span><span class="n">shift</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
<span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="n">value</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">scaffoldScaleDict</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span><span class="nb">zip</span><span class="p">(</span><span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold</span><span class="o">.</span><span class="n">tolist</span><span class="p">(),</span> <span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">genome_scale</span><span class="o">.</span><span class="n">tolist</span><span class="p">()))</span>
<span class="n">display</span><span class="p">(</span><span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">head</span><span class="p">())</span>
<span class="n">scaffoldScaleDict</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>scaffold</th>
      <th>scaffold_size</th>
      <th>genome_scale</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>3825</th>
      <td>Scpiz6a_49</td>
      <td>85027298</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>3824</th>
      <td>Scpiz6a_55.1</td>
      <td>77407970</td>
      <td>85027298.0</td>
    </tr>
    <tr>
      <th>3823</th>
      <td>Scpiz6a_26</td>
      <td>75278123</td>
      <td>162435268.0</td>
    </tr>
    <tr>
      <th>3822</th>
      <td>Scpiz6a_73</td>
      <td>74911545</td>
      <td>237713391.0</td>
    </tr>
    <tr>
      <th>3821</th>
      <td>Scpiz6a_37</td>
      <td>69231668</td>
      <td>312624936.0</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[324]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>{&#39;Scpiz6a_3775&#39;: 1639022565.0,
 &#39;Scpiz6a_1710&#39;: 1639028019.0,
 &#39;Scpiz6a_1711&#39;: 1636688900.0,
 &#39;Scpiz6a_1712&#39;: 1636222427.0,
 &#39;Scpiz6a_1713&#39;: 1639520853.0,
 &#39;Scpiz6a_1714&#39;: 1639254835.0,
 &#39;Scpiz6a_1715&#39;: 1638962284.0,
 &#39;Scpiz6a_1716&#39;: 1638685159.0,
 &#39;Scpiz6a_1717&#39;: 1637338467.0,
 &#39;Scpiz6a_1718&#39;: 1638778443.0,
 &#39;Scpiz6a_1719&#39;: 1637647215.0,
 &#39;Scpiz6a_617&#39;: 1637685041.0,
 &#39;Scpiz6a_1481&#39;: 1637831127.0,
 &#39;Scpiz6a_615&#39;: 1637164824.0,
 &#39;Scpiz6a_614&#39;: 1637186843.0,
 &#39;Scpiz6a_613&#39;: 1634735097.0,
 &#39;Scpiz6a_612&#39;: 1634710713.0,
 &#39;Scpiz6a_1486&#39;: 1638634444.0,
 &#39;Scpiz6a_610&#39;: 1630834129.0,
 &#39;Scpiz6a_1488&#39;: 1638073634.0,
 &#39;Scpiz6a_1489&#39;: 1639352546.0,
 &#39;Scpiz6a_619&#39;: 1637639083.0,
 &#39;Scpiz6a_618&#39;: 1634257302.0,
 &#39;Scpiz6a_1650&#39;: 1639000713.0,
 &#39;Scpiz6a_2061&#39;: 1639465599.0,
 &#39;Scpiz6a_1654&#39;: 1638824618.0,
 &#39;Scpiz6a_103&#39;: 1637491959.0,
 &#39;Scpiz6a_236&#39;: 1638854824.0,
 &#39;Scpiz6a_1628&#39;: 1637469657.0,
 &#39;Scpiz6a_1629&#39;: 1634572756.0,
 &#39;Scpiz6a_1158&#39;: 1635199980.0,
 &#39;Scpiz6a_1159&#39;: 1634700916.0,
 &#39;Scpiz6a_1624&#39;: 1635402501.0,
 &#39;Scpiz6a_1157&#39;: 1633609584.0,
 &#39;Scpiz6a_1154&#39;: 1638864854.0,
 &#39;Scpiz6a_1155&#39;: 1636341192.0,
 &#39;Scpiz6a_1152&#39;: 1634952594.0,
 &#39;Scpiz6a_1153&#39;: 1638658707.0,
 &#39;Scpiz6a_1150&#39;: 1634897027.0,
 &#39;Scpiz6a_1623&#39;: 1635467812.0,
 &#39;Scpiz6a_729&#39;: 1638872646.0,
 &#39;Scpiz6a_2262&#39;: 1637256707.0,
 &#39;Scpiz6a_1398&#39;: 1638815628.0,
 &#39;Scpiz6a_1399&#39;: 1638836947.0,
 &#39;Scpiz6a_1394&#39;: 1637765139.0,
 &#39;Scpiz6a_722&#39;: 1637044331.0,
 &#39;Scpiz6a_721&#39;: 1635975188.0,
 &#39;Scpiz6a_720&#39;: 1636648737.0,
 &#39;Scpiz6a_1390&#39;: 1639041076.0,
 &#39;Scpiz6a_1391&#39;: 1633343192.0,
 &#39;Scpiz6a_1392&#39;: 1635390132.0,
 &#39;Scpiz6a_724&#39;: 1634104860.0,
 &#39;Scpiz6a_105&#39;: 1593914005.0,
 &#39;Scpiz6a_104&#39;: 1633720801.0,
 &#39;Scpiz6a_1899&#39;: 1637713220.0,
 &#39;Scpiz6a_1898&#39;: 1637742552.0,
 &#39;Scpiz6a_266&#39;: 1639380275.0,
 &#39;Scpiz6a_267&#39;: 1638411903.0,
 &#39;Scpiz6a_264&#39;: 1490215262.0,
 &#39;Scpiz6a_265&#39;: 1635914618.0,
 &#39;Scpiz6a_262&#39;: 1639409878.0,
 &#39;Scpiz6a_263&#39;: 1513297986.0,
 &#39;Scpiz6a_260&#39;: 1637056350.0,
 &#39;Scpiz6a_261&#39;: 1634998264.0,
 &#39;Scpiz6a_2407&#39;: 1639312200.0,
 &#39;Scpiz6a_1895&#39;: 1636962637.0,
 &#39;Scpiz6a_268&#39;: 1635542199.0,
 &#39;Scpiz6a_269&#39;: 1614479145.0,
 &#39;Scpiz6a_460&#39;: 1634469674.0,
 &#39;Scpiz6a_461&#39;: 1634518226.0,
 &#39;Scpiz6a_462&#39;: 1583396600.0,
 &#39;Scpiz6a_463&#39;: 1636564490.0,
 &#39;Scpiz6a_464&#39;: 1634152996.0,
 &#39;Scpiz6a_465&#39;: 1636004285.0,
 &#39;Scpiz6a_466&#39;: 1633741570.0,
 &#39;Scpiz6a_467&#39;: 1634759326.0,
 &#39;Scpiz6a_468&#39;: 1636427463.0,
 &#39;Scpiz6a_2498&#39;: 1638604312.0,
 &#39;Scpiz6a_1020&#39;: 1639467615.0,
 &#39;Scpiz6a_1021&#39;: 1637828497.0,
 &#39;Scpiz6a_1026&#39;: 1638288477.0,
 &#39;Scpiz6a_1027&#39;: 1636508830.0,
 &#39;Scpiz6a_1024&#39;: 1637533584.0,
 &#39;Scpiz6a_1892&#39;: 1637362653.0,
 &#39;Scpiz6a_1891&#39;: 1639471644.0,
 &#39;Scpiz6a_751&#39;: 1637334192.0,
 &#39;Scpiz6a_900&#39;: 1639104929.0,
 &#39;Scpiz6a_1480&#39;: 1639148920.0,
 &#39;Scpiz6a_616&#39;: 1638054626.0,
 &#39;Scpiz6a_1482&#39;: 1637927717.0,
 &#39;Scpiz6a_1483&#39;: 1637926421.0,
 &#39;Scpiz6a_1484&#39;: 1637859998.0,
 &#39;Scpiz6a_907&#39;: 1637580439.0,
 &#39;Scpiz6a_1485&#39;: 1635092543.0,
 &#39;Scpiz6a_2618&#39;: 1638819001.0,
 &#39;Scpiz6a_2619&#39;: 1636497297.0,
 &#39;Scpiz6a_611&#39;: 1639415986.0,
 &#39;Scpiz6a_3102&#39;: 1636040380.0,
 &#39;Scpiz6a_2610&#39;: 1636983930.0,
 &#39;Scpiz6a_2611&#39;: 1639194711.0,
 &#39;Scpiz6a_2612&#39;: 1633577153.0,
 &#39;Scpiz6a_97&#39;: 1639064949.0,
 &#39;Scpiz6a_2614&#39;: 1639359753.0,
 &#39;Scpiz6a_3101&#39;: 1638534173.0,
 &#39;Scpiz6a_2616&#39;: 1635634980.0,
 &#39;Scpiz6a_2617&#39;: 1637943245.0,
 &#39;Scpiz6a_594&#39;: 1637782359.0,
 &#39;Scpiz6a_595&#39;: 1637392426.0,
 &#39;Scpiz6a_596&#39;: 1635573967.0,
 &#39;Scpiz6a_597&#39;: 1620810945.0,
 &#39;Scpiz6a_590&#39;: 1637302757.0,
 &#39;Scpiz6a_591&#39;: 1636888975.0,
 &#39;Scpiz6a_592&#39;: 1637543261.0,
 &#39;Scpiz6a_593&#39;: 1457337151.0,
 &#39;Scpiz6a_2980&#39;: 1639158534.0,
 &#39;Scpiz6a_906&#39;: 1639066032.0,
 &#39;Scpiz6a_598&#39;: 1638775056.0,
 &#39;Scpiz6a_599&#39;: 1638516502.0,
 &#39;Scpiz6a_3106&#39;: 1637886118.0,
 &#39;Scpiz6a_2430&#39;: 1635662321.0,
 &#39;Scpiz6a_2431&#39;: 1636986966.0,
 &#39;Scpiz6a_2120&#39;: 1636307007.0,
 &#39;Scpiz6a_2433&#39;: 1636054744.0,
 &#39;Scpiz6a_2126&#39;: 1637607803.0,
 &#39;Scpiz6a_3105&#39;: 1635530230.0,
 &#39;Scpiz6a_2124&#39;: 1632900011.0,
 &#39;Scpiz6a_2794&#39;: 1634783399.0,
 &#39;Scpiz6a_2438&#39;: 1637775741.0,
 &#39;Scpiz6a_2439&#39;: 1638301906.0,
 &#39;Scpiz6a_2128&#39;: 1639296617.0,
 &#39;Scpiz6a_2129&#39;: 1637030789.0,
 &#39;Scpiz6a_2797&#39;: 1639525853.0,
 &#39;Scpiz6a_296&#39;: 1639303895.0,
 &#39;Scpiz6a_2796&#39;: 1635431185.0,
 &#39;Scpiz6a_905&#39;: 1638661013.0,
 &#39;Scpiz6a_2791&#39;: 1639487750.0,
 &#39;Scpiz6a_2790&#39;: 1633882142.0,
 &#39;Scpiz6a_2021&#39;: 1637718561.0,
 &#39;Scpiz6a_2058&#39;: 1638702336.0,
 &#39;Scpiz6a_2020&#39;: 1639292456.0,
 &#39;Scpiz6a_2724&#39;: 1635897973.0,
 &#39;Scpiz6a_289&#39;: 1623755943.0,
 &#39;Scpiz6a_2726&#39;: 1638100123.0,
 &#39;Scpiz6a_2055&#39;: 1635371536.0,
 &#39;Scpiz6a_2052&#39;: 1636493995.0,
 &#39;Scpiz6a_2721&#39;: 1636136613.0,
 &#39;Scpiz6a_2722&#39;: 1635960573.0,
 &#39;Scpiz6a_2723&#39;: 1636971773.0,
 &#39;Scpiz6a_280&#39;: 1637794253.0,
 &#39;Scpiz6a_281&#39;: 1636588884.0,
 &#39;Scpiz6a_282&#39;: 1638656400.0,
 &#39;Scpiz6a_283&#39;: 1634890020.0,
 &#39;Scpiz6a_2728&#39;: 1634703368.0,
 &#39;Scpiz6a_2729&#39;: 1635052399.0,
 &#39;Scpiz6a_286&#39;: 1635134590.0,
 &#39;Scpiz6a_287&#39;: 1637592769.0,
 &#39;Scpiz6a_3231&#39;: 1637060849.0,
 &#39;Scpiz6a_3230&#39;: 1637388182.0,
 &#39;Scpiz6a_3233&#39;: 1635221531.0,
 &#39;Scpiz6a_3232&#39;: 1635236566.0,
 &#39;Scpiz6a_2548&#39;: 1636289834.0,
 &#39;Scpiz6a_2549&#39;: 1635899825.0,
 &#39;Scpiz6a_3237&#39;: 1638161429.0,
 &#39;Scpiz6a_3236&#39;: 1632719930.0,
 &#39;Scpiz6a_3239&#39;: 1634737525.0,
 &#39;Scpiz6a_3238&#39;: 1637536350.0,
 &#39;Scpiz6a_2546&#39;: 1636349713.0,
 &#39;Scpiz6a_2547&#39;: 1638760351.0,
 &#39;Scpiz6a_2540&#39;: 1633632827.0,
 &#39;Scpiz6a_2541&#39;: 1634603406.0,
 &#39;Scpiz6a_2542&#39;: 1634636321.0,
 &#39;Scpiz6a_2543&#39;: 1635306763.0,
 &#39;Scpiz6a_3054&#39;: 1638729710.0,
 &#39;Scpiz6a_3235&#39;: 1638919235.0,
 &#39;Scpiz6a_2640&#39;: 1639378228.0,
 &#39;Scpiz6a_428.1&#39;: 1631638909.0,
 &#39;Scpiz6a_3294&#39;: 1635121361.0,
 &#39;Scpiz6a_132&#39;: 1631155467.0,
 &#39;Scpiz6a_79&#39;: 1635296208.0,
 &#39;Scpiz6a_130&#39;: 1639486744.0,
 &#39;Scpiz6a_131&#39;: 1639469630.0,
 &#39;Scpiz6a_136&#39;: 1570486390.0,
 &#39;Scpiz6a_137&#39;: 1636832959.0,
 &#39;Scpiz6a_134&#39;: 1637921234.0,
 &#39;Scpiz6a_135&#39;: 1637755844.0,
 &#39;Scpiz6a_70&#39;: 1637496132.0,
 &#39;Scpiz6a_71&#39;: 1636120695.0,
 &#39;Scpiz6a_72&#39;: 655146831.0,
 &#39;Scpiz6a_73&#39;: 237713391.0,
 &#39;Scpiz6a_74&#39;: 880550878.0,
 &#39;Scpiz6a_75&#39;: 1145830824.0,
 &#39;Scpiz6a_76&#39;: 1636603481.0,
 &#39;Scpiz6a_77&#39;: 1637158940.0,
 &#39;Scpiz6a_3529&#39;: 1638797613.0,
 &#39;Scpiz6a_3528&#39;: 1637123475.0,
 &#39;Scpiz6a_3400&#39;: 1637150100.0,
 &#39;Scpiz6a_3525&#39;: 1637248036.0,
 &#39;Scpiz6a_3524&#39;: 1638343235.0,
 &#39;Scpiz6a_3527&#39;: 1637011187.0,
 &#39;Scpiz6a_3526&#39;: 1639477686.0,
 &#39;Scpiz6a_3521&#39;: 1636348011.0,
 &#39;Scpiz6a_3520&#39;: 1637652633.0,
 &#39;Scpiz6a_3523&#39;: 1638018986.0,
 &#39;Scpiz6a_3522&#39;: 1638934715.0,
 &#39;Scpiz6a_3401&#39;: 1636752665.0,
 &#39;Scpiz6a_755&#39;: 1636622902.0,
 &#39;Scpiz6a_1071&#39;: 1638077427.0,
 &#39;Scpiz6a_3020&#39;: 1639462575.0,
 &#39;Scpiz6a_832&#39;: 1638871534.0,
 &#39;Scpiz6a_2144&#39;: 1639500817.0,
 &#39;Scpiz6a_868&#39;: 1638983180.0,
 &#39;Scpiz6a_869&#39;: 1637048840.0,
 &#39;Scpiz6a_715&#39;: 1639207426.0,
 &#39;Scpiz6a_864&#39;: 1635141184.0,
 &#39;Scpiz6a_865&#39;: 1637292721.0,
 &#39;Scpiz6a_866&#39;: 1636521981.0,
 &#39;Scpiz6a_867&#39;: 1635367392.0,
 &#39;Scpiz6a_860&#39;: 1635890556.0,
 &#39;Scpiz6a_861&#39;: 1638825741.0,
 &#39;Scpiz6a_862&#39;: 1636848581.0,
 &#39;Scpiz6a_863&#39;: 1638827985.0,
 &#39;Scpiz6a_3419&#39;: 1635781653.0,
 &#39;Scpiz6a_3418&#39;: 1637259595.0,
 &#39;Scpiz6a_3149&#39;: 1639251688.0,
 &#39;Scpiz6a_3148&#39;: 1637816656.0,
 &#39;Scpiz6a_3411&#39;: 1637220452.0,
 &#39;Scpiz6a_3146&#39;: 1636526905.0,
 &#39;Scpiz6a_3145&#39;: 1637278343.0,
 &#39;Scpiz6a_3144&#39;: 1637412173.0,
 &#39;Scpiz6a_3415&#39;: 1635357006.0,
 &#39;Scpiz6a_3142&#39;: 1636505539.0,
 &#39;Scpiz6a_3417&#39;: 1635184816.0,
 &#39;Scpiz6a_3140&#39;: 1634875959.0,
 &#39;Scpiz6a_2940&#39;: 1634705818.0,
 &#39;Scpiz6a_2941&#39;: 1638812253.0,
 &#39;Scpiz6a_2942&#39;: 1634562477.0,
 &#39;Scpiz6a_2943&#39;: 1635603572.0,
 &#39;Scpiz6a_2944&#39;: 1637429050.0,
 &#39;Scpiz6a_2945&#39;: 1638591521.0,
 &#39;Scpiz6a_2946&#39;: 1639428185.0,
 &#39;Scpiz6a_2947&#39;: 1638153952.0,
 &#39;Scpiz6a_2948&#39;: 1635145575.0,
 &#39;Scpiz6a_2949&#39;: 1638265230.0,
 &#39;Scpiz6a_1094&#39;: 1637850822.0,
 &#39;Scpiz6a_3099&#39;: 1636709689.0,
 &#39;Scpiz6a_3098&#39;: 1638560001.0,
 &#39;Scpiz6a_3095&#39;: 1638567020.0,
 &#39;Scpiz6a_3094&#39;: 1638901514.0,
 &#39;Scpiz6a_3097&#39;: 1638010054.0,
 &#39;Scpiz6a_3096&#39;: 1637407948.0,
 &#39;Scpiz6a_3091&#39;: 1638472708.0,
 &#39;Scpiz6a_3090&#39;: 1634892357.0,
 &#39;Scpiz6a_3093&#39;: 1636192815.0,
 &#39;Scpiz6a_3092&#39;: 1638881540.0,
 &#39;Scpiz6a_1817&#39;: 1636523623.0,
 &#39;Scpiz6a_1816&#39;: 1635437310.0,
 &#39;Scpiz6a_1815&#39;: 1636056536.0,
 &#39;Scpiz6a_1814&#39;: 1637405129.0,
 &#39;Scpiz6a_1365&#39;: 1635534224.0,
 &#39;Scpiz6a_1364&#39;: 1638540053.0,
 &#39;Scpiz6a_1811&#39;: 1638608955.0,
 &#39;Scpiz6a_1366&#39;: 1636238037.0,
 &#39;Scpiz6a_1369&#39;: 1638755821.0,
 &#39;Scpiz6a_1368&#39;: 1638322606.0,
 &#39;Scpiz6a_1819&#39;: 1634666292.0,
 &#39;Scpiz6a_1818&#39;: 1639279944.0,
 &#39;Scpiz6a_910&#39;: 1633844828.0,
 &#39;Scpiz6a_1566&#39;: 1636517050.0,
 &#39;Scpiz6a_912&#39;: 1636182319.0,
 &#39;Scpiz6a_913&#39;: 1637707868.0,
 &#39;Scpiz6a_914&#39;: 1634975493.0,
 &#39;Scpiz6a_1562&#39;: 1638234530.0,
 &#39;Scpiz6a_916&#39;: 1636803183.0,
 &#39;Scpiz6a_1560&#39;: 1634288832.0,
 &#39;Scpiz6a_918&#39;: 1637138281.0,
 &#39;Scpiz6a_919&#39;: 1636760597.0,
 &#39;Scpiz6a_1626&#39;: 1639437312.0,
 &#39;Scpiz6a_1569&#39;: 1639269496.0,
 &#39;Scpiz6a_1568&#39;: 1637595505.0,
 &#39;Scpiz6a_3705&#39;: 1639403764.0,
 &#39;Scpiz6a_3704&#39;: 1636621284.0,
 &#39;Scpiz6a_3707&#39;: 1637008166.0,
 &#39;Scpiz6a_1627&#39;: 1639236975.0,
 &#39;Scpiz6a_3701&#39;: 1636141904.0,
 &#39;Scpiz6a_3700&#39;: 1638584532.0,
 &#39;Scpiz6a_3703&#39;: 1635817607.0,
 &#39;Scpiz6a_3702&#39;: 1636634217.0,
 &#39;Scpiz6a_1620&#39;: 1639264265.0,
 &#39;Scpiz6a_3709&#39;: 1638403516.0,
 &#39;Scpiz6a_3708&#39;: 1637766466.0,
 &#39;Scpiz6a_1621&#39;: 1639337063.0,
 &#39;Scpiz6a_1523&#39;: 1639425137.0,
 &#39;Scpiz6a_1969&#39;: 1637667506.0,
 &#39;Scpiz6a_1968&#39;: 1634246374.0,
 &#39;Scpiz6a_1622&#39;: 1637635015.0,
 &#39;Scpiz6a_1522&#39;: 1639033466.0,
 &#39;Scpiz6a_1963&#39;: 1639496800.0,
 &#39;Scpiz6a_1962&#39;: 1639324642.0,
 &#39;Scpiz6a_1961&#39;: 1638793105.0,
 &#39;Scpiz6a_1960&#39;: 1634197357.0,
 &#39;Scpiz6a_1967&#39;: 1635251545.0,
 &#39;Scpiz6a_1918&#39;: 1636873478.0,
 &#39;Scpiz6a_1965&#39;: 1639227492.0,
 &#39;Scpiz6a_1964&#39;: 1635083654.0,
 &#39;Scpiz6a_957&#39;: 1636826703.0,
 &#39;Scpiz6a_1527&#39;: 1638944653.0,
 &#39;Scpiz6a_1526&#39;: 1638593849.0,
 &#39;Scpiz6a_1703&#39;: 1639152127.0,
 &#39;Scpiz6a_1702&#39;: 1638857054.0,
 &#39;Scpiz6a_1701&#39;: 1639502825.0,
 &#39;Scpiz6a_1700&#39;: 1634887679.0,
 &#39;Scpiz6a_1707&#39;: 1637356969.0,
 &#39;Scpiz6a_1706&#39;: 1637053348.0,
 &#39;Scpiz6a_1705&#39;: 1636725635.0,
 &#39;Scpiz6a_1704&#39;: 1636270881.0,
 &#39;Scpiz6a_1709&#39;: 1634598316.0,
 &#39;Scpiz6a_1708&#39;: 1638003665.0,
 &#39;Scpiz6a_604&#39;: 1636189318.0,
 &#39;Scpiz6a_1492&#39;: 1632448361.0,
 &#39;Scpiz6a_1491&#39;: 1637936780.0,
 &#39;Scpiz6a_607&#39;: 1634161274.0,
 &#39;Scpiz6a_1497&#39;: 1636155986.0,
 &#39;Scpiz6a_601&#39;: 1638236992.0,
 &#39;Scpiz6a_1495&#39;: 1637233539.0,
 &#39;Scpiz6a_603&#39;: 1634821633.0,
 &#39;Scpiz6a_2337&#39;: 1639172392.0,
 &#39;Scpiz6a_1499&#39;: 1637359811.0,
 &#39;Scpiz6a_1498&#39;: 1636318994.0,
 &#39;Scpiz6a_608&#39;: 1634496852.0,
 &#39;Scpiz6a_609&#39;: 1633800367.0,
 &#39;Scpiz6a_723&#39;: 1638529465.0,
 &#39;Scpiz6a_1395&#39;: 1637965169.0,
 &#39;Scpiz6a_1396&#39;: 1637970315.0,
 &#39;Scpiz6a_2243&#39;: 1636700104.0,
 &#39;Scpiz6a_1397&#39;: 1637880904.0,
 &#39;Scpiz6a_2242&#39;: 1639349454.0,
 &#39;Scpiz6a_727&#39;: 1639274723.0,
 &#39;Scpiz6a_2241&#39;: 1636129546.0,
 &#39;Scpiz6a_726&#39;: 1636353115.0,
 &#39;Scpiz6a_1149&#39;: 1638260329.0,
 &#39;Scpiz6a_1148&#39;: 1634768976.0,
 &#39;Scpiz6a_3315&#39;: 1637962593.0,
 &#39;Scpiz6a_725&#39;: 1637721231.0,
 &#39;Scpiz6a_1639&#39;: 1639068198.0,
 &#39;Scpiz6a_3740&#39;: 1635225831.0,
 &#39;Scpiz6a_1141&#39;: 1636609957.0,
 &#39;Scpiz6a_1636&#39;: 1634351040.0,
 &#39;Scpiz6a_1143&#39;: 1638448914.0,
 &#39;Scpiz6a_1142&#39;: 1635813838.0,
 &#39;Scpiz6a_1145&#39;: 1638865968.0,
 &#39;Scpiz6a_3743&#39;: 1635675947.0,
 &#39;Scpiz6a_1147&#39;: 1635223682.0,
 &#39;Scpiz6a_1146&#39;: 1636770105.0,
 &#39;Scpiz6a_738&#39;: 1638243142.0,
 &#39;Scpiz6a_739&#39;: 1637059350.0,
 &#39;Scpiz6a_3034&#39;: 1639243288.0,
 &#39;Scpiz6a_3310&#39;: 1638133980.0,
 &#39;Scpiz6a_1389&#39;: 1639275768.0,
 &#39;Scpiz6a_1388&#39;: 1637448685.0,
 &#39;Scpiz6a_730&#39;: 1636722449.0,
 &#39;Scpiz6a_731&#39;: 1637375427.0,
 &#39;Scpiz6a_1385&#39;: 1636712880.0,
 &#39;Scpiz6a_733&#39;: 1636033177.0,
 &#39;Scpiz6a_734&#39;: 1638102639.0,
 &#39;Scpiz6a_735&#39;: 1636111825.0,
 &#39;Scpiz6a_736&#39;: 1635506181.0,
 &#39;Scpiz6a_1380&#39;: 1634397852.0,
 &#39;Scpiz6a_3031&#39;: 1637299891.0,
 &#39;Scpiz6a_3030&#39;: 1638439378.0,
 &#39;Scpiz6a_2328&#39;: 1636997578.0,
 &#39;Scpiz6a_3423&#39;: 1638232066.0,
 &#39;Scpiz6a_35&#39;: 1637515580.0,
 &#39;Scpiz6a_2898&#39;: 1639293497.0,
 &#39;Scpiz6a_455&#39;: 1631374720.0,
 &#39;Scpiz6a_454&#39;: 1636948912.0,
 &#39;Scpiz6a_1037&#39;: 1635158691.0,
 &#39;Scpiz6a_1036&#39;: 1632625465.0,
 &#39;Scpiz6a_1031&#39;: 1637355548.0,
 &#39;Scpiz6a_1030&#39;: 1635059116.0,
 &#39;Scpiz6a_453&#39;: 1634165388.0,
 &#39;Scpiz6a_452&#39;: 1638637915.0,
 &#39;Scpiz6a_3427&#39;: 1637351282.0,
 &#39;Scpiz6a_1039&#39;: 1637827182.0,
 &#39;Scpiz6a_458&#39;: 1616314001.0,
 &#39;Scpiz6a_2917&#39;: 1639382321.0,
 &#39;Scpiz6a_3429&#39;: 1636467502.0,
 &#39;Scpiz6a_3584&#39;: 1638672526.0,
 &#39;Scpiz6a_2609&#39;: 1639160668.0,
 &#39;Scpiz6a_2608&#39;: 1635611436.0,
 &#39;Scpiz6a_2603&#39;: 1635463754.0,
 &#39;Scpiz6a_2602&#39;: 1638145219.0,
 &#39;Scpiz6a_2601&#39;: 1637590031.0,
 &#39;Scpiz6a_2600&#39;: 1636630986.0,
 &#39;Scpiz6a_2607&#39;: 1636793752.0,
 &#39;Scpiz6a_2606&#39;: 1638690892.0,
 &#39;Scpiz6a_2605&#39;: 1638128980.0,
 &#39;Scpiz6a_2604&#39;: 1634927248.0,
 &#39;Scpiz6a_2423&#39;: 1629438344.0,
 &#39;Scpiz6a_2422&#39;: 1636462513.0,
 &#39;Scpiz6a_2421&#39;: 1638299466.0,
 &#39;Scpiz6a_2420&#39;: 1636293274.0,
 &#39;Scpiz6a_2427&#39;: 1637419209.0,
 &#39;Scpiz6a_2426&#39;: 1633793927.0,
 &#39;Scpiz6a_2425&#39;: 1633083921.0,
 &#39;Scpiz6a_2424&#39;: 1639077939.0,
 &#39;Scpiz6a_2429&#39;: 1638229601.0,
 &#39;Scpiz6a_2428&#39;: 1634475163.0,
 &#39;Scpiz6a_1897&#39;: 1638767141.0,
 &#39;Scpiz6a_2041&#39;: 1634626238.0,
 &#39;Scpiz6a_2736&#39;: 1636839212.0,
 &#39;Scpiz6a_2735&#39;: 1632552725.0,
 &#39;Scpiz6a_2042&#39;: 1636168284.0,
 &#39;Scpiz6a_2045&#39;: 1638270129.0,
 &#39;Scpiz6a_2732&#39;: 1635858909.0,
 &#39;Scpiz6a_2731&#39;: 1637668857.0,
 &#39;Scpiz6a_2046&#39;: 1635681780.0,
 &#39;Scpiz6a_2049&#39;: 1638913700.0,
 &#39;Scpiz6a_2048&#39;: 1637210246.0,
 &#39;Scpiz6a_2739&#39;: 1637378263.0,
 &#39;Scpiz6a_2738&#39;: 1636205032.0,
 &#39;Scpiz6a_3045&#39;: 1638975490.0,
 &#39;Scpiz6a_2559&#39;: 1635847671.0,
 &#39;Scpiz6a_2558&#39;: 1635125778.0,
 &#39;Scpiz6a_2557&#39;: 1634541746.0,
 &#39;Scpiz6a_2556&#39;: 1637365493.0,
 &#39;Scpiz6a_2555&#39;: 1633912899.0,
 &#39;Scpiz6a_2554&#39;: 1634648849.0,
 &#39;Scpiz6a_2553&#39;: 1638709190.0,
 &#39;Scpiz6a_2552&#39;: 1639085506.0,
 &#39;Scpiz6a_2551&#39;: 1635989764.0,
 &#39;Scpiz6a_2550&#39;: 1635964228.0,
 &#39;Scpiz6a_2148&#39;: 1638087529.0,
 &#39;Scpiz6a_93&#39;: 1635136790.0,
 &#39;Scpiz6a_90&#39;: 1327567734.0,
 &#39;Scpiz6a_147&#39;: 912569336.0,
 &#39;Scpiz6a_146&#39;: 449721910.0,
 &#39;Scpiz6a_145&#39;: 1636881232.0,
 &#39;Scpiz6a_144&#39;: 1634993721.0,
 &#39;Scpiz6a_2139&#39;: 1635838286.0,
 &#39;Scpiz6a_142&#39;: 1634641345.0,
 &#39;Scpiz6a_141&#39;: 1632082201.0,
 &#39;Scpiz6a_68&#39;: 1606102572.0,
 &#39;Scpiz6a_2135&#39;: 1635821371.0,
 &#39;Scpiz6a_66&#39;: 1355282223.0,
 &#39;Scpiz6a_65&#39;: 561993337.0,
 &#39;Scpiz6a_2136&#39;: 1633052192.0,
 &#39;Scpiz6a_2131&#39;: 1635966055.0,
 &#39;Scpiz6a_62&#39;: 1634894693.0,
 &#39;Scpiz6a_149&#39;: 1636561231.0,
 &#39;Scpiz6a_60&#39;: 1538020002.0,
 &#39;Scpiz6a_3558&#39;: 1636970251.0,
 &#39;Scpiz6a_3559&#39;: 1639013831.0,
 &#39;Scpiz6a_94&#39;: 1636580764.0,
 &#39;Scpiz6a_3550&#39;: 1638414297.0,
 &#39;Scpiz6a_3551&#39;: 1636944328.0,
 &#39;Scpiz6a_3552&#39;: 1634565048.0,
 &#39;Scpiz6a_153&#39;: 1637952282.0,
 &#39;Scpiz6a_3554&#39;: 1635034448.0,
 &#39;Scpiz6a_1028&#39;: 1635774058.0,
 &#39;Scpiz6a_3556&#39;: 1638806624.0,
 &#39;Scpiz6a_3557&#39;: 1638517682.0,
 &#39;Scpiz6a_2140&#39;: 1638954575.0,
 &#39;Scpiz6a_1029&#39;: 1636811034.0,
 &#39;Scpiz6a_3398&#39;: 1634847732.0,
 &#39;Scpiz6a_3399&#39;: 1637430455.0,
 &#39;Scpiz6a_3396&#39;: 1637689076.0,
 &#39;Scpiz6a_3397&#39;: 1635644762.0,
 &#39;Scpiz6a_3394&#39;: 1635392195.0,
 &#39;Scpiz6a_3395&#39;: 1638736532.0,
 &#39;Scpiz6a_3392&#39;: 1636592130.0,
 &#39;Scpiz6a_98&#39;: 1637236443.0,
 &#39;Scpiz6a_3390&#39;: 1637528049.0,
 &#39;Scpiz6a_3391&#39;: 1637105646.0,
 &#39;Scpiz6a_1985&#39;: 1637317061.0,
 &#39;Scpiz6a_99&#39;: 1590008057.0,
 &#39;Scpiz6a_1984&#39;: 1636325831.0,
 &#39;Scpiz6a_2412&#39;: 1638053357.0,
 &#39;Scpiz6a_1987&#39;: 1638297026.0,
 &#39;Scpiz6a_2145&#39;: 1638649476.0,
 &#39;Scpiz6a_1022&#39;: 1636439177.0,
 &#39;Scpiz6a_1454&#39;: 1638749020.0,
 &#39;Scpiz6a_2410&#39;: 1636614812.0,
 &#39;Scpiz6a_1023&#39;: 1638712616.0,
 &#39;Scpiz6a_1981&#39;: 1639140365.0,
 &#39;Scpiz6a_2147&#39;: 1637396664.0,
 &#39;Scpiz6a_1452&#39;: 1637811392.0,
 &#39;Scpiz6a_851&#39;: 1636530183.0,
 &#39;Scpiz6a_850&#39;: 1637821920.0,
 &#39;Scpiz6a_853&#39;: 1638262780.0,
 &#39;Scpiz6a_852&#39;: 1634908676.0,
 &#39;Scpiz6a_855&#39;: 1637630945.0,
 &#39;Scpiz6a_854&#39;: 1633455330.0,
 &#39;Scpiz6a_857&#39;: 1638249287.0,
 &#39;Scpiz6a_856&#39;: 1635871966.0,
 &#39;Scpiz6a_859&#39;: 1632881707.0,
 &#39;Scpiz6a_858&#39;: 1634086614.0,
 &#39;Scpiz6a_1450&#39;: 1637457082.0,
 &#39;Scpiz6a_3178&#39;: 1636022355.0,
 &#39;Scpiz6a_3179&#39;: 1637459879.0,
 &#39;Scpiz6a_3172&#39;: 1638613595.0,
 &#39;Scpiz6a_3173&#39;: 1639055189.0,
 &#39;Scpiz6a_3170&#39;: 1635461725.0,
 &#39;Scpiz6a_3171&#39;: 1636400548.0,
 &#39;Scpiz6a_3176&#39;: 1639490768.0,
 &#39;Scpiz6a_3177&#39;: 1639030199.0,
 &#39;Scpiz6a_3174&#39;: 1639329821.0,
 &#39;Scpiz6a_3175&#39;: 1635987947.0,
 &#39;Scpiz6a_2953&#39;: 1639304934.0,
 &#39;Scpiz6a_2952&#39;: 1637783681.0,
 &#39;Scpiz6a_2951&#39;: 1634502231.0,
 &#39;Scpiz6a_2950&#39;: 1638469142.0,
 &#39;Scpiz6a_2957&#39;: 1635193491.0,
 &#39;Scpiz6a_2956&#39;: 1638743348.0,
 &#39;Scpiz6a_2955&#39;: 1639042163.0,
 &#39;Scpiz6a_2954&#39;: 1634854809.0,
 &#39;Scpiz6a_2959&#39;: 1635693422.0,
 &#39;Scpiz6a_2958&#39;: 1634871267.0,
 &#39;Scpiz6a_1822&#39;: 1637550166.0,
 &#39;Scpiz6a_1823&#39;: 1638460819.0,
 &#39;Scpiz6a_1820&#39;: 1635268609.0,
 &#39;Scpiz6a_1821&#39;: 1634438845.0,
 &#39;Scpiz6a_1826&#39;: 1636503892.0,
 &#39;Scpiz6a_1827&#39;: 1635654525.0,
 &#39;Scpiz6a_1824&#39;: 1635494112.0,
 &#39;Scpiz6a_1825&#39;: 1633713795.0,
 &#39;Scpiz6a_1828&#39;: 1638426251.0,
 &#39;Scpiz6a_1829&#39;: 1639215882.0,
 &#39;Scpiz6a_92&#39;: 1638528287.0,
 &#39;Scpiz6a_3546&#39;: 1639163868.0,
 &#39;Scpiz6a_1552&#39;: 1634843002.0,
 &#39;Scpiz6a_924&#39;: 1636825138.0,
 &#39;Scpiz6a_927&#39;: 1632696671.0,
 &#39;Scpiz6a_926&#39;: 1636336074.0,
 &#39;Scpiz6a_921&#39;: 1637958728.0,
 &#39;Scpiz6a_1557&#39;: 1633171268.0,
 &#39;Scpiz6a_1554&#39;: 1636500596.0,
 &#39;Scpiz6a_922&#39;: 1638230834.0,
 &#39;Scpiz6a_886&#39;: 1638034278.0,
 &#39;Scpiz6a_1558&#39;: 1637698487.0,
 &#39;Scpiz6a_1559&#39;: 1639407840.0,
 &#39;Scpiz6a_929&#39;: 1634950296.0,
 &#39;Scpiz6a_928&#39;: 1636016940.0,
 &#39;Scpiz6a_3630&#39;: 1638525931.0,
 &#39;Scpiz6a_3770&#39;: 1637504472.0,
 &#39;Scpiz6a_3771&#39;: 1637989574.0,
 &#39;Scpiz6a_3772&#39;: 1636814172.0,
 &#39;Scpiz6a_3773&#39;: 1634913330.0,
 &#39;Scpiz6a_3774&#39;: 1633894483.0,
 &#39;Scpiz6a_3633&#39;: 1638600827.0,
 &#39;Scpiz6a_3776&#39;: 1639225382.0,
 &#39;Scpiz6a_61&#39;: 1637565329.0,
 &#39;Scpiz6a_3778&#39;: 1637840322.0,
 &#39;Scpiz6a_3779&#39;: 1639385390.0,
 &#39;Scpiz6a_885&#39;: 1638142723.0,
 &#39;Scpiz6a_1956&#39;: 1639285163.0,
 &#39;Scpiz6a_1957&#39;: 1637042827.0,
 &#39;Scpiz6a_1954&#39;: 1635866374.0,
 &#39;Scpiz6a_1955&#39;: 1635217228.0,
 &#39;Scpiz6a_1952&#39;: 1633535263.0,
 &#39;Scpiz6a_1953&#39;: 1639288291.0,
 &#39;Scpiz6a_1950&#39;: 1637833755.0,
 &#39;Scpiz6a_1951&#39;: 1635747392.0,
 &#39;Scpiz6a_1958&#39;: 1636559601.0,
 &#39;Scpiz6a_1959&#39;: 1635595693.0,
 &#39;Scpiz6a_2132&#39;: 1639059529.0,
 &#39;Scpiz6a_1738&#39;: 1636747902.0,
 &#39;Scpiz6a_1739&#39;: 1638575199.0,
 &#39;Scpiz6a_1736&#39;: 1634938791.0,
 &#39;Scpiz6a_1737&#39;: 1635416874.0,
 &#39;Scpiz6a_1734&#39;: 1637018733.0,
 &#39;Scpiz6a_1735&#39;: 1634520852.0,
 &#39;Scpiz6a_1732&#39;: 1639424121.0,
 &#39;Scpiz6a_1733&#39;: 1638682865.0,
 &#39;Scpiz6a_1730&#39;: 1634986894.0,
 &#39;Scpiz6a_1731&#39;: 1639038902.0,
 &#39;Scpiz6a_639&#39;: 1635997031.0,
 &#39;Scpiz6a_638&#39;: 1634406797.0,
 &#39;Scpiz6a_3047&#39;: 1638408310.0,
 &#39;Scpiz6a_631&#39;: 1634208904.0,
 &#39;Scpiz6a_630&#39;: 1634600862.0,
 &#39;Scpiz6a_633&#39;: 1635984308.0,
 &#39;Scpiz6a_632&#39;: 1636704899.0,
 &#39;Scpiz6a_635&#39;: 1638737669.0,
 &#39;Scpiz6a_634&#39;: 1634127117.0,
 &#39;Scpiz6a_637&#39;: 1635804393.0,
 &#39;Scpiz6a_636&#39;: 1636477456.0,
 &#39;Scpiz6a_2899&#39;: 1637116056.0,
 &#39;Scpiz6a_177.1&#39;: 1631079089.0,
 &#39;Scpiz6a_3609&#39;: 1637579067.0,
 &#39;Scpiz6a_1924&#39;: 1638366208.0,
 &#39;Scpiz6a_3604&#39;: 1638158938.0,
 &#39;Scpiz6a_3605&#39;: 1639298697.0,
 &#39;Scpiz6a_3606&#39;: 1638639072.0,
 &#39;Scpiz6a_3607&#39;: 1633960896.0,
 &#39;Scpiz6a_3600&#39;: 1638845894.0,
 &#39;Scpiz6a_3601&#39;: 1637192703.0,
 &#39;Scpiz6a_3602&#39;: 1636847020.0,
 &#39;Scpiz6a_3603&#39;: 1636086913.0,
 &#39;Scpiz6a_1449&#39;: 1639025838.0,
 &#39;Scpiz6a_3386&#39;: 1637993420.0,
 &#39;Scpiz6a_2624&#39;: 1638228368.0,
 &#39;Scpiz6a_1608&#39;: 1634980059.0,
 &#39;Scpiz6a_1609&#39;: 1635469841.0,
 &#39;Scpiz6a_2419&#39;: 1638718322.0,
 &#39;Scpiz6a_1602&#39;: 1638132730.0,
 &#39;Scpiz6a_1603&#39;: 1635666216.0,
 &#39;Scpiz6a_1600&#39;: 1635398382.0,
 &#39;Scpiz6a_1601&#39;: 1638213546.0,
 &#39;Scpiz6a_1606&#39;: 1638334753.0,
 &#39;Scpiz6a_1607&#39;: 1637468261.0,
 &#39;Scpiz6a_1604&#39;: 1635703108.0,
 &#39;Scpiz6a_1605&#39;: 1637189773.0,
 &#39;Scpiz6a_709&#39;: 1634177644.0,
 &#39;Scpiz6a_708&#39;: 1637739891.0,
 &#39;Scpiz6a_705&#39;: 1638290921.0,
 &#39;Scpiz6a_704&#39;: 1636201546.0,
 &#39;Scpiz6a_707&#39;: 1632263361.0,
 &#39;Scpiz6a_706&#39;: 1639179841.0,
 &#39;Scpiz6a_701&#39;: 1639191528.0,
 &#39;Scpiz6a_700&#39;: 1632935983.0,
 &#39;Scpiz6a_703&#39;: 1636804754.0,
 &#39;Scpiz6a_702&#39;: 1636899788.0,
 &#39;Scpiz6a_423.1&#39;: 1613339411.0,
 &#39;Scpiz6a_448&#39;: 1636666432.0,
 &#39;Scpiz6a_449&#39;: 1633285286.0,
 &#39;Scpiz6a_442&#39;: 1631907437.0,
 &#39;Scpiz6a_443&#39;: 1635047918.0,
 &#39;Scpiz6a_440&#39;: 1636660002.0,
 &#39;Scpiz6a_441&#39;: 1638416691.0,
 &#39;Scpiz6a_446&#39;: 1638405914.0,
 &#39;Scpiz6a_447&#39;: 1621891978.0,
 &#39;Scpiz6a_444&#39;: 1636796898.0,
 &#39;Scpiz6a_445&#39;: 1635962401.0,
 &#39;Scpiz6a_1040&#39;: 1638227134.0,
 &#39;Scpiz6a_55.2&#39;: 1573213224.0,
 &#39;Scpiz6a_55.1&#39;: 85027298.0,
 &#39;Scpiz6a_1043&#39;: 1639365921.0,
 &#39;Scpiz6a_1044&#39;: 1637133841.0,
 &#39;Scpiz6a_1045&#39;: 1636232838.0,
 &#39;Scpiz6a_1046&#39;: 1638196213.0,
 &#39;Scpiz6a_1047&#39;: 1636915195.0,
 &#39;Scpiz6a_1048&#39;: 1635206455.0,
 &#39;Scpiz6a_1049&#39;: 1631752598.0,
 &#39;Scpiz6a_253.1&#39;: 1634017894.0,
 &#39;Scpiz6a_2751&#39;: 1638438185.0,
 &#39;Scpiz6a_2100&#39;: 1638531819.0,
 &#39;Scpiz6a_2750&#39;: 1634628761.0,
 &#39;Scpiz6a_2753&#39;: 1639214826.0,
 &#39;Scpiz6a_2752&#39;: 1637113084.0,
 &#39;Scpiz6a_1421&#39;: 1639508841.0,
 &#39;Scpiz6a_1156&#39;: 1637111597.0,
 &#39;Scpiz6a_217&#39;: 1638833589.0,
 &#39;Scpiz6a_2754&#39;: 1636896702.0,
 &#39;Scpiz6a_2636&#39;: 1637062348.0,
 &#39;Scpiz6a_2637&#39;: 1638377063.0,
 &#39;Scpiz6a_2634&#39;: 1637519740.0,
 &#39;Scpiz6a_2635&#39;: 1635627142.0,
 &#39;Scpiz6a_2632&#39;: 1635304657.0,
 &#39;Scpiz6a_2633&#39;: 1636378571.0,
 &#39;Scpiz6a_2630&#39;: 1638246829.0,
 &#39;Scpiz6a_2631&#39;: 1638346867.0,
 &#39;Scpiz6a_2756&#39;: 1635232275.0,
 &#39;Scpiz6a_2638&#39;: 1637572203.0,
 &#39;Scpiz6a_2639&#39;: 1638536525.0,
 &#39;Scpiz6a_2613&#39;: 1639360781.0,
 &#39;Scpiz6a_2388&#39;: 1638331113.0,
 &#39;Scpiz6a_2389&#39;: 1639361809.0,
 &#39;Scpiz6a_2386&#39;: 1637337042.0,
 &#39;Scpiz6a_2387&#39;: 1636672857.0,
 &#39;Scpiz6a_2384&#39;: 1636104717.0,
 &#39;Scpiz6a_2385&#39;: 1635980663.0,
 &#39;Scpiz6a_2382&#39;: 1634427393.0,
 &#39;Scpiz6a_2383&#39;: 1638293364.0,
 &#39;Scpiz6a_2380&#39;: 1636738367.0,
 &#39;Scpiz6a_2381&#39;: 1636606719.0,
 &#39;Scpiz6a_435.1&#39;: 1629598643.0,
 &#39;Scpiz6a_60.1&#39;: 1632741856.0,
 &#39;Scpiz6a_2702&#39;: 1636412339.0,
 &#39;Scpiz6a_2703&#39;: 1637389597.0,
 &#39;Scpiz6a_2700&#39;: 1638224666.0,
 &#39;Scpiz6a_2701&#39;: 1638686306.0,
 &#39;Scpiz6a_2706&#39;: 1637651279.0,
 &#39;Scpiz6a_2707&#39;: 1637102668.0,
 &#39;Scpiz6a_2704&#39;: 1635971536.0,
 &#39;Scpiz6a_2705&#39;: 1638136478.0,
 &#39;Scpiz6a_2708&#39;: 1638204885.0,
 &#39;Scpiz6a_2709&#39;: 1635794927.0,
 &#39;Scpiz6a_2568&#39;: 1638585697.0,
 &#39;Scpiz6a_2569&#39;: 1637274022.0,
 &#39;Scpiz6a_2358&#39;: 1639485738.0,
 &#39;Scpiz6a_2562&#39;: 1636890521.0,
 &#39;Scpiz6a_2563&#39;: 1637625507.0,
 &#39;Scpiz6a_2560&#39;: 1634447331.0,
 &#39;Scpiz6a_2561&#39;: 1637410765.0,
 &#39;Scpiz6a_2566&#39;: 1638113948.0,
 &#39;Scpiz6a_2567&#39;: 1637763812.0,
 &#39;Scpiz6a_2564&#39;: 1635768358.0,
 &#39;Scpiz6a_2565&#39;: 1636798470.0,
 &#39;Scpiz6a_2098&#39;: 1635599634.0,
 &#39;Scpiz6a_2099&#39;: 1638842540.0,
 &#39;Scpiz6a_2092&#39;: 1639521853.0,
 &#39;Scpiz6a_2093&#39;: 1638450105.0,
 &#39;Scpiz6a_2090&#39;: 1639331891.0,
 &#39;Scpiz6a_3041&#39;: 1638250515.0,
 &#39;Scpiz6a_2096&#39;: 1637476636.0,
 &#39;Scpiz6a_2097&#39;: 1635570008.0,
 &#39;Scpiz6a_2094&#39;: 1638986471.0,
 &#39;Scpiz6a_2095&#39;: 1637540498.0,
 &#39;Scpiz6a_154&#39;: 1635739730.0,
 &#39;Scpiz6a_2149&#39;: 1635548173.0,
 &#39;Scpiz6a_156&#39;: 1634744805.0,
 &#39;Scpiz6a_91&#39;: 1636502245.0,
 &#39;Scpiz6a_150&#39;: 1474288736.0,
 &#39;Scpiz6a_151&#39;: 1638503518.0,
 &#39;Scpiz6a_152&#39;: 1634866573.0,
 &#39;Scpiz6a_95&#39;: 1635674001.0,
 &#39;Scpiz6a_2416&#39;: 1638727434.0,
 &#39;Scpiz6a_2141&#39;: 1635038941.0,
 &#39;Scpiz6a_2142&#39;: 1636484075.0,
 &#39;Scpiz6a_2143&#39;: 1635105862.0,
 &#39;Scpiz6a_158&#39;: 1634376358.0,
 &#39;Scpiz6a_159&#39;: 1636525264.0,
 &#39;Scpiz6a_2146&#39;: 1634673750.0,
 &#39;Scpiz6a_2411&#39;: 1636255341.0,
 &#39;Scpiz6a_2122&#39;: 1638829106.0,
 &#39;Scpiz6a_3549&#39;: 1635056879.0,
 &#39;Scpiz6a_3548&#39;: 1636402235.0,
 &#39;Scpiz6a_3163&#39;: 1637406539.0,
 &#39;Scpiz6a_2123&#39;: 1639411914.0,
 &#39;Scpiz6a_3543&#39;: 1637437475.0,
 &#39;Scpiz6a_3542&#39;: 1639164934.0,
 &#39;Scpiz6a_3541&#39;: 1638939134.0,
 &#39;Scpiz6a_3540&#39;: 1635352840.0,
 &#39;Scpiz6a_3547&#39;: 1635877552.0,
 &#39;Scpiz6a_2432&#39;: 1637626867.0,
 &#39;Scpiz6a_1928&#39;: 1637636371.0,
 &#39;Scpiz6a_3544&#39;: 1638001106.0,
 &#39;Scpiz6a_3389&#39;: 1639282032.0,
 &#39;Scpiz6a_3388&#39;: 1638925871.0,
 &#39;Scpiz6a_2890&#39;: 1639442373.0,
 &#39;Scpiz6a_2891&#39;: 1636693705.0,
 &#39;Scpiz6a_2896&#39;: 1639185156.0,
 &#39;Scpiz6a_2897&#39;: 1638670226.0,
 &#39;Scpiz6a_2894&#39;: 1634792997.0,
 &#39;Scpiz6a_2895&#39;: 1635334055.0,
 &#39;Scpiz6a_3381&#39;: 1634816872.0,
 &#39;Scpiz6a_3380&#39;: 1634515579.0,
 &#39;Scpiz6a_3383&#39;: 1635065821.0,
 &#39;Scpiz6a_3382&#39;: 1612164559.0,
 &#39;Scpiz6a_3385&#39;: 1636060117.0,
 &#39;Scpiz6a_3384&#39;: 1635212922.0,
 &#39;Scpiz6a_3387&#39;: 1634984616.0,
 &#39;Scpiz6a_2127&#39;: 1637074324.0,
 &#39;Scpiz6a_2436&#39;: 1636447523.0,
 &#39;Scpiz6a_2962&#39;: 1637908243.0,
 &#39;Scpiz6a_1151&#39;: 1636276055.0,
 &#39;Scpiz6a_2125&#39;: 1637758500.0,
 &#39;Scpiz6a_3168&#39;: 1636442516.0,
 &#39;Scpiz6a_846&#39;: 1637480819.0,
 &#39;Scpiz6a_847&#39;: 1636720855.0,
 &#39;Scpiz6a_844&#39;: 1637904343.0,
 &#39;Scpiz6a_845&#39;: 1637458481.0,
 &#39;Scpiz6a_842&#39;: 1637114570.0,
 &#39;Scpiz6a_843&#39;: 1636495646.0,
 &#39;Scpiz6a_840&#39;: 1632823512.0,
 &#39;Scpiz6a_841&#39;: 1636118923.0,
 &#39;Scpiz6a_556&#39;: 1639348423.0,
 &#39;Scpiz6a_366&#39;: 1638688600.0,
 &#39;Scpiz6a_848&#39;: 1636510474.0,
 &#39;Scpiz6a_849&#39;: 1635546183.0,
 &#39;Scpiz6a_3165&#39;: 1636909039.0,
 &#39;Scpiz6a_3164&#39;: 1637622786.0,
 &#39;Scpiz6a_3167&#39;: 1638762615.0,
 &#39;Scpiz6a_3166&#39;: 1636719260.0,
 &#39;Scpiz6a_3161&#39;: 1638698906.0,
 &#39;Scpiz6a_3160&#39;: 1636094042.0,
 &#39;Scpiz6a_2968&#39;: 1634415699.0,
 &#39;Scpiz6a_3162&#39;: 1637949702.0,
 &#39;Scpiz6a_2966&#39;: 1634409765.0,
 &#39;Scpiz6a_2967&#39;: 1635811951.0,
 &#39;Scpiz6a_2964&#39;: 1634510256.0,
 &#39;Scpiz6a_2965&#39;: 1634845368.0,
 &#39;Scpiz6a_3169&#39;: 1634623712.0,
 &#39;Scpiz6a_2963&#39;: 1635245131.0,
 &#39;Scpiz6a_2960&#39;: 1635475921.0,
 &#39;Scpiz6a_2961&#39;: 1638275025.0,
 &#39;Scpiz6a_3414&#39;: 1638188774.0,
 &#39;Scpiz6a_2808&#39;: 1639012738.0,
 &#39;Scpiz6a_938&#39;: 1636420747.0,
 &#39;Scpiz6a_939&#39;: 1635027692.0,
 &#39;Scpiz6a_1549&#39;: 1637354126.0,
 &#39;Scpiz6a_1548&#39;: 1634109346.0,
 &#39;Scpiz6a_1545&#39;: 1634354229.0,
 &#39;Scpiz6a_933&#39;: 1635329873.0,
 &#39;Scpiz6a_1547&#39;: 1638509422.0,
 &#39;Scpiz6a_1546&#39;: 1634646349.0,
 &#39;Scpiz6a_1541&#39;: 1633445801.0,
 &#39;Scpiz6a_937&#39;: 1634658824.0,
 &#39;Scpiz6a_934&#39;: 1637778389.0,
 &#39;Scpiz6a_935&#39;: 1637216082.0,
 &#39;Scpiz6a_2414&#39;: 1637493350.0,
 &#39;Scpiz6a_3763&#39;: 1638326254.0,
 &#39;Scpiz6a_3762&#39;: 1637705190.0,
 &#39;Scpiz6a_3761&#39;: 1637832441.0,
 &#39;Scpiz6a_3760&#39;: 1635554142.0,
 &#39;Scpiz6a_3767&#39;: 1637687731.0,
 &#39;Scpiz6a_3766&#39;: 1635373605.0,
 &#39;Scpiz6a_3765&#39;: 1635331964.0,
 &#39;Scpiz6a_3764&#39;: 1638337179.0,
 &#39;Scpiz6a_3769&#39;: 1636097602.0,
 &#39;Scpiz6a_3768&#39;: 1637224818.0,
 &#39;Scpiz6a_1207&#39;: 1637466865.0,
 &#39;Scpiz6a_1941&#39;: 1637757172.0,
 &#39;Scpiz6a_1940&#39;: 1638489323.0,
 &#39;Scpiz6a_1943&#39;: 1639483726.0,
 &#39;Scpiz6a_1942&#39;: 1635490077.0,
 &#39;Scpiz6a_1945&#39;: 1638106411.0,
 &#39;Scpiz6a_1944&#39;: 1637461277.0,
 &#39;Scpiz6a_1947&#39;: 1637332766.0,
 &#39;Scpiz6a_1946&#39;: 1636647126.0,
 &#39;Scpiz6a_1949&#39;: 1636052952.0,
 &#39;Scpiz6a_1948&#39;: 1639102773.0,
 &#39;Scpiz6a_1729&#39;: 1638892642.0,
 &#39;Scpiz6a_1728&#39;: 1634826391.0,
 &#39;Scpiz6a_1721&#39;: 1633781034.0,
 &#39;Scpiz6a_1720&#39;: 1638220961.0,
 &#39;Scpiz6a_1723&#39;: 1634461344.0,
 &#39;Scpiz6a_1722&#39;: 1633787484.0,
 &#39;Scpiz6a_1725&#39;: 1636727228.0,
 &#39;Scpiz6a_1724&#39;: 1635982486.0,
 &#39;Scpiz6a_1727&#39;: 1637848198.0,
 &#39;Scpiz6a_1726&#39;: 1637849510.0,
 &#39;Scpiz6a_628&#39;: 1639021474.0,
 &#39;Scpiz6a_629&#39;: 1638905950.0,
 &#39;Scpiz6a_626&#39;: 1637857378.0,
 &#39;Scpiz6a_627&#39;: 1630211994.0,
 &#39;Scpiz6a_624&#39;: 1637914743.0,
 &#39;Scpiz6a_625&#39;: 1636117150.0,
 &#39;Scpiz6a_622&#39;: 1639238028.0,
 &#39;Scpiz6a_623&#39;: 1637084776.0,
 &#39;Scpiz6a_620&#39;: 1636191067.0,
 &#39;Scpiz6a_621&#39;: 1639076857.0,
 &#39;Scpiz6a_1079&#39;: 1638630973.0,
 &#39;Scpiz6a_3619&#39;: 1638068570.0,
 &#39;Scpiz6a_3618&#39;: 1637664804.0,
 &#39;Scpiz6a_3617&#39;: 1638840304.0,
 &#39;Scpiz6a_3616&#39;: 1635132388.0,
 &#39;Scpiz6a_3615&#39;: 1639404783.0,
 &#39;Scpiz6a_3614&#39;: 1637624147.0,
 &#39;Scpiz6a_3613&#39;: 1636330953.0,
 &#39;Scpiz6a_3612&#39;: 1635336145.0,
 &#39;Scpiz6a_3611&#39;: 1639231709.0,
 &#39;Scpiz6a_3610&#39;: 1637386767.0,
 &#39;Scpiz6a_1839&#39;: 1639081182.0,
 &#39;Scpiz6a_1838&#39;: 1637632303.0,
 &#39;Scpiz6a_1835&#39;: 1638830227.0,
 &#39;Scpiz6a_1834&#39;: 1638867082.0,
 &#39;Scpiz6a_1837&#39;: 1635936726.0,
 &#39;Scpiz6a_1836&#39;: 1635735889.0,
 &#39;Scpiz6a_1831&#39;: 1636961114.0,
 &#39;Scpiz6a_1830&#39;: 1635942242.0,
 &#39;Scpiz6a_1833&#39;: 1638868195.0,
 &#39;Scpiz6a_1832&#39;: 1635591747.0,
 &#39;Scpiz6a_1934&#39;: 1637239347.0,
 &#39;Scpiz6a_1427&#39;: 1638980984.0,
 &#39;Scpiz6a_1784&#39;: 1638628658.0,
 &#39;Scpiz6a_1615&#39;: 1634424509.0,
 &#39;Scpiz6a_1614&#39;: 1636732004.0,
 &#39;Scpiz6a_1617&#39;: 1639370029.0,
 &#39;Scpiz6a_1616&#39;: 1635396320.0,
 &#39;Scpiz6a_1611&#39;: 1635701172.0,
 &#39;Scpiz6a_1610&#39;: 1638931400.0,
 &#39;Scpiz6a_1613&#39;: 1635843919.0,
 &#39;Scpiz6a_1612&#39;: 1636450858.0,
 &#39;Scpiz6a_1930&#39;: 1638317740.0,
 &#39;Scpiz6a_1619&#39;: 1635310975.0,
 &#39;Scpiz6a_1618&#39;: 1634068026.0,
 &#39;Scpiz6a_1423&#39;: 1636459185.0,
 &#39;Scpiz6a_712&#39;: 1637395252.0,
 &#39;Scpiz6a_288&#39;: 1634756910.0,
 &#39;Scpiz6a_710&#39;: 1636490689.0,
 &#39;Scpiz6a_711&#39;: 1636298429.0,
 &#39;Scpiz6a_716&#39;: 1636701703.0,
 &#39;Scpiz6a_717&#39;: 1635182644.0,
 &#39;Scpiz6a_714&#39;: 1638352917.0,
 &#39;Scpiz6a_2725&#39;: 1635730122.0,
 &#39;Scpiz6a_1387&#39;: 1639071447.0,
 &#39;Scpiz6a_718&#39;: 1637286974.0,
 &#39;Scpiz6a_1933&#39;: 1637006655.0,
 &#39;Scpiz6a_2054&#39;: 1639234869.0,
 &#39;Scpiz6a_932&#39;: 1638082480.0,
 &#39;Scpiz6a_324&#39;: 1638398716.0,
 &#39;Scpiz6a_3706&#39;: 1639478693.0,
 &#39;Scpiz6a_1544&#39;: 1638191254.0,
 &#39;Scpiz6a_2359&#39;: 1638728572.0,
 &#39;Scpiz6a_825&#39;: 1639034554.0,
 &#39;Scpiz6a_930&#39;: 1638968890.0,
 &#39;Scpiz6a_2053&#39;: 1636244964.0,
 &#39;Scpiz6a_931&#39;: 1638852593.0,
 &#39;Scpiz6a_2050&#39;: 1637023255.0,
 &#39;Scpiz6a_1171&#39;: 1638240683.0,
 &#39;Scpiz6a_1656&#39;: 1636515406.0,
 &#39;Scpiz6a_936&#39;: 1637641795.0,
 &#39;Scpiz6a_2051&#39;: 1639155331.0,
 &#39;Scpiz6a_1540&#39;: 1635154323.0,
 &#39;Scpiz6a_1543&#39;: 1637963881.0,
 &#39;Scpiz6a_439&#39;: 1639344295.0,
 &#39;Scpiz6a_438&#39;: 1639480707.0,
 &#39;Scpiz6a_437&#39;: 1635787345.0,
 &#39;Scpiz6a_436&#39;: 1638622868.0,
 &#39;Scpiz6a_435&#39;: 1627809148.0,
 &#39;Scpiz6a_434&#39;: 1597630794.0,
 &#39;Scpiz6a_433&#39;: 1636294993.0,
 &#39;Scpiz6a_432&#39;: 1639011645.0,
 &#39;Scpiz6a_431&#39;: 1636465839.0,
 &#39;Scpiz6a_430&#39;: 1638751288.0,
 &#39;Scpiz6a_1053&#39;: 1637551546.0,
 &#39;Scpiz6a_3684&#39;: 1635388068.0,
 &#39;Scpiz6a_1051&#39;: 1636862600.0,
 &#39;Scpiz6a_1050&#39;: 1637399486.0,
 &#39;Scpiz6a_1057&#39;: 1634931871.0,
 &#39;Scpiz6a_1056&#39;: 1636800041.0,
 &#39;Scpiz6a_1055&#39;: 1638684012.0,
 &#39;Scpiz6a_1054&#39;: 1636749490.0,
 &#39;Scpiz6a_284&#39;: 1637644505.0,
 &#39;Scpiz6a_1059&#39;: 1635697298.0,
 &#39;Scpiz6a_1058&#39;: 1638562342.0,
 &#39;Scpiz6a_3686&#39;: 1638264005.0,
 &#39;Scpiz6a_285&#39;: 1637799532.0,
 &#39;Scpiz6a_3687&#39;: 1639460557.0,
 &#39;Scpiz6a_2353&#39;: 1637471053.0,
 &#39;Scpiz6a_1371&#39;: 1639441361.0,
 &#39;Scpiz6a_364&#39;: 1637771768.0,
 &#39;Scpiz6a_3680&#39;: 1639096305.0,
 &#39;Scpiz6a_2352&#39;: 1637465469.0,
 &#39;Scpiz6a_1890&#39;: 1639308050.0,
 &#39;Scpiz6a_3681&#39;: 1638782956.0,
 &#39;Scpiz6a_2225&#39;: 1637923828.0,
 &#39;Scpiz6a_3682&#39;: 1636696906.0,
 &#39;Scpiz6a_3339&#39;: 1638811128.0,
 &#39;Scpiz6a_834&#39;: 1638016434.0,
 &#39;Scpiz6a_381&#39;: 1639479700.0,
 &#39;Scpiz6a_380&#39;: 1634668780.0,
 &#39;Scpiz6a_383&#39;: 1632042295.0,
 &#39;Scpiz6a_382&#39;: 1561455177.0,
 &#39;Scpiz6a_385&#39;: 1639131793.0,
 &#39;Scpiz6a_384&#39;: 1634577889.0,
 &#39;Scpiz6a_387&#39;: 1637865236.0,
 &#39;Scpiz6a_386&#39;: 1638345657.0,
 &#39;Scpiz6a_389&#39;: 1638379473.0,
 &#39;Scpiz6a_388&#39;: 1637637727.0,
 &#39;Scpiz6a_1052&#39;: 1638844776.0,
 &#39;Scpiz6a_543&#39;: 1638166408.0,
 &#39;Scpiz6a_2620&#39;: 1635873830.0,
 &#39;Scpiz6a_2623&#39;: 1635186988.0,
 &#39;Scpiz6a_540&#39;: 1634379467.0,
 &#39;Scpiz6a_547&#39;: 1639018199.0,
 &#39;Scpiz6a_546&#39;: 1637069838.0,
 &#39;Scpiz6a_545&#39;: 1636178814.0,
 &#39;Scpiz6a_544&#39;: 1638935820.0,
 &#39;Scpiz6a_2629&#39;: 1639111388.0,
 &#39;Scpiz6a_2628&#39;: 1634859516.0,
 &#39;Scpiz6a_549&#39;: 1638997431.0,
 &#39;Scpiz6a_548&#39;: 1633364850.0,
 &#39;Scpiz6a_2544&#39;: 1638721741.0,
 &#39;Scpiz6a_2545&#39;: 1638404715.0,
 &#39;Scpiz6a_2399&#39;: 1634528712.0,
 &#39;Scpiz6a_2398&#39;: 1635465783.0,
 &#39;Scpiz6a_2391&#39;: 1638459629.0,
 &#39;Scpiz6a_2390&#39;: 1634567619.0,
 &#39;Scpiz6a_2246&#39;: 1638422668.0,
 &#39;Scpiz6a_2392&#39;: 1639419037.0,
 &#39;Scpiz6a_2395&#39;: 1636904414.0,
 &#39;Scpiz6a_2394&#39;: 1636115376.0,
 &#39;Scpiz6a_2397&#39;: 1637956150.0,
 &#39;Scpiz6a_3335&#39;: 1636635832.0,
 &#39;Scpiz6a_2615&#39;: 1639493785.0,
 &#39;Scpiz6a_469&#39;: 1638902623.0,
 &#39;Scpiz6a_3337&#39;: 1639123210.0,
 &#39;Scpiz6a_2719&#39;: 1636422428.0,
 &#39;Scpiz6a_2718&#39;: 1638320174.0,
 &#39;Scpiz6a_2715&#39;: 1639159601.0,
 &#39;Scpiz6a_2714&#39;: 1636424108.0,
 &#39;Scpiz6a_2717&#39;: 1637679652.0,
 &#39;Scpiz6a_2716&#39;: 1637015715.0,
 &#39;Scpiz6a_2711&#39;: 1636024159.0,
 &#39;Scpiz6a_2710&#39;: 1639141435.0,
 &#39;Scpiz6a_2713&#39;: 1637417802.0,
 &#39;Scpiz6a_2712&#39;: 1637311342.0,
 &#39;Scpiz6a_2575&#39;: 1639256931.0,
 &#39;Scpiz6a_2574&#39;: 1634491458.0,
 &#39;Scpiz6a_2577&#39;: 1637035306.0,
 &#39;Scpiz6a_2576&#39;: 1638607795.0,
 &#39;Scpiz6a_2571&#39;: 1637512805.0,
 &#39;Scpiz6a_2570&#39;: 1638079954.0,
 &#39;Scpiz6a_2573&#39;: 1638800994.0,
 &#39;Scpiz6a_2572&#39;: 1635266482.0,
 &#39;Scpiz6a_30.1&#39;: 510652038.0,
 &#39;Scpiz6a_2579&#39;: 1639009459.0,
 &#39;Scpiz6a_2578&#39;: 1637925125.0,
 ...}
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [325]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="s1">&#39;genome_pos&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">scaffoldScaleDict</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="s1">&#39;height&#39;</span><span class="p">]</span><span class="o">=</span><span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">choice</span><span class="p">(</span><span class="n">a</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mf">0.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">,</span><span class="mf">0.05</span><span class="p">))</span> <span class="k">for</span> <span class="n">n</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">avgCovEqualHetSitesDF</span><span class="p">))]</span>
<span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">scaffoldSizeDict</span><span class="p">[</span><span class="n">x</span><span class="p">])</span>
<span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[325]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>pos</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>animal</th>
      <th>allele_count</th>
      <th>equal</th>
      <th>genome_pos</th>
      <th>height</th>
      <th>chrom_size</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>357652</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
      <td>357652.0</td>
      <td>0.85</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>533563</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
      <td>533563.0</td>
      <td>0.35</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>533564</td>
      <td>10</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
      <td>533564.0</td>
      <td>0.45</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>1076362</td>
      <td>10</td>
      <td>0</td>
      <td>0</td>
      <td>10</td>
      <td>20</td>
      <td>Transversion</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
      <td>1076362.0</td>
      <td>0.45</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>1258626</td>
      <td>10</td>
      <td>0</td>
      <td>10</td>
      <td>0</td>
      <td>20</td>
      <td>Transition</td>
      <td>Atig003</td>
      <td>(10, 10)</td>
      <td>True</td>
      <td>1258626.0</td>
      <td>0.25</td>
      <td>85027298</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [326]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="n">avgCovEqualHetSitesDF</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]</span> <span class="o">==</span><span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;animal&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[326]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>pos</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>cov</th>
      <th>status</th>
      <th>allele_count</th>
      <th>equal</th>
      <th>genome_pos</th>
      <th>height</th>
      <th>chrom_size</th>
    </tr>
    <tr>
      <th>animal</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>A.tig_12512</th>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
      <td>38</td>
    </tr>
    <tr>
      <th>A.tig_12513</th>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
      <td>29</td>
    </tr>
    <tr>
      <th>A.tig_9721</th>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
      <td>952</td>
    </tr>
    <tr>
      <th>A_tigris8450</th>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
      <td>41</td>
    </tr>
    <tr>
      <th>Atig001</th>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
      <td>629</td>
    </tr>
    <tr>
      <th>Atig003</th>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
      <td>610</td>
    </tr>
    <tr>
      <th>Atig_122</th>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
      <td>848</td>
    </tr>
    <tr>
      <th>Atig_4278</th>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
      <td>807</td>
    </tr>
    <tr>
      <th>Atig_6993</th>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
      <td>26</td>
    </tr>
    <tr>
      <th>Atig_9177</th>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
      <td>28</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [327]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="p">,</span><span class="n">axarr</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span>
    <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">10.5</span><span class="p">,</span> <span class="mi">4</span><span class="p">),</span>
    <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
    <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.0</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">}</span>
<span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">()):</span>
    <span class="k">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="p">))</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">avgCovEqualHetSitesDF</span><span class="p">[(</span><span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">avgCovEqualHetSitesDF</span><span class="o">.</span><span class="n">chrom_size</span> <span class="o">&gt;</span> <span class="mf">1e6</span><span class="p">)]</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="s1">&#39;genome_pos&#39;</span><span class="p">,</span> <span class="s1">&#39;height&#39;</span><span class="p">,</span> <span class="n">style</span> <span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">marker</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axarr</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mf">1.5</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">scaffoldSizes</span><span class="p">[</span><span class="n">scaffoldSizes</span><span class="o">.</span><span class="n">scaffold_size</span> <span class="o">&gt;</span> <span class="mf">1e6</span><span class="p">]</span><span class="o">.</span><span class="n">genome_scale</span><span class="p">,</span> <span class="n">ymin</span><span class="o">=-</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">ymax</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span><span class="n">lw</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span><span class="n">linestyles</span><span class="o">=</span><span class="s1">&#39;solid&#39;</span><span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;red&#39;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
598
30440
32832
46773
47695
52945
878
598
716
806
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_228_1.png" src="_images/marmorata_genome_paper_figures_3_228_1.png" />
</div>
</div>
</div>
<div class="section" id="Pysam-10kb-Non-Overlapping-Windows">
<h2>Pysam 10kb Non-Overlapping Windows<a class="headerlink" href="#Pysam-10kb-Non-Overlapping-Windows" title="Permalink to this headline">¶</a></h2>
<p>The following python script will apply a non-overlapping window across
all sequences represented in a profiles file and count the number of
even split heterozygous sites in each window.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># %load ../bin/scan_profile_no_lim.py</span>
<span class="c1">#!/usr/bin/env python</span>
<span class="c1">#Author: Duncan Tormey</span>
<span class="c1">#Email: dut@stowers.org or duncantormey@gmail.com</span>
<span class="c1">##################################################</span>
<span class="c1"># This script takes the  tsv output from         #</span>
<span class="c1"># pysam_profiler.py and applies a 10Kb sliding   #</span>
<span class="c1"># window accross each scaffod, counting sites as #</span>
<span class="c1"># heterozygous if the pass the conditions of the #</span>
<span class="c1"># function is_het. This script is currently hard #</span>
<span class="c1"># coded for my data.                             #</span>
<span class="c1">##################################################</span>

<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">print_function</span>
<span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">division</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="kn">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">multiprocessing</span> <span class="kn">as</span> <span class="nn">mp</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="kn">as</span> <span class="nn">np</span>

<span class="k">def</span> <span class="nf">apply_df</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">func</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">):</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">func</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>


<span class="k">def</span> <span class="nf">apply_by_multiprocessing</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">func</span><span class="p">,</span> <span class="n">workers</span><span class="p">,</span> <span class="o">*</span><span class="n">args</span><span class="p">):</span>
    <span class="n">pool</span> <span class="o">=</span> <span class="n">mp</span><span class="o">.</span><span class="n">Pool</span><span class="p">(</span><span class="n">processes</span><span class="o">=</span><span class="n">workers</span><span class="p">)</span>
    <span class="n">result</span> <span class="o">=</span> <span class="p">[</span><span class="n">pool</span><span class="o">.</span><span class="n">apply_async</span><span class="p">(</span><span class="n">apply_df</span><span class="p">,</span> <span class="n">args</span> <span class="o">=</span> <span class="p">(</span><span class="n">d</span><span class="p">,</span> <span class="n">func</span><span class="p">)</span> <span class="o">+</span> <span class="n">args</span><span class="p">)</span> <span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">array_split</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">workers</span><span class="p">)]</span>
    <span class="n">output</span> <span class="o">=</span> <span class="p">[</span><span class="n">p</span><span class="o">.</span><span class="n">get</span><span class="p">()</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">result</span><span class="p">]</span>
    <span class="n">pool</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>
    <span class="k">return</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">output</span><span class="p">)</span>



<span class="k">def</span> <span class="nf">check_equal_het</span><span class="p">(</span><span class="n">row</span><span class="p">):</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;T&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;C&#39;</span><span class="p">],</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">]]</span>
    <span class="n">non_zero_counts</span> <span class="o">=</span> <span class="p">[</span><span class="n">n</span> <span class="k">for</span> <span class="n">n</span> <span class="ow">in</span> <span class="n">counts</span> <span class="k">if</span> <span class="n">n</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">]</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">)</span><span class="o">==</span><span class="mi">2</span> <span class="ow">and</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">non_zero_counts</span><span class="p">))</span><span class="o">==</span><span class="mi">1</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">True</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">False</span>


<span class="k">def</span> <span class="nf">is_het</span><span class="p">(</span><span class="n">row</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="mi">8</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">False</span>
    <span class="k">elif</span> <span class="n">check_equal_het</span><span class="p">(</span><span class="n">row</span><span class="p">):</span>
        <span class="k">return</span> <span class="bp">True</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="k">return</span> <span class="bp">False</span>


<span class="k">def</span> <span class="nf">load_het_profiles</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">):</span>
    <span class="n">het_profs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">path</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
    <span class="c1">#het_profs[&#39;is_het&#39;] = het_profs.apply(lambda row: is_het(row, avg_cov))</span>
    <span class="n">het_profs</span><span class="p">[</span><span class="s1">&#39;is_het&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">apply_by_multiprocessing</span><span class="p">(</span><span class="n">het_profs</span><span class="p">,</span> <span class="n">is_het</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">)</span> <span class="c1">#this broke with all sites</span>
    <span class="n">het_profs</span><span class="p">[</span><span class="n">het_profs</span><span class="p">[</span><span class="s1">&#39;is_het&#39;</span><span class="p">]</span><span class="o">==</span><span class="bp">True</span><span class="p">]</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">path</span> <span class="o">+</span> <span class="s1">&#39;.is_het8.nolim&#39;</span><span class="p">,</span><span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">het_profs</span>


<span class="k">def</span> <span class="nf">return_window_df</span><span class="p">(</span><span class="n">het_profs</span><span class="p">,</span> <span class="n">scaffold_sizes</span><span class="p">,</span> <span class="n">window</span><span class="o">=</span><span class="mi">10000</span><span class="p">):</span>
    <span class="n">windows</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">append</span> <span class="o">=</span> <span class="n">windows</span><span class="o">.</span><span class="n">append</span>
    <span class="k">for</span> <span class="n">scaffold</span> <span class="ow">in</span> <span class="n">het_profs</span><span class="o">.</span><span class="n">chrom</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
        <span class="n">scaffold_data</span> <span class="o">=</span> <span class="n">het_profs</span><span class="p">[</span><span class="n">het_profs</span><span class="o">.</span><span class="n">chrom</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span>
        <span class="n">scaffold_size</span> <span class="o">=</span> <span class="n">scaffold_sizes</span><span class="p">[</span><span class="n">scaffold_sizes</span><span class="o">.</span><span class="n">scaffold</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">][</span><span class="s1">&#39;scaffold_size&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
        <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="n">scaffold_size</span><span class="p">,</span> <span class="n">window</span><span class="p">):</span>
            <span class="k">if</span> <span class="n">i</span><span class="o">+</span><span class="n">window</span> <span class="o">&lt;=</span> <span class="n">scaffold_size</span><span class="p">:</span>
                <span class="n">num_het</span> <span class="o">=</span> <span class="n">scaffold_data</span><span class="p">[(</span><span class="n">scaffold_data</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span><span class="o">&gt;=</span><span class="n">i</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">scaffold_data</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span><span class="o">&lt;</span><span class="n">i</span><span class="o">+</span><span class="n">window</span><span class="p">)][</span><span class="s1">&#39;is_het&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
                <span class="n">size</span> <span class="o">=</span> <span class="n">window</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">num_het</span> <span class="o">=</span> <span class="n">scaffold_data</span><span class="p">[</span><span class="n">scaffold_data</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="n">i</span><span class="p">][</span><span class="s1">&#39;is_het&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
                <span class="n">size</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">scaffold_data</span><span class="p">[</span><span class="n">scaffold_data</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="n">i</span><span class="p">])</span>

            <span class="n">append</span><span class="p">((</span><span class="n">scaffold</span><span class="p">,</span> <span class="n">i</span><span class="p">,</span> <span class="n">num_het</span><span class="p">,</span> <span class="n">size</span><span class="p">))</span>

    <span class="n">window_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">windows</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">,</span><span class="s1">&#39;window_start&#39;</span><span class="p">,</span> <span class="s1">&#39;het_sites&#39;</span><span class="p">,</span><span class="s1">&#39;window_size&#39;</span><span class="p">])</span>

    <span class="k">return</span> <span class="n">window_df</span>

<span class="k">def</span> <span class="nf">write_window_df</span><span class="p">(</span><span class="n">het_prof_path</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">,</span> <span class="n">scaffold_sizes_path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">window</span><span class="o">=</span><span class="mi">10000</span><span class="p">):</span>
    <span class="n">scaffold_sizes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">scaffold_sizes_path</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;scaffold&#39;</span><span class="p">,</span><span class="s1">&#39;scaffold_size&#39;</span><span class="p">])</span>
    <span class="n">het_profs</span> <span class="o">=</span> <span class="n">load_het_profiles</span><span class="p">(</span><span class="n">het_prof_path</span><span class="p">,</span> <span class="n">cpus</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">)</span>
    <span class="n">window_df</span> <span class="o">=</span> <span class="n">return_window_df</span><span class="p">(</span><span class="n">het_profs</span><span class="p">,</span> <span class="n">scaffold_sizes</span><span class="p">,</span> <span class="n">window</span><span class="p">)</span>
    <span class="n">window_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">het_prof_path</span><span class="o">+</span><span class="s1">&#39;.</span><span class="si">%s</span><span class="s1">kbwindows8.nolim.tsv&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">window</span><span class="o">/</span><span class="mi">1000</span><span class="p">),</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="bp">None</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">scaffold_sizes_path</span> <span class="o">=</span> <span class="s1">&#39;/home/dut/projects/tigris/genome_annotation/fasta/scaffold_sizes.clean.tsv&#39;</span>

    <span class="n">paths</span> <span class="o">=</span><span class="p">[</span><span class="s1">&#39;/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/pysam/Atig001.merged.dedup.realigned.prof.not_hom&#39;</span><span class="p">,</span>
            <span class="s1">&#39;/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/pysam/Atig003.merged.dedup.realigned.prof.not_hom&#39;</span><span class="p">,</span>
            <span class="s1">&#39;/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/pysam/Atig_122.merged.dedup.realigned.prof.not_hom&#39;</span><span class="p">,</span>
            <span class="s1">&#39;/n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/pysam/A_tigris8450.merged.dedup.realigned.prof.not_hom&#39;</span><span class="p">]</span>

    <span class="n">avg_covs</span> <span class="o">=</span> <span class="p">[</span><span class="mi">16</span><span class="p">,</span> <span class="mi">18</span><span class="p">,</span><span class="mi">19</span><span class="p">,</span><span class="mi">18</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">path</span><span class="p">,</span> <span class="n">avg_cov</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">paths</span><span class="p">,</span> <span class="n">avg_covs</span><span class="p">):</span>
        <span class="k">print</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
        <span class="n">write_window_df</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">avg_cov</span><span class="p">,</span> <span class="n">scaffold_sizes_path</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span> <span class="mi">10000</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [150]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">equalWindowPaths</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;Atig001&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig001.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A_tigris8450.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig003&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig003.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_122&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_122.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12512&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_12512.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_12513&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_12513.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;A.tig_9721&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/A.tig_9721.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_4278&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_4278.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_6993&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_6993.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
    <span class="s1">&#39;Atig_9177&#39;</span><span class="p">:</span> <span class="s1">&#39;../data/pysam/Atig_9177.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv&#39;</span><span class="p">,</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [151]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">ret_equal_het_windows</span><span class="p">(</span><span class="n">window_paths</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">sample_name</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">window_paths</span><span class="p">):</span>
        <span class="n">path</span> <span class="o">=</span> <span class="n">window_paths</span><span class="p">[</span><span class="n">sample_name</span><span class="p">]</span>
        <span class="n">window_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">window_df</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">sample_name</span>
        <span class="k">if</span> <span class="n">i</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">df_equal_het_windows</span> <span class="o">=</span> <span class="n">window_df</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">df_equal_het_windows</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">df_equal_het_windows</span><span class="p">,</span> <span class="n">window_df</span><span class="p">])</span>

        <span class="k">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df_equal_het_windows</span><span class="p">),</span> <span class="n">i</span><span class="p">,</span> <span class="n">path</span><span class="p">,</span> <span class="n">sample_name</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">df_equal_het_windows</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [152]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">equalHetWindowsDF</span> <span class="o">=</span> <span class="n">ret_equal_het_windows</span><span class="p">(</span><span class="n">equalWindowPaths</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
163465 0 ../data/pysam/A.tig_9721.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv A.tig_9721
326930 1 ../data/pysam/A_tigris8450.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv A_tigris8450
490395 2 ../data/pysam/A.tig_12512.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv A.tig_12512
653860 3 ../data/pysam/A.tig_12513.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv A.tig_12513
817325 4 ../data/pysam/Atig_4278.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig_4278
980790 5 ../data/pysam/Atig003.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig003
1144255 6 ../data/pysam/Atig001.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig001
1307720 7 ../data/pysam/Atig_6993.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig_6993
1471185 8 ../data/pysam/Atig_9177.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig_9177
1634650 9 ../data/pysam/Atig_122.merged.dedup.realigned.prof.not_hom.10.0kbwindows8.nolim.tsv Atig_122
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [153]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">chrom</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">scaffoldSizeDict</span><span class="p">[</span><span class="n">x</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [154]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">sort_values</span><span class="p">([</span><span class="s1">&#39;animal&#39;</span><span class="p">,</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">,</span> <span class="s1">&#39;window_start&#39;</span><span class="p">],</span> <span class="n">ascending</span><span class="o">=</span><span class="p">[</span><span class="bp">True</span><span class="p">,</span> <span class="bp">False</span><span class="p">,</span> <span class="bp">True</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[154]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>chrom</th>
      <th>window_start</th>
      <th>het_sites</th>
      <th>window_size</th>
      <th>animal</th>
      <th>chrom_size</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_49</td>
      <td>0</td>
      <td>1</td>
      <td>10000</td>
      <td>A.tig_12512</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Scpiz6a_49</td>
      <td>10000</td>
      <td>0</td>
      <td>10000</td>
      <td>A.tig_12512</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Scpiz6a_49</td>
      <td>20000</td>
      <td>0</td>
      <td>10000</td>
      <td>A.tig_12512</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Scpiz6a_49</td>
      <td>30000</td>
      <td>0</td>
      <td>10000</td>
      <td>A.tig_12512</td>
      <td>85027298</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Scpiz6a_49</td>
      <td>40000</td>
      <td>0</td>
      <td>10000</td>
      <td>A.tig_12512</td>
      <td>85027298</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [155]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span><span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>

<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
        <span class="n">rolling_mean_data</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span><span class="o">==</span><span class="n">animal</span><span class="p">]</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="mi">100</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span>
        <span class="n">sns</span><span class="o">.</span><span class="n">distplot</span><span class="p">(</span><span class="n">rolling_mean_data</span><span class="p">,</span>
                     <span class="n">kde</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                     <span class="n">hist_kws</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;alpha&quot;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span>
                               <span class="s2">&quot;color&quot;</span><span class="p">:</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                               <span class="s2">&quot;linewidth&quot;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span>
                               <span class="s2">&quot;histtype&quot;</span><span class="p">:</span> <span class="s2">&quot;step&quot;</span><span class="p">},</span>
                     <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                     <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>


<span class="c1">#ax5.set_title(&#39;1Mb Heterozygosity Distribution&#39;, fontsize=minorFontSize)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Windows&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Mean of Het Sites per 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.32</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>
<span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span> <span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure3C.pdf&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_237_0.png" src="_images/marmorata_genome_paper_figures_3_237_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [156]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">step</span><span class="o">=</span><span class="mi">500</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

    <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                      <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                      <span class="c1">#style=&#39;.&#39;,</span>
                                      <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                      <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                      <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                      <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                      <span class="c1">#markersize=1.0,</span>
                                      <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                     <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                     <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                     <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                    <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">4</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="c1">#fig.savefig(&#39;../fig2/Figure4A.pdf&#39;,bbox_inches=&#39;tight&#39;,dpi=300)</span>
<span class="c1">#fig.savefig(&#39;../fig/Figure4A.png&#39;,bbox_inches=&#39;tight&#39;,dpi=300)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[156]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>&lt;matplotlib.legend.Legend at 0x7f62d6ae72d0&gt;
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_238_1.png" src="_images/marmorata_genome_paper_figures_3_238_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [157]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span><span class="o">=</span><span class="mi">500</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">family</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="n">family_1</span><span class="p">,</span> <span class="n">family_2</span><span class="p">,</span> <span class="n">family_3</span><span class="p">]):</span>
    <span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span> <span class="mf">1.0</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">family</span><span class="p">:</span>
        <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

        <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                          <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                          <span class="c1">#style=&#39;.&#39;,</span>
                                          <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                          <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                          <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="c1">#markersize=1.0,</span>
                                          <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">,</span>
                                          <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                         <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                         <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                        <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
    <span class="c1">#ax.set_title(&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">3</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">4</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#     fig.savefig(&#39;../fig2/Figure4%s.pdf&#39; % letters[i], format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_239_0.png" src="_images/marmorata_genome_paper_figures_3_239_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_239_1.png" src="_images/marmorata_genome_paper_figures_3_239_1.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_239_2.png" src="_images/marmorata_genome_paper_figures_3_239_2.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [158]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span><span class="o">=</span><span class="mi">500</span>
<span class="n">fig</span><span class="p">,</span><span class="n">axarr</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span>
    <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.2</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span>
    <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
    <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.0</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">}</span>
<span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">family</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="n">family_1</span><span class="p">,</span> <span class="n">family_2</span><span class="p">,</span> <span class="n">family_3</span><span class="p">]):</span>
    <span class="n">ax</span><span class="o">=</span><span class="n">axarr</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">family</span><span class="p">:</span>
        <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

        <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                          <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                          <span class="c1">#style=&#39;.&#39;,</span>
                                          <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                          <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                          <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="c1">#markersize=1.0,</span>
                                          <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">,</span>
                                          <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                         <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                         <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                        <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">4.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
    <span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
    <span class="c1">#ax.set_title(&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_ylabel(&#39;Avg. Het. Sites / 10kb&#39;, fontsize=minorFontSize-3)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mf">3.5</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">0.04</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure4A.pdf&#39;</span><span class="p">,</span> <span class="n">format</span><span class="o">=</span><span class="s2">&quot;pdf&quot;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_240_0.png" src="_images/marmorata_genome_paper_figures_3_240_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [216]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span><span class="o">=</span><span class="mi">100</span>
<span class="n">fig</span><span class="p">,</span><span class="n">axarr</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span>
    <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.2</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span>
    <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
    <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.0</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">}</span>
<span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">family</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="n">family_1</span><span class="p">,</span> <span class="n">family_2</span><span class="p">,</span> <span class="n">family_3</span><span class="p">]):</span>
    <span class="n">ax</span><span class="o">=</span><span class="n">axarr</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">family</span><span class="p">:</span>
        <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

        <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                          <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                          <span class="c1">#style=&#39;.&#39;,</span>
                                          <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                          <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                          <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="c1">#markersize=1.0,</span>
                                          <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">,</span>
                                          <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                         <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                         <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                        <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">4.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
    <span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
    <span class="c1">#ax.set_title(&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_ylabel(&#39;Avg. Het. Sites / 10kb&#39;, fontsize=minorFontSize-3)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mf">4.5</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">0.04</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">)</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure4A.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[216]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>&lt;matplotlib.text.Text at 0x7f6350242fd0&gt;
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_241_1.png" src="_images/marmorata_genome_paper_figures_3_241_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [215]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span><span class="o">=</span><span class="mi">50</span>
<span class="n">fig</span><span class="p">,</span><span class="n">axarr</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span>
    <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.2</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span>
    <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
    <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.0</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">}</span>
<span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">family</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="n">family_1</span><span class="p">,</span> <span class="n">family_2</span><span class="p">,</span> <span class="n">family_3</span><span class="p">]):</span>
    <span class="n">ax</span><span class="o">=</span><span class="n">axarr</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">family</span><span class="p">:</span>
        <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

        <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                          <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                          <span class="c1">#style=&#39;.&#39;,</span>
                                          <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                          <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                          <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="c1">#markersize=1.0,</span>
                                          <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">,</span>
                                          <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                         <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                         <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                        <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">4.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
    <span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
    <span class="c1">#ax.set_title(&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_ylabel(&#39;Avg. Het. Sites / 10kb&#39;, fontsize=minorFontSize-3)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mf">4.5</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">0.04</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">)</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure4A.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_242_0.png" src="_images/marmorata_genome_paper_figures_3_242_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [218]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span><span class="o">=</span><span class="mi">10</span>
<span class="n">fig</span><span class="p">,</span><span class="n">axarr</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span>
    <span class="n">sharex</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span>
    <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.2</span><span class="p">,</span> <span class="mi">3</span><span class="p">),</span>
    <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span>
    <span class="n">gridspec_kw</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;wspace&#39;</span><span class="p">:</span><span class="mf">0.0</span><span class="p">,</span> <span class="s1">&#39;hspace&#39;</span><span class="p">:</span><span class="mf">0.1</span><span class="p">}</span>
<span class="p">)</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">family</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="n">family_1</span><span class="p">,</span> <span class="n">family_2</span><span class="p">,</span> <span class="n">family_3</span><span class="p">]):</span>
    <span class="n">ax</span><span class="o">=</span><span class="n">axarr</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
    <span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">family</span><span class="p">:</span>
        <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

        <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                          <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                          <span class="c1">#style=&#39;.&#39;,</span>
                                          <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span>
                                          <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>
                                          <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                          <span class="c1">#markersize=1.0,</span>
                                          <span class="n">alpha</span><span class="o">=</span><span class="mf">0.7</span><span class="p">,</span>
                                          <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                         <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                         <span class="n">color</span><span class="o">=</span><span class="n">family_colors</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                         <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                        <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">6.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
    <span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
    <span class="c1">#ax.set_title(&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_ylabel(&#39;Avg. Het. Sites / 10kb&#39;, fontsize=minorFontSize-3)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mf">6.5</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">0.04</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="s1">&#39;vertical&#39;</span><span class="p">)</span>
<span class="c1">#fig.savefig(&#39;../fig2/Figure4A.pdf&#39;, format=&quot;pdf&quot;,bbox_inches=&#39;tight&#39;)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[218]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>&lt;matplotlib.text.Text at 0x7f63a8f0aad0&gt;
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_243_1.png" src="_images/marmorata_genome_paper_figures_3_243_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [160]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span> <span class="o">=</span> <span class="mi">500</span>
<span class="n">mother</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="s1">&#39;Atig_122&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">fp_animal</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom_size&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mf">5e5</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">ratio</span> <span class="o">=</span> <span class="n">mother</span><span class="o">/</span><span class="n">fp_animal</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                <span class="n">style</span><span class="o">=</span><span class="s1">&#39;-&#39;</span><span class="p">,</span>
                <span class="n">label</span><span class="o">=</span><span class="n">animal</span><span class="p">,</span>
                <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                <span class="n">markersize</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span>
               <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mi">2</span><span class="p">),</span>
               <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Ratio of 122:8450 Average Het sites to per 10kb </span><span class="si">%s</span><span class="s1">Mb Sliding Window&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">step</span><span class="o">/</span><span class="mi">100</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;log2 ratio&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">9</span><span class="p">)</span>
<span class="n">mean_r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">se_r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="c1">#/np.sqrt(len(ratio.dropna()))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">1.0e4</span><span class="p">,</span><span class="mf">0.25</span><span class="p">,</span> <span class="s1">&#39;average log2(ratio) = </span><span class="si">%s</span><span class="se">\n</span><span class="s1">standard deviation log2(ratio) = </span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">mean_r</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">se_r</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/SupplementalFigure7.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/SupplementalFigure7.png&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">300</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_244_0.png" src="_images/marmorata_genome_paper_figures_3_244_0.png" />
</div>
</div>
</div>
<div class="section" id="V2R-Copy-Number-analysis-MAKER-Annotations">
<h2>V2R Copy Number analysis MAKER Annotations<a class="headerlink" href="#V2R-Copy-Number-analysis-MAKER-Annotations" title="Permalink to this headline">¶</a></h2>
<p>This first table I am reading in shows the copy number variation for
every gene annotated in the MAKER annotations.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [161]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">gffCopyNumMarmDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../../dovetail_vomeronasal/data/annotCopyNumsDF.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">gffCopyNumMarmDF</span> <span class="o">=</span> <span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">type</span><span class="o">==</span><span class="s2">&quot;gene&quot;</span><span class="p">]</span>
<span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">cnv</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="s1">&#39;CN1&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
<span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;copies&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;cnv&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">2</span><span class="p">]))</span>
<span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[161]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqnames</th>
      <th>start</th>
      <th>end</th>
      <th>width</th>
      <th>strand</th>
      <th>source</th>
      <th>type</th>
      <th>score</th>
      <th>phase</th>
      <th>ID</th>
      <th>...</th>
      <th>Note</th>
      <th>Dbxref</th>
      <th>Parent</th>
      <th>X_AED</th>
      <th>X_QI</th>
      <th>X_eAED</th>
      <th>Ontology_term</th>
      <th>score.1</th>
      <th>cnv</th>
      <th>copies</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Scpiz6a_1</td>
      <td>69274</td>
      <td>86971.0</td>
      <td>17698</td>
      <td>+</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gene1</td>
      <td>...</td>
      <td>Similar to LYPD6: Ly6/PLAUR domain-containing protein 6 (Homo sapiens)</td>
      <td>c("Gene3D:G3DSA:2.10.60.10", "SUPERFAMILY:SSF57302")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>11</th>
      <td>Scpiz6a_1</td>
      <td>105656</td>
      <td>124326.0</td>
      <td>18671</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gene2</td>
      <td>...</td>
      <td>Similar to MMADHC: Methylmalonic aciduria and homocystinuria type D homolog, mitochondrial (Gallus gallus)</td>
      <td>c("InterPro:IPR019362", "Pfam:PF10229")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>GO:0009235</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>40</th>
      <td>Scpiz6a_1</td>
      <td>458899</td>
      <td>468953.0</td>
      <td>10055</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gene3</td>
      <td>...</td>
      <td>Similar to Rnd3: Rho-related GTP-binding protein RhoE (Rattus norvegicus)</td>
      <td>c("Gene3D:G3DSA:3.40.50.300", "InterPro:IPR001806", "InterPro:IPR003578", "InterPro:IPR003579", "InterPro:IPR027417", "Pfam:PF00071", "ProSiteProfiles:PS51420", "SMART:SM00174", "SMART:SM00175", "SUPERFAMILY:SSF52540")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0005525", "GO:0005622", "GO:0007264", "GO:0015031")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>50</th>
      <td>Scpiz6a_1</td>
      <td>609021</td>
      <td>609749.0</td>
      <td>729</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gene4</td>
      <td>...</td>
      <td>Protein of unknown function</td>
      <td>c("Gene3D:G3DSA:1.10.443.10", "InterPro:IPR011010", "InterPro:IPR013762", "SUPERFAMILY:SSF56349")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0003677", "GO:0006310", "GO:0015074")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
    </tr>
    <tr>
      <th>54</th>
      <td>Scpiz6a_1</td>
      <td>677616</td>
      <td>678006.0</td>
      <td>391</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gene5</td>
      <td>...</td>
      <td>Similar to rgr-1: Mediator of RNA polymerase II transcription subunit 14 (Caenorhabditis briggsae)</td>
      <td>character(0)</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>character(0)</td>
      <td>1246.0</td>
      <td>CN1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 22 columns</p>
</div></div>
</div>
<p>Next I add a column to contain the gene symbol.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [162]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;Note&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;Similar to &#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[162]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>0     LYPD6
11    MMADHC
40    Rnd3
50    Protein of unknown function
54    rgr-1
Name: gene_symbol, dtype: object
</pre></div>
</div>
</div>
<p>Here I number of times each gene symbol occurs genome wide. Only gene
symbols that occur more than 10 times are shown.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [163]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">3.2</span><span class="p">))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">value_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s2">&quot;Protein of unknown function&quot;</span><span class="p">][</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">value_counts</span><span class="p">[</span><span class="n">value_counts</span><span class="o">.</span><span class="n">gene_symbol</span> <span class="o">&gt;</span> <span class="mi">10</span><span class="p">]</span>
<span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Genome Wide Counts&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Homologous Gene Symbol&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/vmnr_maker_all_counts.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_251_0.png" src="_images/marmorata_genome_paper_figures_3_251_0.png" />
</div>
</div>
<p>Here are the numbers from that plot. Vmn2r26 occurs 309 times genome
wide.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [164]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[164]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Vmn2r26</th>
      <td>309</td>
    </tr>
    <tr>
      <th>ZSCAN31</th>
      <td>55</td>
    </tr>
    <tr>
      <th>ZBED9</th>
      <td>35</td>
    </tr>
    <tr>
      <th>ZNF420</th>
      <td>28</td>
    </tr>
    <tr>
      <th>Znf24</th>
      <td>28</td>
    </tr>
    <tr>
      <th>TRIM27</th>
      <td>21</td>
    </tr>
    <tr>
      <th>cysS</th>
      <td>18</td>
    </tr>
    <tr>
      <th>rgr-1</th>
      <td>17</td>
    </tr>
    <tr>
      <th>Trim27</th>
      <td>14</td>
    </tr>
    <tr>
      <th>OR5V1</th>
      <td>13</td>
    </tr>
    <tr>
      <th>ZSCAN2</th>
      <td>13</td>
    </tr>
    <tr>
      <th>infB</th>
      <td>13</td>
    </tr>
    <tr>
      <th>At1g06800</th>
      <td>12</td>
    </tr>
    <tr>
      <th>ZFP2</th>
      <td>11</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Here I generate the same plot as above, but only for scaffold 45 (which
is the scaffold that shows increased apparent heterozygosity). Only
symbols 4 or more occurences are shown.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [165]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">3.2</span><span class="p">,</span> <span class="mf">3.2</span><span class="p">))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">value_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">gffCopyNumMarmDF</span><span class="p">[(</span><span class="n">gffCopyNumMarmDF</span><span class="p">[</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s2">&quot;Protein of unknown function&quot;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span><span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)][</span><span class="s1">&#39;gene_symbol&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">value_counts</span><span class="p">[</span><span class="n">value_counts</span><span class="o">.</span><span class="n">gene_symbol</span> <span class="o">&gt;</span> <span class="mi">4</span><span class="p">]</span>
<span class="n">data</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Scaffold 45 Counts&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Homologous Gene Symbol&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/vmnr_maker_45_counts.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_255_0.png" src="_images/marmorata_genome_paper_figures_3_255_0.png" />
</div>
</div>
<p>Here is the data for this plot. Vmn2r26 occurs 177 time on scaffold 45.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [166]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">data</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[166]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>gene_symbol</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Vmn2r26</th>
      <td>167</td>
    </tr>
    <tr>
      <th>OR14I1</th>
      <td>10</td>
    </tr>
    <tr>
      <th>OR11G2</th>
      <td>6</td>
    </tr>
    <tr>
      <th>PRSS27</th>
      <td>5</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>Here I save the subsetted table for all Vmn2r26 that are on scaffold 45.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [167]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrGffCopyDF</span> <span class="o">=</span> <span class="n">gffCopyNumMarmDF</span><span class="p">[(</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">Note</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39;Vmn2r26&#39;</span><span class="p">,</span><span class="n">case</span><span class="o">=</span><span class="bp">False</span><span class="p">))</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span><span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">vmnrGffCopyDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[167]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>index</th>
      <th>seqnames</th>
      <th>start</th>
      <th>end</th>
      <th>width</th>
      <th>strand</th>
      <th>source</th>
      <th>type</th>
      <th>score</th>
      <th>phase</th>
      <th>...</th>
      <th>Dbxref</th>
      <th>Parent</th>
      <th>X_AED</th>
      <th>X_QI</th>
      <th>X_eAED</th>
      <th>Ontology_term</th>
      <th>score.1</th>
      <th>cnv</th>
      <th>copies</th>
      <th>gene_symbol</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>465598</td>
      <td>Scpiz6a_45</td>
      <td>5125710</td>
      <td>5235885.0</td>
      <td>110176</td>
      <td>+</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>c("Gene3D:G3DSA:3.40.50.2300", "InterPro:IPR000337", "InterPro:IPR001828", "InterPro:IPR004073", "InterPro:IPR011500", "InterPro:IPR017978", "InterPro:IPR017979", "InterPro:IPR028082", "PRINTS:PR00248", "PRINTS:PR01535", "Pfam:PF00003", "Pfam:PF01094", "Pfam:PF07562", "ProSitePatterns:PS00981", "ProSiteProfiles:PS50259", "SUPERFAMILY:SSF53822", "SUPERFAMILY:SSF81665")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0004930", "GO:0007186", "GO:0016021")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>Vmn2r26</td>
    </tr>
    <tr>
      <th>1</th>
      <td>465885</td>
      <td>Scpiz6a_45</td>
      <td>5243577</td>
      <td>5320649.0</td>
      <td>77073</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>c("Gene3D:G3DSA:3.40.50.2300", "InterPro:IPR000337", "InterPro:IPR001828", "InterPro:IPR004073", "InterPro:IPR011500", "InterPro:IPR017978", "InterPro:IPR017979", "InterPro:IPR028082", "PRINTS:PR00248", "PRINTS:PR01535", "Pfam:PF00003", "Pfam:PF01094", "Pfam:PF07562", "ProSitePatterns:PS00981", "ProSiteProfiles:PS50259", "SUPERFAMILY:SSF53822", "SUPERFAMILY:SSF81665")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0004930", "GO:0007186", "GO:0016021")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>Vmn2r26</td>
    </tr>
    <tr>
      <th>2</th>
      <td>466138</td>
      <td>Scpiz6a_45</td>
      <td>5332681</td>
      <td>5367671.0</td>
      <td>34991</td>
      <td>-</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>c("Gene3D:G3DSA:3.40.50.2300", "InterPro:IPR000337", "InterPro:IPR001828", "InterPro:IPR004073", "InterPro:IPR011500", "InterPro:IPR017978", "InterPro:IPR017979", "InterPro:IPR028082", "PRINTS:PR00248", "PRINTS:PR01535", "Pfam:PF00003", "Pfam:PF01094", "Pfam:PF07562", "ProSitePatterns:PS00981", "ProSiteProfiles:PS50259", "SUPERFAMILY:SSF53822")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0004930", "GO:0007186", "GO:0016021")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>Vmn2r26</td>
    </tr>
    <tr>
      <th>3</th>
      <td>466383</td>
      <td>Scpiz6a_45</td>
      <td>5374537</td>
      <td>5392755.0</td>
      <td>18219</td>
      <td>+</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>c("Gene3D:G3DSA:3.40.50.2300", "InterPro:IPR000337", "InterPro:IPR001828", "InterPro:IPR004073", "InterPro:IPR011500", "InterPro:IPR017978", "InterPro:IPR017979", "InterPro:IPR028082", "PRINTS:PR00248", "PRINTS:PR01535", "Pfam:PF00003", "Pfam:PF01094", "Pfam:PF07562", "ProSitePatterns:PS00981", "ProSiteProfiles:PS50259", "SUPERFAMILY:SSF53822")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0004930", "GO:0007186", "GO:0016021")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>Vmn2r26</td>
    </tr>
    <tr>
      <th>4</th>
      <td>466470</td>
      <td>Scpiz6a_45</td>
      <td>5509516</td>
      <td>5545895.0</td>
      <td>36380</td>
      <td>+</td>
      <td>maker</td>
      <td>gene</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>c("Gene3D:G3DSA:3.40.50.2300", "InterPro:IPR000337", "InterPro:IPR001828", "InterPro:IPR004073", "InterPro:IPR011500", "InterPro:IPR017978", "InterPro:IPR017979", "InterPro:IPR028082", "PRINTS:PR00248", "PRINTS:PR01535", "Pfam:PF00003", "Pfam:PF01094", "Pfam:PF07562", "ProSitePatterns:PS00981", "ProSiteProfiles:PS50259", "SUPERFAMILY:SSF53822")</td>
      <td>character(0)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>c("GO:0004930", "GO:0007186", "GO:0016021")</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>Vmn2r26</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 24 columns</p>
</div></div>
</div>
<p>In this next section I compare the compy number estimations in 8450 to
the number of annotations present in the genome assembly</p>
<p>There are 323 copies of Vmn2r26 in 8450, according to the MAKER gene
annotations and the copy number variation analysis.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [168]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">gffCopyNumMarmDF</span><span class="p">[(</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">type</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gffCopyNumMarmDF</span><span class="o">.</span><span class="n">Note</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39;Vmn2r26&#39;</span><span class="p">,</span><span class="n">case</span><span class="o">=</span><span class="bp">False</span><span class="p">))]</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[168]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>323
</pre></div>
</div>
</div>
<p>Of the 323 copies, 177 reside on scaffold 45.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [169]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrGffCopyDF</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[169]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>177
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [170]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrGffCopyDF</span><span class="o">.</span><span class="n">cnv</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[170]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>&lt;matplotlib.axes._subplots.AxesSubplot at 0x7f64f1eadf50&gt;
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_265_1.png" src="_images/marmorata_genome_paper_figures_3_265_1.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [171]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrDF</span> <span class="o">=</span> <span class="n">gffMarmDF</span><span class="p">[(</span><span class="n">gffMarmDF</span><span class="o">.</span><span class="n">feature</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gffMarmDF</span><span class="o">.</span><span class="n">gene_symbol</span><span class="o">==</span><span class="s1">&#39;Vmn2r26&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gffMarmDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span><span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;width&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;end&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">]</span>
<span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;broken_bars&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">vmnrDF</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">]</span><span class="o">/</span><span class="mi">10000</span><span class="p">,</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;width&#39;</span><span class="p">]</span><span class="o">/</span><span class="mi">10000</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">vmnrDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[171]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>seqid</th>
      <th>source</th>
      <th>feature</th>
      <th>start</th>
      <th>end</th>
      <th>score</th>
      <th>strand</th>
      <th>phase</th>
      <th>attributes</th>
      <th>gene_attributes</th>
      <th>gene_symbol</th>
      <th>gene_id</th>
      <th>name</th>
      <th>count_column</th>
      <th>width</th>
      <th>broken_bars</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>482606</th>
      <td>Scpiz6a_45</td>
      <td>MAKER2</td>
      <td>gene</td>
      <td>5125710.0</td>
      <td>5235885.0</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>ID=gene17679;Name=ATIG_00000233;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-52.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822,SUPERFAMILY:SSF81665;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>ID=gene17679;Name=ATIG_00000233;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-52.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822,SUPERFAMILY:SSF81665;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>Vmn2r26</td>
      <td>gene17679</td>
      <td>ATIG_00000233</td>
      <td>(Scpiz6a_45, Vmn2r26)</td>
      <td>110175.0</td>
      <td>(512.571, 11.0175)</td>
    </tr>
    <tr>
      <th>482894</th>
      <td>Scpiz6a_45</td>
      <td>MAKER2</td>
      <td>gene</td>
      <td>5243577.0</td>
      <td>5320649.0</td>
      <td>.</td>
      <td>-</td>
      <td>.</td>
      <td>ID=gene17680;Name=ATIG_00000234;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-53.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822,SUPERFAMILY:SSF81665;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>ID=gene17680;Name=ATIG_00000234;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-53.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822,SUPERFAMILY:SSF81665;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>Vmn2r26</td>
      <td>gene17680</td>
      <td>ATIG_00000234</td>
      <td>(Scpiz6a_45, Vmn2r26)</td>
      <td>77072.0</td>
      <td>(524.3577, 7.7072)</td>
    </tr>
    <tr>
      <th>483148</th>
      <td>Scpiz6a_45</td>
      <td>MAKER2</td>
      <td>gene</td>
      <td>5332681.0</td>
      <td>5367671.0</td>
      <td>.</td>
      <td>-</td>
      <td>.</td>
      <td>ID=gene17681;Name=ATIG_00000235;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-53.1;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>ID=gene17681;Name=ATIG_00000235;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-53.1;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>Vmn2r26</td>
      <td>gene17681</td>
      <td>ATIG_00000235</td>
      <td>(Scpiz6a_45, Vmn2r26)</td>
      <td>34990.0</td>
      <td>(533.2681, 3.499)</td>
    </tr>
    <tr>
      <th>483394</th>
      <td>Scpiz6a_45</td>
      <td>MAKER2</td>
      <td>gene</td>
      <td>5374537.0</td>
      <td>5392755.0</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>ID=gene17682;Name=ATIG_00000236;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-55.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>ID=gene17682;Name=ATIG_00000236;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-55.0;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>Vmn2r26</td>
      <td>gene17682</td>
      <td>ATIG_00000236</td>
      <td>(Scpiz6a_45, Vmn2r26)</td>
      <td>18218.0</td>
      <td>(537.4537, 1.8218)</td>
    </tr>
    <tr>
      <th>483482</th>
      <td>Scpiz6a_45</td>
      <td>MAKER2</td>
      <td>gene</td>
      <td>5509516.0</td>
      <td>5545895.0</td>
      <td>.</td>
      <td>+</td>
      <td>.</td>
      <td>ID=gene17683;Name=ATIG_00000237;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-55.1;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>ID=gene17683;Name=ATIG_00000237;Alias=maker-Scpiz6a_45-exonerate_protein2genome-gene-55.1;Note=Similar to Vmn2r26: Vomeronasal type-2 receptor 26 (Mus musculus);Dbxref=Gene3D:G3DSA:3.40.50.2300,InterPro:IPR000337,InterPro:IPR001828,InterPro:IPR004073,InterPro:IPR011500,InterPro:IPR017978,InterPro:IPR017979,InterPro:IPR028082,PRINTS:PR00248,PRINTS:PR01535,Pfam:PF00003,Pfam:PF01094,Pfam:PF07562,ProSitePatterns:PS00981,ProSiteProfiles:PS50259,SUPERFAMILY:SSF53822;Ontology_term=GO:0004930,GO:0007186,GO:0016021</td>
      <td>Vmn2r26</td>
      <td>gene17683</td>
      <td>ATIG_00000237</td>
      <td>(Scpiz6a_45, Vmn2r26)</td>
      <td>36379.0</td>
      <td>(550.9516, 3.6379)</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [172]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span> <span class="mf">1.93</span><span class="p">),</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="mi">111</span><span class="p">)</span>
<span class="n">step</span><span class="o">=</span><span class="mi">100</span>
<span class="n">scaffold</span><span class="o">=</span><span class="s1">&#39;Scpiz6a_45&#39;</span>
<span class="n">lns</span> <span class="o">=</span> <span class="bp">None</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="n">scaffold_het_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">scaffold</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>
    <span class="n">ln1</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
           <span class="s1">&#39;.&#39;</span><span class="p">,</span>
            <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
            <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
            <span class="n">markersize</span><span class="o">=</span><span class="mf">2.5</span><span class="p">,</span>
                 <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

    <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                     <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                     <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span>
                   <span class="n">rasterized</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">lns</span><span class="p">:</span>
        <span class="n">lns</span> <span class="o">=</span> <span class="n">ln1</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">lns</span><span class="o">+=</span><span class="n">ln1</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">2.7</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">5.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Avg. Het Sites / 10Kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Window Start Position (Mb)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%d</span><span class="s1">Mb Sliding Window Scaffold: </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">step</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000</span><span class="p">,</span> <span class="n">scaffold</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">min</span><span class="p">(),</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">max</span><span class="p">())</span>

<span class="n">labs</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_label</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">lns</span><span class="p">]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">lns</span><span class="p">,</span> <span class="n">labs</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.20</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span><span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>



<span class="n">chrom</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">add_collection</span><span class="p">(</span><span class="n">BrokenBarHCollection</span><span class="p">(</span>
        <span class="p">[(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">min</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="p">,</span> <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">max</span><span class="p">()</span><span class="o">-</span><span class="mi">105</span><span class="p">)],</span>
        <span class="p">(</span><span class="o">-</span><span class="mf">2.2</span><span class="p">,</span><span class="mf">1.5</span><span class="p">),</span>
        <span class="n">facecolors</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="s1">&#39;A_tigris8450&#39;</span><span class="p">],</span>
        <span class="n">alpha</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span>
        <span class="n">linewidths</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                                      <span class="p">)</span>
    <span class="p">)</span>


<span class="n">ax</span><span class="o">.</span><span class="n">add_collection</span><span class="p">(</span><span class="n">BrokenBarHCollection</span><span class="p">(</span>
        <span class="n">vmnrDF</span><span class="p">[</span><span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;feature&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">,</span>
        <span class="p">(</span><span class="o">-</span><span class="mf">2.2</span><span class="p">,</span><span class="mf">1.5</span><span class="p">),</span>
        <span class="n">facecolors</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;black&#39;</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrDF</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">))],</span>
        <span class="n">linewidths</span><span class="o">=</span><span class="p">[</span><span class="mf">0.5</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrDF</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">))]</span>
                                      <span class="p">)</span>
    <span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig2/Figure4B.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_267_0.png" src="_images/marmorata_genome_paper_figures_3_267_0.png" />
</div>
</div>
</div>
<div class="section" id="V2R-Exonerate-Copy-Number-analysis">
<h2>V2R Exonerate Copy Number analysis<a class="headerlink" href="#V2R-Exonerate-Copy-Number-analysis" title="Permalink to this headline">¶</a></h2>
<p>Instead of using the maker annotations I also tried identifying v2r
genes by aligning the mouse ortholog to the genome with exonerate.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [173]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../../dovetail_vomeronasal/data/vmnrCopyNumsDF.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [174]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[174]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Unnamed: 0</th>
      <th>seqnames</th>
      <th>start</th>
      <th>end</th>
      <th>width</th>
      <th>strand</th>
      <th>source</th>
      <th>type</th>
      <th>score</th>
      <th>phase</th>
      <th>...</th>
      <th>similarity</th>
      <th>insertions</th>
      <th>deletions</th>
      <th>intron_id</th>
      <th>splice_site</th>
      <th>alignment_id</th>
      <th>Query</th>
      <th>Align</th>
      <th>frameshifts</th>
      <th>cnv</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64910441</td>
      <td>2105</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>683.0</td>
      <td>NaN</td>
      <td>...</td>
      <td>63.30</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>cds</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>exon</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>69.05</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64908465</td>
      <td>2</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>splice5</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1.0</td>
      <td>GT</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64909578</td>
      <td>1115</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>intron</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>1.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 24 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [175]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;copies&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;cnv&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="mi">2</span><span class="p">]))</span>
<span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;broken_bars&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">row</span><span class="p">:</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">]</span><span class="o">/</span><span class="mi">10000</span><span class="p">,</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;width&#39;</span><span class="p">]</span><span class="o">/</span><span class="mi">10000</span><span class="p">),</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[175]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Unnamed: 0</th>
      <th>seqnames</th>
      <th>start</th>
      <th>end</th>
      <th>width</th>
      <th>strand</th>
      <th>source</th>
      <th>type</th>
      <th>score</th>
      <th>phase</th>
      <th>...</th>
      <th>deletions</th>
      <th>intron_id</th>
      <th>splice_site</th>
      <th>alignment_id</th>
      <th>Query</th>
      <th>Align</th>
      <th>frameshifts</th>
      <th>cnv</th>
      <th>copies</th>
      <th>broken_bars</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64910441</td>
      <td>2105</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>683.0</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.2105)</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>cds</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.0127)</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>exon</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>0.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.0127)</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64908465</td>
      <td>2</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>splice5</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>1.0</td>
      <td>GT</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8464, 0.0002)</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64909578</td>
      <td>1115</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>intron</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>1.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8464, 0.1115)</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 26 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [176]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">exon_lengths</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">exon_length</span><span class="o">=</span> <span class="bp">None</span>
<span class="k">for</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="mi">8</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">exon_length</span><span class="p">:</span>
            <span class="n">i</span><span class="o">=</span><span class="mi">1</span>
            <span class="n">exon_length</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="k">continue</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">exon_length</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon_length</span><span class="p">]</span><span class="o">*</span><span class="n">i</span>
            <span class="n">exon_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">exon_length</span><span class="p">)</span>
            <span class="n">exon_length</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">i</span><span class="o">=</span><span class="mi">0</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="mi">8</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;exon&quot;</span><span class="p">:</span>
        <span class="n">exon_length</span><span class="o">+=</span><span class="n">row</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span>
    <span class="n">i</span><span class="o">+=</span><span class="mi">1</span>

<span class="n">exon_length</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon_length</span><span class="p">]</span><span class="o">*</span><span class="n">i</span>
<span class="n">exon_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">exon_length</span><span class="p">)</span>

<span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;sum_exon_lengths&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">exon_lengths</span>
<span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[176]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Unnamed: 0</th>
      <th>seqnames</th>
      <th>start</th>
      <th>end</th>
      <th>width</th>
      <th>strand</th>
      <th>source</th>
      <th>type</th>
      <th>score</th>
      <th>phase</th>
      <th>...</th>
      <th>intron_id</th>
      <th>splice_site</th>
      <th>alignment_id</th>
      <th>Query</th>
      <th>Align</th>
      <th>frameshifts</th>
      <th>cnv</th>
      <th>copies</th>
      <th>broken_bars</th>
      <th>sum_exon_lengths</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64910441</td>
      <td>2105</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>gene</td>
      <td>683.0</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.2105)</td>
      <td>990</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>cds</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.0127)</td>
      <td>990</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>Scpiz6a_49</td>
      <td>64908337</td>
      <td>64908463</td>
      <td>127</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>exon</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8337, 0.0127)</td>
      <td>990</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64908465</td>
      <td>2</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>splice5</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>1.0</td>
      <td>GT</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8464, 0.0002)</td>
      <td>990</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>Scpiz6a_49</td>
      <td>64908464</td>
      <td>64909578</td>
      <td>1115</td>
      <td>+</td>
      <td>exonerate:protein2genome:local</td>
      <td>intron</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>...</td>
      <td>1.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>CN1</td>
      <td>1</td>
      <td>(6490.8464, 0.1115)</td>
      <td>990</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 27 columns</p>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [177]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span> <span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">1000</span><span class="p">)</span> <span class="o">&amp;</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[177]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>196
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [178]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span> <span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">1000</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)]</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[178]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>206
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [179]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">1000</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[179]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>205
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [180]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">((</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">1000</span><span class="p">))]</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[180]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>215
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [181]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span> <span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">800</span><span class="p">)</span> <span class="o">&amp;</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[181]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>309
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [182]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">seqnames</span> <span class="o">==</span> <span class="s1">&#39;Scpiz6a_45&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">800</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;gene&#39;</span><span class="p">)]</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[182]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>319
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [183]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">800</span><span class="p">)])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[183]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>466
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [184]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="p">[(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">((</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span><span class="o">&gt;</span><span class="mi">800</span><span class="p">))]</span><span class="o">.</span><span class="n">copies</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[184]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>478
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [185]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">sorted</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;exon&#39;</span><span class="p">][</span><span class="s1">&#39;broken_bars&#39;</span><span class="p">],</span> <span class="n">key</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">])[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[185]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>(738.6374, 0.0002)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [186]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="nb">sorted</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="n">vmnrCopyNumbers</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">][</span><span class="s1">&#39;broken_bars&#39;</span><span class="p">],</span> <span class="n">key</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="p">[</span><span class="mi">1</span><span class="p">])[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[186]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>(9.7566, 0.0072)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [187]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">cnv</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[187]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>array([&#39;CN1&#39;, &#39;CN3&#39;, &#39;CN4&#39;], dtype=object)
</pre></div>
</div>
</div>
</div>
<div class="section" id="V2R-Tree">
<h2>V2R Tree<a class="headerlink" href="#V2R-Tree" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [188]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/single_vomeronasal_proteins alignment FastTree Tree_names.newick.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[188]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_286_0.png" src="_images/marmorata_genome_paper_figures_3_286_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [189]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/single_vomeronasal_proteins alignment FastTree Tree.newick.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[189]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_287_0.png" src="_images/marmorata_genome_paper_figures_3_287_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [190]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../fig/single_vomeronasal_proteins alignment FastTree Tree.newick.pdf ../fig2/Figure7C.pdf
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="stderr output_area docutils container">
<div class="highlight"><pre>
cp: target ‘../fig2/Figure7C.pdf’ is not a directory
</pre></div></div>
</div>
</div>
<div class="section" id="Embryonic-FACs-Analysis">
<h2>Embryonic FACs Analysis<a class="headerlink" href="#Embryonic-FACs-Analysis" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [191]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../fig/14-Jun-2017-Layout2_real_percentages.pdf.pdf ../fig2/Figure5.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [192]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/14-Jun-2017-Layout2_real_percentages.jpg&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[192]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_291_0.jpeg" src="_images/marmorata_genome_paper_figures_3_291_0.jpeg" />
</div>
</div>
</div>
<div class="section" id="Blood-FACs-analysis">
<h2>Blood FACs analysis<a class="headerlink" href="#Blood-FACs-analysis" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [193]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../fig/blood_ploidy_1.pdf ../fig2/Figure6A.pdf
cp ../fig/blood_ploidy_2.pdf ../fig2/Figure6B.pdf
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [194]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/blood_ploidy2.png.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[194]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_294_0.png" src="_images/marmorata_genome_paper_figures_3_294_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [195]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">Image</span><span class="p">(</span><span class="s1">&#39;../fig/blood_ploidy1.png&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[195]:
</pre></div>
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_295_0.png" src="_images/marmorata_genome_paper_figures_3_295_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [196]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
cp ../data/8449-8450<span class="se">\ </span>FACS.xlsx ../data/supplemental_data/supplemental_table_8.xlsx
</pre></div>
</div>
</div>
</div>
<div class="section" id="Wild-Caught-Microsatellite-Analysis">
<h2>Wild Caught Microsatellite Analysis<a class="headerlink" href="#Wild-Caught-Microsatellite-Analysis" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [197]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildExcelFile</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">ExcelFile</span><span class="p">(</span><span class="s1">&#39;../data/20180122_Wild_marmoratus_genotyping_results.xlsx&#39;</span><span class="p">)</span>
<span class="n">wildExcelFile</span><span class="o">.</span><span class="n">sheet_names</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[197]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>[u&#39;MOLGN_15379_15380_15381&#39;,
 u&#39;MOLGN_15411&#39;,
 u&#39;MOLGN_15403_15404&#39;,
 u&#39;MOLGN_15396_15397&#39;,
 u&#39;MOLGN_15393_15394_15395&#39;,
 u&#39;MOLGN_15299&#39;]
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [198]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildGenotypeDF</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">wildExcelFile</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="n">sheet</span><span class="p">)</span> <span class="k">for</span> <span class="n">sheet</span> <span class="ow">in</span> <span class="n">wildExcelFile</span><span class="o">.</span><span class="n">sheet_names</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">drop</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [199]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
<span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">columns</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[199]:
</pre></div>
</div>
<div class="output_area highlight-none"><div class="highlight"><pre>
<span></span>Index([  u&#39;Allele 1&#39;,   u&#39;Allele 2&#39;,   u&#39;Allele 3&#39;,   u&#39;Allele 4&#39;,
         u&#39;Allele 5&#39;,   u&#39;Comments&#39;,   u&#39;Height 1&#39;,   u&#39;Height 2&#39;,
         u&#39;Height 3&#39;,   u&#39;Height 4&#39;,   u&#39;Height 5&#39;,         u&#39;MS&#39;,
        u&#39;Sample ID&#39;,     u&#39;Size 1&#39;,     u&#39;Size 2&#39;,     u&#39;Size 3&#39;,
           u&#39;Size 4&#39;,     u&#39;Size 5&#39;, u&#39;Unnamed: 4&#39;, u&#39;Unnamed: 7&#39;,
       u&#39;Unnamed: 9&#39;,   u&#39;comments&#39;],
      dtype=&#39;object&#39;)
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [200]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildGenotypeDF</span> <span class="o">=</span> <span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s1">&#39;Sample.*|Size.*|Height.*&#39;</span><span class="p">)</span>
<span class="n">wildGenotypeDF</span><span class="p">[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">wildGenotypeDF</span><span class="p">[</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;-&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[200]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>1</th>
      <td>5001.0</td>
      <td>5348.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Ai5013-LizardMS</td>
      <td>213.60</td>
      <td>220.39</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>2</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Ai5043-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2724.0</td>
      <td>2731.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Cvanu24-LizardMS</td>
      <td>200.00</td>
      <td>203.92</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2936.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Cvanu7-LizardMS</td>
      <td>331.63</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [201]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">wildGenotypeDF</span><span class="p">[</span><span class="n">wildGenotypeDF</span><span class="p">[</span><span class="s1">&#39;Size 3&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">notnull</span><span class="p">()]</span><span class="o">.</span><span class="n">sample_name</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">display</span><span class="p">(</span><span class="n">wildGenotypeDF</span><span class="p">[</span><span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="n">sample</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1464</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-A105-LizardMS_F07.fsa</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1465</th>
      <td>9703.0</td>
      <td>7719.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-Ai5013-LizardMS_F08.fsa</td>
      <td>226.05</td>
      <td>255.51</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1466</th>
      <td>11107.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-Ai5043-LizardMS_F12.fsa</td>
      <td>175.63</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1467</th>
      <td>32656.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-Cvanu24-LizardMS_F06.fsa</td>
      <td>199.82</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1468</th>
      <td>1345.0</td>
      <td>4682.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-Cvanu7-LizardMS_F05.fsa</td>
      <td>342.13</td>
      <td>351.25</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1469</th>
      <td>359.0</td>
      <td>322.0</td>
      <td>6870.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-D106-LizardMS_F09.fsa</td>
      <td>264.53</td>
      <td>285.45</td>
      <td>457.81</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1470</th>
      <td>3881.0</td>
      <td>2537.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>not sure about 146 as single peak only</td>
      <td>TJH3028-D107-LizardMS_F10.fsa</td>
      <td>140.76</td>
      <td>145.97</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1471</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-D111-LizardMS_F11.fsa</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1472</th>
      <td>30216.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-MS1-LizardMS_F01.fsa</td>
      <td>234.96</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1473</th>
      <td>32588.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-MS6-LizardMS_F02.fsa</td>
      <td>176.08</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1474</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-MS7-LizardMS_F03.fsa</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
    <tr>
      <th>1475</th>
      <td>32086.0</td>
      <td>1549.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028-MS8-LizardMS_F04.fsa</td>
      <td>113.19</td>
      <td>138.90</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>TJH3028</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2712</th>
      <td>64.0</td>
      <td>39.0</td>
      <td>58.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-A105-LizardMS</td>
      <td>219.37</td>
      <td>227.77</td>
      <td>238.69</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2713</th>
      <td>26543.0</td>
      <td>22494.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-Ai5013-LizardMS</td>
      <td>217.46</td>
      <td>236.64</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2714</th>
      <td>4418.0</td>
      <td>4625.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-Ai5043-LizardMS</td>
      <td>175.80</td>
      <td>181.56</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2715</th>
      <td>31654.0</td>
      <td>23268.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-Cvanu24-LizardMS</td>
      <td>199.91</td>
      <td>203.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2716</th>
      <td>5253.0</td>
      <td>3638.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-Cvanu7-LizardMS</td>
      <td>331.55</td>
      <td>334.64</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2717</th>
      <td>6457.0</td>
      <td>5573.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-D106-LizardMS</td>
      <td>294.97</td>
      <td>303.13</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2718</th>
      <td>374.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-D107-LizardMS</td>
      <td>144.13</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2719</th>
      <td>5985.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-D111-LizardMS</td>
      <td>150.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2720</th>
      <td>15727.0</td>
      <td>12459.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-MS1-LizardMS</td>
      <td>217.79</td>
      <td>234.96</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2721</th>
      <td>32488.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-MS6-LizardMS</td>
      <td>174.11</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2722</th>
      <td>22967.0</td>
      <td>5195.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-MS7-LizardMS</td>
      <td>274.52</td>
      <td>285.27</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
    <tr>
      <th>2723</th>
      <td>31064.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075-MS8-LizardMS</td>
      <td>112.98</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104075</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2736</th>
      <td>78.0</td>
      <td>109.0</td>
      <td>83.0</td>
      <td>58.0</td>
      <td>NaN</td>
      <td>104082-A105-LizardMS</td>
      <td>193.93</td>
      <td>219.15</td>
      <td>235.55</td>
      <td>255.87</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2737</th>
      <td>32477.0</td>
      <td>31434.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-Ai5013-LizardMS</td>
      <td>220.19</td>
      <td>236.54</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2738</th>
      <td>18103.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-Ai5043-LizardMS</td>
      <td>175.62</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2739</th>
      <td>32158.0</td>
      <td>25910.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-Cvanu24-LizardMS</td>
      <td>199.91</td>
      <td>201.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2740</th>
      <td>32364.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-Cvanu7-LizardMS</td>
      <td>331.32</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2741</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2742</th>
      <td>4094.0</td>
      <td>3748.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-D107-LizardMS</td>
      <td>149.80</td>
      <td>154.15</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2743</th>
      <td>7918.0</td>
      <td>5367.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-D111-LizardMS</td>
      <td>150.09</td>
      <td>202.98</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2744</th>
      <td>15817.0</td>
      <td>14636.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-MS1-LizardMS</td>
      <td>216.85</td>
      <td>235.03</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2745</th>
      <td>32128.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-MS6-LizardMS</td>
      <td>173.95</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2746</th>
      <td>32685.0</td>
      <td>24497.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-MS7-LizardMS</td>
      <td>227.08</td>
      <td>258.10</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
    <tr>
      <th>2747</th>
      <td>31102.0</td>
      <td>31552.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082-MS8-LizardMS</td>
      <td>113.03</td>
      <td>117.37</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104082</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2772</th>
      <td>60.0</td>
      <td>62.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-A105-LizardMS</td>
      <td>221.68</td>
      <td>231.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2773</th>
      <td>13684.0</td>
      <td>24650.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-Ai5013-LizardMS</td>
      <td>224.20</td>
      <td>233.49</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2774</th>
      <td>1147.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-Ai5043-LizardMS</td>
      <td>175.66</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2775</th>
      <td>18467.0</td>
      <td>25741.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-Cvanu24-LizardMS</td>
      <td>201.85</td>
      <td>203.79</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2776</th>
      <td>5604.0</td>
      <td>4056.0</td>
      <td>2654.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-Cvanu7-LizardMS</td>
      <td>315.33</td>
      <td>331.53</td>
      <td>334.65</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2777</th>
      <td>6119.0</td>
      <td>11117.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-D106-LizardMS</td>
      <td>286.83</td>
      <td>319.68</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2778</th>
      <td>68.0</td>
      <td>103.0</td>
      <td>68.0</td>
      <td>34.0</td>
      <td>NaN</td>
      <td>104142-D107-LizardMS</td>
      <td>139.42</td>
      <td>144.03</td>
      <td>297.12</td>
      <td>300.45</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2779</th>
      <td>3001.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-D111-LizardMS</td>
      <td>150.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2780</th>
      <td>32672.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-MS1-LizardMS</td>
      <td>234.84</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2781</th>
      <td>32323.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-MS6-LizardMS</td>
      <td>173.92</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2782</th>
      <td>20527.0</td>
      <td>7502.0</td>
      <td>4529.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-MS7-LizardMS</td>
      <td>227.22</td>
      <td>254.25</td>
      <td>269.70</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
    <tr>
      <th>2783</th>
      <td>31030.0</td>
      <td>19019.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142-MS8-LizardMS</td>
      <td>112.19</td>
      <td>117.40</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104142</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2784</th>
      <td>81.0</td>
      <td>86.0</td>
      <td>33.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-A105-LizardMS</td>
      <td>215.96</td>
      <td>223.75</td>
      <td>239.83</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2785</th>
      <td>32341.0</td>
      <td>31900.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-Ai5013-LizardMS</td>
      <td>217.94</td>
      <td>225.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2786</th>
      <td>15320.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-Ai5043-LizardMS</td>
      <td>178.43</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2787</th>
      <td>31667.0</td>
      <td>22448.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-Cvanu24-LizardMS</td>
      <td>199.82</td>
      <td>201.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2788</th>
      <td>18418.0</td>
      <td>9184.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-Cvanu7-LizardMS</td>
      <td>331.52</td>
      <td>343.05</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2789</th>
      <td>26369.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-D106-LizardMS</td>
      <td>298.78</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2790</th>
      <td>951.0</td>
      <td>1260.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-D107-LizardMS</td>
      <td>139.42</td>
      <td>140.67</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2791</th>
      <td>8774.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-D111-LizardMS</td>
      <td>150.10</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2792</th>
      <td>18387.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-MS1-LizardMS</td>
      <td>216.82</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2793</th>
      <td>32387.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-MS6-LizardMS</td>
      <td>173.97</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2794</th>
      <td>24150.0</td>
      <td>19251.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-MS7-LizardMS</td>
      <td>250.33</td>
      <td>254.26</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
    <tr>
      <th>2795</th>
      <td>31138.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148-MS8-LizardMS</td>
      <td>113.06</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104148</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2796</th>
      <td>136.0</td>
      <td>106.0</td>
      <td>202.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-A105-LizardMS</td>
      <td>201.15</td>
      <td>205.06</td>
      <td>226.95</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2797</th>
      <td>9322.0</td>
      <td>8966.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-Ai5013-LizardMS</td>
      <td>241.50</td>
      <td>247.02</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2798</th>
      <td>2751.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-Ai5043-LizardMS</td>
      <td>181.31</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2799</th>
      <td>16578.0</td>
      <td>14548.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-Cvanu24-LizardMS</td>
      <td>199.91</td>
      <td>201.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2800</th>
      <td>1693.0</td>
      <td>1309.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-Cvanu7-LizardMS</td>
      <td>331.53</td>
      <td>338.83</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2801</th>
      <td>2721.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-D106-LizardMS</td>
      <td>298.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2802</th>
      <td>330.0</td>
      <td>302.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-D107-LizardMS</td>
      <td>127.67</td>
      <td>154.15</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2803</th>
      <td>1910.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-D111-LizardMS</td>
      <td>150.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2804</th>
      <td>8394.0</td>
      <td>8372.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-MS1-LizardMS</td>
      <td>216.75</td>
      <td>234.91</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2805</th>
      <td>13653.0</td>
      <td>10479.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-MS6-LizardMS</td>
      <td>172.11</td>
      <td>174.08</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2806</th>
      <td>13723.0</td>
      <td>7639.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-MS7-LizardMS</td>
      <td>219.50</td>
      <td>254.22</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
    <tr>
      <th>2807</th>
      <td>9612.0</td>
      <td>6378.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192-MS8-LizardMS</td>
      <td>117.41</td>
      <td>130.13</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104192</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2820</th>
      <td>53.0</td>
      <td>52.0</td>
      <td>58.0</td>
      <td>68.0</td>
      <td>NaN</td>
      <td>104245-A105-LizardMS</td>
      <td>193.81</td>
      <td>195.76</td>
      <td>200.44</td>
      <td>217.25</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2821</th>
      <td>29737.0</td>
      <td>19672.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-Ai5013-LizardMS</td>
      <td>216.28</td>
      <td>243.08</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2822</th>
      <td>3069.0</td>
      <td>2911.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-Ai5043-LizardMS</td>
      <td>178.60</td>
      <td>181.50</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2823</th>
      <td>32635.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-Cvanu24-LizardMS</td>
      <td>199.74</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2824</th>
      <td>16668.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-Cvanu7-LizardMS</td>
      <td>331.71</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2825</th>
      <td>19040.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-D106-LizardMS</td>
      <td>307.32</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2826</th>
      <td>667.0</td>
      <td>572.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-D107-LizardMS</td>
      <td>126.64</td>
      <td>157.21</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2827</th>
      <td>6616.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-D111-LizardMS</td>
      <td>225.94</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2828</th>
      <td>32718.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-MS1-LizardMS</td>
      <td>216.79</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2829</th>
      <td>31649.0</td>
      <td>18384.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-MS6-LizardMS</td>
      <td>174.15</td>
      <td>176.17</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2830</th>
      <td>14797.0</td>
      <td>19173.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-MS7-LizardMS</td>
      <td>265.88</td>
      <td>266.74</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
    <tr>
      <th>2831</th>
      <td>30208.0</td>
      <td>21609.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245-MS8-LizardMS</td>
      <td>113.29</td>
      <td>117.48</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104245</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2832</th>
      <td>49.0</td>
      <td>82.0</td>
      <td>68.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-A105-LizardMS</td>
      <td>198.13</td>
      <td>226.43</td>
      <td>245.61</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2833</th>
      <td>32579.0</td>
      <td>27497.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-Ai5013-LizardMS</td>
      <td>224.24</td>
      <td>243.35</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2834</th>
      <td>945.0</td>
      <td>936.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-Ai5043-LizardMS</td>
      <td>175.68</td>
      <td>178.56</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2835</th>
      <td>31628.0</td>
      <td>15008.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-Cvanu24-LizardMS</td>
      <td>203.88</td>
      <td>207.85</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2836</th>
      <td>23000.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-Cvanu7-LizardMS</td>
      <td>331.52</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2837</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2838</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-D107-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2839</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-D111-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2840</th>
      <td>18385.0</td>
      <td>15933.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-MS1-LizardMS</td>
      <td>216.83</td>
      <td>235.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2841</th>
      <td>32572.0</td>
      <td>29951.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-MS6-LizardMS</td>
      <td>157.85</td>
      <td>174.31</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2842</th>
      <td>26503.0</td>
      <td>11294.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-MS7-LizardMS</td>
      <td>258.06</td>
      <td>290.12</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
    <tr>
      <th>2843</th>
      <td>30039.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250-MS8-LizardMS</td>
      <td>113.15</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104250</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2868</th>
      <td>71.0</td>
      <td>58.0</td>
      <td>54.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-A105-LizardMS</td>
      <td>192.93</td>
      <td>233.97</td>
      <td>238.74</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2869</th>
      <td>13574.0</td>
      <td>8755.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-Ai5013-LizardMS</td>
      <td>204.80</td>
      <td>240.35</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2870</th>
      <td>6048.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-Ai5043-LizardMS</td>
      <td>175.56</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2871</th>
      <td>25610.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-Cvanu24-LizardMS</td>
      <td>203.75</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2872</th>
      <td>7732.0</td>
      <td>5801.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-Cvanu7-LizardMS</td>
      <td>332.51</td>
      <td>334.56</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2873</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2874</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-D107-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2875</th>
      <td>5111.0</td>
      <td>4661.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-D111-LizardMS</td>
      <td>241.06</td>
      <td>249.01</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2876</th>
      <td>21010.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-MS1-LizardMS</td>
      <td>234.98</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2877</th>
      <td>32584.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-MS6-LizardMS</td>
      <td>171.91</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2878</th>
      <td>26582.0</td>
      <td>14283.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-MS7-LizardMS</td>
      <td>250.30</td>
      <td>269.72</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
    <tr>
      <th>2879</th>
      <td>30847.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423-MS8-LizardMS</td>
      <td>110.77</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104423</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2880</th>
      <td>118.0</td>
      <td>45.0</td>
      <td>83.0</td>
      <td>76.0</td>
      <td>NaN</td>
      <td>104426-A105-LizardMS</td>
      <td>192.77</td>
      <td>207.78</td>
      <td>211.79</td>
      <td>219.65</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2881</th>
      <td>22638.0</td>
      <td>21094.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-Ai5013-LizardMS</td>
      <td>212.62</td>
      <td>220.58</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2882</th>
      <td>13331.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-Ai5043-LizardMS</td>
      <td>175.59</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2883</th>
      <td>32510.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-Cvanu24-LizardMS</td>
      <td>203.58</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2884</th>
      <td>28604.0</td>
      <td>21127.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-Cvanu7-LizardMS</td>
      <td>332.51</td>
      <td>334.65</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2885</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2886</th>
      <td>99.0</td>
      <td>82.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-D107-LizardMS</td>
      <td>269.94</td>
      <td>272.94</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2887</th>
      <td>2847.0</td>
      <td>2594.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-D111-LizardMS</td>
      <td>253.29</td>
      <td>269.73</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2888</th>
      <td>31870.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-MS1-LizardMS</td>
      <td>235.06</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2889</th>
      <td>32565.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-MS6-LizardMS</td>
      <td>171.98</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2890</th>
      <td>29906.0</td>
      <td>3804.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-MS7-LizardMS</td>
      <td>254.17</td>
      <td>285.30</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
    <tr>
      <th>2891</th>
      <td>29604.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426-MS8-LizardMS</td>
      <td>110.78</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104426</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2892</th>
      <td>48.0</td>
      <td>105.0</td>
      <td>93.0</td>
      <td>153.0</td>
      <td>151</td>
      <td>104479-A105-LizardMS</td>
      <td>201.95</td>
      <td>205.23</td>
      <td>217.9</td>
      <td>234.79</td>
      <td>239.57</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2893</th>
      <td>21389.0</td>
      <td>13995.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-Ai5013-LizardMS</td>
      <td>221.16</td>
      <td>246.99</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2894</th>
      <td>2248.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-Ai5043-LizardMS</td>
      <td>178.51</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2895</th>
      <td>32575.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-Cvanu24-LizardMS</td>
      <td>199.73</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2896</th>
      <td>5873.0</td>
      <td>5266.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-Cvanu7-LizardMS</td>
      <td>331.62</td>
      <td>334.65</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2897</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-D106-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2898</th>
      <td>167.0</td>
      <td>126.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-D107-LizardMS</td>
      <td>126.51</td>
      <td>139.42</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2899</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-D111-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2900</th>
      <td>30043.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-MS1-LizardMS</td>
      <td>234.91</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2901</th>
      <td>32562.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-MS6-LizardMS</td>
      <td>174.14</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2902</th>
      <td>21475.0</td>
      <td>16363.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-MS7-LizardMS</td>
      <td>258.07</td>
      <td>261.97</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
    <tr>
      <th>2903</th>
      <td>30969.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479-MS8-LizardMS</td>
      <td>113.02</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104479</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2904</th>
      <td>50.0</td>
      <td>79.0</td>
      <td>73.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-A105-LizardMS</td>
      <td>204.15</td>
      <td>233.58</td>
      <td>238.33</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2905</th>
      <td>31844.0</td>
      <td>29471.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-Ai5013-LizardMS</td>
      <td>235.37</td>
      <td>240.15</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2906</th>
      <td>9667.0</td>
      <td>9326.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-Ai5043-LizardMS</td>
      <td>175.68</td>
      <td>178.54</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2907</th>
      <td>32610.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-Cvanu24-LizardMS</td>
      <td>199.73</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2908</th>
      <td>17600.0</td>
      <td>17465.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-Cvanu7-LizardMS</td>
      <td>317.48</td>
      <td>331.55</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2909</th>
      <td>15628.0</td>
      <td>12419.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-D106-LizardMS</td>
      <td>290.84</td>
      <td>319.59</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2910</th>
      <td>43.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-D107-LizardMS</td>
      <td>182.48</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2911</th>
      <td>14296.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-D111-LizardMS</td>
      <td>150.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2912</th>
      <td>29710.0</td>
      <td>22374.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-MS1-LizardMS</td>
      <td>217.73</td>
      <td>235.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2913</th>
      <td>31625.0</td>
      <td>21138.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-MS6-LizardMS</td>
      <td>172.01</td>
      <td>174.13</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2914</th>
      <td>32724.0</td>
      <td>13099.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-MS7-LizardMS</td>
      <td>235.63</td>
      <td>261.93</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
    <tr>
      <th>2915</th>
      <td>31883.0</td>
      <td>25148.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480-MS8-LizardMS</td>
      <td>108.93</td>
      <td>117.43</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>104480</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [202]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">display</span><span class="p">(</span><span class="n">wildGenotypeDF</span><span class="p">[</span><span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">sample_name</span> <span class="o">==</span> <span class="s1">&#39;ESP9281&#39;</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Height 1</th>
      <th>Height 2</th>
      <th>Height 3</th>
      <th>Height 4</th>
      <th>Height 5</th>
      <th>Sample ID</th>
      <th>Size 1</th>
      <th>Size 2</th>
      <th>Size 3</th>
      <th>Size 4</th>
      <th>Size 5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2544</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-A105-LizardMS_A07.fsa</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2545</th>
      <td>26417.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-Ai5013-LizardMS_A08.fsa</td>
      <td>253.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2546</th>
      <td>13201.0</td>
      <td>12823.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-Ai5043-LizardMS_A12.fsa</td>
      <td>173.03</td>
      <td>175.87</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2547</th>
      <td>35701.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-Cvanu24-LizardMS_A06.fsa</td>
      <td>200.26</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2548</th>
      <td>1783.0</td>
      <td>5034.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-Cvanu7-LizardMS_A05.fsa</td>
      <td>337.95</td>
      <td>338.99</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2549</th>
      <td>11356.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-D106-LizardMS_A09.fsa</td>
      <td>398.02</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2550</th>
      <td>572.0</td>
      <td>333.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-D107-LizardMS_A10.fsa</td>
      <td>140.65</td>
      <td>189.82</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2551</th>
      <td>3970.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-D111-LizardMS_A11.fsa</td>
      <td>150.09</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2552</th>
      <td>32654.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-MS1-LizardMS_A01.fsa</td>
      <td>235.00</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2553</th>
      <td>32531.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-MS6-LizardMS_A02.fsa</td>
      <td>175.99</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2554</th>
      <td>38004.0</td>
      <td>15960.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-MS7-LizardMS_A03.fsa</td>
      <td>227.41</td>
      <td>286.21</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
    <tr>
      <th>2555</th>
      <td>35119.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281-MS8-LizardMS_A04.fsa</td>
      <td>113.66</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>ESP9281</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [203]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">call_float</span><span class="p">(</span><span class="n">x</span><span class="p">):</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
        <span class="k">return</span> <span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
    <span class="k">except</span> <span class="ne">ValueError</span><span class="p">:</span>
        <span class="k">return</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>

<span class="k">for</span> <span class="n">column</span> <span class="ow">in</span> <span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span><span class="n">regex</span><span class="o">=</span><span class="s1">&#39;Height|Size&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
    <span class="n">wildGenotypeDF</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">=</span> <span class="n">wildGenotypeDF</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">call_float</span><span class="p">)</span>

<span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="n">c</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="s1">&#39;_&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>
<span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[203]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>height_1</th>
      <th>height_2</th>
      <th>height_3</th>
      <th>height_4</th>
      <th>height_5</th>
      <th>sample_id</th>
      <th>size_1</th>
      <th>size_2</th>
      <th>size_3</th>
      <th>size_4</th>
      <th>size_5</th>
      <th>sample_name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-A105-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>1</th>
      <td>5001.0</td>
      <td>5348.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Ai5013-LizardMS</td>
      <td>213.60</td>
      <td>220.39</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>2</th>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Ai5043-LizardMS</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2724.0</td>
      <td>2731.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Cvanu24-LizardMS</td>
      <td>200.00</td>
      <td>203.92</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2936.0</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50-Cvanu7-LizardMS</td>
      <td>331.63</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>20545_50</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [204]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildGenotypeDF</span><span class="p">[</span><span class="o">~</span><span class="n">wildGenotypeDF</span><span class="o">.</span><span class="n">sample_id</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39;A105&#39;</span><span class="p">)][[</span><span class="s1">&#39;sample_id&#39;</span><span class="p">,</span> <span class="s1">&#39;size_1&#39;</span><span class="p">,</span> <span class="s1">&#39;size_2&#39;</span><span class="p">,</span> <span class="s1">&#39;height_1&#39;</span><span class="p">,</span> <span class="s1">&#39;height_2&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">to_excel</span><span class="p">(</span><span class="s1">&#39;../data/cleanWildCaughtGenotypes.xls&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [205]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildMarmPopulation</span> <span class="o">=</span> <span class="n">bliz</span><span class="o">.</span><span class="n">MicrosatellitePopulation</span><span class="p">(</span><span class="s1">&#39;../data/cleanWildCaughtGenotypes.xls&#39;</span><span class="p">,</span> <span class="n">split_id</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [206]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildMarmPopulation</span><span class="o">.</span><span class="n">calc_internal_relatedness</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [207]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">wildMarmPopulation</span><span class="o">.</span><span class="n">ir_df</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>Out[207]:
</pre></div>
</div>
<div class="output_area docutils container">
<div>
<style>
    .dataframe thead tr:only-child th {
        text-align: right;
    }

    .dataframe thead th {
        text-align: left;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sample_name</th>
      <th>homozygosity_by_loci</th>
      <th>internal_relatedness</th>
      <th>num_hom_loci</th>
      <th>total_loci</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>ASH95383</td>
      <td>1.000000</td>
      <td>1.000000</td>
      <td>3</td>
      <td>3</td>
    </tr>
    <tr>
      <th>1</th>
      <td>ESP9281</td>
      <td>0.945705</td>
      <td>0.716907</td>
      <td>9</td>
      <td>11</td>
    </tr>
    <tr>
      <th>2</th>
      <td>ASH77912</td>
      <td>0.979907</td>
      <td>0.693688</td>
      <td>8</td>
      <td>10</td>
    </tr>
    <tr>
      <th>3</th>
      <td>ASH79240</td>
      <td>0.910078</td>
      <td>0.618572</td>
      <td>7</td>
      <td>10</td>
    </tr>
    <tr>
      <th>4</th>
      <td>ASH92211</td>
      <td>0.924920</td>
      <td>0.607237</td>
      <td>8</td>
      <td>11</td>
    </tr>
    <tr>
      <th>5</th>
      <td>ASH276</td>
      <td>0.897886</td>
      <td>0.591083</td>
      <td>2</td>
      <td>3</td>
    </tr>
    <tr>
      <th>6</th>
      <td>DED30</td>
      <td>0.870799</td>
      <td>0.585639</td>
      <td>8</td>
      <td>11</td>
    </tr>
    <tr>
      <th>7</th>
      <td>TJH3049</td>
      <td>0.943729</td>
      <td>0.585429</td>
      <td>8</td>
      <td>11</td>
    </tr>
    <tr>
      <th>8</th>
      <td>ASH74810</td>
      <td>0.919498</td>
      <td>0.579133</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>9</th>
      <td>ESP9339</td>
      <td>0.955960</td>
      <td>0.576756</td>
      <td>8</td>
      <td>11</td>
    </tr>
    <tr>
      <th>10</th>
      <td>MSB74811</td>
      <td>0.943110</td>
      <td>0.576405</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>11</th>
      <td>DJL302</td>
      <td>0.969596</td>
      <td>0.574503</td>
      <td>8</td>
      <td>11</td>
    </tr>
    <tr>
      <th>12</th>
      <td>20545_50</td>
      <td>0.727726</td>
      <td>0.561240</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>13</th>
      <td>MSB74845</td>
      <td>0.949749</td>
      <td>0.550232</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>14</th>
      <td>ASH257</td>
      <td>0.841659</td>
      <td>0.535554</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>15</th>
      <td>ASH95278</td>
      <td>0.852358</td>
      <td>0.534167</td>
      <td>7</td>
      <td>10</td>
    </tr>
    <tr>
      <th>16</th>
      <td>ASH50544</td>
      <td>0.878625</td>
      <td>0.520383</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>17</th>
      <td>ASH9319</td>
      <td>0.835545</td>
      <td>0.493239</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>18</th>
      <td>ASH3107</td>
      <td>0.779146</td>
      <td>0.489776</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>19</th>
      <td>ESP9348</td>
      <td>0.879563</td>
      <td>0.479330</td>
      <td>7</td>
      <td>10</td>
    </tr>
    <tr>
      <th>20</th>
      <td>ASH95229</td>
      <td>0.858512</td>
      <td>0.469257</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>21</th>
      <td>TJH3054_1</td>
      <td>0.734402</td>
      <td>0.457206</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>22</th>
      <td>MSB74683</td>
      <td>0.830661</td>
      <td>0.453333</td>
      <td>6</td>
      <td>9</td>
    </tr>
    <tr>
      <th>23</th>
      <td>104245</td>
      <td>0.602340</td>
      <td>0.434585</td>
      <td>6</td>
      <td>11</td>
    </tr>
    <tr>
      <th>24</th>
      <td>TJH3052</td>
      <td>0.894342</td>
      <td>0.433347</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>25</th>
      <td>DL931</td>
      <td>0.810170</td>
      <td>0.432749</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>26</th>
      <td>DED18</td>
      <td>0.885275</td>
      <td>0.432286</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>27</th>
      <td>TJH3471</td>
      <td>0.894827</td>
      <td>0.426000</td>
      <td>6</td>
      <td>10</td>
    </tr>
    <tr>
      <th>28</th>
      <td>ASH18536</td>
      <td>0.862690</td>
      <td>0.419264</td>
      <td>4</td>
      <td>7</td>
    </tr>
    <tr>
      <th>29</th>
      <td>TJH2868</td>
      <td>0.873408</td>
      <td>0.413793</td>
      <td>7</td>
      <td>11</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>213</th>
      <td>ASH79183</td>
      <td>0.722844</td>
      <td>-0.045206</td>
      <td>3</td>
      <td>7</td>
    </tr>
    <tr>
      <th>214</th>
      <td>ASH207</td>
      <td>0.565846</td>
      <td>-0.046467</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>215</th>
      <td>ASH273</td>
      <td>0.679125</td>
      <td>-0.052038</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>216</th>
      <td>TJH2805</td>
      <td>0.707373</td>
      <td>-0.055218</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>217</th>
      <td>ESP9250</td>
      <td>0.617087</td>
      <td>-0.055492</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>218</th>
      <td>DED54</td>
      <td>0.688651</td>
      <td>-0.060361</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>219</th>
      <td>TJH2928</td>
      <td>0.534696</td>
      <td>-0.063050</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>220</th>
      <td>TJH2874</td>
      <td>0.595521</td>
      <td>-0.068655</td>
      <td>3</td>
      <td>9</td>
    </tr>
    <tr>
      <th>221</th>
      <td>ESP9286</td>
      <td>0.641650</td>
      <td>-0.071637</td>
      <td>3</td>
      <td>10</td>
    </tr>
    <tr>
      <th>222</th>
      <td>ASH1028</td>
      <td>0.520457</td>
      <td>-0.072709</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>223</th>
      <td>ASH248</td>
      <td>0.562779</td>
      <td>-0.080929</td>
      <td>2</td>
      <td>8</td>
    </tr>
    <tr>
      <th>224</th>
      <td>TJH2875</td>
      <td>0.614083</td>
      <td>-0.084542</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>225</th>
      <td>TJH2923</td>
      <td>0.680991</td>
      <td>-0.085451</td>
      <td>3</td>
      <td>10</td>
    </tr>
    <tr>
      <th>226</th>
      <td>TJH3032</td>
      <td>0.634081</td>
      <td>-0.086504</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>227</th>
      <td>ASH95434</td>
      <td>0.641563</td>
      <td>-0.089494</td>
      <td>3</td>
      <td>10</td>
    </tr>
    <tr>
      <th>228</th>
      <td>104077</td>
      <td>0.302531</td>
      <td>-0.099657</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>229</th>
      <td>ESP9236</td>
      <td>0.664756</td>
      <td>-0.100248</td>
      <td>3</td>
      <td>10</td>
    </tr>
    <tr>
      <th>230</th>
      <td>MTH543</td>
      <td>0.339005</td>
      <td>-0.100433</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>231</th>
      <td>ASH253</td>
      <td>0.637554</td>
      <td>-0.100766</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>232</th>
      <td>ASH147</td>
      <td>0.562765</td>
      <td>-0.103567</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>233</th>
      <td>ASH193</td>
      <td>0.338987</td>
      <td>-0.104663</td>
      <td>2</td>
      <td>9</td>
    </tr>
    <tr>
      <th>234</th>
      <td>ASH219_50</td>
      <td>0.509851</td>
      <td>-0.122351</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>235</th>
      <td>TJH2930</td>
      <td>0.612763</td>
      <td>-0.124122</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>236</th>
      <td>ESP9212</td>
      <td>0.617907</td>
      <td>-0.128924</td>
      <td>3</td>
      <td>11</td>
    </tr>
    <tr>
      <th>237</th>
      <td>ASH229</td>
      <td>0.494106</td>
      <td>-0.136949</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>238</th>
      <td>ASH250</td>
      <td>0.557310</td>
      <td>-0.185525</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>239</th>
      <td>TJH2884</td>
      <td>0.545215</td>
      <td>-0.198337</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>240</th>
      <td>GDC5008</td>
      <td>0.543326</td>
      <td>-0.200414</td>
      <td>2</td>
      <td>10</td>
    </tr>
    <tr>
      <th>241</th>
      <td>ASH182</td>
      <td>0.528233</td>
      <td>-0.211511</td>
      <td>1</td>
      <td>5</td>
    </tr>
    <tr>
      <th>242</th>
      <td>ASH252</td>
      <td>0.592701</td>
      <td>-0.214024</td>
      <td>2</td>
      <td>8</td>
    </tr>
  </tbody>
</table>
<p>243 rows × 5 columns</p>
</div></div>
</div>
</div>
<div class="section" id="Merge-figures">
<h2>Merge figures<a class="headerlink" href="#Merge-figures" title="Permalink to this headline">¶</a></h2>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [208]:
</pre></div>
</div>
<div class="input_area highlight-bash"><div class="highlight"><pre>
<span></span>%%bash
<span class="nb">cd</span> ../fig2
rm merged_figures.pdf
pdfunite *.pdf merged_figures.pdf
</pre></div>
</div>
</div>
</div>
<div class="section" id="Grouped-Figures">
<h2>Grouped Figures<a class="headerlink" href="#Grouped-Figures" title="Permalink to this headline">¶</a></h2>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [209]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">marm_ir</span> <span class="o">=</span> <span class="n">marmPopulation</span><span class="o">.</span><span class="n">ir_df</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;ticks&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">1.2</span><span class="p">))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">distplot</span><span class="p">(</span><span class="n">marm_ir</span><span class="p">[</span><span class="s1">&#39;internal_relatedness&#39;</span><span class="p">],</span> <span class="n">kde</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">rug</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkred&#39;</span><span class="p">,</span> <span class="n">hist_kws</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;alpha&#39;</span><span class="p">:</span><span class="mf">0.6</span><span class="p">})</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mf">8.5</span><span class="p">))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Animals&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Internal Relatedness&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax.set_title(&#39;Internal Relatedness of 44 Aspidoscelis marmorata&#39;)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">arrow</span><span class="p">(</span><span class="o">.</span><span class="mi">978</span><span class="p">,</span><span class="mf">2.8</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">head_width</span><span class="o">=</span><span class="mf">0.035</span><span class="p">,</span> <span class="n">head_length</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;#3498db&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">arrow</span><span class="p">(</span><span class="mf">0.355</span> <span class="p">,</span><span class="mf">2.8</span><span class="p">,</span><span class="mf">0.0</span><span class="p">,</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.01</span><span class="p">,</span> <span class="n">head_width</span><span class="o">=</span><span class="mf">0.035</span><span class="p">,</span> <span class="n">head_length</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s2">&quot;#9b59b6&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mf">1.1</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#0.336735</span>
<span class="c1">#ax.spines[&#39;top&#39;].set_visible(False)</span>
<span class="c1">#ax.spines[&#39;right&#39;].set_visible(False)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_bottom</span><span class="p">()</span>
<span class="n">ax</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_left</span><span class="p">()</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/internal_relatedness.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_312_0.png" src="_images/marmorata_genome_paper_figures_3_312_0.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [210]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.2</span><span class="p">,</span><span class="mf">2.5</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.2</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.40</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">])</span>
<span class="n">ax2</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">])</span>
<span class="c1">#############################################################################################</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">nxDF</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="s1">&#39;perc&#39;</span><span class="p">,</span><span class="s1">&#39;size&#39;</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax1</span><span class="p">)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">get_major_formatter</span><span class="p">()</span><span class="o">.</span><span class="n">set_scientific</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">FuncFormatter</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">,</span> <span class="n">p</span><span class="p">:</span> <span class="n">format</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="mi">1000000</span><span class="p">),</span> <span class="s1">&#39;,&#39;</span><span class="p">)))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Scaffold Size (MB)&#39;</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Percentage of Total Bases in Genome (%)&#39;</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">120</span><span class="p">,</span><span class="mi">20</span><span class="p">))</span>
<span class="c1">#ax1.set_title(&#39;${Aspidoscelis\ marmoratus}$ Genome Continuity&#39;,fontsize=majorFontSize)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span><span class="n">ymin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">ymax</span><span class="o">=</span><span class="mi">90000000</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">hlines</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="mi">32220929</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">,</span><span class="n">xmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">xmax</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="s1">&#39;N50 = 32.22 MB&#39;</span><span class="p">,</span><span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">15</span><span class="p">,</span><span class="mi">28000000</span><span class="p">),</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">vlines</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;g&#39;</span><span class="p">,</span><span class="n">ymin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">ymax</span><span class="o">=</span><span class="mi">90000000</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">hlines</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="mi">8340160</span><span class="p">,</span><span class="n">color</span> <span class="o">=</span><span class="s1">&#39;g&#39;</span><span class="p">,</span><span class="n">xmin</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">xmax</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span><span class="n">linestyle</span><span class="o">=</span><span class="s1">&#39;--&#39;</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="s1">&#39;N90 = 8.34 MB&#39;</span><span class="p">,</span><span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="mi">55</span><span class="p">,</span><span class="mi">4500000</span><span class="p">),</span><span class="n">fontsize</span> <span class="o">=</span> <span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_bottom</span><span class="p">()</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_left</span><span class="p">()</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.22</span><span class="p">,</span> <span class="mf">1.05</span><span class="p">,</span> <span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax1</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
<span class="c1">#############################################################################################</span>
<span class="n">organism_map</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Gallus_gallus&#39;</span><span class="p">:</span> <span class="s1">&#39;Chicken&#39;</span><span class="p">,</span> <span class="s1">&#39;2_Python_molurus_bivittatus-5&#39;</span><span class="p">:</span> <span class="s1">&#39;Burmese python&#39;</span><span class="p">,</span> <span class="s1">&#39;Aspidoscelis_marmorata&#39;</span><span class="p">:</span> <span class="s1">&#39;Aspidoscelis marmoratus&#39;</span><span class="p">,</span>
                <span class="s1">&#39;Pelodiscus_sinensis&#39;</span><span class="p">:</span> <span class="s1">&#39;Chinese softshell turtle&#39;</span><span class="p">,</span> <span class="s1">&#39;Homo_sapiens&#39;</span><span class="p">:</span> <span class="s1">&#39;Human&#39;</span><span class="p">,</span> <span class="s1">&#39;Canis_familiaris&#39;</span><span class="p">:</span> <span class="s1">&#39;Dog&#39;</span><span class="p">,</span> <span class="s1">&#39;Mus_musculus&#39;</span><span class="p">:</span> <span class="s1">&#39;Mouse&#39;</span><span class="p">,</span>
                <span class="s1">&#39;Danio_rerio&#39;</span><span class="p">:</span> <span class="s1">&#39;Zebrafish&#39;</span><span class="p">,</span> <span class="s1">&#39;Anolis_carolinensis&#39;</span><span class="p">:</span> <span class="s1">&#39;Carolina anole lizard&#39;</span><span class="p">}</span>

<span class="n">markers</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;--o&#39;</span><span class="p">,</span> <span class="s1">&#39;--v&#39;</span><span class="p">,</span> <span class="s1">&#39;--^&#39;</span><span class="p">,</span> <span class="s1">&#39;--&lt;&#39;</span><span class="p">,</span> <span class="s1">&#39;--&gt;&#39;</span><span class="p">,</span> <span class="s1">&#39;--8&#39;</span><span class="p">,</span> <span class="s1">&#39;--s&#39;</span><span class="p">,</span> <span class="s1">&#39;--p&#39;</span><span class="p">,</span> <span class="s1">&#39;--h&#39;</span><span class="p">,</span> <span class="s1">&#39;--H&#39;</span><span class="p">,</span> <span class="s1">&#39;--D&#39;</span><span class="p">,</span> <span class="s1">&#39;--d&#39;</span><span class="p">]</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">organism</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">gcSTD</span><span class="o">.</span><span class="n">organism</span><span class="o">.</span><span class="n">unique</span><span class="p">()):</span>
    <span class="c1">#name = organism.replace(&#39;2_&#39;,&#39;&#39;).replace(&#39;-5&#39;,&#39;&#39;).replace(&#39;_&#39;, &#39; &#39;)</span>
    <span class="n">name</span> <span class="o">=</span> <span class="n">organism_map</span><span class="p">[</span><span class="n">organism</span><span class="p">]</span>
    <span class="n">gc_dist</span> <span class="o">=</span> <span class="n">gcSTD</span><span class="p">[</span><span class="n">gcSTD</span><span class="o">.</span><span class="n">organism</span> <span class="o">==</span><span class="n">organism</span><span class="p">]</span>
    <span class="n">gc_dist</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;window_size&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;window_size&#39;</span><span class="p">],</span>
            <span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;std_dev&#39;</span><span class="p">],</span>
            <span class="n">markers</span><span class="p">[</span><span class="n">i</span><span class="p">],</span>
            <span class="n">label</span><span class="o">=</span><span class="n">name</span><span class="p">,</span>
            <span class="n">markersize</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span>
            <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
<span class="k">print</span><span class="p">(</span><span class="n">organism_map</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s2">&quot;log&quot;</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([</span><span class="mi">10</span><span class="p">]</span> <span class="o">+</span> <span class="n">gc_dist</span><span class="p">[</span><span class="s1">&#39;window_size&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">tolist</span><span class="p">())</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">matplotlib</span><span class="o">.</span><span class="n">ticker</span><span class="o">.</span><span class="n">ScalarFormatter</span><span class="p">())</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="mi">2770</span><span class="p">,</span><span class="mi">330000</span><span class="p">)</span>
<span class="n">xtick_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="nb">int</span><span class="p">(</span><span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="mf">1000.0</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax2</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">xtick_labels</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">11</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax2</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">% G</span><span class="s1">C Standard Deviation&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;log(Window Size(kb))&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax2.set_title(&#39;Standard Deviation of GC Content for Multiple Window Sizes&#39;, fontsize=majorFontSize)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span><span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.9</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">get_xaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_bottom</span><span class="p">()</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">get_yaxis</span><span class="p">()</span><span class="o">.</span><span class="n">tick_left</span><span class="p">()</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.22</span><span class="p">,</span> <span class="mf">1.05</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax2</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
<span class="c1">#############################################################################################</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/N50_plot_and_GC_std.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/N50_plot_and_GC_std.png&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="stderr output_area docutils container">
<div class="highlight"><pre>
/home/dut/anaconda3/envs/anaconda2/lib/python2.7/site-packages/ipykernel/__main__.py:41: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{&#39;Anolis_carolinensis&#39;: &#39;Carolina anole lizard&#39;, &#39;Canis_familiaris&#39;: &#39;Dog&#39;, &#39;2_Python_molurus_bivittatus-5&#39;: &#39;Burmese python&#39;, &#39;Gallus_gallus&#39;: &#39;Chicken&#39;, &#39;Mus_musculus&#39;: &#39;Mouse&#39;, &#39;Danio_rerio&#39;: &#39;Zebrafish&#39;, &#39;Aspidoscelis_marmorata&#39;: &#39;Aspidoscelis marmoratus&#39;, &#39;Pelodiscus_sinensis&#39;: &#39;Chinese softshell turtle&#39;, &#39;Homo_sapiens&#39;: &#39;Human&#39;}
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_313_2.png" src="_images/marmorata_genome_paper_figures_3_313_2.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [211]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">x_buffer</span> <span class="o">=</span> <span class="mi">10000</span>
<span class="n">num_clusters</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>


<span class="n">sns</span><span class="o">.</span><span class="n">set</span><span class="p">(</span><span class="n">font_scale</span><span class="o">=</span><span class="mf">1.0</span><span class="p">,</span> <span class="n">style</span><span class="o">=</span><span class="s1">&#39;white&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">3.5</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="n">num_clusters</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
<span class="n">base_cluster_length</span> <span class="o">=</span> <span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">stop</span><span class="o">.</span><span class="n">max</span><span class="p">()</span> <span class="o">+</span> <span class="n">x_buffer</span>
<span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">scaffold</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">([</span><span class="s1">&#39;Scpiz6a_30.1&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_86&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_37&#39;</span><span class="p">,</span><span class="s1">&#39;Scpiz6a_1&#39;</span><span class="p">]):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">i</span><span class="p">,</span> <span class="p">:])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4</span><span class="p">)</span>
    <span class="c1">#base_cluster_length = hoxGeneCoordsDF[hoxGeneCoordsDF.seqid == scaffold].stop.max() + x_buffer</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="n">x_buffer</span><span class="p">,</span> <span class="n">base_cluster_length</span> <span class="o">+</span> <span class="n">x_buffer</span><span class="p">)</span>
    <span class="n">cluster_patch</span> <span class="o">=</span> <span class="n">coord_to_square</span><span class="p">(</span><span class="n">coord</span> <span class="o">=</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="n">base_cluster_length</span><span class="p">),</span> <span class="n">level</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;lightgrey&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.15</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">cluster_patch</span><span class="p">)</span>
    <span class="n">counter</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">hoxGeneCoordsDF</span><span class="p">[</span><span class="n">hoxGeneCoordsDF</span><span class="o">.</span><span class="n">seqid</span> <span class="o">==</span> <span class="n">scaffold</span><span class="p">]</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
        <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;strand&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;+&#39;</span><span class="p">:</span>
            <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;firebrick&#39;</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;midnightblue&#39;</span>
        <span class="n">evidence_square</span> <span class="o">=</span> <span class="n">coord_to_square</span><span class="p">(</span><span class="n">coord</span> <span class="o">=</span> <span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">],</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;stop&#39;</span><span class="p">]),</span> <span class="n">level</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">color</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mf">0.45</span><span class="p">)</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">add_patch</span><span class="p">(</span><span class="n">evidence_square</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">counter</span> <span class="o">%</span> <span class="mi">2</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
            <span class="n">text_div</span> <span class="o">=</span> <span class="mf">15.0</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">text_div</span> <span class="o">=</span> <span class="mf">1.3</span>

        <span class="n">ax</span><span class="o">.</span><span class="n">annotate</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;gene&#39;</span><span class="p">],</span> <span class="n">xy</span><span class="o">=</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="p">((</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;stop&#39;</span><span class="p">]</span> <span class="o">-</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;start&#39;</span><span class="p">])</span> <span class="o">/</span> <span class="n">text_div</span><span class="p">),</span>
                                   <span class="mf">4.0</span><span class="p">),</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;left&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">40</span><span class="p">)</span>
        <span class="n">counter</span><span class="o">+=</span><span class="mi">1</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;right&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;top&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;left&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">([])</span>
    <span class="n">xticks</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">([])</span>
<span class="n">ax</span><span class="o">.</span><span class="n">spines</span><span class="p">[</span><span class="s1">&#39;bottom&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">set_visible</span><span class="p">(</span><span class="bp">True</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">280000</span><span class="p">,</span><span class="mi">20000</span><span class="p">))</span>
<span class="n">x1kb_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="mi">1000</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">x1kb_labels</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;scale (KB)&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/hox_clusters.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/hox_clusters.png&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span><span class="n">dpi</span><span class="o">=</span><span class="mi">1000</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_314_0.png" src="_images/marmorata_genome_paper_figures_3_314_0.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [212]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">6.4</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.3</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.30</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
<span class="n">column</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">]</span>

<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">()):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">column</span><span class="p">[</span><span class="n">i</span><span class="p">]])</span>
    <span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
    <span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
    <span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
    <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
    <span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

    <span class="n">ax</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span><span class="n">legend</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span>  <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ticks</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
    <span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
    <span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax.set_title( &#39;Animal:&#39;%change_name(animal), fontsize=minorFontSize)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">140000000</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mi">26</span><span class="p">,</span><span class="mi">100000000</span><span class="p">,</span><span class="s1">&#39;avg. cov = </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="nb">round</span><span class="p">(</span><span class="n">cov_mean</span><span class="p">,</span><span class="mi">2</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.18</span><span class="p">,</span> <span class="mf">1.08</span><span class="p">,</span> <span class="n">letters</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_coverage_distributions.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_coverage_distributions.png&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span><span class="n">dpi</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

<span class="ansi-red-fg">IndexError</span>Traceback (most recent call last)
<span class="ansi-green-fg">&lt;ipython-input-212-b27acee3a0dd&gt;</span> in <span class="ansi-cyan-fg">&lt;module&gt;</span><span class="ansi-blue-fg">()</span>
<span class="ansi-green-intense-fg ansi-bold">      8</span>
<span class="ansi-green-intense-fg ansi-bold">      9</span> <span class="ansi-green-fg">for</span> i<span class="ansi-blue-fg">,</span> animal <span class="ansi-green-fg">in</span> enumerate<span class="ansi-blue-fg">(</span>profCountDF<span class="ansi-blue-fg">.</span>animal<span class="ansi-blue-fg">.</span>unique<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">:</span>
<span class="ansi-green-fg">---&gt; 10</span><span class="ansi-red-fg">     </span>ax <span class="ansi-blue-fg">=</span> plt<span class="ansi-blue-fg">.</span>subplot<span class="ansi-blue-fg">(</span>gs<span class="ansi-blue-fg">[</span>row<span class="ansi-blue-fg">[</span>i<span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">,</span> column<span class="ansi-blue-fg">[</span>i<span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">)</span>
<span class="ansi-green-intense-fg ansi-bold">     11</span>     outer_lim <span class="ansi-blue-fg">=</span> <span class="ansi-cyan-fg">50</span>
<span class="ansi-green-intense-fg ansi-bold">     12</span>     cov_dist <span class="ansi-blue-fg">=</span> profCountDF<span class="ansi-blue-fg">[</span>profCountDF<span class="ansi-blue-fg">.</span>animal <span class="ansi-blue-fg">==</span> animal<span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">&#39;cov&#39;</span><span class="ansi-blue-fg">,</span><span class="ansi-blue-fg">&#39;occurence&#39;</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">.</span>groupby<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">&#39;cov&#39;</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">&#39;occurence&#39;</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">.</span>sum<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>reset_index<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>

<span class="ansi-red-fg">IndexError</span>: list index out of range
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/marmorata_genome_paper_figures_3_316_1.png" src="_images/marmorata_genome_paper_figures_3_316_1.png" />
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>

<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">6.4</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.6</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.30</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
<span class="n">column</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
<span class="n">covValuesDict</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;A_tigris8450&#39;</span><span class="p">:</span><span class="mi">18</span><span class="p">,</span>
<span class="s1">&#39;Atig_122&#39;</span><span class="p">:</span><span class="mi">18</span><span class="p">,</span>
<span class="s1">&#39;Atig003&#39;</span><span class="p">:</span><span class="mi">18</span><span class="p">,</span>
<span class="s1">&#39;Atig001&#39;</span><span class="p">:</span><span class="mi">16</span><span class="p">,</span>
<span class="p">}</span>

<span class="n">letters</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="s1">&#39;C&#39;</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">]</span>


<span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">animal</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">()):</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">column</span><span class="p">[</span><span class="n">i</span><span class="p">]])</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.18</span><span class="p">,</span> <span class="mf">1.08</span><span class="p">,</span> <span class="n">letters</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span><span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
    <span class="n">cov</span> <span class="o">=</span> <span class="n">covValuesDict</span><span class="p">[</span><span class="n">animal</span><span class="p">]</span>
    <span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
    <span class="n">ax</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
    <span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>

    <span class="c1">#legend</span>
    <span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
    <span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
    <span class="n">l</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span>
                  <span class="p">)</span>
    <span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

    <span class="c1">#label bars</span>
    <span class="n">rects</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">patches</span>

    <span class="c1"># Now make some labels</span>
    <span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span><span class="k">str</span>(i)[:4] for i,t in zip(subed.tn_tv_ratio, subed.total)]

    <span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
        <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
        <span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">25</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>

    <span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites In Genome&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#plt.xticks(rotation=45)</span>


    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4500000</span><span class="p">,</span><span class="mi">500000</span><span class="p">))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_tntv.pdf&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_tntv.png&#39;</span><span class="p">,</span><span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span><span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.1</span><span class="p">,</span><span class="n">dpi</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>

<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s1">&#39;whitegrid&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mf">5.8</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.9</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.90</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">])</span>
<span class="n">ax1_5</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">])</span>
<span class="n">ax2</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">2</span><span class="p">:])</span>
<span class="n">ax3</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">:])</span>
<span class="n">ax4</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">2</span><span class="p">])</span>
<span class="n">ax5</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">:])</span>
<span class="n">all_axes</span> <span class="o">=</span> <span class="p">[</span><span class="n">ax1</span><span class="p">,</span> <span class="n">ax1_5</span><span class="p">,</span> <span class="n">ax2</span><span class="p">,</span> <span class="n">ax3</span><span class="p">,</span> <span class="n">ax4</span><span class="p">,</span> <span class="n">ax5</span><span class="p">]</span>

<span class="c1">######################################################################################################################################################</span>

<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">,</span> <span class="p">{</span><span class="s1">&#39;axes.grid&#39;</span> <span class="p">:</span> <span class="bp">False</span><span class="p">})</span>
<span class="n">animal</span><span class="o">=</span><span class="s2">&quot;A_tigris8450&quot;</span>
<span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

<span class="n">ax1</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span><span class="n">legend</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax1</span><span class="p">,</span>  <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">])</span>

<span class="n">ticks</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
<span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
<span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax1.set_title(&quot;Distribution of Coverage: %s&quot; % change_name(animal))</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">140000000</span><span class="p">)</span>

<span class="c1">#add fill</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax1</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>

<span class="c1">#add figure label</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">6</span><span class="p">})</span>

<span class="n">ax1</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax1</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="c1">####################################################################################################################################################</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">,</span> <span class="p">{</span><span class="s1">&#39;axes.grid&#39;</span> <span class="p">:</span> <span class="bp">False</span><span class="p">})</span>

<span class="n">animal</span><span class="o">=</span><span class="s2">&quot;Atig_122&quot;</span>
<span class="n">outer_lim</span> <span class="o">=</span> <span class="mi">50</span>
<span class="c1">#profCountDF, animal</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">profCountDF</span><span class="p">[</span><span class="n">profCountDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">][[</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="s1">&#39;occurence&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;cov&#39;</span><span class="p">)[</span><span class="s1">&#39;occurence&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
<span class="n">cov_mean</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">*</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="p">)</span><span class="o">/</span><span class="n">cov_dist</span><span class="o">.</span><span class="n">occurence</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;occurence&#39;</span><span class="p">][</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="n">outer_lim</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">cov_dist</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="p">[</span><span class="n">cov_dist</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="n">outer_lim</span><span class="p">]</span>

<span class="n">ax1_5</span> <span class="o">=</span> <span class="n">cov_dist</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;occurence&#39;</span><span class="p">,</span><span class="n">kind</span><span class="o">=</span><span class="s1">&#39;bar&#39;</span><span class="p">,</span><span class="n">legend</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax1_5</span><span class="p">,</span>  <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">edgecolor</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">])</span>
<span class="n">ticks</span> <span class="o">=</span> <span class="n">ax1_5</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklocs</span><span class="p">()</span>
<span class="n">ticklabels</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_text</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">ax1_5</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">get_ticklabels</span><span class="p">()]</span>
<span class="n">ticklabels</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="sa">r</span><span class="s1">&#39;$\geq</span><span class="si">%s</span><span class="s1">$&#39;</span><span class="o">%</span><span class="k">str</span>(outer_lim)
<span class="n">ax1_5</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticks</span><span class="p">(</span><span class="n">ticks</span><span class="p">[::</span><span class="mi">5</span><span class="p">])</span>
<span class="n">ax1_5</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_ticklabels</span><span class="p">(</span><span class="n">ticklabels</span><span class="p">[::</span><span class="mi">5</span><span class="p">],</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="n">ax1_5</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Sites&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1_5</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#ax1_5.set_title(change_name(animal))</span>
<span class="n">ax1_5</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">140000000</span><span class="p">)</span>

<span class="c1">#add fill</span>
<span class="n">ax1_5</span><span class="o">.</span><span class="n">fill_between</span><span class="p">((</span><span class="n">cov_mean</span><span class="p">,</span> <span class="n">ax1_5</span><span class="o">.</span><span class="n">get_xlim</span><span class="p">()[</span><span class="mi">1</span><span class="p">]),</span><span class="n">ax1_5</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">0</span><span class="p">],</span><span class="n">ax1_5</span><span class="o">.</span><span class="n">get_ylim</span><span class="p">()[</span><span class="mi">1</span><span class="p">],</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;grey&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>

<span class="n">ax1_5</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">6</span><span class="p">})</span>

<span class="n">ax1_5</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax1</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>


<span class="c1">####################################################################################################################################################</span>
<span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">,</span> <span class="p">{</span><span class="s1">&#39;axes.grid&#39;</span> <span class="p">:</span> <span class="bp">True</span><span class="p">})</span>

<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">perCovEqualRatesDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">ax2</span> <span class="o">=</span> <span class="n">perCovEqualRatesDF</span><span class="p">[(</span><span class="n">perCovEqualRatesDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">perCovEqualRatesDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">2</span><span class="p">)]</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s1">&#39;cov&#39;</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="s1">&#39;het_per_10kb&#39;</span><span class="p">,</span>
                                                                                                           <span class="n">ylim</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mf">0.5</span><span class="p">,</span><span class="mi">18</span><span class="p">),</span> <span class="n">xlim</span> <span class="o">=</span> <span class="p">(</span><span class="mf">3.5</span><span class="p">,</span><span class="mf">100.5</span><span class="p">),</span>
                                                                                                           <span class="n">style</span><span class="o">=</span><span class="s1">&#39;--o&#39;</span><span class="p">,</span> <span class="n">yticks</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">18</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span>
                                                                                                           <span class="n">logy</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span><span class="n">xticks</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">102</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span>
                                                                                                           <span class="n">label</span><span class="o">=</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax2</span><span class="p">,</span>
                                                                                                           <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span>
                                                                                                           <span class="n">markersize</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">linewidth</span><span class="o">=</span><span class="mf">0.8</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Equal Het. Sites per 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Coverage&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
    <span class="c1">#ax2.set_title(&#39;Rate of Even Split Heterozygous Sites vs Coverage&#39;, fontsize=minorFontSize)</span>
    <span class="n">ax2</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.32</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span> <span class="n">prop</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;size&#39;</span><span class="p">:</span><span class="mi">8</span><span class="p">})</span>



<span class="n">ax2</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax2</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>
<span class="c1">#ax2.yaxis.set_major_formatter(ScalarFormatter())</span>
<span class="c1">#ax2.yaxis.set_major_formatter(ticker.FuncFormatter(lambda y,pos: (&#39;{{:.{:1d}f}}&#39;.format(int(np.maximum(-np.log10(y),0)))).format(y)))</span>

<span class="n">ax2</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax2</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="c1">###################################################################################################################################################</span>

<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;Atig_122&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>
<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
<span class="n">ax3</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax3</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax3</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">l</span> <span class="o">=</span> <span class="n">ax3</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span>
              <span class="p">)</span>
<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax3</span><span class="o">.</span><span class="n">patches</span>

<span class="c1"># Now make some labels</span>
<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span><span class="k">str</span>(i)[:4] for i,t in zip(subed.tn_tv_ratio, subed.total)]

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax3</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">25</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax3</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="c1">#plt.xticks(rotation=45)</span>


<span class="n">ax3</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4500000</span><span class="p">,</span><span class="mi">500000</span><span class="p">))</span>


<span class="n">ax3</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;D&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax3</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="c1">##################################################################################################################################################</span>


<span class="n">animal</span> <span class="o">=</span> <span class="s1">&#39;A_tigris8450&#39;</span>
<span class="n">cov</span> <span class="o">=</span> <span class="mi">18</span>
<span class="n">subed</span> <span class="o">=</span> <span class="n">tntvDF</span><span class="p">[(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">tntvDF</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">cov</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">tntvDF</span><span class="o">.</span><span class="n">definable</span><span class="o">==</span><span class="bp">True</span><span class="p">)]</span>
<span class="n">ax4</span><span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transitions</span> <span class="o">+</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;lightblue&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax4</span><span class="p">)</span>

<span class="c1">#Plot 2 - overlay - &quot;bottom&quot; series</span>
<span class="n">bottom_plot</span> <span class="o">=</span> <span class="n">sns</span><span class="o">.</span><span class="n">barplot</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">allele_count</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">subed</span><span class="o">.</span><span class="n">transversions</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">&quot;purple&quot;</span><span class="p">,</span><span class="n">ax</span><span class="o">=</span><span class="n">ax4</span><span class="p">)</span>

<span class="c1">#legend</span>
<span class="n">topbar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s2">&quot;lightblue&quot;</span><span class="p">,</span> <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">bottombar</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">Rectangle</span><span class="p">((</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">),</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">fc</span><span class="o">=</span><span class="s1">&#39;purple&#39;</span><span class="p">,</span>  <span class="n">edgecolor</span> <span class="o">=</span> <span class="s1">&#39;none&#39;</span><span class="p">)</span>
<span class="n">l</span> <span class="o">=</span> <span class="n">ax4</span><span class="o">.</span><span class="n">legend</span><span class="p">([</span><span class="n">bottombar</span><span class="p">,</span> <span class="n">topbar</span><span class="p">],</span> <span class="p">[</span><span class="s1">&#39;Transversions&#39;</span><span class="p">,</span> <span class="s1">&#39;Transitions&#39;</span><span class="p">],</span> <span class="n">loc</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">ncol</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span>
              <span class="p">)</span>
<span class="n">l</span><span class="o">.</span><span class="n">draw_frame</span><span class="p">(</span><span class="bp">False</span><span class="p">)</span>

<span class="c1">#label bars</span>
<span class="n">rects</span> <span class="o">=</span> <span class="n">ax4</span><span class="o">.</span><span class="n">patches</span>

<span class="c1"># Now make some labels</span>
<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span><span class="k">str</span>(i)[:4] for i,t in zip(subed.tn_tv_ratio, subed.total)]

<span class="k">for</span> <span class="n">rect</span><span class="p">,</span> <span class="n">label</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">rects</span><span class="p">,</span> <span class="n">labels</span><span class="p">):</span>
    <span class="n">height</span> <span class="o">=</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_height</span><span class="p">()</span>
    <span class="n">ax4</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="n">rect</span><span class="o">.</span><span class="n">get_x</span><span class="p">()</span> <span class="o">+</span> <span class="n">rect</span><span class="o">.</span><span class="n">get_width</span><span class="p">()</span><span class="o">/</span><span class="mi">2</span><span class="p">,</span> <span class="n">height</span> <span class="o">+</span> <span class="mi">25</span><span class="p">,</span> <span class="n">label</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="o">-</span><span class="mi">2</span><span class="p">)</span>

<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Number of Sites&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">bottom_plot</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Genotype&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax4</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">ax4</span><span class="o">.</span><span class="n">get_xticklabels</span><span class="p">(),</span><span class="n">rotation</span><span class="o">=</span><span class="mi">45</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="n">ax4</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1"> Genotypes at </span><span class="si">%s</span><span class="s1">X&#39;</span><span class="o">%</span><span class="p">(</span><span class="n">change_name</span><span class="p">(</span><span class="n">animal</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">cov</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax4</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax4</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="nb">xrange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">4500000</span><span class="p">,</span><span class="mi">500000</span><span class="p">))</span>


<span class="n">ax4</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;E&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax4</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>

<span class="c1">##################################################################################################################################################</span>

<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">animal_ids</span><span class="p">:</span>
    <span class="n">sns</span><span class="o">.</span><span class="n">distplot</span><span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span><span class="o">==</span><span class="n">animal</span><span class="p">]</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="mi">100</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">dropna</span><span class="p">(),</span>
                 <span class="n">kde</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                 <span class="n">hist_kws</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;alpha&quot;</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="p">:</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span><span class="s2">&quot;linewidth&quot;</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span><span class="s2">&quot;histtype&quot;</span><span class="p">:</span> <span class="s2">&quot;step&quot;</span><span class="p">},</span>
                 <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                 <span class="n">ax</span><span class="o">=</span><span class="n">ax5</span><span class="p">)</span>


<span class="c1">#ax5.set_title(&#39;1Mb Heterozygosity Distribution&#39;, fontsize=minorFontSize)</span>
<span class="n">ax5</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Number of Windows&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax5</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Mean of Het Sites per 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax5</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax5</span><span class="o">.</span><span class="n">ticklabel_format</span><span class="p">(</span><span class="n">style</span><span class="o">=</span><span class="s1">&#39;sci&#39;</span><span class="p">,</span><span class="n">scilimits</span><span class="o">=</span><span class="p">(</span><span class="o">-</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">),</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span><span class="n">useOffset</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

<span class="n">ax5</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.15</span><span class="p">,</span> <span class="mf">1.3</span><span class="p">,</span> <span class="s1">&#39;F&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax5</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>


<span class="c1">#####################################################################################################################################################</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/heterozygosity_qc.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/heterozygosity_qc.png&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.2</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>

<span></span><span class="n">sns</span><span class="o">.</span><span class="n">set_style</span><span class="p">(</span><span class="s2">&quot;whitegrid&quot;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">=</span><span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span> <span class="mf">3.8</span><span class="p">))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.9</span><span class="p">,</span><span class="n">wspace</span><span class="o">=</span><span class="mf">0.25</span><span class="p">)</span>
<span class="n">gs</span> <span class="o">=</span> <span class="n">gridspec</span><span class="o">.</span><span class="n">GridSpec</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">ax1</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="p">:])</span>
<span class="c1">#ax2 = plt.subplot(gs[1, 0])</span>
<span class="n">ax3</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="p">:])</span>
<span class="c1">#ax4 = plt.subplot(gs[2, 0])</span>


<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>

<span class="n">step</span><span class="o">=</span><span class="mi">500</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">five_mb_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>

    <span class="n">ax1</span> <span class="o">=</span> <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                                      <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                                      <span class="n">style</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span>
                                      <span class="n">ax</span><span class="o">=</span><span class="n">ax1</span><span class="p">,</span>
                                      <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                      <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                                      <span class="n">markersize</span><span class="o">=</span><span class="mf">2.0</span><span class="p">,</span>
                                      <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

    <span class="n">ax1</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                     <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">five_mb_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                     <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span>


<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax1</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Genome Wide Heterozygosity 5Mb Sliding Window&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span> <span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Avg. Het. Sites / 10kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">4</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.25</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span><span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span><span class="n">fontsize</span> <span class="o">=</span> <span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.25</span><span class="p">,</span> <span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax1</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
      <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>
<span class="c1">#######################################################################################################################################################################################</span>

<span class="n">step</span><span class="o">=</span><span class="mi">100</span>
<span class="n">scaffold</span><span class="o">=</span><span class="s1">&#39;Scpiz6a_45&#39;</span>
<span class="n">lns</span> <span class="o">=</span> <span class="bp">None</span>
<span class="k">for</span> <span class="n">animal</span> <span class="ow">in</span> <span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="n">scaffold_het_rolling</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;animal&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">animal</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="s1">&#39;chrom&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">scaffold</span><span class="p">)]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span>
    <span class="n">ln1</span> <span class="o">=</span> <span class="n">ax3</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
           <span class="s1">&#39;.&#39;</span><span class="p">,</span>
            <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
            <span class="n">label</span><span class="o">=</span><span class="n">id_to_name</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
            <span class="n">markersize</span><span class="o">=</span><span class="mf">2.5</span><span class="p">)</span>

    <span class="n">ax3</span><span class="o">.</span><span class="n">fill_between</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="p">,</span>
                     <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">-</span><span class="mi">2</span><span class="o">*</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="o">*</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="o">/</span><span class="n">np</span><span class="o">.</span><span class="n">sqrt</span><span class="p">(</span><span class="n">step</span><span class="p">),</span>
                     <span class="n">color</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="n">animal</span><span class="p">],</span>
                     <span class="n">alpha</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">lns</span><span class="p">:</span>
        <span class="n">lns</span> <span class="o">=</span> <span class="n">ln1</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">lns</span><span class="o">+=</span><span class="n">ln1</span>

<span class="n">ax3</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="o">-</span><span class="mf">2.7</span><span class="p">,</span> <span class="mi">5</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_yticks</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mf">5.0</span><span class="p">,</span><span class="mf">1.0</span><span class="p">))</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_yticklabels</span><span class="p">(</span><span class="n">ax3</span><span class="o">.</span><span class="n">get_yticks</span><span class="p">(),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;Avg. Het Sites / 10Kb&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>


<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax3</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Window Start Position (Mb)&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>

<span class="n">ax3</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%d</span><span class="s1">Mb Sliding Window Scaffold: </span><span class="si">%s</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">step</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000</span><span class="p">,</span> <span class="n">scaffold</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">min</span><span class="p">(),</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">max</span><span class="p">())</span>

<span class="n">labs</span> <span class="o">=</span> <span class="p">[</span><span class="n">l</span><span class="o">.</span><span class="n">get_label</span><span class="p">()</span> <span class="k">for</span> <span class="n">l</span> <span class="ow">in</span> <span class="n">lns</span><span class="p">]</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">lns</span><span class="p">,</span> <span class="n">labs</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.25</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">),</span><span class="n">markerscale</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax3</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mf">0.1</span><span class="p">,</span> <span class="mf">1.25</span><span class="p">,</span> <span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">transform</span><span class="o">=</span><span class="n">ax3</span><span class="o">.</span><span class="n">transAxes</span><span class="p">,</span>
         <span class="n">fontsize</span><span class="o">=</span><span class="n">majorFontSize</span><span class="p">,</span> <span class="n">fontweight</span><span class="o">=</span><span class="s1">&#39;bold&#39;</span><span class="p">,</span> <span class="n">va</span><span class="o">=</span><span class="s1">&#39;top&#39;</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>


<span class="n">chrom</span> <span class="o">=</span> <span class="n">ax3</span><span class="o">.</span><span class="n">add_collection</span><span class="p">(</span><span class="n">BrokenBarHCollection</span><span class="p">(</span>
        <span class="p">[(</span><span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">min</span><span class="p">()</span><span class="o">+</span><span class="mi">2</span><span class="p">,</span> <span class="n">scaffold_het_rolling</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">max</span><span class="p">()</span><span class="o">-</span><span class="mi">105</span><span class="p">)],</span>
        <span class="p">(</span><span class="o">-</span><span class="mf">2.2</span><span class="p">,</span><span class="mf">1.5</span><span class="p">),</span>
        <span class="n">facecolors</span><span class="o">=</span><span class="n">color_ids</span><span class="p">[</span><span class="s1">&#39;A_tigris8450&#39;</span><span class="p">],</span>
        <span class="n">alpha</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span>
        <span class="n">linewidths</span><span class="o">=</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
                                      <span class="p">)</span>
    <span class="p">)</span>


<span class="n">ax3</span><span class="o">.</span><span class="n">add_collection</span><span class="p">(</span><span class="n">BrokenBarHCollection</span><span class="p">(</span>
        <span class="n">vmnrDF</span><span class="p">[</span><span class="n">vmnrDF</span><span class="p">[</span><span class="s1">&#39;feature&#39;</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;gene&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">,</span>
        <span class="p">(</span><span class="o">-</span><span class="mf">2.2</span><span class="p">,</span><span class="mf">1.5</span><span class="p">),</span>
        <span class="n">facecolors</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;black&#39;</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">))],</span>
        <span class="n">linewidths</span><span class="o">=</span><span class="p">[</span><span class="mf">0.5</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">xrange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">vmnrCopyNumbers</span><span class="o">.</span><span class="n">broken_bars</span><span class="p">))]</span>
                                      <span class="p">)</span>
    <span class="p">)</span>



<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/genome_wide_heterozygosity.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/genome_wide_heterozygosity.png&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1">#head -n 1 A_tigris8450.merged.dedup.realigned.prof &gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_45.prof</span>
<span class="c1">#nohup cat A_tigris8450.merged.dedup.realigned.prof | awk &#39;$1==&quot;Scpiz6a_45&quot;&#39; &gt;&gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_45.prof &amp;</span>
<span class="c1">#head -n 1 A_tigris8450.merged.dedup.realigned.prof &gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_122.prof</span>
<span class="c1">#awk &#39;$1==&quot;Scpiz6a_122&quot;&#39; A_tigris8450.merged.dedup.realigned.prof  &gt;&gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_122.prof &amp;</span>
<span class="c1">#head -n 1 A_tigris8450.merged.dedup.realigned.prof &gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_47.prof</span>
<span class="c1">#awk &#39;$1==&quot;Scpiz6a_47&quot;&#39; A_tigris8450.merged.dedup.realigned.prof  &gt;&gt; A_tigris8450.merged.dedup.realigned.Scpiz6a_47.prof &amp;</span>
<span class="c1">#covScaffold_45_8450 = pd.read_csv(&#39;../data/pysam/A_tigris8450.merged.dedup.realigned.Scpiz6a_45.prof&#39;,sep=&#39;\t&#39;)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="c1"># step=1000</span>
<span class="c1"># rolling = covScaffold_45_8450.rolling(step,)[&#39;cov&#39;]</span>

<span class="c1"># ax = rolling.std().plot(title=&#39;Scaffold 45 8450 Standard Deviation Coverage %sbp Sliding Window&#39; % str(step),</span>
<span class="c1">#                          legend=False,</span>
<span class="c1">#                          rot = 90,</span>
<span class="c1">#                          style=&#39;-&#39;,</span>
<span class="c1">#                          #label=animal,</span>
<span class="c1">#                          color=&#39;darkblue&#39;,</span>
<span class="c1">#                          markersize=1.0,</span>
<span class="c1">#                          alpha=0.5,</span>
<span class="c1">#                          figsize=(20,8),</span>
<span class="c1">#                         label=&#39;Rolling Average&#39;)</span>

<span class="c1"># # ax.fill_between(rolling.mean().index,</span>
<span class="c1"># #                     rolling.mean()-2*five_mb_rolling.std()/np.sqrt(step),</span>
<span class="c1"># #                     rolling.mean()+2*five_mb_rolling.std()/np.sqrt(step),</span>
<span class="c1"># #                     color=&#39;blue&#39;,</span>
<span class="c1"># #                     alpha=0.3)</span>
<span class="c1"># print(covScaffold_45_8450[&#39;cov&#39;].mean())</span>
<span class="c1"># ax.hlines(y=covScaffold_45_8450[&#39;cov&#39;].mean(),xmin=ax.get_xlim()[0],xmax=ax.get_xlim()[1],label=&#39;Scaffold Average&#39;)</span>

<span class="c1"># megabase_labels = [str((x/1e6)) for x in ax.get_xticks()]</span>
<span class="c1"># ax.set_xticklabels(megabase_labels,rotation=90)</span>
<span class="c1"># ax.set_xlabel(&#39;Window Start Position (Mb)&#39;)</span>
<span class="c1"># ax.set_ylabel(&#39;Standard Deviation Coverage&#39;)</span>
<span class="c1"># ax.set_ylim(-5, 250)</span>
<span class="c1"># ax.legend(markerscale=10.0)</span>
<span class="c1"># #ax.set_yscale(&#39;log&#39;)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span><span class="n">step</span> <span class="o">=</span> <span class="mi">500</span>
<span class="n">mother</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="s1">&#39;Atig_122&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">fp_animal</span> <span class="o">=</span> <span class="n">equalHetWindowsDF</span><span class="p">[</span><span class="n">equalHetWindowsDF</span><span class="o">.</span><span class="n">animal</span> <span class="o">==</span> <span class="s1">&#39;A_tigris8450&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span><span class="o">.</span><span class="n">het_sites</span><span class="o">.</span><span class="n">rolling</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="n">step</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">ratio</span> <span class="o">=</span> <span class="n">mother</span><span class="o">/</span><span class="n">fp_animal</span>
<span class="k">print</span><span class="p">(</span><span class="nb">min</span><span class="p">(</span><span class="n">ratio</span><span class="o">.</span><span class="n">dropna</span><span class="p">()),</span><span class="nb">max</span><span class="p">(</span><span class="n">ratio</span><span class="o">.</span><span class="n">dropna</span><span class="p">()))</span>
<span class="n">ax</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">legend</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span>
                <span class="n">rot</span> <span class="o">=</span> <span class="mi">90</span><span class="p">,</span>
                <span class="n">style</span><span class="o">=</span><span class="s1">&#39;.&#39;</span><span class="p">,</span>
                <span class="n">label</span><span class="o">=</span><span class="n">animal</span><span class="p">,</span>
                <span class="n">color</span><span class="o">=</span><span class="s1">&#39;black&#39;</span><span class="p">,</span>
                <span class="n">markersize</span><span class="o">=</span><span class="mf">2.0</span><span class="p">,</span>
               <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">6.4</span><span class="p">,</span><span class="mi">2</span><span class="p">),</span>
               <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s1">&#39;Ratio of 122:8450 Average Het sites to per 10kb </span><span class="si">%s</span><span class="s1">Mb Sliding Window&#39;</span> <span class="o">%</span> <span class="nb">str</span><span class="p">(</span><span class="n">step</span><span class="o">/</span><span class="mi">100</span><span class="p">),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">megabase_labels</span> <span class="o">=</span> <span class="p">[</span><span class="nb">str</span><span class="p">((</span><span class="n">x</span><span class="o">*</span><span class="mi">10000</span><span class="o">/</span><span class="mi">1000000000</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_xticks</span><span class="p">()]</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xticklabels</span><span class="p">(</span><span class="n">megabase_labels</span><span class="p">,</span><span class="n">rotation</span><span class="o">=</span><span class="mi">90</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Window Start Position (Gb)&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s1">&#39;log2 ratio&#39;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">9</span><span class="p">)</span>
<span class="n">mean_r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">mean</span><span class="p">()</span>
<span class="n">se_r</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">ratio</span><span class="p">)</span><span class="o">.</span><span class="n">std</span><span class="p">()</span><span class="c1">#/np.sqrt(len(ratio.dropna()))</span>
<span class="n">ax</span><span class="o">.</span><span class="n">text</span><span class="p">(</span><span class="mf">1.0e4</span><span class="p">,</span><span class="mf">0.25</span><span class="p">,</span> <span class="s1">&#39;average log2(ratio) = </span><span class="si">%s</span><span class="se">\n</span><span class="s1">standard deviation log2(ratio) = </span><span class="si">%s</span><span class="s1">&#39;</span><span class="o">%</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">mean_r</span><span class="p">),</span> <span class="nb">str</span><span class="p">(</span><span class="n">se_r</span><span class="p">)),</span> <span class="n">fontsize</span><span class="o">=</span><span class="n">minorFontSize</span><span class="p">)</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ax</span><span class="o">.</span><span class="n">get_figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_log2_ratio.pdf&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="s1">&#39;../fig/supplemental_log2_ratio.png&#39;</span><span class="p">,</span> <span class="n">bbox_inches</span><span class="o">=</span><span class="s1">&#39;tight&#39;</span><span class="p">,</span> <span class="n">pad_inches</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">500</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none"><div class="highlight"><pre>
<span></span>In [ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython2"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">A marmoratus Genome Project</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Genome Paper Figures: DMT analysis Version 2</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Imports">Imports</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Naming-Colors-and-Constants">Naming Colors and Constants</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Accesory-Functions">Accesory Functions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Initial-Microsatellite-Genotyping-of-8450-and-122">Initial Microsatellite Genotyping of 8450 and 122</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Population-Microsatellite-Analysis-and-Figures">Population Microsatellite Analysis and Figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Genomic-Sequencing-Data">Genomic Sequencing Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Genome-Size-Prediction-with-FACs">Genome Size Prediction with FACs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#All-Scaffolds-Genome-Stats">All Scaffolds Genome Stats</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Large-Scaffold-Genome-Stats">Large Scaffold Genome Stats</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Isochore-Analysis">Isochore Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Busco-Analysis-(Done-by-Rutendo-Sigauke)">Busco Analysis (Done by Rutendo Sigauke)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Synteny-Analysis-(Done-by-Rutendo-Sigauke)">Synteny Analysis (Done by Rutendo Sigauke)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Repeat-Analysis-(Done-by-Rutendo-Sigauke)">Repeat Analysis (Done by Rutendo Sigauke)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Load-Annotations">Load Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#HOX-Gene-Analysis">HOX Gene Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#GATK">GATK</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#Running-GATK">Running GATK</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#GATK-Run-on-8450,-001-and-003">GATK Run on 8450, 001 and 003</a></li>
<li class="toctree-l4"><a class="reference internal" href="#GATK-Run-on-122">GATK Run on 122</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#GATK-Initial-Results-Figure-(Generated-by-Morgan-Weichert)">GATK Initial Results Figure (Generated by Morgan Weichert)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#Partial-GATK-pipeline-Run-on-Additional-Animals">Partial GATK pipeline Run on Additional Animals</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#Coverage-Accross-Profiles-and-Affect-on-Heterozygosity">Coverage Accross Profiles and Affect on Heterozygosity</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Confidence-in-Allele-Support-Inferred-from-Transition-Transversion-Ratio">Confidence in Allele Support Inferred from Transition Transversion Ratio</a></li>
<li class="toctree-l2"><a class="reference internal" href="#General-Genotype-Support">General Genotype Support</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Distribution-of-Genome-Average-Even-Heterozygous-Sites">Distribution of Genome Average Even Heterozygous Sites</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Pysam-10kb-Non-Overlapping-Windows">Pysam 10kb Non-Overlapping Windows</a></li>
<li class="toctree-l2"><a class="reference internal" href="#V2R-Copy-Number-analysis-MAKER-Annotations">V2R Copy Number analysis MAKER Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#V2R-Exonerate-Copy-Number-analysis">V2R Exonerate Copy Number analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#V2R-Tree">V2R Tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Embryonic-FACs-Analysis">Embryonic FACs Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Blood-FACs-analysis">Blood FACs analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Wild-Caught-Microsatellite-Analysis">Wild Caught Microsatellite Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Merge-figures">Merge figures</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Grouped-Figures">Grouped Figures</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Welcome to A marmoratus Genome Project’s documentation!</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017, Duncan Tormey.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/marmorata_genome_paper_figures_3.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>