#!/bin/bash
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_4278.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_4278.final_raw_var.vcf &
proc0=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12512.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12512.final_raw_var.vcf &
proc1=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_12513.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12513.final_raw_var.vcf &
proc2=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/A.tig_9721.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_9721.final_raw_var.vcf &
proc3=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_6993.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_6993.final_raw_var.vcf &
proc4=$!
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -I /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/recal_bams/Atig_9177.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_9177.final_raw_var.vcf &
proc5=$!
wait "$proc0" "$proc1" "$proc2" "$proc3" "$proc4" "$proc5"
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_4278.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12512.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_12513.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/A.tig_9721.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_6993.final_raw_var.vcf --variant /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/hc_variant_calling/Atig_9177.final_raw_var.vcf -allSites -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf -selectType SNP -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_snps.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf -selectType INDEL -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_indels.vcf
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/reference/tigris_scaffolds_filt_10000.fa -V /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_indels.vcf
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.gvcf | grep -v '\./\.'  > /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.gvcf
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_snps.vcf | grep 'default_snp_filter' | cut -f 1,2 > /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_snps.tsv
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.final_filt_indels.vcf   | grep 'default_indel_filter' | cut -f 1,2 > /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_indels.tsv
cat /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_indels.tsv /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_snps.tsv > /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_positions.tsv
vcftools --vcf /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.gvcf --recode --out /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/jg_Atig_4278_A.tig_12512_A.tig_12513_A.tig_9721_Atig_6993_Atig_9177.no_call_removed.filt --exclude-positions /n/projects/dut/a_marmorata/parthenogen_heterozygosity/data/gatk_MOLNG-2139_MOLNG-2140/final_variant_calling/joint_genotypes/bad_positions.tsv
