#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450_final.plots) &
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122_final.plots) &
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003_final.plots) &
proc3=$!
wait "$proc1" "$proc2" "$proc3"
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001.merged.dedup.realigned.recal_final.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001_final.plots) &
proc4=$!
wait "$proc1" "$proc2" "$proc3" "$proc4"
