#!/bin/bash
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/A_tigris8450.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf &
#proc0=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig_122.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf &
#proc1=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig003.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf &
#proc2=$!
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller --variant_index_type LINEAR --variant_index_parameter 128000 -ERC GVCF -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/recal_bams/Atig001.merged.dedup.realigned.recal_final.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf &
#proc3=$!
#wait "$proc0" "$proc1" "$proc2" "$proc3"
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T GenotypeGVCFs -nt 24 -ploidy 2 -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/A_tigris8450.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig_122.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig003.final_raw_var.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/Atig001.final_raw_var.vcf -allSites -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_snps.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_snps.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_indels.vcf
#java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_indels.vcf
#cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.gvcf | grep -v '\./\.'  > /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_snps.vcf | grep 'default_snp_filter' | cut -f 1,2 > /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_snps.tsv
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/A_tigris8450_Atig_122_Atig003_Atig001.final_filt_indels.vcf | grep 'default_indel_filter' | cut -f 1,2 > /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_indels.tsv
cat /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_indels.tsv /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_snps.tsv > /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_positions.tsv
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.gvcf --recode --out /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/joint_genotypes/jg_A_tigris8450_Atig_122_Atig003_Atig001.no_call_removed.filt --exclude-positions /home/dut/projects/tigris/heterozygosity/gatk6/final_variant_calling/hc_variant_calling/bad_positions.tsv
