#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_indels.vcf) &
proc0=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_indels.vcf) &
proc1=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_indels.vcf) &
proc2=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_indels.vcf) &
proc3=$!
wait "$proc0" "$proc1" "$proc2" "$proc3"
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.0_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.0_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/A_tigris8450.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_0.plots) &
proc4=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig_122.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_0.plots) &
proc5=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig003.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_0.plots) &
proc6=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/preprocessing/realigned_merged_bams/Atig001.merged.dedup.realigned.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_0.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_0.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_0.plots) &
proc7=$!
wait "$proc4" "$proc5" "$proc6" "$proc7"
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_indels.vcf) &
proc8=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_indels.vcf) &
proc9=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_indels.vcf) &
proc10=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_indels.vcf) &
proc11=$!
wait "$proc8" "$proc9" "$proc10" "$proc11"
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.1_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.1_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_1.plots) &
proc12=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_1.plots) &
proc13=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_1.plots) &
proc14=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_1.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_1.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_1.plots) &
proc15=$!
wait "$proc12" "$proc13" "$proc14" "$proc15"
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_indels.vcf) &
proc16=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_indels.vcf) &
proc17=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_indels.vcf) &
proc18=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_indels.vcf) &
proc19=$!
wait "$proc16" "$proc17" "$proc18" "$proc19"
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.2_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.2_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2
