#!/bin/bash
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450_2.plots) &
proc20=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122_2.plots) &
proc21=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003_2.plots) &
proc22=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T PrintReads -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.recal_2.bam; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.bam -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_2.recode.vcf -knownSites /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_2.recode.vcf -BQSR /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -o /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.after_table; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T AnalyzeCovariates -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -before /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.before_table -after /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.after_table -plots /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001_2.plots) &
proc23=$!
wait "$proc20" "$proc21" "$proc22" "$proc23"
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/A_tigris8450.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_indels.vcf) &
proc24=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig_122.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_indels.vcf) &
proc25=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig003.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_indels.vcf) &
proc26=$!
(java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T HaplotypeCaller -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -I /home/dut/projects/tigris/heterozygosity/gatk6/training/recal_bams/Atig001.merged.dedup.realigned.recal_0.recal_1.recal_2.bam  -stand_call_conf 30 -stand_emit_conf 30 -mbq 17  -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf -selectType SNP -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_snps.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_snp_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_snps.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T SelectVariants -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_var.vcf -selectType INDEL -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_indels.vcf; java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -T VariantFiltration -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -V /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_raw_indels.vcf --filterExpression "QD < 2.0 || FS > 200.0 || ReadPosRankSum < -20.0" --filterName "default_indel_filter" -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_indels.vcf) &
proc27=$!
wait "$proc24" "$proc25" "$proc26" "$proc27"
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_snps.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_snps.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.vcf --excludeNonVariants --minimumN 1
java -Xmx3g -jar /home/dut/bin/GenomeAnalysisTK-3.5/GenomeAnalysisTK.jar -R /home/dut/projects/tigris/heterozygosity/gatk6/reference/tigris_scaffolds_filt_10000.fa -T CombineVariants --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/A_tigris8450.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig_122.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig003.3_filt_indels.vcf --variant /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/Atig001.3_filt_indels.vcf -o /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.vcf --excludeNonVariants --minimumN 1
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3.vcf --remove-filtered LowQual --remove-filtered default_snp_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_snps_3
vcftools --vcf /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3.vcf --remove-filtered LowQual --remove-filtered default_indel_filter --recode --recode-INFO-all --out /home/dut/projects/tigris/heterozygosity/gatk6/training/variant_calling/all_indels_3
